From b3340c4d8913afac0cc33a7c53c2beb5aa14af5c Mon Sep 17 00:00:00 2001
From: Max Horn <max@quendi.de>
Date: Fri, 29 Apr 2022 13:21:58 +0200
Subject: [PATCH] kernel: fix FreeBSD compiler warning

---
 src/libgap-api.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/libgap-api.c b/src/libgap-api.c
index 9ccf59bfe..e14f9682f 100644
--- a/src/libgap-api.c
+++ b/src/libgap-api.c
@@ -611,7 +611,7 @@ void GAP_LeaveStack_(void)
 
 void GAP_EnterDebugMessage_(char * message, char * file, int line)
 {
-    fprintf(stderr, "%s: %d; %s:%d\n", message, EnterStackCount, file, line);
+    fprintf(stderr, "%s: %d; %s:%d\n", message, (int)EnterStackCount, file, line);
 }
 
 int GAP_Error_Prejmp_(const char * file, int line)
-- 
2.36.0

