diff --git a/include/liblas/detail/endian.hpp b/include/liblas/detail/endian.hpp
index cd94997c..946bcb5a 100644
--- a/include/liblas/detail/endian.hpp
+++ b/include/liblas/detail/endian.hpp
@@ -88,7 +88,7 @@
    || defined(_M_ALPHA) || defined(__amd64) \
    || defined(__amd64__) || defined(_M_AMD64) \
    || defined(__x86_64) || defined(__x86_64__) \
-   || defined(_M_X64)
+   || defined(_M_X64) || defined(__aarch64__) || defined(__arm__)
 
 # define LIBLAS_LITTLE_ENDIAN
 # define LIBLAS_BYTE_ORDER 1234
