diff --git a/Makefile b/Makefile
index 1787450..91b42b3 100644
--- a/Makefile
+++ b/Makefile
@@ -1,11 +1,14 @@
 include Makefile.inc
+
+EXE = luna$(exeext)
+
 ifeq ($(ARCH),WINDOWS)
-  TARGETS = luna destrat behead tocol
+  TARGETS = $(EXE) libluna destrat behead tocol
 else	
   ifdef WASM
-    TARGETS=luna libluna destrat
+    TARGETS=$(EXE) libluna destrat
   else
-    TARGETS = luna libluna destrat regional behead dmerge tocol fixrows cgi-mapper simassoc
+    TARGETS = $(EXE) libluna destrat regional behead dmerge tocol fixrows cgi-mapper simassoc
   endif
 endif
 
@@ -55,7 +58,7 @@ DEPS := $(OBJS:.o=.d)
 
 all : $(TARGETS)
 
-luna: main.o $(OBJS)
+$(EXE): main.o $(OBJS)
 	$(CXX) -o $@ $^ $(LDFLAGS)
 
 
@@ -67,13 +70,8 @@ libluna: libluna.a $(SHARED_LIB)
 
 # shared library (libluna)
 
-ifeq ($(ARCH),MAC)
 $(SHARED_LIB) : $(OBJS)
-	$(LD) -dynamiclib $(LDFLAGS) -o $(SHARED_LIB) $(OBJS)
-else ifeq ($(ARCH),LINUX)
-$(SHARED_LIB) : $(OBJS)
-	$(LD) -shared $(LDFLAGS) -o $(SHARED_LIB) $(OBJS)
-endif
+	$(LD) -shared -o $(SHARED_LIB) $(OBJS) $(LDFLAGS)
 
 # objects
 libluna.a : $(OBJS)
diff --git a/Makefile.inc b/Makefile.inc
index a8e1a6f..c7cafdb 100644
--- a/Makefile.inc
+++ b/Makefile.inc
@@ -15,22 +15,11 @@ endif
 DEP_INCLUDES=
 DEP_LIB=
 
-ifdef ZLIB
-DEP_INCLUDES += -I$(ZLIB)/include
-DEP_LIB += -L$(ZLIB)/lib
-endif
+DEP_INCLUDES += -I$(includedir)
+DEP_LIB += -L$(libdir)
 
-ifdef FFTW
-DEP_INCLUDES += -I$(FFTW)/include
-DEP_LIB = -L$(FFTW)/lib
 ifeq ($(ARCH),LINUX)
-DEP_LIB	+= -Wl,-rpath,$(FFTW)/lib
-endif
-endif
-
-ifdef LGBM_PATH
-DEP_INCLUDES +=-I$(LGBM_PATH)/include
-DEP_LIB +=-L$(LGBM_PATH)
+DEP_LIB	+= -Wl,-rpath,$(libdir)
 endif
 
 
@@ -38,16 +27,10 @@ endif
 ## Compiler flags
 ##
 
-ifeq ($(WASM),1)
- CXX = em++
- CC = emcc
-else
- CXX = g++
- CC = gcc
-endif
-
+CFLAGS += -D$(ARCH)
+CXXFLAGS += -D$(ARCH)
 
-CXXFLAGS=-O2 -std=gnu++11 -MMD -MP -I. -I.. 
+CXXFLAGS+=-O2 -std=gnu++11 -MMD -MP -I. -I.. 
 #CXXFLAGS=-O0 -std=gnu++11 -MMD -MP -I. -I.. -pg
 
 CXXFLAGS +=$(DEP_INCLUDES)
@@ -58,7 +41,7 @@ endif
 
 
 ifeq ($(ARCH),MAC)
-CXXFLAGS+=-I/usr/local/include
+CXXFLAGS+=-I$(includedir)
 endif
 
 ifeq ($(ARCH),WINDOWS)
@@ -69,41 +52,18 @@ ifeq ($(LGBM),1)
 CXXFLAGS+=-DHAS_LGBM
 endif
 
-CFLAGS=-O2 -DSQLITE_THREADSAFE=0 -DSQLITE_OMIT_LOAD_EXTENSION
+CFLAGS+=-O2 -DSQLITE_THREADSAFE=0 -DSQLITE_OMIT_LOAD_EXTENSION
 
 
 ##
 ## Linker flags
 ##
 
-ifeq ($(WASM),1)
- LD = em++
-else
- LD = g++
-endif
-
+LD = $(CXX)
 
 LDFLAGS = $(DEP_LIB) -L. -L..
 
-# assuming mingw64 for windows build:
-
-ifeq ($(ARCH),WINDOWS)
- LDFLAGS += -L/usr/local/lib -L/mingw64/lib/ -L/mingw64/bin/ -lfftw3 -lz -static -static-libgcc -static-libstdc++
-else
- ifeq ($(WASM),1)
-  LDFLAGS += -L${WASM_DEPS} -lfftw3 -s USE_ZLIB=1
- else
-  LDFLAGS += -L/usr/local/lib -lfftw3 -lz
- endif
-endif
-
-ifdef LGBM_PATH
-ifeq ($(ARCH),WINDOWS)
-LDFLAGS += -L/mingw64/bin/
-else
-LDFLAGS += -Wl,-rpath,${LGBM_PATH}
-endif
-endif
+LDFLAGS += -L$(libdir) -lfftw3 -lz
 
 ifeq ($(LGBM),1)
 ifeq ($(ARCH),WINDOWS)
@@ -151,7 +111,7 @@ endif
 ## Shared library name
 ##
 
-SHARED_LIB=libluna.dylib
+SHARED_LIB=libluna.$(dlext)
 
 ##
 ## Misc
diff --git a/dsp/firWindowing/Makefile~ b/dsp/firWindowing/Makefile~
index 44504f2..6b7da26 100644
--- a/dsp/firWindowing/Makefile~
+++ b/dsp/firWindowing/Makefile~
@@ -1,4 +1,3 @@
-CC			= g++
 CFLAGS	+= -Wall
 LIBRARY	= -lm -lfftw3
 
diff --git a/sstore/make.sh b/sstore/make.sh
index 4d9147c..fce6e60 100644
--- a/sstore/make.sh
+++ b/sstore/make.sh
@@ -1,11 +1,11 @@
 
 make
-g++ -O2 -c loadss.cpp
-g++ -O2 -o loadss loadss.o sstore.o ../db/sqlwrap.o ../db/sqlite3.o ../helper/helper.o ../defs/defs.o ../miscmath/crandom.o
+${CXX} -O2 -c loadss.cpp
+${CXX} -O2 -o loadss loadss.o sstore.o ../db/sqlwrap.o ../db/sqlite3.o ../helper/helper.o ../defs/defs.o ../miscmath/crandom.o
 
 # simple interval loader :  loadints name < {ints}  
-g++ -O2 -c loadints.cpp
-g++ -O2 -o loadints loadints.o sstore.o ../db/sqlwrap.o ../db/sqlite3.o ../helper/helper.o ../defs/defs.o ../miscmath/crandom.o
+${CXX} -O2 -c loadints.cpp
+${CXX} -O2 -o loadints loadints.o sstore.o ../db/sqlwrap.o ../db/sqlite3.o ../helper/helper.o ../defs/defs.o ../miscmath/crandom.o
 
-g++ -O2 -c tabless.cpp
-g++ -O2 -o tabless tabless.o sstore.o ../db/sqlwrap.o ../db/sqlite3.o ../helper/helper.o ../defs/defs.o ../miscmath/crandom.o
+${CXX} -O2 -c tabless.cpp
+${CXX} -O2 -o tabless tabless.o sstore.o ../db/sqlwrap.o ../db/sqlite3.o ../helper/helper.o ../defs/defs.o ../miscmath/crandom.o
diff --git a/tinyxml/Makefile.orginal b/tinyxml/Makefile.orginal
index 5ffb952..011cc95 100644
--- a/tinyxml/Makefile.orginal
+++ b/tinyxml/Makefile.orginal
@@ -19,9 +19,7 @@ TINYXML_USE_STL := NO
 
 #****************************************************************************
 
-CC     := gcc
-CXX    := g++
-LD     := g++
+LD     := $(CXX)
 AR     := ar rc
 RANLIB := ranlib
 
@@ -37,19 +35,19 @@ DEBUG_LDFLAGS    := -g
 RELEASE_LDFLAGS  :=
 
 ifeq (YES, ${DEBUG})
-   CFLAGS       := ${DEBUG_CFLAGS}
-   CXXFLAGS     := ${DEBUG_CXXFLAGS}
-   LDFLAGS      := ${DEBUG_LDFLAGS}
+   CFLAGS       += ${DEBUG_CFLAGS}
+   CXXFLAGS     += ${DEBUG_CXXFLAGS}
+   LDFLAGS      += ${DEBUG_LDFLAGS}
 else
-   CFLAGS       := ${RELEASE_CFLAGS}
-   CXXFLAGS     := ${RELEASE_CXXFLAGS}
-   LDFLAGS      := ${RELEASE_LDFLAGS}
+   CFLAGS       += ${RELEASE_CFLAGS}
+   CXXFLAGS     += ${RELEASE_CXXFLAGS}
+   LDFLAGS      += ${RELEASE_LDFLAGS}
 endif
 
 ifeq (YES, ${PROFILE})
-   CFLAGS   := ${CFLAGS} -pg -O3
-   CXXFLAGS := ${CXXFLAGS} -pg -O3
-   LDFLAGS  := ${LDFLAGS} -pg
+   CFLAGS   += ${CFLAGS} -pg -O3
+   CXXFLAGS += ${CXXFLAGS} -pg -O3
+   LDFLAGS  += ${LDFLAGS} -pg
 endif
 
 #****************************************************************************
@@ -74,8 +72,8 @@ INCS :=
 # Makefile code common to all platforms
 #****************************************************************************
 
-CFLAGS   := ${CFLAGS}   ${DEFS}
-CXXFLAGS := ${CXXFLAGS} ${DEFS}
+CFLAGS   += ${DEFS}
+CXXFLAGS += ${DEFS}
 
 #****************************************************************************
 # Targets of the build
