diff --git a/Makefile b/Makefile
index 5d7a2ed..67bfa9c 100644
--- a/Makefile
+++ b/Makefile
@@ -1,8 +1,11 @@
 include Makefile.inc
+
+EXE = luna$(exeext)
+
 ifeq ($(ARCH),WINDOWS)
-  TARGETS = luna destrat behead tocol
+  TARGETS = $(EXE) destrat behead tocol
 else
-  TARGETS = luna libluna destrat behead dmerge tocol fixrows cgi-mapper
+  TARGETS = $(EXE) libluna destrat behead dmerge tocol fixrows cgi-mapper
 endif
 
 SRCS = globals.cpp eval.cpp cmddefs.cpp \
@@ -46,7 +49,7 @@ DEPS := $(OBJS:.o=.d)
 
 all : $(TARGETS)
 
-luna: main.o $(OBJS)
+$(EXE): main.o $(OBJS)
 	$(CXX) -o $@ $^ $(LDFLAGS)
 
 libluna: libluna.a $(SHARED_LIB)
diff --git a/Makefile.inc b/Makefile.inc
index 8840537..252c1ea 100644
--- a/Makefile.inc
+++ b/Makefile.inc
@@ -15,17 +15,13 @@ endif
 DEP_INCLUDES=
 DEP_LIB=
 
-ifdef ZLIB
-DEP_INCLUDES += -I$(ZLIB)/include
-DEP_LIB += -L$(ZLIB)/lib
-endif
+DEP_INCLUDES += -I$(includedir)
+DEP_LIB += -L$(libdir)
 
-ifdef FFTW
-DEP_INCLUDES += -I$(FFTW)/include
-DEP_LIB = -L$(FFTW)/lib
+DEP_INCLUDES += -I$(includedir)
+DEP_LIB = -L$(libdir)
 ifeq ($(ARCH),LINUX)
-DEP_LIB	+= -Wl,-rpath,$(FFTW)/lib
-endif
+DEP_LIB	+= -Wl,-rpath,$(libdir)
 endif
 
 
@@ -33,24 +29,23 @@ endif
 ## Compiler flags
 ##
 
-CXX = g++
-
-CC = gcc
+CFLAGS += -D$(ARCH)
+CXXFLAGS += -D$(ARCH)
 
-CXXFLAGS=-O2 -std=gnu++11 -MMD -MP -I. -I..
+CXXFLAGS+=-O2 -std=gnu++11 -MMD -MP -I. -I..
 #CXXFLAGS=-O0 -std=gnu++11 -MMD -MP -I. -I.. -pg
 
 CXXFLAGS +=$(DEP_INCLUDES)
 
 ifeq ($(ARCH),MAC)
-CXXFLAGS += -I/usr/local/include
+CXXFLAGS += -I$(includedir)
 endif
 
 ifeq ($(ARCH),WINDOWS)
 CXXFLAGS += -DWINDOWS
 endif
 
-CFLAGS = -O2 -DSQLITE_THREADSAFE=0 -DSQLITE_OMIT_LOAD_EXTENSION
+CFLAGS += -O2 -DSQLITE_THREADSAFE=0 -DSQLITE_OMIT_LOAD_EXTENSION
 #CFLAGS = -O0 -g -DSQLITE_THREADSAFE=0 -DSQLITE_OMIT_LOAD_EXTENSION
 
 
@@ -58,17 +53,11 @@ CFLAGS = -O2 -DSQLITE_THREADSAFE=0 -DSQLITE_OMIT_LOAD_EXTENSION
 ## Linker flags
 ##
 
-LD = g++ 
+LD = $(CXX)
 
 LDFLAGS = $(DEP_LIB) -L. -L..
 
-# assuming mingw64 for windows build:
-
-ifeq ($(ARCH),WINDOWS)
-LDFLAGS += -L/usr/local/lib -L/mingw64/lib/ -L/mingw64/bin/ -lfftw3 -lz -static -static-libgcc -static-libstdc++
-else
-LDFLAGS += -L/usr/local/lib -lfftw3 -lz
-endif
+LDFLAGS += -L$(libdir) -lfftw3 -lz
 
 ##
 ## Static binary: 'make static STATIC=1', otherwise set PIC code
diff --git a/dsp/firWindowing/Makefile~ b/dsp/firWindowing/Makefile~
index 44504f2..6b7da26 100644
--- a/dsp/firWindowing/Makefile~
+++ b/dsp/firWindowing/Makefile~
@@ -1,4 +1,3 @@
-CC			= g++
 CFLAGS	+= -Wall
 LIBRARY	= -lm -lfftw3
 
diff --git a/sstore/make.sh b/sstore/make.sh
index 4d9147c..fce6e60 100644
--- a/sstore/make.sh
+++ b/sstore/make.sh
@@ -1,11 +1,11 @@
 
 make
-g++ -O2 -c loadss.cpp
-g++ -O2 -o loadss loadss.o sstore.o ../db/sqlwrap.o ../db/sqlite3.o ../helper/helper.o ../defs/defs.o ../miscmath/crandom.o
+${CXX} -O2 -c loadss.cpp
+${CXX} -O2 -o loadss loadss.o sstore.o ../db/sqlwrap.o ../db/sqlite3.o ../helper/helper.o ../defs/defs.o ../miscmath/crandom.o
 
 # simple interval loader :  loadints name < {ints}  
-g++ -O2 -c loadints.cpp
-g++ -O2 -o loadints loadints.o sstore.o ../db/sqlwrap.o ../db/sqlite3.o ../helper/helper.o ../defs/defs.o ../miscmath/crandom.o
+${CXX} -O2 -c loadints.cpp
+${CXX} -O2 -o loadints loadints.o sstore.o ../db/sqlwrap.o ../db/sqlite3.o ../helper/helper.o ../defs/defs.o ../miscmath/crandom.o
 
-g++ -O2 -c tabless.cpp
-g++ -O2 -o tabless tabless.o sstore.o ../db/sqlwrap.o ../db/sqlite3.o ../helper/helper.o ../defs/defs.o ../miscmath/crandom.o
+${CXX} -O2 -c tabless.cpp
+${CXX} -O2 -o tabless tabless.o sstore.o ../db/sqlwrap.o ../db/sqlite3.o ../helper/helper.o ../defs/defs.o ../miscmath/crandom.o
diff --git a/tinyxml/Makefile.orginal b/tinyxml/Makefile.orginal
index 5ffb952..011cc95 100644
--- a/tinyxml/Makefile.orginal
+++ b/tinyxml/Makefile.orginal
@@ -19,9 +19,7 @@ TINYXML_USE_STL := NO
 
 #****************************************************************************
 
-CC     := gcc
-CXX    := g++
-LD     := g++
+LD     := $(CXX)
 AR     := ar rc
 RANLIB := ranlib
 
@@ -37,19 +35,19 @@ DEBUG_LDFLAGS    := -g
 RELEASE_LDFLAGS  :=
 
 ifeq (YES, ${DEBUG})
-   CFLAGS       := ${DEBUG_CFLAGS}
-   CXXFLAGS     := ${DEBUG_CXXFLAGS}
-   LDFLAGS      := ${DEBUG_LDFLAGS}
+   CFLAGS       += ${DEBUG_CFLAGS}
+   CXXFLAGS     += ${DEBUG_CXXFLAGS}
+   LDFLAGS      += ${DEBUG_LDFLAGS}
 else
-   CFLAGS       := ${RELEASE_CFLAGS}
-   CXXFLAGS     := ${RELEASE_CXXFLAGS}
-   LDFLAGS      := ${RELEASE_LDFLAGS}
+   CFLAGS       += ${RELEASE_CFLAGS}
+   CXXFLAGS     += ${RELEASE_CXXFLAGS}
+   LDFLAGS      += ${RELEASE_LDFLAGS}
 endif
 
 ifeq (YES, ${PROFILE})
-   CFLAGS   := ${CFLAGS} -pg -O3
-   CXXFLAGS := ${CXXFLAGS} -pg -O3
-   LDFLAGS  := ${LDFLAGS} -pg
+   CFLAGS   += ${CFLAGS} -pg -O3
+   CXXFLAGS += ${CXXFLAGS} -pg -O3
+   LDFLAGS  += ${LDFLAGS} -pg
 endif
 
 #****************************************************************************
@@ -74,8 +72,8 @@ INCS :=
 # Makefile code common to all platforms
 #****************************************************************************
 
-CFLAGS   := ${CFLAGS}   ${DEFS}
-CXXFLAGS := ${CXXFLAGS} ${DEFS}
+CFLAGS   += ${DEFS}
+CXXFLAGS += ${DEFS}
 
 #****************************************************************************
 # Targets of the build
