diff --git a/Makefile b/Makefile
index 5a56fad..20c08d5 100644
--- a/Makefile
+++ b/Makefile
@@ -31,7 +31,7 @@ luna : main.o globals.o eval.o cmddefs.o $(OBJLIBS)
 	$(LD) $(LDFLAGS) -o luna $(OBJS) $(LIBS)
 
 static : main.o globals.o eval.o cmddefs.o $(OBJLIBS)
-	g++ -static -static-libgcc -static-libstdc++ -L/usr/local/lib	\
+	g++ -static -static-libgcc -static-libstdc++ -L$(LIBDIR)	\
 	-o luna main.o globals.o eval.o cmddefs.o libspindles.a liblwprep.a	\
 	libartifacts.a libtimeline.a libannot.a libedf.a		\
 	libintervals.a libcwt.a libdsp.a libstaging.a libclocs.a	\
diff --git a/Makefile.inc b/Makefile.inc
index d7269b1..0da8cd7 100644
--- a/Makefile.inc
+++ b/Makefile.inc
@@ -54,7 +54,7 @@ endif
 CXXFLAGS += $(DEP_INCLUDES) 
 
 ifeq ($(ARCH),MAC)
-CXXFLAGS += -I/usr/local/include
+CXXFLAGS += -I$(PREFIX)/include
 endif
 
 CFLAGS = -O2 -DSQLITE_THREADSAFE=0 -DSQLITE_OMIT_LOAD_EXTENSION
@@ -68,7 +68,7 @@ LD = g++
 LDFLAGS = $(DEP_LIB)
 
 ifndef WINDOWS
-LDFLAGS += -L/usr/local/lib
+LDFLAGS += -L$(LIBDIR)
 endif
 
 
