diff --git a/cmake/modules/FindAndVerifyLAPACK.cmake b/cmake/modules/FindAndVerifyLAPACK.cmake
index 1292536de..faf290cbb 100644
--- a/cmake/modules/FindAndVerifyLAPACK.cmake
+++ b/cmake/modules/FindAndVerifyLAPACK.cmake
@@ -142,10 +142,12 @@ include(CheckFunctionExists)
 set(CMAKE_REQUIRED_LIBRARIES "${BLAS_LINKER_FLAGS}" "${BLAS_LIBRARIES}")
 check_function_exists(dgemm BLAS_NO_USE_UNDERSCORE)
 check_function_exists(dgemm_ BLAS_USE_UNDERSCORE)
+check_function_exists(dgemm_64_ BLAS_USE_UNDERSCORE_64)
 
 set(CMAKE_REQUIRED_LIBRARIES "${LAPACK_LINKER_FLAGS}" "${LAPACK_LIBRARIES}")
 check_function_exists(dlacpy LAPACK_NO_USE_UNDERSCORE)
 check_function_exists(dlacpy_ LAPACK_USE_UNDERSCORE)
+check_function_exists(dlacpy_64_ LAPACK_USE_UNDERSCORE_64)
 
 # If both dgemm and dgemm_ are found, don't use the suffix
 if (BLAS_NO_USE_UNDERSCORE)
@@ -153,7 +155,10 @@ if (BLAS_NO_USE_UNDERSCORE)
   message(STATUS "Using BLAS with no symbol mangling.")
 elseif (BLAS_USE_UNDERSCORE)
   set(${UPPER_PROJECT_NAME}_BLAS_SUFFIX "_")
-  message(STATUS "Using BLAS with trailing underscore.")
+  message(STATUS "Using BLAS with trailing _.")
+elseif (BLAS_USE_UNDERSCORE_64)
+  set(${UPPER_PROJECT_NAME}_BLAS_SUFFIX "_64_")
+  message(STATUS "Using BLAS with trailing _64_.")
 else ()
   message(FATAL_ERROR "Could not determine BLAS suffix!")
 endif ()
@@ -164,7 +169,10 @@ if (LAPACK_NO_USE_UNDERSCORE)
   message(STATUS "Using LAPACK with no symbol mangling.")
 elseif (LAPACK_USE_UNDERSCORE)
   set(${UPPER_PROJECT_NAME}_LAPACK_SUFFIX "_")
-  message(STATUS "Using LAPACK with trailing underscore.")
+  message(STATUS "Using LAPACK with trailing _.")
+elseif (LAPACK_USE_UNDERSCORE_64)
+  set(${UPPER_PROJECT_NAME}_LAPACK_SUFFIX "_64_")
+  message(STATUS "Using LAPACK with trailing _64_.")
 else ()
   message(STATUS
     "Could not find LAPACK's copy function. Disabling LAPACK support.")
