diff --git a/CHOLMOD/SuiteSparse_metis/GKlib/random.c b/CHOLMOD/SuiteSparse_metis/GKlib/random.c
index b6462a060..fd89a0ca9 100644
--- a/CHOLMOD/SuiteSparse_metis/GKlib/random.c
+++ b/CHOLMOD/SuiteSparse_metis/GKlib/random.c
@@ -71,8 +71,8 @@ GK_MKRANDOM(gk_z,   size_t, ssize_t)
 #elif defined ( HAVE_KEYWORD__THREAD )
 
     // gcc and many other compilers support the __thread keyword
-    __thread static uint64_t mt[NN];
-    __thread static int mti=NN+1;
+    static __thread uint64_t mt[NN];
+    static __thread int mti=NN+1;
 
 #elif defined ( HAVE_KEYWORD__DECLSPEC_THREAD )
 
