diff --git a/bindings/C/adios2/c/adios2_c_adios.cpp b/bindings/C/adios2/c/adios2_c_adios.cpp
index 2f6dd6a61..cfe19f21b 100644
--- a/bindings/C/adios2/c/adios2_c_adios.cpp
+++ b/bindings/C/adios2/c/adios2_c_adios.cpp
@@ -48,7 +48,7 @@ adios2_adios *adios2_init_serial()

 adios2_adios *adios2_init_config_serial(const char *config_file)
 {
-    return adios2_init_config_glue_serial("", adios2_debug_mode_off, "C");
+    return adios2_init_config_glue_serial(config_file, adios2_debug_mode_off, "C");
 }

 adios2_io *adios2_declare_io(adios2_adios *adios, const char *name)
