diff --git a/source/adios2/engine/ssc/SscReader.cpp b/source/adios2/engine/ssc/SscReader.cpp
index e8461258b..ff13c69c2 100644
--- a/source/adios2/engine/ssc/SscReader.cpp
+++ b/source/adios2/engine/ssc/SscReader.cpp
@@ -150,7 +150,7 @@ StepStatus SscReader::BeginStep(const StepMode stepMode,
                 {
                     helper::Log("Engine", "SSCReader", "BeginStep",
                                 "unknown data type", 0, m_Comm.Rank(), 0,
-                                m_Verbosity, helper::LogMode::ERROR);
+                                m_Verbosity, helper::LogMode::FATALERROR);
                 }
             }
         }
@@ -252,7 +252,7 @@ void SscReader::PerformGets()
                         {
                             helper::Log("Engine", "SSCReader", "PerformGets",
                                         "unknown data type", 0, m_Comm.Rank(),
-                                        0, m_Verbosity, helper::LogMode::ERROR);
+                                        0, m_Verbosity, helper::LogMode::FATALERROR);
                         }
                     }
                 }
diff --git a/source/adios2/engine/ssc/SscReader.tcc b/source/adios2/engine/ssc/SscReader.tcc
index 6271b69b9..9dd70e191 100644
--- a/source/adios2/engine/ssc/SscReader.tcc
+++ b/source/adios2/engine/ssc/SscReader.tcc
@@ -154,7 +154,7 @@ void SscReader::GetDeferredCommon(Variable<T> &variable, T *data)
                     {
                         helper::Log("Engine", "SSCReader", "GetDeferredCommon",
                                     "unknown ShapeID", 0, m_Comm.Rank(), 0,
-                                    m_Verbosity, helper::LogMode::ERROR);
+                                    m_Verbosity, helper::LogMode::FATALERROR);
                     }
                 }
             }
diff --git a/source/adios2/helper/adiosCommDummy.cpp b/source/adios2/helper/adiosCommDummy.cpp
index 9c0234952..2f09b97ce 100644
--- a/source/adios2/helper/adiosCommDummy.cpp
+++ b/source/adios2/helper/adiosCommDummy.cpp
@@ -25,7 +25,7 @@ void CommDummyError(const std::string &msg)
     helper::Log("Helper", "adiosCommDummy", "CommDummyError",
                 "CommDummy: a function returned error code '" + msg +
                     "'. Aborting!",
-                helper::LogMode::ERROR);
+                helper::LogMode::FATALERROR);
     std::abort();
 }
 }
diff --git a/source/adios2/helper/adiosLog.cpp b/source/adios2/helper/adiosLog.cpp
index f080797b3..f13769b84 100644
--- a/source/adios2/helper/adiosLog.cpp
+++ b/source/adios2/helper/adiosLog.cpp
@@ -54,7 +54,7 @@ std::string MakeMessage(const std::string &component, const std::string &source,
     {
         m << warningColor << " [ADIOS2 WARNING]" << defaultColor;
     }
-    else if (mode == ERROR)
+    else if (mode == FATALERROR)
     {
         m << errorColor << " [ADIOS2 ERROR]" << defaultColor;
     }
@@ -102,7 +102,7 @@ void Log(const std::string &component, const std::string &source,
     if ((logRank >= 0 && commRank >= 0 && logRank != commRank) ||
         priority > verbosity ||
         (messages.find(message) != messages.end() &&
-         (mode == LogMode::ERROR || mode == LogMode::WARNING)))
+         (mode == LogMode::FATALERROR || mode == LogMode::WARNING)))
     {
         return;
     }
@@ -115,7 +115,7 @@ void Log(const std::string &component, const std::string &source,
     {
         std::cout << m;
     }
-    else if (mode == ERROR)
+    else if (mode == FATALERROR)
     {
         std::cerr << m;
     }
diff --git a/source/adios2/helper/adiosLog.h b/source/adios2/helper/adiosLog.h
index 570ffd68a..74dc73da9 100644
--- a/source/adios2/helper/adiosLog.h
+++ b/source/adios2/helper/adiosLog.h
@@ -21,7 +21,7 @@ namespace helper
 enum LogMode : char
 {
     EXCEPTION = 'x',
-    ERROR = 'e',
+    FATALERROR = 'e',
     WARNING = 'w',
     INFO = 'i'
 };
@@ -36,7 +36,7 @@ std::string MakeMessage(const std::string &component, const std::string &source,
  * @param source: class name of component
  * @param activity: function name where this is called
  * @param message: text message
- * @param mode: INFO, WARNING or ERROR
+ * @param mode: INFO, WARNING or FATALERROR
  */
 void Log(const std::string &component, const std::string &source,
          const std::string &activity, const std::string &message,
@@ -50,7 +50,7 @@ void Log(const std::string &component, const std::string &source,
  * @param message: text message
  * @param priority: only print if(priority<=verbosity)
  * @param verbosity: engine parameter for engine wide verbosity level
- * @param mode: INFO, WARNING or ERROR
+ * @param mode: INFO, WARNING or FATALERROR
  */
 void Log(const std::string &component, const std::string &source,
          const std::string &activity, const std::string &message,
@@ -66,7 +66,7 @@ void Log(const std::string &component, const std::string &source,
  * @param commRank: current MPI rank
  * @param priority: only print if(priority<=verbosity)
  * @param verbosity: engine parameter for engine wide verbosity level
- * @param mode: INFO, WARNING or ERROR
+ * @param mode: INFO, WARNING or FATALERROR
  */
 void Log(const std::string &component, const std::string &source,
          const std::string &activity, const std::string &message,
diff --git a/source/adios2/helper/adiosSystem.cpp b/source/adios2/helper/adiosSystem.cpp
index 7ff903b84..92e6f040b 100644
--- a/source/adios2/helper/adiosSystem.cpp
+++ b/source/adios2/helper/adiosSystem.cpp
@@ -97,25 +97,25 @@ int ExceptionToError(const std::string &function)
     catch (std::invalid_argument &e)
     {
         helper::Log("Helper", "adiosSystem", "ExceptionToError",
-                    function + ": " + e.what(), helper::LogMode::ERROR);
+                    function + ": " + e.what(), helper::LogMode::FATALERROR);
         return 1;
     }
     catch (std::system_error &e)
     {
         helper::Log("Helper", "adiosSystem", "ExceptionToError",
-                    function + ": " + e.what(), helper::LogMode::ERROR);
+                    function + ": " + e.what(), helper::LogMode::FATALERROR);
         return 2;
     }
     catch (std::runtime_error &e)
     {
         helper::Log("Helper", "adiosSystem", "ExceptionToError",
-                    function + ": " + e.what(), helper::LogMode::ERROR);
+                    function + ": " + e.what(), helper::LogMode::FATALERROR);
         return 3;
     }
     catch (std::exception &e)
     {
         helper::Log("Helper", "adiosSystem", "ExceptionToError",
-                    function + ": " + e.what(), helper::LogMode::ERROR);
+                    function + ": " + e.what(), helper::LogMode::FATALERROR);
         return 4;
     }
 }
diff --git a/source/adios2/operator/compress/CompressBlosc.cpp b/source/adios2/operator/compress/CompressBlosc.cpp
index 8233fc3f3..e414b64ac 100644
--- a/source/adios2/operator/compress/CompressBlosc.cpp
+++ b/source/adios2/operator/compress/CompressBlosc.cpp
@@ -141,7 +141,7 @@ size_t CompressBlosc::Operate(const char *dataIn, const Dims &blockStart,
                         "ADIOS WARNING: Unknown parameter keyword '" + key +
                             "' with value '" + value +
                             "' passed to Blosc compression operator.",
-                        helper::ERROR);
+                        helper::FATALERROR);
         }
     }
 
diff --git a/source/adios2/toolkit/burstbuffer/FileDrainerSingleThread.cpp b/source/adios2/toolkit/burstbuffer/FileDrainerSingleThread.cpp
index a0452d64c..8e4f6c5a1 100644
--- a/source/adios2/toolkit/burstbuffer/FileDrainerSingleThread.cpp
+++ b/source/adios2/toolkit/burstbuffer/FileDrainerSingleThread.cpp
@@ -220,7 +220,7 @@ void FileDrainerSingleThread::DrainThread()
                     helper::Log("BurstBuffer", "FileDrainerSingleThread",
                                 "DrainThread",
                                 "ADIOS THREAD ERROR: " + std::string(e.what()),
-                                helper::LogMode::ERROR);
+                                helper::LogMode::FATALERROR);
                 }
             }
             break;
diff --git a/source/adios2/toolkit/query/JsonWorker.cpp b/source/adios2/toolkit/query/JsonWorker.cpp
index 4b842267f..8b0fc3d1a 100644
--- a/source/adios2/toolkit/query/JsonWorker.cpp
+++ b/source/adios2/toolkit/query/JsonWorker.cpp
@@ -105,7 +105,7 @@ void JsonWorker::ParseJson()
         {
             helper::Log("Query", "JsonWorker", "ParseJson",
                         "No such variable: " + varName.dump(),
-                        helper::LogMode::ERROR);
+                        helper::LogMode::FATALERROR);
             return nullptr;
         }
 
diff --git a/source/adios2/toolkit/query/Query.cpp b/source/adios2/toolkit/query/Query.cpp
index 0ed551a03..54d48c7a6 100644
--- a/source/adios2/toolkit/query/Query.cpp
+++ b/source/adios2/toolkit/query/Query.cpp
@@ -244,7 +244,7 @@ bool QueryVar::IsSelectionValid(adios2::Dims &shape) const
         helper::Log(
             "Query", "QueryVar", "IsSelectionValid",
             "Query selection dimension is different from shape dimension",
-            helper::LogMode::ERROR);
+            helper::LogMode::FATALERROR);
         return false; // different dimension
     }
 
diff --git a/source/adios2/toolkit/query/Worker.cpp b/source/adios2/toolkit/query/Worker.cpp
index 8541cbadb..be6787c8d 100644
--- a/source/adios2/toolkit/query/Worker.cpp
+++ b/source/adios2/toolkit/query/Worker.cpp
@@ -73,7 +73,7 @@ QueryVar *Worker::GetBasicVarQuery(adios2::core::IO &currentIO,
     {
         helper::Log("Query", "Worker", "GetBasicVarQuery",
                     "No such variable: " + variableName,
-                    helper::LogMode::ERROR);
+                    helper::LogMode::FATALERROR);
         return nullptr;
     }
 #define declare_type(T)                                                        \
diff --git a/source/adios2/toolkit/query/XmlWorker.cpp b/source/adios2/toolkit/query/XmlWorker.cpp
index 40e7c855d..d49c8b6a0 100644
--- a/source/adios2/toolkit/query/XmlWorker.cpp
+++ b/source/adios2/toolkit/query/XmlWorker.cpp
@@ -150,7 +150,7 @@ QueryVar *XmlWorker::ParseVarNode(const pugi::xml_node &node,
     {
         helper::Log("Query", "XmlWorker", "ParseVarNode",
                     "No such variable: " + variableName,
-                    helper::LogMode::ERROR);
+                    helper::LogMode::FATALERROR);
         helper::Throw<std::ios_base::failure>(
             "Toolkit", "query::XmlWorker", "ParseVarNode",
             "variable: " + variableName + " not found");
diff --git a/source/utils/adios_reorganize/Reorganize.cpp b/source/utils/adios_reorganize/Reorganize.cpp
index 80db2a04f..d8bb1348c 100644
--- a/source/utils/adios_reorganize/Reorganize.cpp
+++ b/source/utils/adios_reorganize/Reorganize.cpp
@@ -597,7 +597,7 @@ int Reorganize::ProcessMetadata(core::Engine &rStream, core::IO &io,
                     "write buffer size needs to hold about " +
                         std::to_string(bufsize) + " bytes but max is set to " +
                         std::to_string(max_write_buffer_size),
-                    m_Rank, m_Rank, 0, 0, helper::LogMode::ERROR);
+                    m_Rank, m_Rank, 0, 0, helper::LogMode::FATALERROR);
         return 1;
     }
 
@@ -608,7 +608,7 @@ int Reorganize::ProcessMetadata(core::Engine &rStream, core::IO &io,
                         std::to_string(largest_block) +
                         " bytes but max is set to " +
                         std::to_string(max_read_buffer_size),
-                    m_Rank, m_Rank, 0, 0, helper::LogMode::ERROR);
+                    m_Rank, m_Rank, 0, 0, helper::LogMode::FATALERROR);
         return 1;
     }
     return retval;
@@ -628,7 +628,7 @@ int Reorganize::ReadWrite(core::Engine &rStream, core::Engine &wStream,
                 std::to_string(nvars) +
                 ") to read does not match the number of processed variables (" +
                 std::to_string(varinfo.size()) + ")",
-            m_Rank, m_Rank, 0, 0, helper::LogMode::ERROR);
+            m_Rank, m_Rank, 0, 0, helper::LogMode::FATALERROR);
     }
 
     /*
