--- a/source/h5vol/H5VolUtil.h
+++ a/source/h5vol/H5VolUtil.h
@@ -37,9 +27,9 @@
 // h5 uses hsize_t for dimensions (unsigned long long)
 // adios uses size_t
 //
-void gUtilConvert(hsize_t *fromH5, size_t *to, uint ndims);
+void gUtilConvert(hsize_t *fromH5, size_t *to, hsize_t ndims);
 
-int gUtilADIOS2GetShape(hid_t space_id, size_t *shape, uint ndims);
+int gUtilADIOS2GetShape(hid_t space_id, size_t *shape, hsize_t ndims);
 
 int gUtilADIOS2GetBlockInfo(hid_t hyperSlab_id, size_t *start, size_t *count,
                             hsize_t ndims);
--- a/source/h5vol/H5VolUtil.c
+++ a/source/h5vol/H5VolUtil.c
@@ -56,16 +56,16 @@
     return p;
 }
 
-void gUtilConvert(hsize_t *fromH5, size_t *to, uint ndims)
+void gUtilConvert(hsize_t *fromH5, size_t *to, hsize_t ndims)
 {
-    uint i = 0;
+    hsize_t i = 0;
     for (i = 0; i < ndims; i++)
     {
         to[i] = fromH5[i];
     }
 }
 
-int gUtilADIOS2GetShape(hid_t space_id, size_t *shape, uint ndims)
+int gUtilADIOS2GetShape(hid_t space_id, size_t *shape, hsize_t ndims)
 {
     if (gUtilADIOS2IsScalar(space_id))
     {
