diff --git a/src/runtime_src/tools/xclbinutil/CMakeLists.txt b/src/runtime_src/tools/xclbinutil/CMakeLists.txt
index 4c8a89232..58a211012 100644
--- a/src/runtime_src/tools/xclbinutil/CMakeLists.txt
+++ b/src/runtime_src/tools/xclbinutil/CMakeLists.txt
@@ -79,9 +79,14 @@ set(XCLBINUTIL_SRCS ${XCLBINUTIL_MAIN_FILE} ${XCLBINUTIL_FILES_SRCS})
 
 add_executable(${XCLBINUTIL_NAME} ${XCLBINUTIL_SRCS})
 
+message(STATUS "RapidJSON include dir:" ${RapidJSON_INCLUDE_DIRS})
+target_include_directories(${XCLBINUTIL_NAME} PRIVATE ${RapidJSON_INCLUDE_DIRS}/..)
+
 # Signing xclbin images currently is not support on windows
 if(NOT WIN32)
   target_link_libraries(${XCLBINUTIL_NAME} PRIVATE crypto)
+else()
+  target_link_libraries(${XCLBINUTIL_NAME} PRIVATE wsock32 ws2_32)
 endif()
 
 # Add compile definitions
