diff --git a/build/targets/all.cmake b/build/targets/all.cmake
index 4b4473824..c724b9da2 100644
--- a/build/targets/all.cmake
+++ b/build/targets/all.cmake
@@ -9,7 +9,7 @@ set ( INCLUDE_OSC         ON  CACHE STRING  "Include Faust OSC static library" F
 set ( INCLUDE_HTTP        ON  CACHE STRING  "Include Faust HTTPD static library" FORCE )
 
 set ( OSCDYNAMIC          ON  CACHE STRING  "Include Faust OSC dynamic library" FORCE )
-set ( HTTPDYNAMIC         ON  CACHE STRING  "Include Faust HTTP dynamic library" FORCE )
+set ( HTTPDYNAMIC         OFF  CACHE STRING  "Include Faust HTTP dynamic library" FORCE )
 
-set ( INCLUDE_ITP         ON  CACHE STRING  "Include Faust Machine library" FORCE )
-set ( ITPDYNAMIC          ON  CACHE STRING  "Include Faust Machine library" FORCE )
+set ( INCLUDE_ITP         OFF  CACHE STRING  "Include Faust Machine library" FORCE )
+set ( ITPDYNAMIC          OFF  CACHE STRING  "Include Faust Machine library" FORCE )
