diff --git a/compiler/errors/exception.hh b/compiler/errors/exception.hh
index 2a295092e..b7aa18574 100644
--- a/compiler/errors/exception.hh
+++ b/compiler/errors/exception.hh
@@ -29,7 +29,7 @@ Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
 //#include <io.h>
 #endif
 
-#if !defined(EMCC) && !defined(WIN32) && !defined(ANDROID)
+#if !defined(EMCC) && !defined(WIN32) && !defined(ANDROID) && !defined(ALPINE)
 #include <execinfo.h>
 #endif
 
