diff --git a/LLVMExports-release.cmake b/LLVMExports-release.cmake 
index 726572400..e3a4db76e 100644
--- a/LLVMExports-release.cmake
+++ b/LLVMExports-release.cmake
@@ -552,7 +552,6 @@ set_target_properties(LLVMIntelJITEvents PROPERTIES
   )

 list(APPEND _IMPORT_CHECK_TARGETS LLVMIntelJITEvents )
-list(APPEND _IMPORT_CHECK_FILES_FOR_LLVMIntelJITEvents "${_IMPORT_PREFIX}/lib/libLLVMIntelJITEvents.a" )

 # Import target "LLVMTarget" for configuration "Release"
 set_property(TARGET LLVMTarget APPEND PROPERTY IMPORTED_CONFIGURATIONS RELEASE)
