diff --git a/LLVMExports-release.cmake b/LLVMExports-release.cmake
index cbcda57a..9e7ec7f0 100644
--- a/LLVMExports-release.cmake
+++ b/LLVMExports-release.cmake
@@ -592,7 +592,6 @@ set_target_properties(LLVMIntelJITEvents PROPERTIES
   )
 
 list(APPEND _IMPORT_CHECK_TARGETS LLVMIntelJITEvents )
-list(APPEND _IMPORT_CHECK_FILES_FOR_LLVMIntelJITEvents "${_IMPORT_PREFIX}/lib/libLLVMIntelJITEvents.a" )
 
 # Import target "LLVMPerfJITEvents" for configuration "Release"
 set_property(TARGET LLVMPerfJITEvents APPEND PROPERTY IMPORTED_CONFIGURATIONS RELEASE)
@@ -1053,7 +1052,6 @@ set_target_properties(mlir-tblgen PROPERTIES
   )
 
 list(APPEND _IMPORT_CHECK_TARGETS mlir-tblgen )
-list(APPEND _IMPORT_CHECK_FILES_FOR_mlir-tblgen "${_IMPORT_PREFIX}/tools/mlir-tblgen" )
 
 # Import target "mlir-linalg-ods-gen" for configuration "Release"
 set_property(TARGET mlir-linalg-ods-gen APPEND PROPERTY IMPORTED_CONFIGURATIONS RELEASE)
@@ -1062,7 +1060,6 @@ set_target_properties(mlir-linalg-ods-gen PROPERTIES
   )
 
 list(APPEND _IMPORT_CHECK_TARGETS mlir-linalg-ods-gen )
-list(APPEND _IMPORT_CHECK_FILES_FOR_mlir-linalg-ods-gen "${_IMPORT_PREFIX}/tools/mlir-linalg-ods-gen" )
 
 # Import target "MLIRSupportIndentedOstream" for configuration "Release"
 set_property(TARGET MLIRSupportIndentedOstream APPEND PROPERTY IMPORTED_CONFIGURATIONS RELEASE)
@@ -1103,7 +1100,6 @@ set_target_properties(mlir-cpu-runner PROPERTIES
   )
 
 list(APPEND _IMPORT_CHECK_TARGETS mlir-cpu-runner )
-list(APPEND _IMPORT_CHECK_FILES_FOR_mlir-cpu-runner "${_IMPORT_PREFIX}/tools/mlir-cpu-runner" )
 
 # Import target "mlir-opt" for configuration "Release"
 set_property(TARGET mlir-opt APPEND PROPERTY IMPORTED_CONFIGURATIONS RELEASE)
@@ -1112,7 +1108,6 @@ set_target_properties(mlir-opt PROPERTIES
   )
 
 list(APPEND _IMPORT_CHECK_TARGETS mlir-opt )
-list(APPEND _IMPORT_CHECK_FILES_FOR_mlir-opt "${_IMPORT_PREFIX}/tools/mlir-opt" )
 
 # Import target "mlir-reduce" for configuration "Release"
 set_property(TARGET mlir-reduce APPEND PROPERTY IMPORTED_CONFIGURATIONS RELEASE)
@@ -1121,7 +1116,6 @@ set_target_properties(mlir-reduce PROPERTIES
   )
 
 list(APPEND _IMPORT_CHECK_TARGETS mlir-reduce )
-list(APPEND _IMPORT_CHECK_FILES_FOR_mlir-reduce "${_IMPORT_PREFIX}/tools/mlir-reduce" )
 
 # Import target "mlir-translate" for configuration "Release"
 set_property(TARGET mlir-translate APPEND PROPERTY IMPORTED_CONFIGURATIONS RELEASE)
@@ -1130,7 +1124,6 @@ set_target_properties(mlir-translate PROPERTIES
   )
 
 list(APPEND _IMPORT_CHECK_TARGETS mlir-translate )
-list(APPEND _IMPORT_CHECK_FILES_FOR_mlir-translate "${_IMPORT_PREFIX}/tools/mlir-translate" )
 
 # Import target "bugpoint" for configuration "Release"
 set_property(TARGET bugpoint APPEND PROPERTY IMPORTED_CONFIGURATIONS RELEASE)
