diff --git a/build/misc/llvm.cmake b/build/misc/llvm.cmake
index 97b409d46..88bd7630f 100644
--- a/build/misc/llvm.cmake
+++ b/build/misc/llvm.cmake
@@ -104,7 +104,6 @@ macro (llvm_cmake)
 		message(STATUS "Found LLVM ${LLVM_PACKAGE_VERSION}")
 		message(STATUS "Using LLVMConfig.cmake in: ${LLVM_DIR}")
 		# Find the libraries that correspond to the LLVM components that we wish to use
-		llvm_map_components_to_libnames(LLVM_LIBS all)
 #		list(REMOVE_ITEM LLVM_LIBS LTO)
 	else()
 		llvm_config()
