diff --git a/build/http/CMakeLists.txt b/build/http/CMakeLists.txt
index 4862c8252..ac94ef74f 100644
--- a/build/http/CMakeLists.txt
+++ b/build/http/CMakeLists.txt
@@ -165,7 +165,7 @@ if (HTTPDYNAMIC)
        elseif(MSYS)
                target_link_libraries (httpdynamic ${LIBMICROHTTPD__LDFLAGS})
        else()
-               target_link_libraries (httpdynamic microhttpd)
+               target_link_libraries (httpdynamic ws2_32 microhttpd)
                set_target_properties (httpdynamic PROPERTIES
                        COMPILE_FLAGS  "${LIBMICROHTTPD__CFLAGS}"
                        LINK_FLAGS     "${LIBMICROHTTPD__LDFLAGS}")
