diff --git a/coders/Makefile.am b/coders/Makefile.am
index d2b7b4828..b78748524 100644
--- a/coders/Makefile.am
+++ b/coders/Makefile.am
@@ -1390,7 +1390,7 @@ coders_url_la_SOURCES      = coders/url.c
 coders_url_la_CPPFLAGS     = $(MAGICK_CODER_CPPFLAGS)
 coders_url_la_LDFLAGS      = $(MODULECOMMONFLAGS)
 coders_url_la_LIBADD       = $(MAGICKCORE_LIBS) $(XML_LIBS) $(LZMA_LIBS) $(ZLIB_LIBS) $(MATH_LIBS)
-if WIN32_NATIVE_BUILD
+ifeq ($(MAGICK_TARGET_OS),Windows_NT)
 coders_url_la_LIBADD += -lurlmon
 endif
 
