--- a/src/hdf5_drv/H5FDsilo.c
+++ b/src/hdf5_drv/H5FDsilo.c
@@ -191,7 +191,6 @@
 #endif /* HDassert */
 #ifndef HDopen
     #ifdef _WIN32
-        typedef int mode_t;
         #define HDopen(S,F,M)           my_sopen_s(S,F,M)
     #else
         #ifdef _O_BINARY
--- a/src/silo/silo_win32_compatibility.h
+++ b/src/silo/silo_win32_compatibility.h
@@ -3,8 +3,8 @@
 #define SILO_WIN32_COMPATIBILITY
 #include <io.h>  /* Include Windows IO */
 #include <errno.h>
-#include <sys\types.h>
-#include <sys\stat.h>
+#include <sys/types.h>
+#include <sys/stat.h>
 
 
 #define access    _access
