From 0bb184678bfa92ef77e9f1365ec3adc33cfdc3bf Mon Sep 17 00:00:00 2001
From: KristofferC <kcarlsson89@gmail.com>
Date: Mon, 13 Dec 2021 18:02:20 +0100
Subject: [PATCH] also look for shared libraries on Windows

---
 CMakeLists.txt | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index b394f7ccc..e974d8322 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -59,6 +59,11 @@ endif()
 set(CMAKE_C_STANDARD 11)
 set(CMAKE_C_VISIBILITY_PRESET hidden)
 
+if(WIN32)
+  # Ensure on Windows we also look for shared libraries `*.dll`, instead of just import libraries
+  set(CMAKE_FIND_LIBRARY_SUFFIXES ".$ENV{dlext}" ${CMAKE_FIND_LIBRARY_SUFFIXES})
+endif()
+
 message(STATUS "Requested multimedia engine: ${USE_MULTIMEDIA}")
 message(STATUS "Requested build mode: ${CMAKE_BUILD_TYPE}")
 
-- 
2.30.2

