diff --git a/CMakeLists.txt b/CMakeLists.txt
index d9b9f0f79..240bf1ee9 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -592,25 +592,27 @@
 target_compile_options(notcurses-static
   PRIVATE
     -fPIE
 )
 
+if(${USE_CXX})
 target_compile_options(notcurses++-static
   PRIVATE
   ${NCPP_COMPILE_OPTIONS}
   -fPIE
 )
 
 target_compile_definitions(notcurses++
   PUBLIC
     ${NCPP_COMPILE_DEFINITIONS_PUBLIC}
 )
 
 target_compile_definitions(notcurses++-static
   PUBLIC
     ${NCPP_COMPILE_DEFINITIONS_PUBLIC}
 )
+endif()
 
 file(GLOB NCPP_HEADERS
   CONFIGURE_DEPENDS
   LIST_DIRECTORIES false
   ${PROJECT_SOURCE_DIR}/include/ncpp/*.hh)
