diff --git a/src/component_types/base_packed_component.h b/src/component_types/base_packed_component.h
index bbf39b0..551a85a 100644
--- a/src/component_types/base_packed_component.h
+++ b/src/component_types/base_packed_component.h
@@ -119,7 +119,7 @@ public:
   static unsigned log2(unsigned v){
          // taken from
          // http://graphics.stanford.edu/~seander/bithacks.html#IntegerLogLookup
-         static const char LogTable256[256] =
+         static const signed char LogTable256[256] =
          {
          #define LT(n) n, n, n, n, n, n, n, n, n, n, n, n, n, n, n, n
              -1, 0, 1, 1, 2, 2, 2, 2, 3, 3, 3, 3, 3, 3, 3, 3,
