diff --git a/src/esmumps/CMakeLists.txt b/src/esmumps/CMakeLists.txt
index 62114b5..b4de67b 100644
--- a/src/esmumps/CMakeLists.txt
+++ b/src/esmumps/CMakeLists.txt
@@ -136,5 +136,17 @@ install(TARGETS esmumps
   PUBLIC_HEADER DESTINATION include)
 
+# ptesmumps.h
+add_custom_command(OUTPUT ${GENERATED_INCLUDE_DIR}/ptesmumps.h
+  COMMAND ${CMAKE_COMMAND} -E copy ${CMAKE_CURRENT_SOURCE_DIR}/library.h ${GENERATED_INCLUDE_DIR}/ptesmumps.h
+  DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/library.h
+  COMMENT "Copy ptesmumps.h to include directory")
+
+add_custom_target(ptesmumps_h
+  DEPENDS "${GENERATED_INCLUDE_DIR}/ptesmumps.h")
+
+set(ptesmumps_headers ${esmumps_headers})
+list(REMOVE_ITEM ptesmumps_headers "${GENERATED_INCLUDE_DIR}/esmumps.h")
+list(APPEND ptesmumps_headers "${GENERATED_INCLUDE_DIR}/ptesmumps.h")
 # Dummy PTEsMUMPS library file
-add_library(ptesmumps ${esmumps_src} ${esmumps_headers})
+add_library(ptesmumps ${esmumps_src} ${ptesmumps_headers})
 
@@ -145,4 +145,4 @@ install(TARGETS esmumps
-set_target_properties(ptesmumps PROPERTIES PUBLIC_HEADER "${GENERATED_INCLUDE_DIR}/esmumps.h")
+set_target_properties(ptesmumps PROPERTIES PUBLIC_HEADER "${GENERATED_INCLUDE_DIR}/ptesmumps.h")
 
 target_include_directories(ptesmumps PRIVATE
   $<BUILD_INTERFACE:${LIBSCOTCH_INCLUDE_DIR}>
