diff --git a/src/libscotch/common_thread.h b/src/libscotch/common_thread.h
index 235d920..ed47fe1 100644
--- a/src/libscotch/common_thread.h
+++ b/src/libscotch/common_thread.h
@@ -78,11 +78,11 @@ typedef struct ThreadContext_ {
   volatile unsigned int         bainnum;          /*+ Number of barrier instance          +*/
   pthread_mutex_t               lockdat;          /*+ Lock for updating status            +*/
   pthread_cond_t                conddat;          /*+ Wakeup condition for slave threads  +*/
-  union {                                         /*+ Context save area for main thread   +*/
 #ifdef COMMON_PTHREAD_AFFINITY_LINUX
+  union {                                         /*+ Context save area for main thread   +*/
     cpu_set_t                   cpusdat;          /*+ Original thread mask of main thread +*/
-#endif /* COMMON_PTHREAD_AFFINITY_LINUX */
   }                             savedat;
+#endif /* COMMON_PTHREAD_AFFINITY_LINUX */
 #endif /* COMMON_PTHREAD */
 } ThreadContext;
 
