diff --git a/src/check/CMakeLists.txt b/src/check/CMakeLists.txt
index 3f67aac..d45f685 100644
--- a/src/check/CMakeLists.txt
+++ b/src/check/CMakeLists.txt
@@ -130,7 +130,7 @@ if(BUILD_LIBSCOTCHMETIS)
     target_include_directories(test_libmetis_dualv${version} PRIVATE ${GENERATED_INCLUDE_DIR})
     target_link_libraries(test_libmetis_dualv${version} scotchmetisv${version} scotch scotcherr)
     add_test(NAME test_libmetis_dualv${version} COMMAND test_libmetis_dualv${version})
-    if(${CMAKE_Fortran_COMPILER_SUPPORTS_F90}) # Safeguard to avoid x77 compilers
+    if(${CMAKE_Fortran_COMPILER_SUPPORTS_F90} AND NOT WIN32) # Safeguard to avoid x77 compilers
       add_executable(test_libmetis_dual_f_v${version} test_libmetis_dual_f.f90)
       target_compile_definitions(test_libmetis_dual_f_v${version} PRIVATE SCOTCH_METIS_VERSION=${version})
       target_include_directories(test_libmetis_dual_f_v${version} PRIVATE ${GENERATED_INCLUDE_DIR})
