diff --git a/src/scotch/dgmap.c b/src/scotch/dgmap.c
index 572d936..21a20e8 100644
--- a/src/scotch/dgmap.c
+++ b/src/scotch/dgmap.c
@@ -56,7 +56,9 @@
 
 #define SCOTCH_PTSCOTCH
 
+#if ((defined COMMON_TIMING_OLD) || (defined HAVE_SYS_TIME_H))
 #include <sys/time.h>
+#endif /* ((defined COMMON_TIMING_OLD) || (defined HAVE_SYS_TIME_H)) */
 
 #include "module.h"
 #include "common.h"
diff --git a/src/scotch/dgord.c b/src/scotch/dgord.c
index 5a8b464..f1232b7 100644
--- a/src/scotch/dgord.c
+++ b/src/scotch/dgord.c
@@ -57,7 +57,9 @@
 
 #define SCOTCH_PTSCOTCH
 
+#if ((defined COMMON_TIMING_OLD) || (defined HAVE_SYS_TIME_H))
 #include <sys/time.h>
+#endif /* ((defined COMMON_TIMING_OLD) || (defined HAVE_SYS_TIME_H)) */
 
 #include "module.h"
 #include "common.h"
diff --git a/src/check/test_common_file_compress.c b/src/check/test_common_file_compress.c
index 1e67f37..0e9f411 100644
--- a/src/check/test_common_file_compress.c
+++ b/src/check/test_common_file_compress.c
@@ -57,7 +57,9 @@
 #include <stdio.h>
 #include <sys/types.h>
 #include <sys/stat.h>
+#if ((! defined COMMON_OS_WINDOWS) && (! defined HAVE_NOT_UNISTD_H))
 #include <unistd.h>
+#endif /* ((! defined COMMON_OS_WINDOWS) && (! defined HAVE_NOT_UNISTD_H)) */
 
 #include "../libscotch/module.h"
 #include "../libscotch/common.h"
diff --git a/src/check/test_common_random.c b/src/check/test_common_random.c
index f03f496..a6ddf15 100644
--- a/src/check/test_common_random.c
+++ b/src/check/test_common_random.c
@@ -59,7 +59,9 @@
 #include <stdio.h>
 #include <sys/types.h>
 #include <sys/stat.h>
+#if ((! defined COMMON_OS_WINDOWS) && (! defined HAVE_NOT_UNISTD_H))
 #include <unistd.h>
+#endif /* ((! defined COMMON_OS_WINDOWS) && (! defined HAVE_NOT_UNISTD_H)) */
 
 #include "../libscotch/module.h"
 #include "../libscotch/common.h"
@@ -141,8 +143,11 @@ char *              argv[])
         SCOTCH_errorPrint ("Could not save random state");
         break;
     }
-
+#ifndef _WIN32
     sleep (2);                                    /* Next run will not get the same time() value */
+#else
+    Sleep (2000);
+#endif
   }
   else {                                          /* Second pass */
     char *              charptr;

diff --git a/src/check/test_fibo.c b/src/check/test_fibo.c
index b301c2d..f6459b8 100644
--- a/src/check/test_fibo.c
+++ b/src/check/test_fibo.c
@@ -59,7 +59,9 @@
 #include <stdio.h>
 #include <sys/types.h>
 #include <sys/stat.h>
+#if ((! defined COMMON_OS_WINDOWS) && (! defined HAVE_NOT_UNISTD_H))
 #include <unistd.h>
+#endif /* ((! defined COMMON_OS_WINDOWS) && (! defined HAVE_NOT_UNISTD_H)) */
 
 #include "../libscotch/module.h"
 #include "../libscotch/common.h"
diff --git a/src/check/test_scotch_dgraph_band.c b/src/check/test_scotch_dgraph_band.c
index 639188e..942b87a 100644
--- a/src/check/test_scotch_dgraph_band.c
+++ b/src/check/test_scotch_dgraph_band.c
@@ -55,10 +55,14 @@
 #include <stdint.h>
 #endif /* (((defined __STDC_VERSION__) && (__STDC_VERSION__ >= 199901L)) || (defined HAVE_STDINT_H)) */
 #include <stdlib.h>
+#if ((defined COMMON_TIMING_OLD) || (defined HAVE_SYS_TIME_H))
 #include <sys/time.h>
+#endif /* ((defined COMMON_TIMING_OLD) || (defined HAVE_SYS_TIME_H)) */
 #include <sys/types.h>
 #include <pthread.h>
+#if ((! defined COMMON_OS_WINDOWS) && (! defined HAVE_NOT_UNISTD_H))
 #include <unistd.h>
+#endif /* ((! defined COMMON_OS_WINDOWS) && (! defined HAVE_NOT_UNISTD_H)) */
 
 #include "ptscotch.h"
 
diff --git a/src/check/test_scotch_dgraph_check.c b/src/check/test_scotch_dgraph_check.c
index dcc7acb..04ea117 100644
--- a/src/check/test_scotch_dgraph_check.c
+++ b/src/check/test_scotch_dgraph_check.c
@@ -55,10 +55,14 @@
 #include <stdint.h>
 #endif /* (((defined __STDC_VERSION__) && (__STDC_VERSION__ >= 199901L)) || (defined HAVE_STDINT_H)) */
 #include <stdlib.h>
+#if ((defined COMMON_TIMING_OLD) || (defined HAVE_SYS_TIME_H))
 #include <sys/time.h>
+#endif /* ((defined COMMON_TIMING_OLD) || (defined HAVE_SYS_TIME_H)) */
 #include <sys/types.h>
 #include <pthread.h>
+#if ((! defined COMMON_OS_WINDOWS) && (! defined HAVE_NOT_UNISTD_H))
 #include <unistd.h>
+#endif /* ((! defined COMMON_OS_WINDOWS) && (! defined HAVE_NOT_UNISTD_H)) */
 
 #include "ptscotch.h"
 
diff --git a/src/check/test_scotch_dgraph_coarsen.c b/src/check/test_scotch_dgraph_coarsen.c
index 85c816e..4fa9aea 100644
--- a/src/check/test_scotch_dgraph_coarsen.c
+++ b/src/check/test_scotch_dgraph_coarsen.c
@@ -55,10 +55,14 @@
 #include <stdint.h>
 #endif /* (((defined __STDC_VERSION__) && (__STDC_VERSION__ >= 199901L)) || (defined HAVE_STDINT_H)) */
 #include <stdlib.h>
+#if ((defined COMMON_TIMING_OLD) || (defined HAVE_SYS_TIME_H))
 #include <sys/time.h>
+#endif /* ((defined COMMON_TIMING_OLD) || (defined HAVE_SYS_TIME_H)) */
 #include <sys/types.h>
 #include <pthread.h>
+#if ((! defined COMMON_OS_WINDOWS) && (! defined HAVE_NOT_UNISTD_H))
 #include <unistd.h>
+#endif /* ((! defined COMMON_OS_WINDOWS) && (! defined HAVE_NOT_UNISTD_H)) */
 
 #include "ptscotch.h"
 
diff --git a/src/check/test_scotch_dgraph_grow.c b/src/check/test_scotch_dgraph_grow.c
index 50d0a9c..e61eb2f 100644
--- a/src/check/test_scotch_dgraph_grow.c
+++ b/src/check/test_scotch_dgraph_grow.c
@@ -58,10 +58,14 @@
 #endif /* (((defined __STDC_VERSION__) && (__STDC_VERSION__ >= 199901L)) || (defined HAVE_STDINT_H)) */
 #include <stdlib.h>
 #include <string.h>
+#if ((defined COMMON_TIMING_OLD) || (defined HAVE_SYS_TIME_H))
 #include <sys/time.h>
+#endif /* ((defined COMMON_TIMING_OLD) || (defined HAVE_SYS_TIME_H)) */
 #include <sys/types.h>
 #include <pthread.h>
+#if ((! defined COMMON_OS_WINDOWS) && (! defined HAVE_NOT_UNISTD_H))
 #include <unistd.h>
+#endif /* ((! defined COMMON_OS_WINDOWS) && (! defined HAVE_NOT_UNISTD_H)) */
 
 #include "scotch.h"
 #include "ptscotch.h"
diff --git a/src/check/test_scotch_dgraph_induce.c b/src/check/test_scotch_dgraph_induce.c
index fc1757a..308fe69 100644
--- a/src/check/test_scotch_dgraph_induce.c
+++ b/src/check/test_scotch_dgraph_induce.c
@@ -57,7 +57,9 @@
 #endif /* (((defined __STDC_VERSION__) && (__STDC_VERSION__ >= 199901L)) || (defined HAVE_STDINT_H)) */
 #include <stdlib.h>
 #include <string.h>
+#if ((! defined COMMON_OS_WINDOWS) && (! defined HAVE_NOT_UNISTD_H))
 #include <unistd.h>
+#endif /* ((! defined COMMON_OS_WINDOWS) && (! defined HAVE_NOT_UNISTD_H)) */
 
 #include "scotch.h"
 #include "ptscotch.h"
diff --git a/src/check/test_scotch_dgraph_redist.c b/src/check/test_scotch_dgraph_redist.c
index 7758f91..b5a5d7c 100644
--- a/src/check/test_scotch_dgraph_redist.c
+++ b/src/check/test_scotch_dgraph_redist.c
@@ -53,10 +53,14 @@
 #include <stdint.h>
 #endif /* (((defined __STDC_VERSION__) && (__STDC_VERSION__ >= 199901L)) || (defined HAVE_STDINT_H)) */
 #include <stdlib.h>
+#if ((defined COMMON_TIMING_OLD) || (defined HAVE_SYS_TIME_H))
 #include <sys/time.h>
+#endif /* ((defined COMMON_TIMING_OLD) || (defined HAVE_SYS_TIME_H)) */
 #include <sys/types.h>
 #include <pthread.h>
+#if ((! defined COMMON_OS_WINDOWS) && (! defined HAVE_NOT_UNISTD_H))
 #include <unistd.h>
+#endif /* ((! defined COMMON_OS_WINDOWS) && (! defined HAVE_NOT_UNISTD_H)) */
 
 #include "ptscotch.h"
 
