diff --git a/lib/cpp/src/thrift/transport/TSocket.cpp b/lib/cpp/src/thrift/transport/TSocket.cpp
index 9991fd6bd..e46daeba8 100644
--- a/lib/cpp/src/thrift/transport/TSocket.cpp
+++ b/lib/cpp/src/thrift/transport/TSocket.cpp
@@ -432,6 +432,10 @@ void TSocket::local_open() {
   TWinsockSingleton::create();
 #endif // _WIN32

+#if __FreeBSD__
+  #define EAI_NODATA   EAI_NONAME
+#endif // __FreeBSD__
+
   if (isOpen()) {
     return;
   }
