diff --git a/arch/posix/ua_architecture.h b/arch/posix/ua_architecture.h
index c42768548..8e6159818 100644
--- a/arch/posix/ua_architecture.h
+++ b/arch/posix/ua_architecture.h
@@ -14,6 +14,7 @@

 #include <errno.h>
 #include <arpa/inet.h>
+#include <netinet/in.h>
 #include <netinet/tcp.h>
 #include <netdb.h>
 #include <sys/ioctl.h>
diff --git a/include/open62541/config.h.in b/include/open62541/config.h.in
index 5b20497ed..7758f2ab3 100644
--- a/include/open62541/config.h.in
+++ b/include/open62541/config.h.in
@@ -127,7 +127,7 @@
  * header. */
 #ifdef UA_ARCHITECTURE_POSIX
 # if !defined(_XOPEN_SOURCE)
-#  define _XOPEN_SOURCE 600
+// #  define _XOPEN_SOURCE 600
 # endif
 # ifndef _DEFAULT_SOURCE
 #  define _DEFAULT_SOURCE
