diff --git a/configure b/configure
index 463a500..100da1f 100755
--- a/configure
+++ b/configure
@@ -35178,6 +35178,8 @@ main ()
 
         locale_t t;
         strtod_l(NULL, NULL, t);
+	     strtol_l(NULL, NULL, 0, t);
+        strtoul_l(NULL, NULL, 0, t);
 
   ;
   return 0;
diff --git a/configure.in b/configure.in
index c8b4eae..944f7cb 100644
--- a/configure.in
+++ b/configure.in
@@ -6094,6 +6094,8 @@ if test "$wxUSE_XLOCALE" = "yes" ; then
         [
         locale_t t;
         strtod_l(NULL, NULL, t);
+        strtol_l(NULL, NULL, 0, t);
+        strtoul_l(NULL, NULL, 0, t);
         ],
         wx_cv_type_locale_t=yes,
         wx_cv_type_locale_t=no