`diff --git a/CMakeLists.txt b/CMakeLists.txt
index ae32db39..43b3a75a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -83,8 +83,7 @@ if( CMAKE_SOURCE_DIR STREQUAL CMAKE_CURRENT_SOURCE_DIR )
       "--bindir"
       "--libdir"
       "--includedir"
-      "--prefix"
-      "--src-root")
+      "--prefix")
     execute_process(
       COMMAND ${CONFIG_COMMAND}
       RESULT_VARIABLE HAD_ERROR
diff --git a/runtime/libpgmath/cmake/modules/HandleOutOfTreeLLVM.cmake b/runtime/libpgmath/cmake/modules/HandleOutOfTreeLLVM.cmake
index 83948b14..3e16e263 100644
--- a/runtime/libpgmath/cmake/modules/HandleOutOfTreeLLVM.cmake
+++ b/runtime/libpgmath/cmake/modules/HandleOutOfTreeLLVM.cmake
@@ -14,8 +14,7 @@ macro(find_llvm_parts)
     set(LIBCXX_USING_INSTALLED_LLVM 1)
     set(CONFIG_COMMAND ${LLVM_CONFIG_PATH}
       "--includedir"
-      "--prefix"
-      "--src-root")
+      "--prefix")
     execute_process(
       COMMAND ${CONFIG_COMMAND}
       RESULT_VARIABLE HAD_ERROR
