commit 2d70a9559ec55f97d0b9c153a1c00ef8ac803ae9 (HEAD -> master)
Author: Keno Fischer <keno@juliacomputing.com>
Date:   Sun Aug 14 17:54:01 2022 -0400

    Correct flang2 install dir

diff --git a/tools/flang2/flang2exe/CMakeLists.txt b/tools/flang2/flang2exe/CMakeLists.txt
index a9649e99..5aeeb048 100644
--- a/tools/flang2/flang2exe/CMakeLists.txt
+++ b/tools/flang2/flang2exe/CMakeLists.txt
@@ -160,7 +160,7 @@ target_include_directories(flang2
 
 # Install flang2 executable
 install(TARGETS flang2
-        RUNTIME DESTINATION ${LLVM_TOOLS_BINARY_DIR})
+        RUNTIME DESTINATION bin)
 
 # Local Variables:
 # mode: cmake

