diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9b0b3c61..e85bb70a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -24,6 +24,9 @@ include(toolchain-options)
 set(THREADS_PREFER_PTHREAD_FLAG ON)
 find_package(Threads REQUIRED)
 
+set(CMAKE_SKIP_BUILD_RPATH TRUE)
+set(CMAKE_BUILD_RPATH_USE_ORIGIN TRUE)
+set(CMAKE_INSTALL_RPATH "$ORIGIN/../../lib:$ORIGIN/../../hip/lib")
 
 # ########################################################################
 # Main
