diff --git a/CMakeLists.txt b/CMakeLists.txt
index 403bbf6..72bfcc0 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -106,7 +106,8 @@ elseif (APPLE)
 
     # Enable dispatch queue if the libraries are avaliable
     if (NETWORK_LIB AND SECURITY_LIB)
-        list(APPEND PLATFORM_LIBS "-framework Security -framework Network")
+        list(APPEND PLATFORM_LIBS "-framework" "Security")
+        list(APPEND PLATFORM_LIBS "-framework" "Network")
         list(APPEND EVENT_LOOP_DEFINES "DISPATCH_QUEUE")
     endif ()
 
