diff --git a/source/windows/secure_channel_tls_handler.c b/source/windows/secure_channel_tls_handler.c
index 9ede8de..6158382 100644
--- a/source/windows/secure_channel_tls_handler.c
+++ b/source/windows/secure_channel_tls_handler.c
@@ -23,6 +23,21 @@
 #include <schannel.h>
 #include <security.h>
 
+// https://learn.microsoft.com/en-us/windows/win32/api/schannel/ns-schannel-schannel_cred
+#define SP_PROT_TLS1_0_CLIENT SP_PROT_TLS1_CLIENT
+#define SP_PROT_TLS1_1_CLIENT 0x00000200
+#define SP_PROT_TLS1_0_SERVER SP_PROT_TLS1_SERVER
+#define SP_PROT_TLS1_1_SERVER 0x00000100
+#define SCH_USE_STRONG_CRYPTO 0x00400000
+// https://learn.microsoft.com/en-us/windows/win32/api/sspi/ns-sspi-secbuffer
+#define SECBUFFER_ALERT 17
+// https://learn.microsoft.com/en-us/windows/win32/api/sspi/ne-sspi-sec_application_protocol_negotiation_ext
+typedef enum _SEC_APPLICATION_PROTOCOL_NEGOTIATION_EXT {
+    SecApplicationProtocolNegotiationExt_None,
+    SecApplicationProtocolNegotiationExt_NPN,
+    SecApplicationProtocolNegotiationExt_ALPN
+} SEC_APPLICATION_PROTOCOL_NEGOTIATION_EXT,*PSEC_APPLICATION_PROTOCOL_NEGOTIATION_EXT;
+
 #include <errno.h>
 #include <inttypes.h>
 #include <math.h>
