From 9fa5180a357af59f4167d194ebf83105e2d550de Mon Sep 17 00:00:00 2001
From: acxz <17132214+acxz@users.noreply.github.com>
Date: Sat, 22 May 2021 21:06:19 -0400
Subject: [PATCH] update tools path to bin for newer mkl (2020.1.217)

---
 cmake/FindMKL.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cmake/FindMKL.cmake b/cmake/FindMKL.cmake
index 964ec34..8ecc7a4 100644
--- a/cmake/FindMKL.cmake
+++ b/cmake/FindMKL.cmake
@@ -95,7 +95,7 @@ else()
     set(MKL_INCLUDE_DIR ${MKL_ROOT_DIR}/include)
 
     # set arguments to call the MKL provided tool for linking
-    set(MKL_LINK_TOOL ${MKL_ROOT_DIR}/tools/mkl_link_tool)
+    set(MKL_LINK_TOOL ${MKL_ROOT_DIR}/bin/mkl_link_tool)
 
     if (WIN32)
         set(MKL_LINK_TOOL ${MKL_LINK_TOOL}.exe)
