From 04200d5aa625fc86c2d81ffbf9dd5c70816fe4ce Mon Sep 17 00:00:00 2001
From: luau-project <luau.project@gmail.com>
Date: Tue, 2 Jan 2024 14:29:48 -0300
Subject: [PATCH] Fixing multiarch lib install dir on Debian-based distros

---
 cmake/cminpack_utils.cmake | 8 ++------
 1 file changed, 2 insertions(+), 6 deletions(-)

diff --git a/cmake/cminpack_utils.cmake b/cmake/cminpack_utils.cmake
index e380d5c..75bdb1b 100644
--- a/cmake/cminpack_utils.cmake
+++ b/cmake/cminpack_utils.cmake
@@ -8,12 +8,8 @@ macro(GET_OS_INFO)
     if(NOT DEFINED CMINPACK_LIB_INSTALL_DIR)
     set(CMINPACK_LIB_INSTALL_DIR "lib")
     if(OS_LINUX)
-        if(${CMAKE_SYSTEM_PROCESSOR} STREQUAL "x86_64")
-            set(CMINPACK_LIB_INSTALL_DIR "lib64")
-        else(${CMAKE_SYSTEM_PROCESSOR} STREQUAL "x86_64")
-            set(CMINPACK_LIB_INSTALL_DIR "lib")
-        endif(${CMAKE_SYSTEM_PROCESSOR} STREQUAL "x86_64")
-        message (STATUS "Operating system is Linux")
+        include(GNUInstallDirs)
+	set(CMINPACK_LIB_INSTALL_DIR ${CMAKE_INSTALL_LIBDIR})
     elseif(OS_BSD)
         message (STATUS "Operating system is BSD")
     elseif(OS_WIN)
