From e98c2b60d6d8bcdb9f79c34eba5cdf0240d6864b Mon Sep 17 00:00:00 2001
From: Ian McInerney <i.mcinerney17@imperial.ac.uk>
Date: Tue, 28 May 2024 15:43:51 +0100
Subject: [PATCH] Fix CMake usage for freebsd

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1be7326..c2df39e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -127,7 +127,7 @@ foreach (cminpack_lib ${cminpack_libs})
     )
 
   if (${CMAKE_SYSTEM_NAME} STREQUAL "FreeBSD")
-    target_link_libraries (${cminpack_lib} m)
+    target_link_libraries (${cminpack_lib} PUBLIC m)
   endif ()
 
   include (CheckLibraryExists)
-- 
2.45.0

