diff --git a/include/miopen/config.h.in b/include/miopen/config.h.in
index 1849b4344..d0c6f2a9b 100644
--- a/include/miopen/config.h.in
+++ b/include/miopen/config.h.in
@@ -117,4 +117,13 @@
     ((HIP_PACKAGE_VERSION_MAJOR * 1000ULL + HIP_PACKAGE_VERSION_MINOR) * 1000000 + \
      HIP_PACKAGE_VERSION_PATCH)
 
+/// WORKAROUND_BOOST_ISSUE_392
+/// Workaround for https://github.com/boostorg/config/issues/392#issuecomment-1109889533
+/// See also https://github.com/ROCmSoftwarePlatform/MIOpen/pull/1490#issuecomment-1109928102,
+/// https://github.com/ROCmSoftwarePlatform/MIOpen/pull/1543
+/// TODO: Remove the W/A as soon we switch to the properly fixed boost.
+#if MIOPEN_BACKEND_HIP
+#include <hip/hip_version.h>
+#endif
+
 #endif
