diff --git a/Makefile.in b/Makefile.in
index 8ded8f6..950de39 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -6,8 +6,8 @@ KEXT_SOURCES = src/normaliz.cc
 
 # workaround for GAP 4.11: we need _POSIX_C_SOURCE=1 to avoid an error during compilation;
 # see https://github.com/gap-packages/NormalizInterface/pull/91 for details
-KEXT_CXXFLAGS = @CPPFLAGS@ @NORMALIZ_CPPFLAGS@ -std=c++11 -D_POSIX_C_SOURCE=1
-KEXT_LDFLAGS = @LDFLAGS@ -lstdc++ -R@NORMALIZ_LDPATH@ @NORMALIZ_LDFLAGS@ -lnormaliz
+KEXT_CXXFLAGS = @CPPFLAGS@ @NORMALIZ_CPPFLAGS@ -std=c++11
+KEXT_LDFLAGS = @LDFLAGS@ -lstdc++ -lgmp @NORMALIZ_LDFLAGS@ -lnormaliz
 
 KEXT_USE_AUTOCONF = 1
 
