diff --git a/frame/compat/cblas/src/cblas_f77.h b/frame/compat/cblas/src/cblas_f77.h
index ce9400f31..b04b60963 100644
--- a/frame/compat/cblas/src/cblas_f77.h
+++ b/frame/compat/cblas/src/cblas_f77.h
@@ -247,200 +247,200 @@
 /*
  * Level 1 BLAS
  */
-#define F77_xerbla     xerbla_
-#define F77_srotg      srotg_blis_impl
-#define F77_srotmg     srotmg_blis_impl
-#define F77_srot       srot_blis_impl
-#define F77_srotm      srotm_blis_impl
-#define F77_drotg      drotg_blis_impl
-#define F77_drotmg     drotmg_blis_impl
-#define F77_drot       drot_blis_impl
-#define F77_drotm      drotm_blis_impl
-#define F77_crotg      crotg_blis_impl
-#define F77_csrot      csrot_blis_impl
-#define F77_zrotg      zrotg_blis_impl
-#define F77_zdrot      zdrot_blis_impl
-#define F77_sswap      sswap_blis_impl
-#define F77_scopy      scopy_blis_impl
-#define F77_saxpy      saxpy_blis_impl
-#define F77_isamax_sub isamaxsub_blis_impl
-#define F77_dswap      dswap_blis_impl
-#define F77_dcopy      dcopy_blis_impl
-#define F77_daxpy      daxpy_blis_impl
-#define F77_idamax_sub idamaxsub_blis_impl
-#define F77_cswap      cswap_blis_impl
-#define F77_ccopy      ccopy_blis_impl
-#define F77_caxpy      caxpy_blis_impl
-#define F77_icamax_sub icamaxsub_blis_impl
-#define F77_zswap      zswap_blis_impl
-#define F77_zcopy      zcopy_blis_impl
-#define F77_zaxpy      zaxpy_blis_impl
-#define F77_izamax_sub izamaxsub_blis_impl
-#define F77_sdot_sub   sdotsub_blis_impl
-#define F77_ddot_sub   ddotsub_blis_impl
-#define F77_dsdot_sub  dsdotsub_blis_impl
-#define F77_sscal      sscal_blis_impl
-#define F77_dscal      dscal_blis_impl
-#define F77_cscal      cscal_blis_impl
-#define F77_zscal      zscal_blis_impl
-#define F77_csscal     csscal_blis_impl
-#define F77_zdscal     zdscal_blis_impl
-#define F77_cdotu_sub  cdotusub_blis_impl
-#define F77_cdotc_sub  cdotcsub_blis_impl
-#define F77_zdotu_sub  zdotusub_blis_impl
-#define F77_zdotc_sub  zdotcsub_blis_impl
-#define F77_snrm2_sub  snrm2sub_blis_impl
-#define F77_sasum_sub  sasumsub_blis_impl
-#define F77_dnrm2_sub  dnrm2sub_blis_impl
-#define F77_dasum_sub  dasumsub_blis_impl
-#define F77_scnrm2_sub scnrm2sub_blis_impl
-#define F77_scasum_sub scasumsub_blis_impl
-#define F77_dznrm2_sub dznrm2sub_blis_impl
-#define F77_dzasum_sub dzasumsub_blis_impl
-#define F77_sdsdot_sub sdsdotsub_blis_impl
+#define F77_xerbla     xerbla_64_
+#define F77_srotg      srotg_blis_64_
+#define F77_srotmg     srotmg_blis_64_
+#define F77_srot       srot_blis_64_
+#define F77_srotm      srotm_blis_64_
+#define F77_drotg      drotg_blis_64_
+#define F77_drotmg     drotmg_blis_64_
+#define F77_drot       drot_blis_64_
+#define F77_drotm      drotm_blis_64_
+#define F77_crotg      crotg_blis_64_
+#define F77_csrot      csrot_blis_64_
+#define F77_zrotg      zrotg_blis_64_
+#define F77_zdrot      zdrot_blis_64_
+#define F77_sswap      sswap_blis_64_
+#define F77_scopy      scopy_blis_64_
+#define F77_saxpy      saxpy_blis_64_
+#define F77_isamax_sub isamaxsub_blis_64_
+#define F77_dswap      dswap_blis_64_
+#define F77_dcopy      dcopy_blis_64_
+#define F77_daxpy      daxpy_blis_64_
+#define F77_idamax_sub idamaxsub_blis_64_
+#define F77_cswap      cswap_blis_64_
+#define F77_ccopy      ccopy_blis_64_
+#define F77_caxpy      caxpy_blis_64_
+#define F77_icamax_sub icamaxsub_blis_64_
+#define F77_zswap      zswap_blis_64_
+#define F77_zcopy      zcopy_blis_64_
+#define F77_zaxpy      zaxpy_blis_64_
+#define F77_izamax_sub izamaxsub_blis_64_
+#define F77_sdot_sub   sdotsub_blis_64_
+#define F77_ddot_sub   ddotsub_blis_64_
+#define F77_dsdot_sub  dsdotsub_blis_64_
+#define F77_sscal      sscal_blis_64_
+#define F77_dscal      dscal_blis_64_
+#define F77_cscal      cscal_blis_64_
+#define F77_zscal      zscal_blis_64_
+#define F77_csscal     csscal_blis_64_
+#define F77_zdscal     zdscal_blis_64_
+#define F77_cdotu_sub  cdotusub_blis_64_
+#define F77_cdotc_sub  cdotcsub_blis_64_
+#define F77_zdotu_sub  zdotusub_blis_64_
+#define F77_zdotc_sub  zdotcsub_blis_64_
+#define F77_snrm2_sub  snrm2sub_blis_64_
+#define F77_sasum_sub  sasumsub_blis_64_
+#define F77_dnrm2_sub  dnrm2sub_blis_64_
+#define F77_dasum_sub  dasumsub_blis_64_
+#define F77_scnrm2_sub scnrm2sub_blis_64_
+#define F77_scasum_sub scasumsub_blis_64_
+#define F77_dznrm2_sub dznrm2sub_blis_64_
+#define F77_dzasum_sub dzasumsub_blis_64_
+#define F77_sdsdot_sub sdsdotsub_blis_64_
 /*
 * Level 2 BLAS
 */
-#define F77_ssymv ssymv_blis_impl
-#define F77_ssbmv ssbmv_blis_impl
-#define F77_sspmv sspmv_blis_impl
-#define F77_sger  sger_blis_impl
-#define F77_ssyr  ssyr_blis_impl
-#define F77_sspr  sspr_blis_impl
-#define F77_ssyr2 ssyr2_blis_impl
-#define F77_sspr2 sspr2_blis_impl
-#define F77_dsymv dsymv_blis_impl
-#define F77_dsbmv dsbmv_blis_impl
-#define F77_dspmv dspmv_blis_impl
-#define F77_dger  dger_blis_impl
-#define F77_dsyr  dsyr_blis_impl
-#define F77_dspr  dspr_blis_impl
-#define F77_dsyr2 dsyr2_blis_impl
-#define F77_dspr2 dspr2_blis_impl
-#define F77_chemv chemv_blis_impl
-#define F77_chbmv chbmv_blis_impl
-#define F77_chpmv chpmv_blis_impl
-#define F77_cgeru cgeru_blis_impl
-#define F77_cgerc cgerc_blis_impl
-#define F77_cher  cher_blis_impl
-#define F77_chpr  chpr_blis_impl
-#define F77_cher2 cher2_blis_impl
-#define F77_chpr2 chpr2_blis_impl
-#define F77_zhemv zhemv_blis_impl
-#define F77_zhbmv zhbmv_blis_impl
-#define F77_zhpmv zhpmv_blis_impl
-#define F77_zgeru zgeru_blis_impl
-#define F77_zgerc zgerc_blis_impl
-#define F77_zher  zher_blis_impl
-#define F77_zhpr  zhpr_blis_impl
-#define F77_zher2 zher2_blis_impl
-#define F77_zhpr2 zhpr2_blis_impl
-#define F77_sgemv sgemv_blis_impl
-#define F77_sgbmv sgbmv_blis_impl
-#define F77_strmv strmv_blis_impl
-#define F77_stbmv stbmv_blis_impl
-#define F77_stpmv stpmv_blis_impl
-#define F77_strsv strsv_blis_impl
-#define F77_stbsv stbsv_blis_impl
-#define F77_stpsv stpsv_blis_impl
-#define F77_dgemv dgemv_blis_impl
-#define F77_dgbmv dgbmv_blis_impl
-#define F77_dtrmv dtrmv_blis_impl
-#define F77_dtbmv dtbmv_blis_impl
-#define F77_dtpmv dtpmv_blis_impl
-#define F77_dtrsv dtrsv_blis_impl
-#define F77_dtbsv dtbsv_blis_impl
-#define F77_dtpsv dtpsv_blis_impl
-#define F77_cgemv cgemv_blis_impl
-#define F77_cgbmv cgbmv_blis_impl
-#define F77_ctrmv ctrmv_blis_impl
-#define F77_ctbmv ctbmv_blis_impl
-#define F77_ctpmv ctpmv_blis_impl
-#define F77_ctrsv ctrsv_blis_impl
-#define F77_ctbsv ctbsv_blis_impl
-#define F77_ctpsv ctpsv_blis_impl
-#define F77_zgemv zgemv_blis_impl
-#define F77_zgbmv zgbmv_blis_impl
-#define F77_ztrmv ztrmv_blis_impl
-#define F77_ztbmv ztbmv_blis_impl
-#define F77_ztpmv ztpmv_blis_impl
-#define F77_ztrsv ztrsv_blis_impl
-#define F77_ztbsv ztbsv_blis_impl
-#define F77_ztpsv ztpsv_blis_impl
+#define F77_ssymv ssymv_blis_64_
+#define F77_ssbmv ssbmv_blis_64_
+#define F77_sspmv sspmv_blis_64_
+#define F77_sger  sger_blis_64_
+#define F77_ssyr  ssyr_blis_64_
+#define F77_sspr  sspr_blis_64_
+#define F77_ssyr2 ssyr2_blis_64_
+#define F77_sspr2 sspr2_blis_64_
+#define F77_dsymv dsymv_blis_64_
+#define F77_dsbmv dsbmv_blis_64_
+#define F77_dspmv dspmv_blis_64_
+#define F77_dger  dger_blis_64_
+#define F77_dsyr  dsyr_blis_64_
+#define F77_dspr  dspr_blis_64_
+#define F77_dsyr2 dsyr2_blis_64_
+#define F77_dspr2 dspr2_blis_64_
+#define F77_chemv chemv_blis_64_
+#define F77_chbmv chbmv_blis_64_
+#define F77_chpmv chpmv_blis_64_
+#define F77_cgeru cgeru_blis_64_
+#define F77_cgerc cgerc_blis_64_
+#define F77_cher  cher_blis_64_
+#define F77_chpr  chpr_blis_64_
+#define F77_cher2 cher2_blis_64_
+#define F77_chpr2 chpr2_blis_64_
+#define F77_zhemv zhemv_blis_64_
+#define F77_zhbmv zhbmv_blis_64_
+#define F77_zhpmv zhpmv_blis_64_
+#define F77_zgeru zgeru_blis_64_
+#define F77_zgerc zgerc_blis_64_
+#define F77_zher  zher_blis_64_
+#define F77_zhpr  zhpr_blis_64_
+#define F77_zher2 zher2_blis_64_
+#define F77_zhpr2 zhpr2_blis_64_
+#define F77_sgemv sgemv_blis_64_
+#define F77_sgbmv sgbmv_blis_64_
+#define F77_strmv strmv_blis_64_
+#define F77_stbmv stbmv_blis_64_
+#define F77_stpmv stpmv_blis_64_
+#define F77_strsv strsv_blis_64_
+#define F77_stbsv stbsv_blis_64_
+#define F77_stpsv stpsv_blis_64_
+#define F77_dgemv dgemv_blis_64_
+#define F77_dgbmv dgbmv_blis_64_
+#define F77_dtrmv dtrmv_blis_64_
+#define F77_dtbmv dtbmv_blis_64_
+#define F77_dtpmv dtpmv_blis_64_
+#define F77_dtrsv dtrsv_blis_64_
+#define F77_dtbsv dtbsv_blis_64_
+#define F77_dtpsv dtpsv_blis_64_
+#define F77_cgemv cgemv_blis_64_
+#define F77_cgbmv cgbmv_blis_64_
+#define F77_ctrmv ctrmv_blis_64_
+#define F77_ctbmv ctbmv_blis_64_
+#define F77_ctpmv ctpmv_blis_64_
+#define F77_ctrsv ctrsv_blis_64_
+#define F77_ctbsv ctbsv_blis_64_
+#define F77_ctpsv ctpsv_blis_64_
+#define F77_zgemv zgemv_blis_64_
+#define F77_zgbmv zgbmv_blis_64_
+#define F77_ztrmv ztrmv_blis_64_
+#define F77_ztbmv ztbmv_blis_64_
+#define F77_ztpmv ztpmv_blis_64_
+#define F77_ztrsv ztrsv_blis_64_
+#define F77_ztbsv ztbsv_blis_64_
+#define F77_ztpsv ztpsv_blis_64_
 /*
 * Level 3 BLAS
 */
-#define F77_chemm  chemm_blis_impl
-#define F77_cherk  cherk_blis_impl
-#define F77_cher2k cher2k_blis_impl
-#define F77_zhemm  zhemm_blis_impl
-#define F77_zherk  zherk_blis_impl
-#define F77_zher2k zher2k_blis_impl
-#define F77_sgemm  sgemm_blis_impl
-#define F77_ssymm  ssymm_blis_impl
-#define F77_ssyrk  ssyrk_blis_impl
-#define F77_ssyr2k ssyr2k_blis_impl
-#define F77_strmm  strmm_blis_impl
-#define F77_strsm  strsm_blis_impl
-#define F77_dgemm  dgemm_blis_impl
-#define F77_dsymm  dsymm_blis_impl
-#define F77_dsyrk  dsyrk_blis_impl
-#define F77_dsyr2k dsyr2k_blis_impl
-#define F77_dtrmm  dtrmm_blis_impl
-#define F77_dtrsm  dtrsm_blis_impl
-#define F77_cgemm  cgemm_blis_impl
-#define F77_csymm  csymm_blis_impl
-#define F77_csyrk  csyrk_blis_impl
-#define F77_csyr2k csyr2k_blis_impl
-#define F77_ctrmm  ctrmm_blis_impl
-#define F77_ctrsm  ctrsm_blis_impl
-#define F77_zgemm  zgemm_blis_impl
-#define F77_zsymm  zsymm_blis_impl
-#define F77_zsyrk  zsyrk_blis_impl
-#define F77_zsyr2k zsyr2k_blis_impl
-#define F77_ztrmm  ztrmm_blis_impl
-#define F77_ztrsm  ztrsm_blis_impl
-#define F77_dgemmt  dgemmt_blis_impl
-#define F77_sgemmt  sgemmt_blis_impl
-#define F77_cgemmt  cgemmt_blis_impl
-#define F77_zgemmt  zgemmt_blis_impl
-#define F77_dzgemm  dzgemm_blis_impl
+#define F77_chemm  chemm_blis_64_
+#define F77_cherk  cherk_blis_64_
+#define F77_cher2k cher2k_blis_64_
+#define F77_zhemm  zhemm_blis_64_
+#define F77_zherk  zherk_blis_64_
+#define F77_zher2k zher2k_blis_64_
+#define F77_sgemm  sgemm_blis_64_
+#define F77_ssymm  ssymm_blis_64_
+#define F77_ssyrk  ssyrk_blis_64_
+#define F77_ssyr2k ssyr2k_blis_64_
+#define F77_strmm  strmm_blis_64_
+#define F77_strsm  strsm_blis_64_
+#define F77_dgemm  dgemm_blis_64_
+#define F77_dsymm  dsymm_blis_64_
+#define F77_dsyrk  dsyrk_blis_64_
+#define F77_dsyr2k dsyr2k_blis_64_
+#define F77_dtrmm  dtrmm_blis_64_
+#define F77_dtrsm  dtrsm_blis_64_
+#define F77_cgemm  cgemm_blis_64_
+#define F77_csymm  csymm_blis_64_
+#define F77_csyrk  csyrk_blis_64_
+#define F77_csyr2k csyr2k_blis_64_
+#define F77_ctrmm  ctrmm_blis_64_
+#define F77_ctrsm  ctrsm_blis_64_
+#define F77_zgemm  zgemm_blis_64_
+#define F77_zsymm  zsymm_blis_64_
+#define F77_zsyrk  zsyrk_blis_64_
+#define F77_zsyr2k zsyr2k_blis_64_
+#define F77_ztrmm  ztrmm_blis_64_
+#define F77_ztrsm  ztrsm_blis_64_
+#define F77_dgemmt  dgemmt_blis_64_
+#define F77_sgemmt  sgemmt_blis_64_
+#define F77_cgemmt  cgemmt_blis_64_
+#define F77_zgemmt  zgemmt_blis_64_
+#define F77_dzgemm  dzgemm_blis_64_
 
 /*
 * Aux Function
 */
-#define F77_scabs1 scabs1_
-#define F77_dcabs1 dcabs1_
+#define F77_scabs1 scabs1_64_
+#define F77_dcabs1 dcabs1_64_
 
 /*
  * -- BLAS Extension APIs --
  */
 
-#define F77_saxpby     saxpby_blis_impl
-#define F77_daxpby     daxpby_blis_impl
-#define F77_caxpby     caxpby_blis_impl
-#define F77_zaxpby     zaxpby_blis_impl
-#define F77_cgemm3m    cgemm3m_blis_impl
-#define F77_zgemm3m    zgemm3m_blis_impl
+#define F77_saxpby     saxpby_blis_64_
+#define F77_daxpby     daxpby_blis_64_
+#define F77_caxpby     caxpby_blis_64_
+#define F77_zaxpby     zaxpby_blis_64_
+#define F77_cgemm3m    cgemm3m_blis_64_
+#define F77_zgemm3m    zgemm3m_blis_64_
 
-#define F77_isamin_sub isaminsub_blis_impl
-#define F77_idamin_sub idaminsub_blis_impl
-#define F77_icamin_sub icaminsub_blis_impl
-#define F77_izamin_sub izaminsub_blis_impl
+#define F77_isamin_sub isaminsub_blis_64_
+#define F77_idamin_sub idaminsub_blis_64_
+#define F77_icamin_sub icaminsub_blis_64_
+#define F77_izamin_sub izaminsub_blis_64_
 
 // -- Batch APIs --
-#define F77_sgemm_batch  sgemm_batch_
-#define F77_dgemm_batch  dgemm_batch_
-#define F77_cgemm_batch  cgemm_batch_
-#define F77_zgemm_batch  zgemm_batch_
+#define F77_sgemm_batch  sgemm_batch_64_
+#define F77_dgemm_batch  dgemm_batch_64_
+#define F77_cgemm_batch  cgemm_batch_64_
+#define F77_zgemm_batch  zgemm_batch_64_
 
 // -- Pack-Compute APIs --
-#define F77_sgemm_pack_get_size  sgemm_pack_get_size_blis_impl
-#define F77_dgemm_pack_get_size  dgemm_pack_get_size_blis_impl
-#define F77_sgemm_pack  sgemm_pack_blis_impl
-#define F77_dgemm_pack  dgemm_pack_blis_impl
-#define F77_sgemm_compute  sgemm_compute_blis_impl
-#define F77_dgemm_compute  dgemm_compute_blis_impl
+#define F77_sgemm_pack_get_size  sgemm_pack_get_size_blis_64_
+#define F77_dgemm_pack_get_size  dgemm_pack_get_size_blis_64_
+#define F77_sgemm_pack  sgemm_pack_blis_64_
+#define F77_dgemm_pack  dgemm_pack_blis_64_
+#define F77_sgemm_compute  sgemm_compute_blis_64_
+#define F77_dgemm_compute  dgemm_compute_blis_64_
 #endif
 
 #endif /*  CBLAS_F77_H */
diff --git a/frame/include/bli_macro_defs.h b/frame/include/bli_macro_defs.h
index e35e48c8a..b18a3efb3 100644
--- a/frame/include/bli_macro_defs.h
+++ b/frame/include/bli_macro_defs.h
@@ -92,10 +92,10 @@
 #define PASTEF772(ch1,ch2,name)                            ch1 ## ch2 ## name
 #define PASTEF773(ch1,ch2,ch3,name)                 ch1 ## ch2 ## ch3 ## name
 #else
-#define PASTEF770(name)                                            name ## _
-#define PASTEF77(ch1,name)                                  ch1 ## name ## _
-#define PASTEF772(ch1,ch2,name)                      ch1 ## ch2 ## name ## _
-#define PASTEF773(ch1,ch2,ch3,name)           ch1 ## ch2 ## ch3 ## name ## _
+#define PASTEF770(name)                                            name ## _64_
+#define PASTEF77(ch1,name)                                  ch1 ## name ## _64_
+#define PASTEF772(ch1,ch2,name)                      ch1 ## ch2 ## name ## _64_
+#define PASTEF773(ch1,ch2,ch3,name)           ch1 ## ch2 ## ch3 ## name ## _64_
 #endif
 
 // Macros to define names _blis_impl suffix, *_blis_impl is the blis
