--- a/src/Interfaces/NomadStdCInterface.cpp
+++ b/src/Interfaces/NomadStdCInterface.cpp
@@ -380,6 +380,9 @@
         // reset parameters in case someone wants to restart an optimization again
         nomad_problem->p->resetToDefaultValues();
 
+        NOMAD::OutputQueue::Flush();
+        NOMAD::CacheBase::getInstance()->clear();
+
         return 0;
     }
 
@@ -387,5 +390,9 @@
     {
         printf("NOMAD exception (report to developper):\n%s\n", e.what());
     }
+
+    NOMAD::OutputQueue::Flush();
+    NOMAD::CacheBase::getInstance()->clear();
+
     return -1;
 }
