--- a/ext/sgtelib/CMakeLists.txt
+++ b/ext/sgtelib/CMakeLists.txt
@@ -98,9 +98,9 @@
     $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/src>
     $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}/sgtelib>
     )
-#if(OpenMP_CXX_FOUND)
-#    target_link_libraries(sgtelib PRIVATE OpenMP::OpenMP_CXX)
-#endif()
+if(OpenMP_CXX_FOUND)
+    target_link_libraries(sgtelib PRIVATE OpenMP::OpenMP_CXX)
+endif()
 
 # build executable sgtelib_main
 add_executable(sgtelib_main ${SGTELIB_SOURCES})
@@ -108,9 +108,9 @@
     $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/src>
     $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}/sgtelib>
     )
-#if(OpenMP_CXX_FOUND)
-#    target_link_libraries(sgtelib_main PRIVATE OpenMP::OpenMP_CXX)
-#endif()
+if(OpenMP_CXX_FOUND)
+    target_link_libraries(sgtelib_main PRIVATE OpenMP::OpenMP_CXX)
+endif()
 
 # installing executables and libraries
 install(TARGETS sgtelib_main sgtelib 
