--- a/m4/ax_lib_sqlite3.m4
+++ b/m4/ax_lib_sqlite3.m4
@@ -77,7 +77,7 @@
             unset ac_cv_lib_sqlite3_sqlite3_open
             saved_LIBS="$LIBS"
             LIBS=""
-            AC_CHECK_LIB(sqlite3,sqlite3_open,LIB_SQLITE3_FOUND=yes,LIB_SQLITE3_FOUND=no,-L$ac_sqlite3_path/lib)
+            AC_CHECK_LIB(sqlite3,sqlite3_open,LIB_SQLITE3_FOUND=yes,LIB_SQLITE3_FOUND=no,-L$ac_sqlite3_path/lib ${EXTRA_SQLITE3_LIBS})
             LIBS="$saved_LIBS"
             if test "$LIB_SQLITE3_FOUND" = "yes"; then
                 ac_sqlite3_ldflags="-L$ac_sqlite3_path/lib"
@@ -101,7 +101,7 @@
                         unset ac_cv_lib_sqlite3_sqlite3_open
                         saved_LIBS="$LIBS"
                         LIBS=""
-                        AC_CHECK_LIB(sqlite3,sqlite3_open,LIB_SQLITE3_FOUND=yes,LIB_SQLITE3_FOUND=no,-L$ac_sqlite3_path_tmp/lib)
+                        AC_CHECK_LIB(sqlite3,sqlite3_open,LIB_SQLITE3_FOUND=yes,LIB_SQLITE3_FOUND=no,-L$ac_sqlite3_path_tmp/lib ${EXTRA_SQLITE3_LIBS})
                         LIBS="$saved_LIBS"
                         if test "$LIB_SQLITE3_FOUND" = "yes"; then
                             ac_sqlite3_ldflags="-L$ac_sqlite3_path_tmp/lib"
