diff --git before/WCPG-0.9/autogen.sh after/WCPG-0.9/autogen.sh
old mode 100644
new mode 100755
diff --git before/WCPG-0.9/configure.ac after/WCPG-0.9/configure.ac
index bff551e..1c27c1c 100644
--- before/WCPG-0.9/configure.ac
+++ after/WCPG-0.9/configure.ac
@@ -351,27 +351,6 @@ else
 fi
 
 
-# Check that dynamic libraries are available at runtime
-AC_MSG_CHECKING([if libgmp is found at runtime])
-AC_RUN_IFELSE([AC_LANG_PROGRAM([[#include <gmp.h>]],
-                               [[mpz_t a; mpz_init(a); mpz_clear(a); return 0;]])],
-	      [AC_MSG_RESULT(yes)],
-              [AC_MSG_WARN([libgmp is found at compile time but not at run time. Checking /etc/ld.so.conf and running ldconfig could fix the problem.])])
-
-
-AC_MSG_CHECKING([if libmpfr is found at runtime])
-AC_RUN_IFELSE([AC_LANG_PROGRAM([[#include <mpfr.h>]],
-                               [[mpfr_t a; mpfr_init2(a, 53); mpfr_clear(a); return 0;]])],
-	      [AC_MSG_RESULT(yes)],
-              [AC_MSG_WARN([libmpfr is found at compile time but not at run time. Checking /etc/ld.so.conf and running ldconfig could fix the problem.])])
-
-
-AC_MSG_CHECKING([if libmpfi is found at runtime])
-AC_RUN_IFELSE([AC_LANG_PROGRAM([[#include <mpfi.h>]],
-                               [[mpfi_t a; mpfi_init(a); mpfi_clear(a); return 0;]])],
-	      [AC_MSG_RESULT(yes)],
-              [AC_MSG_ERROR([libmpfi is found at compile time but not at run time. Checking /etc/ld.so.conf and running ldconfig could fix the problem.])])
-
 #Check for -Wl,-export-dynamic etc.
 CFLAGS="$AM_CFLAGS -fPIC $USER_CFLAGS"
 AC_MSG_CHECKING([for -fPIC support])
