--- a/Makefile.in
+++ b/Makefile.in
@@ -257,7 +257,7 @@
 XFAIL_TESTS = $(am__EXEEXT_6) $(am__EXEEXT_6)
 check_PROGRAMS = $(am__EXEEXT_2) $(am__EXEEXT_4)
 @WIN32_NATIVE_BUILD_TRUE@am__append_1 = -lurlmon
-@WIN32_NATIVE_BUILD_TRUE@am__append_2 = -lws2_32
+@WIN32_NATIVE_BUILD_TRUE@am__append_2 = -lws2_32 -lurlmon
 @WIN32_NATIVE_BUILD_TRUE@@WITH_MODULES_FALSE@am__append_3 = -lurlmon
 @MAGICKCORE_ZERO_CONFIGURATION_SUPPORT_TRUE@am__append_4 = MagickCore/threshold-map.h
 @MAGICKCORE_ZERO_CONFIGURATION_SUPPORT_TRUE@am__append_5 = MagickCore/threshold-map.h
