--- a/plugins/EstimationPlugin/src/base/CMakeLists.txt	2020-04-29 20:19:59.000000000 +0200
+++ b/plugins/EstimationPlugin/src/base/CMakeLists.txt	2020-08-23 08:38:26.000000000 +0200
@@ -192,7 +192,6 @@
 TARGET_LINK_LIBRARIES(${TargetName} PRIVATE XercesC::XercesC)
 if(WIN32)
   TARGET_COMPILE_DEFINITIONS(${TargetName} PUBLIC -DXERCES_STATIC_LIBRARY)
-  TARGET_LINK_LIBRARIES(${TargetName} PUBLIC Ws2_32)
 endif()
 
 # Set RPATH to find shared libraries in default locations on Mac/Linux
