--- mlpack-4.6.0/src/mlpack/methods/CMakeLists.txt	2025-04-03 15:47:01.000000000 +0000
+++ mlpack-4.6.0-new/src/mlpack/methods/CMakeLists.txt	2025-04-04 16:49:40.234267529 +0000
@@ -27,6 +27,7 @@
 add_all_bindings(hmm hmm_loglik "Misc. / Other")
 add_all_bindings(hmm hmm_viterbi "Misc. / Other")
 add_all_bindings(hoeffding_trees hoeffding_tree "Classification")
+add_all_bindings(preprocess image_converter "Preprocessing")
 add_all_bindings(kde kde "Misc. / Other")
 add_all_bindings(kernel_pca kernel_pca "Transformations")
 add_all_bindings(kmeans kmeans "Clustering")
@@ -88,11 +89,6 @@
 add_cli_executable(preprocess preprocess_imputer)
 add_markdown_docs(preprocess preprocess_imputer "cli" "")
 
-# The image converter is only enabled if STB is available.
-if (STB_AVAILABLE)
-  add_all_bindings(preprocess image_converter "Preprocessing")
-endif()
-
 # Range search provides a vector of vector of results, and this is only
 # supported for the CLI bindings.
 add_category(range_search "Geometry")
