diff --git a/CMakeLists.txt b/CMakeLists.txt
index 843d334..e292821 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -118,17 +118,17 @@ set(PKG_CONFIG_DIR ${libdir}/pkgconfig)
 #
 #   Fortran Mangling, MPI Tests and BLACS settings
 #
-include(FortranMangling)
-COMPILE(install_COMPILED)
+#include(FortranMangling)
+#COMPILE(install_COMPILED)
 
-FORTRAN_MANGLING(CDEFS)
-#MESSAGE(STATUS "Setting CDEFS = ${CDEFS}")
-#set(CDEFS ${CDEFS} CACHE STRING "Fortran Mangling" FORCE)
-MESSAGE(STATUS "=========")
+#FORTRAN_MANGLING(CDEFS)
+##MESSAGE(STATUS "Setting CDEFS = ${CDEFS}")
+##set(CDEFS ${CDEFS} CACHE STRING "Fortran Mangling" FORCE)
+#MESSAGE(STATUS "=========")
 
 # --------------------------------------------------
 # Compiler Flags
-ADD_DEFINITIONS( "-D${CDEFS}")
+ADD_DEFINITIONS( "-D$ENV{CDEFS}")
 
 # --------------------------------------------------
 # ScaLAPACK needs BLAS and LAPACK
