--- a/src/Makefile
+++ b/src/Makefile
@@ -31,7 +31,10 @@
 include $(topdir)/Makefile.inc
 
 $(incdir)/mumps_int_def.h: build_mumps_int_def 
-	./build_mumps_int_def > $(incdir)/mumps_int_def.h
+    echo "#if ! defined(MUMPS_INT_H)" > $(incdir)/mumps_int_def.h
+    echo "#   define MUMPS_INT_H" >> $(incdir)/mumps_int_def.h
+    echo "#   define MUMPS_INTSIZE32" >> $(incdir)/mumps_int_def.h
+    echo "#endif" >> $(incdir)/mumps_int_def.h
 build_mumps_int_def:build_mumps_int_def.o
 	$(CC) $(OPTC) $(OPTL) build_mumps_int_def.o -o build_mumps_int_def
 
