--- a/PORD/lib/Makefile
+++ b/PORD/lib/Makefile
@@ -6,6 +6,8 @@
 # Otherwise, adequate variables for CC, CFLAGS, AR and
 # RANLIB must be passed to make.
 #
+topdir = ../..
+include $(topdir)/Makefile.inc
 
 INCLUDES = -I../include
 
@@ -29,7 +31,7 @@
 	$(RANLIB) $@
 
 libpord$(PLAT)$(LIBEXT_SHARED):$(OBJS)
-	$(CC) -shared $(OBJS) -o libpord.so
+	$(CC) -shared $(OBJS) -o libpord$(LIBEXT_SHARED)
 
 clean:
 	rm -f *.o
