diff --git a/sunrpc/rpc/types.h b/sunrpc/rpc/types.h
index cf50141400..4adc625243 100644
--- a/sunrpc/rpc/types.h
+++ b/sunrpc/rpc/types.h
@@ -68,7 +68,7 @@ typedef unsigned long rpcport_t;
 #include <sys/types.h>
 #endif
 
-#ifndef __u_char_defined
+#if 0
 typedef __u_char u_char;
 typedef __u_short u_short;
 typedef __u_int u_int;
@@ -78,7 +78,7 @@ typedef __u_quad_t u_quad_t;
 typedef __fsid_t fsid_t;
 # define __u_char_defined
 #endif
-#ifndef __daddr_t_defined
+#if 0
 typedef __daddr_t daddr_t;
 # if ! defined(caddr_t) && ! defined(__caddr_t_defined)
 typedef __caddr_t caddr_t;
