--- a/nccl/src/init.cc
+++ b/nccl/src/init.cc
@@ -107,7 +107,6 @@ static ncclResult_t setCpuStackSize() {
       // Change the default pthread stack size (via a nonportable API, which will become necessary if we switch
       // to C++ threads).
       PTHREADCHECK(pthread_attr_setstacksize(&attr, SAFE_STACK_SIZE), "pthread_attr_setstacksize");
-      PTHREADCHECK(pthread_setattr_default_np(&attr), "pthread_setattr_default_np");
     }

     PTHREADCHECK(pthread_attr_destroy(&attr), "pthread_attr_destroy");
