diff --git src/kernel/rational/givrational.h.orig src/kernel/rational/givrational.h
index 2d933ea5..acfead23 100644
--- src/kernel/rational/givrational.h.orig
+++ src/kernel/rational/givrational.h
@@ -36,7 +36,7 @@ namespace Givaro {
     const Integer round (const Rational &r) ;
     const Integer trunc (const Rational &r) ;
     const Rational abs  (const Rational &r) ;
-    const Rational pow (const Rational& n, uint l);
+    const Rational pow (const Rational& n, unsigned int l);
     const Rational pow (const Rational& n, uint64_t l);
     uint64_t length (const Rational& r) ;
     int sign   (const Rational& r) ;
