--- /dev/null
+++ b/cmake/dependencies/CMakeLists.txt
@@ -235,13 +235,13 @@
   set(GMP OFF)
   set(PAPILO OFF)
   set(ZIMPL OFF)
   set(IPOPT OFF)
   set(AMPL OFF)
 
-  set(TPI "tny" CACHE STRING "Scip param")
+  set(TPI "none" CACHE STRING "Scip param")
   set(EXPRINT "none" CACHE STRING "Scip param")
   set(LPS "none" CACHE STRING "Scip param")
   set(SYM "none" CACHE STRING "Scip param")
   FetchContent_Declare(
     scip
     GIT_REPOSITORY "https://github.com/scipopt/scip.git"
