diff --git a/CMakeLists.txt b/CMakeLists.txt
index 95610d55..9d272e38 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -490,10 +490,10 @@ endif(USE_HDFS)
 
 if(WIN32)
     if(MINGW OR CYGWIN)
-      TARGET_LINK_LIBRARIES(lightgbm Ws2_32)
-      TARGET_LINK_LIBRARIES(_lightgbm Ws2_32)
-      TARGET_LINK_LIBRARIES(lightgbm IPHLPAPI)
-      TARGET_LINK_LIBRARIES(_lightgbm IPHLPAPI)
+      TARGET_LINK_LIBRARIES(lightgbm ws2_32)
+      TARGET_LINK_LIBRARIES(_lightgbm ws2_32)
+      TARGET_LINK_LIBRARIES(lightgbm iphlpapi)
+      TARGET_LINK_LIBRARIES(_lightgbm iphlpapi)
     endif(MINGW OR CYGWIN)
 endif(WIN32)
