--- CMakeLists.txt	2025-09-27 09:47:06.614112874 +0200
+++ CMakeLists.txt	2025-09-29 17:54:55.334763931 +0200
@@ -71,8 +71,16 @@
 	endif()
 elseif(APPLE)
 	list(APPEND FIDO_SOURCES hid_osx.c)
+	
+#elseif(CMAKE_SYSTEM_NAME STREQUAL "Linux")
+#if(WITH_LIBUDEV)
+#    list(APPEND FIDO_SOURCES hid_linux.c)
+#endif()
+#list(APPEND FIDO_SOURCES hid_unix.c)
+
 elseif(CMAKE_SYSTEM_NAME STREQUAL "Linux")
-	list(APPEND FIDO_SOURCES hid_linux.c hid_unix.c)
+    list(APPEND FIDO_SOURCES hid_linux.c hid_unix.c)
+
 elseif(CMAKE_SYSTEM_NAME STREQUAL "NetBSD")
 	list(APPEND FIDO_SOURCES hid_netbsd.c hid_unix.c)
 elseif(CMAKE_SYSTEM_NAME STREQUAL "OpenBSD")
@@ -92,7 +100,7 @@
 list(APPEND COMPAT_SOURCES
 	../openbsd-compat/bsd-asprintf.c
 	../openbsd-compat/bsd-getpagesize.c
-	../openbsd-compat/clock_gettime.c
+	#../openbsd-compat/clock_gettime.c
 	../openbsd-compat/endian_win32.c
 	../openbsd-compat/explicit_bzero.c
 	../openbsd-compat/explicit_bzero_win32.c
@@ -103,6 +111,14 @@
 )
 
 if(WIN32)
+    list(APPEND COMPAT_SOURCES ../openbsd-compat/clock_gettime.c)
+endif()
+# If one must support old/unsupported Linux platforms, can add here:
+# if(SOME_OLD_LINUX)
+#     list(APPEND COMPAT_SOURCES ../openbsd-compat/clock_gettime.c)
+# endif()
+
+if(WIN32)
 	list(APPEND BASE_LIBRARIES wsock32 ws2_32 bcrypt setupapi hid)
 	if(USE_PCSC)
 		list(APPEND BASE_LIBRARIES winscard)
