--- vc-intrinsics/CMakeLists.txt	2020-08-20 12:46:58.661604144 +0200
+++ vc-intrinsics/CMakeLists.txt	2020-08-20 12:47:18.015427529 +0200
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.13.4)
+cmake_minimum_required(VERSION 3.13.0)

 set(LLVM_GENX_INTRINSICS_VERSION 1.0)

--- vc-intrinsics/GenXIntrinsics/CMakeLists.txt	2020-08-20 12:47:03.225053032 +0200
+++ vc-intrinsics/GenXIntrinsics/CMakeLists.txt	2020-08-20 12:47:09.055200659 +0200
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.13.4)
+cmake_minimum_required(VERSION 3.13.0)

 set(GENX_INTRINSICS_MAIN_INCLUDE_DIR ${CMAKE_CURRENT_SOURCE_DIR}/include)
