--- intel-graphics-compiler/IGC/VectorCompiler/lib/GenXCodeGen/GenXWrapper.cpp	2020-08-20 12:51:11.977651573 +0200
+++ intel-graphics-compiler/IGC/VectorCompiler/lib/GenXCodeGen/GenXWrapper.cpp	2020-08-20 12:51:42.218286697 +0200
@@ -101,7 +101,7 @@

 #if defined(__linux__)
   // Hack to workaround cmoc crashes during loading of SPIRV library
-  static auto DeepBindHack = dlopen(SpirvLibName, RTLD_NOW | RTLD_DEEPBIND);
+  static auto DeepBindHack = dlopen(SpirvLibName, RTLD_NOW);
 #endif // __linux__

   using DL = sys::DynamicLibrary;
