From 3a15f35ccae10e0e8d21acb2cdb2790b2955a5be Mon Sep 17 00:00:00 2001
From: Julian P Samaroo <jpsamaroo@jpsamaroo.me>
Date: Mon, 7 Mar 2022 12:37:32 -0600
Subject: [PATCH] Add musl support

---
 source/inc/ze_util.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/source/inc/ze_util.h b/source/inc/ze_util.h
index 1581c83..1a65e4d 100644
--- a/source/inc/ze_util.h
+++ b/source/inc/ze_util.h
@@ -25,7 +25,7 @@
 #  define HMODULE void*
 #  define MAKE_LIBRARY_NAME(NAME, VERSION)    "lib" NAME ".so." VERSION
 #  define MAKE_LAYER_NAME(NAME)    "lib" NAME ".so." L0_VALIDATION_LAYER_SUPPORTED_VERSION
-#  define LOAD_DRIVER_LIBRARY(NAME) dlopen(NAME, RTLD_LAZY|RTLD_LOCAL|RTLD_DEEPBIND)
+#  define LOAD_DRIVER_LIBRARY(NAME) dlopen(NAME, RTLD_LAZY|RTLD_LOCAL)
 #  define FREE_DRIVER_LIBRARY(LIB)  if(LIB) dlclose(LIB)
 #  define GET_FUNCTION_PTR(LIB, FUNC_NAME) dlsym(LIB, FUNC_NAME)
 #endif
-- 
2.35.0

