diff --git a/CMakeLists.txt b/CMakeLists.txt
index ff9aa8e..e3e5e97 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -39,7 +39,7 @@ endfunction()
 set(SLEEF_ALL_SUPPORTED_EXTENSIONS
   AVX512FNOFMA AVX512F AVX2 AVX2128 FMA4 AVX SSE4 SSE2  # x86
   SVENOFMA SVE ADVSIMDNOFMA ADVSIMD                     # Aarch64
-  NEON32 NEON32VFPV4                                    # Aarch32
+  NEON32                                                # Aarch32
   VSX VSXNOFMA VSX3 VSX3NOFMA                           # PPC64
   VXE VXENOFMA VXE2 VXE2NOFMA	                        # IBM Z
   PUREC_SCALAR PURECFMA_SCALAR                          # Generic type
@@ -49,7 +49,7 @@ set(SLEEF_ALL_SUPPORTED_EXTENSIONS
 set(SLEEF_SUPPORTED_LIBM_EXTENSIONS
   AVX512FNOFMA AVX512F AVX2 AVX2128 FMA4 AVX SSE4 SSE2  # x86
   SVENOFMA SVE ADVSIMDNOFMA ADVSIMD                     # Aarch64
-  NEON32 NEON32VFPV4                                    # Aarch32
+  NEON32                                                # Aarch32
   VSX VSXNOFMA VSX3 VSX3NOFMA                           # PPC64
   VXE VXENOFMA VXE2 VXE2NOFMA	                        # IBM Z
   PUREC_SCALAR PURECFMA_SCALAR                          # Generic type
diff --git a/Configure.cmake b/Configure.cmake
index 8141e8e..87b2cd5 100644
--- a/Configure.cmake
+++ b/Configure.cmake
@@ -102,7 +102,6 @@ elseif(SLEEF_TARGET_PROCESSOR MATCHES "aarch64|arm64")
 elseif(CMAKE_SYSTEM_PROCESSOR MATCHES "arm")
   set(SLEEF_ARCH_AARCH32 ON CACHE INTERNAL "True for Aarch32 architecture.")
   set(COMPILER_SUPPORTS_NEON32 1)
-  set(COMPILER_SUPPORTS_NEON32VFPV4 1)
 
   set(CLANG_FLAGS_ENABLE_PURECFMA_SCALAR "-mfpu=vfpv4;-fno-strict-aliasing")
   set(CLANG_FLAGS_ENABLE_PUREC_SCALAR "-fno-strict-aliasing")
@@ -148,7 +147,6 @@ set(CLANG_FLAGS_ENABLE_AVX2128 "-mavx2;-mfma")
 set(CLANG_FLAGS_ENABLE_AVX512F "-mavx512f")
 set(CLANG_FLAGS_ENABLE_AVX512FNOFMA "-mavx512f")
 set(CLANG_FLAGS_ENABLE_NEON32 "--target=arm-linux-gnueabihf;-mcpu=cortex-a8")
-set(CLANG_FLAGS_ENABLE_NEON32VFPV4 "-march=armv7-a;-mfpu=neon-vfpv4")
 # Arm AArch64 vector extensions.
 set(CLANG_FLAGS_ENABLE_SVE "-march=armv8-a+sve")
 set(CLANG_FLAGS_ENABLE_SVENOFMA "-march=armv8-a+sve")
