--- a/configure.ac
+++ b/configure.ac
@@ -303,7 +303,7 @@
       SHRLIB="libwcs.$LIBVER.dylib"
       SONAME="libwcs.$SHVER.dylib"
       SHRLD="$SHRLD -dynamiclib -single_module"
-      SHRLD="$SHRLD -compatibility_version $SHVER -current_version $LIBVER"
+      SHRLD="$SHRLD -compatibility_version $SHVER -current_version $LIBVER -install_name \$(SONAME)"
       SHRLN="libwcs.dylib"
 
       case "$host_cpu" in
--- a/configure
+++ b/configure
@@ -6971,7 +6971,7 @@
       SHRLIB="libwcs.$LIBVER.dylib"
       SONAME="libwcs.$SHVER.dylib"
       SHRLD="$SHRLD -dynamiclib -single_module"
-      SHRLD="$SHRLD -compatibility_version $SHVER -current_version $LIBVER"
+      SHRLD="$SHRLD -compatibility_version $SHVER -current_version $LIBVER -install_name \$(SONAME)"
       SHRLN="libwcs.dylib"
 
       case "$host_cpu" in
