--- dmlc-core-old/CMakeLists.txt
+++ dmlc-core/CMakeLists.txt
@@ -119,7 +119,7 @@
 endif()

 if(WIN32 AND (NOT MSVC))  # On Windows, link Shlwapi.lib for non-MSVC compilers
-  target_link_libraries(dmlc PRIVATE Shlwapi)
+  target_link_libraries(dmlc PRIVATE shlwapi)
 endif()

 # Check location of clock_gettime; if it's in librt, link it
--- dmlc-core-old/include/dmlc/filesystem.h
+++ dmlc-core/include/dmlc/filesystem.h
@@ -18,8 +18,8 @@
 #ifdef _WIN32
 #define NOMINMAX
 #include <windows.h>
-#include <Shlwapi.h>
-#pragma comment(lib, "Shlwapi.lib")
+#include <shlwapi.h>
+#pragma comment(lib, "shlwapi.lib")
 #else  // _WIN32
 #include <unistd.h>
 #include <sys/stat.h>
--- dmlc-core-old/src/io/local_filesys.cc
+++ dmlc-core/src/io/local_filesys.cc
@@ -13,7 +13,7 @@
 }
 #define stat_struct stat
 #else  // _WIN32
-#include <Windows.h>
+#include <windows.h>
 #define stat _stat64
 #define stat_struct __stat64
 #endif  // _WIN32
