+++ xgboost/src/common/io.h
@@ -384,7 +384,7 @@
    * @param length    See the `length` parameter of `mmap` for details.
    */
   explicit PrivateMmapConstStream(std::string path, std::size_t offset, std::size_t length)
-      : AlignedResourceReadStream{std::make_shared<MmapResource>(path, offset, length)} {}
+      : AlignedResourceReadStream{std::shared_ptr<MmapResource>(new MmapResource(path, offset, length))} {}
   ~PrivateMmapConstStream() noexcept(false) override;
 };
 
