From f2a451df6a5c17473d803f8161d9714d6d412266 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Mos=C3=A8=20Giordano?= <mose@gnu.org>
Date: Tue, 21 Dec 2021 17:15:28 +0100
Subject: [PATCH 2/3] Fix signature also of functions in `FbcReactionPlugin.h`

---
 .../fbc/extension/FbcReactionPlugin.h         | 22 +++++++++----------
 1 file changed, 11 insertions(+), 11 deletions(-)

diff --git a/src/sbml/packages/fbc/extension/FbcReactionPlugin.h b/src/sbml/packages/fbc/extension/FbcReactionPlugin.h
index b9bf4ec6c..b21675d2e 100644
--- a/src/sbml/packages/fbc/extension/FbcReactionPlugin.h
+++ b/src/sbml/packages/fbc/extension/FbcReactionPlugin.h
@@ -733,7 +733,7 @@ BEGIN_C_DECLS
  */
 LIBSBML_EXTERN
 char *
-FbcReactionPlugin_getUpperFluxBound(SBasePlugin_t * fbc);
+FbcReactionPlugin_getUpperFluxBound(FbcSBasePlugin_t * fbc);
 
 
 /**
@@ -749,7 +749,7 @@ FbcReactionPlugin_getUpperFluxBound(SBasePlugin_t * fbc);
  */
 LIBSBML_EXTERN
 int
-FbcReactionPlugin_isSetUpperFluxBound(SBasePlugin_t * fbc);
+FbcReactionPlugin_isSetUpperFluxBound(FbcSBasePlugin_t * fbc);
 
 
 /**
@@ -769,7 +769,7 @@ FbcReactionPlugin_isSetUpperFluxBound(SBasePlugin_t * fbc);
  */
 LIBSBML_EXTERN
 int
-FbcReactionPlugin_setUpperFluxBound(SBasePlugin_t * fbc, const char * UpperFluxBound);
+FbcReactionPlugin_setUpperFluxBound(FbcSBasePlugin_t * fbc, const char * UpperFluxBound);
 
 
 /**
@@ -785,7 +785,7 @@ FbcReactionPlugin_setUpperFluxBound(SBasePlugin_t * fbc, const char * UpperFluxB
  */
 LIBSBML_EXTERN
 int
-FbcReactionPlugin_unsetUpperFluxBound(SBasePlugin_t * fbc);
+FbcReactionPlugin_unsetUpperFluxBound(FbcSBasePlugin_t * fbc);
 
 
 /**
@@ -799,7 +799,7 @@ FbcReactionPlugin_unsetUpperFluxBound(SBasePlugin_t * fbc);
  */
 LIBSBML_EXTERN
 char *
-FbcReactionPlugin_getLowerFluxBound(SBasePlugin_t * fbc);
+FbcReactionPlugin_getLowerFluxBound(FbcSBasePlugin_t * fbc);
 
 
 /**
@@ -815,7 +815,7 @@ FbcReactionPlugin_getLowerFluxBound(SBasePlugin_t * fbc);
  */
 LIBSBML_EXTERN
 int
-FbcReactionPlugin_isSetLowerFluxBound(SBasePlugin_t * fbc);
+FbcReactionPlugin_isSetLowerFluxBound(FbcSBasePlugin_t * fbc);
 
 
 /**
@@ -835,7 +835,7 @@ FbcReactionPlugin_isSetLowerFluxBound(SBasePlugin_t * fbc);
  */
 LIBSBML_EXTERN
 int
-FbcReactionPlugin_setLowerFluxBound(SBasePlugin_t * fbc, const char * LowerFluxBound);
+FbcReactionPlugin_setLowerFluxBound(FbcSBasePlugin_t * fbc, const char * LowerFluxBound);
 
 
 /**
@@ -851,7 +851,7 @@ FbcReactionPlugin_setLowerFluxBound(SBasePlugin_t * fbc, const char * LowerFluxB
  */
 LIBSBML_EXTERN
 int
-FbcReactionPlugin_unsetUpperFluxBound(SBasePlugin_t * fbc);
+FbcReactionPlugin_unsetUpperFluxBound(FbcSBasePlugin_t * fbc);
 
 
 /**
@@ -867,7 +867,7 @@ FbcReactionPlugin_unsetUpperFluxBound(SBasePlugin_t * fbc);
  */
 LIBSBML_EXTERN
 int
-FbcReactionPlugin_isSetGeneProductAssociation(SBasePlugin_t * fbc);
+FbcReactionPlugin_isSetGeneProductAssociation(FbcSBasePlugin_t * fbc);
 
 
 /**
@@ -881,7 +881,7 @@ FbcReactionPlugin_isSetGeneProductAssociation(SBasePlugin_t * fbc);
  */
 LIBSBML_EXTERN
 GeneProductAssociation_t*
-FbcReactionPlugin_getGeneProductAssociation(SBasePlugin_t * fbc);
+FbcReactionPlugin_getGeneProductAssociation(FbcSBasePlugin_t * fbc);
 
 
 /**
@@ -898,7 +898,7 @@ FbcReactionPlugin_getGeneProductAssociation(SBasePlugin_t * fbc);
  */
 LIBSBML_EXTERN
 int
-FbcReactionPlugin_setGeneProductAssociation(SBasePlugin_t * fbc, 
+FbcReactionPlugin_setGeneProductAssociation(FbcSBasePlugin_t * fbc, 
                                             GeneProductAssociation_t* gpa);
 
 
-- 
2.33.1

