diff --git before/src/SConstruct after/src/SConstruct
index abec2115a..d141e7b52 100644
--- before/src/SConstruct
+++ after/src/SConstruct
@@ -10,7 +10,7 @@ if sys.version_info.major == 2:
 sys.path.insert(0,'./framework')
 import bldutil, configure, setenv, rsf.doc
 
-env = Environment()
+env = Environment(ENV = os.environ)
 
 if os.path.isfile('config.py'):
     import config
diff --git before/src/pens/fonts/SConstruct after/src/pens/fonts/SConstruct
index f5860e0fb..9e733a2b1 100644
--- before/src/pens/fonts/SConstruct
+++ after/src/pens/fonts/SConstruct
@@ -4,9 +4,9 @@ import bldutil
 
 try: # distributed version
     Import('env root incdir')
-    env = env.Clone()
+    env = env.Clone(CC=os.environ['HOSTCC'], LIBS='')
 except: # local version
-    env = bldutil.Debug()
+    env = bldutil.Debug(CC=os.environ['HOSTCC'], LIBS='')
     root = None
     
 src = Glob('[a-z]*.vplot_font')
@@ -27,11 +27,8 @@ for font in src:
        
     aux = ' '.join(['%s_%s' % (base,x) for x in attr])
 
-    ldpath = ''
-    for key in list(os.environ.keys()):
-        if key[-15:] == 'LD_LIBRARY_PATH':
-            ldpath += '%s=\'%s\' ' % (key,os.environ[key])
-                          
+    ldpath = ldpath = '%s=\'%s\' ' % ("LD_LIBRARY_PATH", os.environ['libdir'])
+
     env.Command([include,bin],[prog,font],
                 '%s ${SOURCES[0].abspath} %s '
                 '< ${SOURCES[1]} > ${TARGETS[0]} && '
diff --git before/src/trip/iwave/base/lib/utils.cc after/src/trip/iwave/base/lib/utils.cc
index c11205b1d..140fee5f7 100644
--- before/src/trip/iwave/base/lib/utils.cc
+++ after/src/trip/iwave/base/lib/utils.cc
@@ -6,7 +6,7 @@ Igor Terentyev.
 
 #include "utils.h"
 
-#ifndef sun
+#if !defined(sun) && !defined(__aarch64__) && !defined(__arm__)
 #include <mm_malloc.h>
 #endif
 
@@ -37,7 +37,7 @@ static MPI_Comm m_rcm = 0;
 
 void* usermalloc_(size_t size)
 {
-#ifdef sun
+#if defined(sun) || defined(__aarch64__) || defined(__arm__)
     return malloc(size);
 #else
     return _mm_malloc(size, 16);
@@ -47,7 +47,7 @@ void* usermalloc_(size_t size)
 
 void userfree_(void *ptr)
 {
-#ifdef sun
+#if defined(sun) || defined(__aarch64__) || defined(__arm__)
     free(ptr); 
 #else
     _mm_free(ptr);
diff --git before/src/user/cram/SConstruct after/src/user/cram/SConstruct
index 94d9369c3..08d11eebf 100644
--- before/src/user/cram/SConstruct
+++ after/src/user/cram/SConstruct
@@ -27,6 +27,8 @@ except: # local version
 if sys.platform.startswith('linux'):
     env.Prepend(CPPDEFINES=['LINUX'])
 
+env.Prepend(CPPDEFINES=['_GNU_SOURCE'])
+
 src = Glob('[a-z]*.c')
 
 sse = None
diff --git before/src/user/mccowan/SConstruct after/src/user/mccowan/SConstruct
index 182c79615..1c05782b7 100644
--- before/src/user/mccowan/SConstruct
+++ after/src/user/mccowan/SConstruct
@@ -18,6 +18,7 @@ src = Glob('[a-z]*.c')
 
 env.Prepend(CPPPATH=['../../include'],
             LIBPATH=['../../lib'],
+            CPPDEFINES=['_GNU_SOURCE'],
             LIBS=[env.get('DYNLIB','')+'rsf'])
 
 for source in src:
