diff --git a/scip/CMakeLists.txt b/scip/CMakeLists.txt
index f7831d9ce8..54915b438f 100644
--- a/scip/CMakeLists.txt
+++ b/scip/CMakeLists.txt
@@ -125,24 +125,18 @@ unset(BLISS_TARGET)
 if(SYM STREQUAL "bliss")
     message(STATUS "Support SYM: bliss")
     set(sym symmetry/compute_symmetry_bliss.cpp)
-
-    # modify configuration for bliss
-    set(BUILD_SHARED_LIBS OFF)
-    set(TMPFLAGS ${CMAKE_C_FLAGS})
-    set(TMXFLAGS ${CMAKE_CXX_FLAGS})
-    set(CMAKE_C_FLAGS -w)
-    set(CMAKE_CXX_FLAGS -w)
-
-    add_subdirectory(src/bliss EXCLUDE_FROM_ALL)
-
-    # undo modification
-    set(CMAKE_C_FLAGS ${TMPFLAGS})
-    set(CMAKE_CXX_FLAGS ${TMXFLAGS})
-    set(BUILD_SHARED_LIBS ${SHARED})
-
-    set(BLISS_TARGET libbliss)
-    set(SYM_LIBRARIES libbliss)
-    set(SYM_PIC_LIBRARIES libbliss)
+    find_package(BLISS)
+    if(BLISS_FOUND)
+        message(STATUS "BLISS found")
+        include_directories(${BLISS_INCLUDE_DIRS})
+        set(SYM_LIBRARIES ${BLISS_LIBRARIES})
+        set(SYM_PIC_LIBRARIES ${BLISS_LIBRARIES})
+        if(BLISS_DEFINITIONS STREQUAL "")
+            add_compile_options("$<$<COMPILE_LANGUAGE:CXX>:${BLISS_DEFINITIONS}>")
+        endif()
+    else()
+       message(FATAL_ERROR "BLISS not found, not using vendored BLISS")
+    endif()
 elseif(SYM STREQUAL "none")
     message(STATUS "Support SYM: OFF")
     set(sym symmetry/compute_symmetry_none.cpp)
