--- papilo/binaries/CMakeLists.txt	2024-04-03 13:39:12.041041231 +0100
+++ papilo/binaries/CMakeLists.txt	2024-04-03 15:50:40.070943339 +0100
@@ -69,6 +69,17 @@
    set(PAPILO_HAVE_ROUNDINGSAT 0)
 endif()

+# Enable the Fortran language support when the clusol target is included.
+# Even though this was done in the papilo CMake to create the target, it must be
+# done here as well, since the language support is contained to the calling directory
+# and lower, and since the papilo/binaries CMake file can be called from the top-level
+# CMake, which doesn't have Fortran language enabled, Fortran might not be enabled here yet.
+if(TARGET clusol)
+   include(CheckLanguage)
+   check_language(Fortran)
+   enable_language(Fortran)
+endif()
+
 include_directories(${SCIP_INCLUDE_DIRS} ${SOPLEX_INCLUDE_DIRS} ${HIGHS_INCLUDE_DIRS} ${CMAKE_CURRENT_BINARY_DIR})
 find_package(Boost REQUIRED OPTIONAL_COMPONENTS iostreams serialization program_options)

