diff --git a/CMakeLists.txt b/CMakeLists.txt
index 748f412..4d3bffe 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -2,6 +2,8 @@ cmake_minimum_required(VERSION 2.8.12 FATAL_ERROR)
 
 project(ffts C ASM)
 
+set(CMAKE_C_STANDARD 90)
+
 # TODO: to support AutoConfigure building, this should came from "template" file
 set(FFTS_MAJOR 0)
 set(FFTS_MINOR 9)
@@ -530,7 +532,7 @@ if(ENABLE_STATIC)
   install( TARGETS ffts_static DESTINATION ${LIB_INSTALL_DIR} )
 endif(ENABLE_STATIC)
 
-if(ENABLE_STATIC OR ENABLE_SHARED)
+if(FALSE)
   find_path(MPFR_INCLUDES
     NAMES mpfr.h
     PATHS ${INCLUDE_INSTALL_DIR}
@@ -589,7 +591,7 @@ if(ENABLE_STATIC OR ENABLE_SHARED)
   )
 
   set_target_properties(ffts_cpu_test PROPERTIES COMPILE_DEFINITIONS FFTS_BUILDING_CPU_TEST)
-endif(ENABLE_STATIC OR ENABLE_SHARED)
+endif(FALSE)
 
 # generate packageconfig file
 if(UNIX)
