diff --git a/Code/MinimalLib/CMakeLists.txt b/Code/MinimalLib/CMakeLists.txt
index 21cc55d02..5b0efea9d 100644
--- a/Code/MinimalLib/CMakeLists.txt
+++ b/Code/MinimalLib/CMakeLists.txt
@@ -21,30 +21,30 @@ endif(RDK_BUILD_MINIMAL_LIB)
 if(RDK_BUILD_CFFI_LIB)
 
     set(CMAKE_C_STANDARD 99)
-    if((NOT MSVC) OR (MSVC AND RDK_INSTALL_DLLS_MSVC))
-        set(LIBS_TO_USE 
+    if((NOT WIN32) OR (WIN32 AND RDK_INSTALL_DLLS_MSVC))
+        set(LIBS_TO_USE
             MolStandardize_static DistGeomHelpers_static ForceFieldHelpers_static DistGeometry_static
             ForceField_static Alignment_static
-            MolInterchange_static Abbreviations_static CIPLabeler_static 
-            MolDraw2D_static Depictor_static 
-            RDInchiLib_static SubstructMatch_static FileParsers_static 
-            SmilesParse_static GraphMol_static RingDecomposerLib_static RDGeometryLib_static RDGeneral_static )
-        if(RDK_URF_LIBS)
+            MolInterchange_static Abbreviations_static CIPLabeler_static
+            MolDraw2D_static Depictor_static
+            RDInchiLib_static SubstructMatch_static FileParsers_static
+            SmilesParse_static GraphMol_static RDGeometryLib_static RDGeneral_static )
+        if(RDK_USE_URF)
             list(APPEND LIBS_TO_USE RingDecomposerLib_static)
         endif()
     else()
-        set(LIBS_TO_USE 
+        set(LIBS_TO_USE
         MolStandardize DistGeomHelpers ForceFieldHelpers DistGeometry
         ForceField Alignment
-        MolInterchange Abbreviations CIPLabeler 
-        MolDraw2D Depictor 
-        RDInchiLib SubstructMatch FileParsers 
-        SmilesParse GraphMol RingDecomposerLib RDGeometryLib RDGeneral )
-        if(RDK_URF_LIBS)
+        MolInterchange Abbreviations CIPLabeler
+        MolDraw2D Depictor
+        RDInchiLib SubstructMatch FileParsers
+        SmilesParse GraphMol RDGeometryLib RDGeneral )
+        if(RDK_USE_URF)
             list(APPEND LIBS_TO_USE RingDecomposerLib)
         endif()
     endif()
-    
+
     add_library(rdkitcffi SHARED cffiwrapper.cpp)
     target_link_libraries(rdkitcffi PUBLIC rdkit_base)
     target_link_libraries(rdkitcffi PUBLIC ${LIBS_TO_USE})
