diff --git a/Code/MinimalLib/CMakeLists.txt b/Code/MinimalLib/CMakeLists.txt
index 21cc55d02..c6b7fbec1 100644
--- a/Code/MinimalLib/CMakeLists.txt
+++ b/Code/MinimalLib/CMakeLists.txt
@@ -21,8 +21,8 @@ endif(RDK_BUILD_MINIMAL_LIB)
 if(RDK_BUILD_CFFI_LIB)
 
     set(CMAKE_C_STANDARD 99)
-    if((NOT MSVC) OR (MSVC AND RDK_INSTALL_DLLS_MSVC))
-        set(LIBS_TO_USE 
+    if((NOT WIN32) OR (WIN32 AND RDK_INSTALL_DLLS_MSVC))
+        set(LIBS_TO_USE
             MolStandardize_static DistGeomHelpers_static ForceFieldHelpers_static DistGeometry_static
             ForceField_static Alignment_static
             MolInterchange_static Abbreviations_static CIPLabeler_static 
diff --git a/Code/GraphMol/CMakeLists.txt b/Code/GraphMol/CMakeLists.txt
index 417a9b875..7bc62a77c 100644
--- a/Code/GraphMol/CMakeLists.txt
+++ b/Code/GraphMol/CMakeLists.txt
@@ -13,7 +13,7 @@ rdkit_library(GraphMol
 target_compile_definitions(GraphMol PRIVATE RDKIT_GRAPHMOL_BUILD)
 if(RDK_USE_URF)
   target_link_libraries(GraphMol PUBLIC ${RDK_URF_LIBS})
-  if(NOT MSVC AND RDK_INSTALL_STATIC_LIBS)
+  if(NOT WIN32 AND RDK_INSTALL_STATIC_LIBS)
     target_link_libraries(GraphMol_static PUBLIC ${RDK_URF_LIBS}_static)
   endif()
 endif()
