--- a/Makefile
+++ b/Makefile
@@ -102,7 +102,7 @@
 ifeq ($(OS),Darwin) #MAC
 LDFLAGS+= -L/usr/local/cuda/lib -lcuda -lcudart -lcublas -lcurand
 else
-LDFLAGS+= -L/usr/local/cuda/lib64 -lcuda -lcudart -lcublas -lcurand
+LDFLAGS+= -L/usr/local/cuda/lib64 -L/usr/local/cuda/lib64/stubs -lcuda -lcudart -lcublas -lcurand
 endif
 endif
 
@@ -113,7 +113,7 @@
 LDFLAGS+= -L/usr/local/cuda/lib -lcudnn
 else
 CFLAGS+= -DCUDNN -I/usr/local/cudnn/include
-LDFLAGS+= -L/usr/local/cudnn/lib64 -lcudnn
+LDFLAGS+= -L/usr/local/cudnn/lib -lcudnn
 endif
 endif
 
