diff --git a/generator.cpp b/generator.cpp
index 7afeb8e..327c3ad 100644
--- a/generator.cpp
+++ b/generator.cpp
@@ -81,7 +81,7 @@ void generate_opening_book() {
 int main(int argc, char** argv) {
   if(argc > 1) {
     int depth = atoi(argv[1]);
-    char pos_str[depth + 1] = {0};
+    char *pos_str = new char[depth + 1](); // zero initialized
     explore(Position(), pos_str, depth);
   } else generate_opening_book();
 }
