diff --git a/Makefile b/Makefile
index d9a8128..9dc569a 100644
--- a/Makefile
+++ b/Makefile
@@ -1,4 +1,4 @@
-CFLAGS = -Iinclude -std=c++0x -O2 -Wall
+CXXFLAGS = -Iinclude -std=c++0x -O2 -Wall
 LFLAGS = -lm
 SHAREFLAGS = -shared -fPIC
 BUILDDIR = .build
@@ -8,11 +8,12 @@ STATIC = bin/MQLib.a
 SRCS = $(shell find $(SRCDIR) -name "*.cpp")
 OBJS = $(shell echo "$(SRCS)" | sed -e "s/ $(SRCDIR)/ $(BUILDDIR)/g" -e "s/^$(SRCDIR)/$(BUILDDIR)/g" -e "s/\.cpp/.o/g")
 DEPS = $(shell echo "$(OBJS)" | sed -e "s/\.o/.P/g")
+CXX ?= g++
 
 all: $(EXECUTABLE) $(STATIC)
 
 $(EXECUTABLE): $(OBJS)
-	g++ $(LFLAGS) -o $(EXECUTABLE) $(OBJS)
+	$(CXX) $(LFLAGS) -o $(EXECUTABLE) $(OBJS)
 
 $(STATIC): $(OBJS)
 	@type ar >/dev/null 2>&1 || { echo >&2 "ar required for building static library but it's not installed.  Aborting."; exit 1; }
@@ -20,8 +21,8 @@ $(STATIC): $(OBJS)
 
 ### Conversion from .d to .P from http://mad-scientist.net/make/autodep.html
 $(BUILDDIR)/%.o : $(SRCDIR)/%.cpp
-	@type g++ >/dev/null 2>&1 || { echo >&2 "g++ required for compilation but it's not installed.  Aborting."; exit 1; }
-	g++ -c -MD $(CFLAGS) -o $@ $<
+	@type $(CXX) >/dev/null 2>&1 || { echo >&2 "$(CXX) required for compilation but it's not installed.  Aborting."; exit 1; }
+	$(CXX) -c -MD $(CXXFLAGS) -o $@ $<
 	@cp $(BUILDDIR)/$(*).d $(BUILDDIR)/$(*).P; \
 	sed -e 's/#.*//' -e 's/^[^:]*: *//' -e 's/ *\\$$//' -e '/^$$/ d' -e 's/$$/ :/' < $(BUILDDIR)/$(*).d >> $(BUILDDIR)/$(*).P; \
 	rm -f $(BUILDDIR)/$(*).d
