--- a/src/dinf.cpp
+++ b/src/dinf.cpp
@@ -568,7 +568,7 @@
 					elevDEM->getdxdyc(j,tempdxc,tempdyc);
 		        
 					
-					float DXX[3] = {0,tempdxc,tempdyc};//tardemlib.cpp ln 1291
+					float DXX[3] = {0,(float)tempdxc,(float)tempdyc};//tardemlib.cpp ln 1291
 					float DD = sqrt(tempdxc*tempdxc+tempdyc*tempdyc);//tardemlib.cpp ln 1293
 					SET2(j,i,DXX,DD, elevDEM,flowDir,slope);//i=y in function form old code j is x switched on purpose
 					//  Use SET2 from serial code here modified to get what it has as felevg.d from elevDEM partition
@@ -799,7 +799,7 @@
 				//  direction based on the artificial elevations
 
 	elevDEM->getdxdyc(j,tempdxc,tempdyc);
-	float DXX[3] = {0,tempdxc,tempdyc};//tardemlib.cpp ln 1291
+	float DXX[3] = {0,(float)tempdxc,(float)tempdyc};//tardemlib.cpp ln 1291
 	float DD = sqrt(tempdxc*tempdxc+tempdyc*tempdyc);//tardemlib.cpp ln 1293
 
 			SET2(j,i,DXX,DD,elevDEM,elev2,flowDir,dn);	//use new elevations to calculate flowDir.	
