diff --git a/public/libbacktrace/elf.cpp b/public/libbacktrace/elf.cpp
--- a/public/libbacktrace/elf.cpp
+++ b/public/libbacktrace/elf.cpp
@@ -41,6 +41,29 @@
 #include <algorithm>

 #ifdef HAVE_DL_ITERATE_PHDR
+
+#ifndef ElfW
+#  if defined(FREEBSD)
+#    if __ELF_WORD_SIZE == 32
+#      define ElfW(type) Elf32_##type
+#    else
+#      define ElfW(type) Elf64_##type
+#    endif
+#  elif defined(NETBSD) || defined(OPENBSD)
+#    if ELFSIZE == 32
+#      define ElfW(type) Elf32_##type
+#    else
+#      define ElfW(type) Elf64_##type
+#    endif
+#  else
+#    if !defined(ELF_CLASS) || ELF_CLASS == ELFCLASS32
+#      define ElfW(type) Elf32_##type
+#    else
+#      define ElfW(type) Elf64_##type
+#    endif
+#  endif
+#endif
+
 #include <link.h>
 #endif
