diff --git before/qrupdate-ng/CMakeLists.txt after/qrupdate-ng/CMakeLists.txt
index 180a93b..0b25fb6 100644
--- before/qrupdate-ng/CMakeLists.txt
+++ after/qrupdate-ng/CMakeLists.txt
@@ -1,7 +1,7 @@
 CMAKE_MINIMUM_REQUIRED(VERSION 3.1.0)
 CMAKE_POLICY(SET CMP0048 NEW)
 PROJECT(qrupdate-ng VERSION 1.1.5 LANGUAGES Fortran)
-ENABLE_TESTING()
+#ENABLE_TESTING()
 
 # Options
 OPTION(BUILD_SHARED_LIBS "Build shared libraries" ON)
@@ -36,10 +36,10 @@ ENDIF()
 
 
 
-FIND_PACKAGE(BLAS REQUIRED)
+FIND_PACKAGE(OpenBLAS REQUIRED CONFIG)
 MESSAGE(STATUS "BLAS Vendor: ${BLA_VENDOR}")
 IF(NOT BLA_VENDOR STREQUAL "IBMESSL")
-	FIND_PACKAGE(LAPACK REQUIRED)
+	FIND_PACKAGE(LAPACK REQUIRED CONFIG)
 ELSE()
 	SET(LAPACK_MISSING_TARGET_NAME lapack_missing)
 	INCLUDE(${CMAKE_SOURCE_DIR}/lapack-missing/LapackMissing.cmake)
@@ -49,7 +49,7 @@ ENDIF()
 
 
 ADD_SUBDIRECTORY(src)
-ADD_SUBDIRECTORY(test)
+#ADD_SUBDIRECTORY(test)
 
 INCLUDE(CMakePackageConfigHelpers)
 WRITE_BASIC_PACKAGE_VERSION_FILE(
