diff --git before/skia/BUILD.gn after/skia/BUILD.gn
index 33f3208..b7efcf6 100644
--- before/skia/BUILD.gn
+++ after/skia/BUILD.gn
@@ -173,7 +173,7 @@ opts("hsw") {
   if (is_win) {
     cflags = [ "/arch:AVX2" ]
   } else {
-    cflags = [ "-march=haswell" ]
+    cflags = [ "-mavx2" ]
   }
 }
 
@@ -183,7 +183,7 @@ opts("skx") {
   if (is_win) {
     cflags = [ "/arch:AVX512" ]
   } else {
-    cflags = [ "-march=skylake-avx512" ]
+    cflags = [ "-mavx512" ]
   }
 }
 
diff --git before/skia/gn/core.gni after/skia/gn/core.gni
index 06a7bb1..0323fe9 100644
--- before/skia/gn/core.gni
+++ after/skia/gn/core.gni
@@ -187,6 +187,8 @@ skia_core_sources = [
   "$_include/private/chromium/SkDiscardableMemory.h",
   "$_include/private/chromium/SkPMColor.h",
   "$_include/private/chromium/Slug.h",
+ "$_src/base/sk_capi.cpp",
+  "$_src/base/sk_capi.h",
   "$_src/base/SkArenaAlloc.cpp",
   "$_src/base/SkArenaAlloc.h",
   "$_src/base/SkArenaAllocList.h",
@@ -847,3 +849,4 @@ skia_no_slug_srcs = []
 skia_core_sources += skia_pathops_sources
 
 skia_core_public += skia_pathops_public
+
