diff --git a/CMakeLists.txt b/CMakeLists.txt
index 252b8bb..1437a6f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -33,8 +33,10 @@ find_package(PkgConfig REQUIRED)
 pkg_check_modules(GIVARO REQUIRED IMPORTED_TARGET givaro)
 pkg_check_modules(FFLAS_FFPACK REQUIRED IMPORTED_TARGET fflas-ffpack)
 
-# set(CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/config;${CMAKE_MODULE_PATH}")
-# find_package(PAPI)
+set(CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/config;${CMAKE_MODULE_PATH}")
+find_package(argp)
+include_directories(${ARGP_INCLUDE_PATH})
+target_link_libraries(${ARGP_LIBRARIES})
 
 if (build_type STREQUAL cov)
     add_compile_options(-Wall -O0 --coverage -Wno-pragmas -fopenmp)
