diff -u givaro.orig/macros/gmp-check.m4 givaro/macros/gmp-check.m4
--- givaro.orig/macros/gmp-check.m4	2023-12-21 13:31:35
+++ givaro/macros/gmp-check.m4	2024-08-04 15:03:36
@@ -105,21 +105,6 @@
 		exit 1
 	])
 
-	AC_MSG_CHECKING([whether gmp version is at least $min_gmp_release])
-	AC_TRY_RUN(
-		[ 
-			#include <cstddef>
-			#include <gmp.h>
-			int main () {
-				return (__GNU_MP_RELEASE < $min_gmp_release);
-			}
-		],
-		[ AC_MSG_RESULT(yes)
-		],
-		[ AC_MSG_RESULT(no)
-		  AC_MSG_ERROR(your GMP is too old. GMP release >= $min_gmp_release needed)
-		  exit 1]
-	)
 	AC_LANG_POP([C++])
 	
 	AC_SUBST(GMP_CFLAGS)
