From 4b335e8f6de21d8b79646c1415594531d930768a Mon Sep 17 00:00:00 2001
From: Max Horn <max@quendi.de>
Date: Fri, 15 Oct 2021 12:47:54 +0200
Subject: [PATCH] Makefile.subdirs: don't assign ; to DEPFLAGS[123] /
 QUIET_{CC,CXX}

---
 Makefile.subdirs | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/Makefile.subdirs b/Makefile.subdirs
index 0fbc73364..0e811cd1c 100644
--- a/Makefile.subdirs
+++ b/Makefile.subdirs
@@ -2,8 +2,8 @@ ifeq ($(strip $(VERBOSE)),)
 	QUIET_CC  = @echo '   ' CC  ' ' $@;
 	QUIET_CXX = @echo '   ' CXX ' ' $@;
 else
-	QUIET_CC  = ;
-	QUIET_CXX = ;
+	QUIET_CC  =
+	QUIET_CXX =
 endif
 
 ifeq ($(WANTDEPS), 1)
@@ -11,9 +11,9 @@ ifeq ($(WANTDEPS), 1)
 	DEPFLAGS2 = -MMD -MP -MF "$(BUILD_DIR)/$(MOD_DIR)_$*.d" -MT "$(BUILD_DIR)/$(MOD_DIR)_$*.d" -MT "$@";
 	DEPFLAGS3 = -MMD -MP -MF "$(BUILD_DIR)/$*.d" -MT "$(BUILD_DIR)/$*.d" -MT "$@";
 else
-    DEPFLAGS1 = ; 
-    DEPFLAGS2 = ;
-    DEPFLAGS3 = ;
+	DEPFLAGS1 =
+	DEPFLAGS2 =
+	DEPFLAGS3 =
 endif
 
 AT=@
-- 
2.33.0

