diff --git a/pdal/DynamicLibrary.cpp b/pdal/DynamicLibrary.cpp
index eccbc2a3a..c664f6854 100755
--- a/pdal/DynamicLibrary.cpp
+++ b/pdal/DynamicLibrary.cpp
@@ -117,7 +117,7 @@ void *DynamicLibrary::getSymbol(const std::string& symbol)
 
     void *sym;
 #ifdef _WIN32
-    sym = ::GetProcAddress((HMODULE)m_handle, symbol.c_str());
+    sym = (void *)::GetProcAddress((HMODULE)m_handle, symbol.c_str());
 #else
     sym = ::dlsym(m_handle, symbol.c_str());
 #endif
