diff --git a/cmake/macros.cmake b/cmake/macros.cmake
index 3028823..40c0ae1 100644
--- a/cmake/macros.cmake
+++ b/cmake/macros.cmake
@@ -109,11 +109,7 @@ macro(PDAL_ADD_PLUGIN _name _type _shortname)
     set(multiValueArgs FILES LINK_WITH INCLUDES SYSTEM_INCLUDES)
     cmake_parse_arguments(PDAL_ADD_PLUGIN "${options}" "${oneValueArgs}"
         "${multiValueArgs}" ${ARGN})
-    if(WIN32)
-        set(${_name} "libpdal_plugin_${_type}_${_shortname}")
-    else()
-        set(${_name} "pdal_plugin_${_type}_${_shortname}")
-    endif()
+    set(${_name} "pdal_plugin_${_type}_${_shortname}")
 
     if (WIN32)
         list(APPEND ${PDAL_ADD_PLUGIN_FILES} ${PDAL_TARGET_OBJECTS})
