diff --git a/vendor/arbiter/CMakeLists.txt b/vendor/arbiter/CMakeLists.txt
index f7f60ff22..1e129d264 100644
--- a/vendor/arbiter/CMakeLists.txt
+++ b/vendor/arbiter/CMakeLists.txt
@@ -7,6 +7,8 @@ if (EXISTS)
     return()
 endif()
 
+
+
 PDAL_ADD_FREE_LIBRARY(${PDAL_ARBITER_LIB_NAME} STATIC arbiter.cpp)
 target_include_directories(${PDAL_ARBITER_LIB_NAME}
     PRIVATE
@@ -15,13 +17,14 @@ target_include_directories(${PDAL_ARBITER_LIB_NAME}
 target_link_libraries(${PDAL_ARBITER_LIB_NAME}
     PRIVATE
         ${CURL_LIBRARIES}
+        -lshlwapi
 )
 target_compile_definitions(${PDAL_ARBITER_LIB_NAME}
     PRIVATE
         ARBITER_CURL)
 
 if (MSVC)
-    target_link_libraries(${PDAL_ARBITER_LIB_NAME} PRIVATE Shlwapi.lib)
+    target_link_libraries(${PDAL_ARBITER_LIB_NAME} PRIVATE shlwapi.lib)
     target_compile_definitions(${PDAL_ARBITER_LIB_NAME}
         PRIVATE
             ARBITER_DLL_EXPORT)
@@ -37,6 +40,10 @@ if (ZLIB_FOUND)
     target_compile_definitions(${PDAL_ARBITER_LIB_NAME}
         PRIVATE
             ARBITER_ZLIB)
+    target_link_libraries(
+            ${PDAL_ARBITER_LIB_NAME}
+            PRIVATE
+            ${ZLIB_LIBRARIES})
 endif()
 if (OPENSSL_FOUND)
     target_include_directories(${PDAL_ARBITER_LIB_NAME}
diff --git a/vendor/arbiter/arbiter.cpp b/vendor/arbiter/arbiter.cpp
index 5ad2affa3..6389715ef 100644
--- a/vendor/arbiter/arbiter.cpp
+++ b/vendor/arbiter/arbiter.cpp
@@ -4816,7 +4816,7 @@ std::unique_ptr<std::string> env(const std::string& var)
 {
     std::unique_ptr<std::string> result;
 
-#ifndef ARBITER_WINDOWS
+#ifndef _MSC_VER
     if (const char* c = getenv(var.c_str())) result.reset(new std::string(c));
 #else
     char* c(nullptr);
diff --git a/vendor/arbiter/arbiter.hpp b/vendor/arbiter/arbiter.hpp
index 90e55283f..b9a78ede2 100644
--- a/vendor/arbiter/arbiter.hpp
+++ b/vendor/arbiter/arbiter.hpp
@@ -3130,10 +3130,15 @@ inline std::string decompress(const char* data, std::size_t size)
 
 #pragma once
 
-#if defined(_WIN32) || defined(WIN32) || defined(_MSC_VER)
+#if defined(_WIN32) || defined(WIN32) || defined(_MSC_VER) || defined \
+    (__MINGW32__)
 #define ARBITER_WINDOWS
 #endif
 
+#ifdef __MINGW32__
+#include <sec_api/stdlib_s.h>
+#endif
+
 #ifndef ARBITER_DLL
 #if defined(ARBITER_WINDOWS)
 #if defined(ARBITER_DLL_EXPORT)

