diff --git a/filters/GreedyProjection.cpp b/filters/GreedyProjection.cpp
index 6f18700fc..ce3fa8cff 100644
--- a/filters/GreedyProjection.cpp
+++ b/filters/GreedyProjection.cpp
@@ -42,6 +42,10 @@
 
 #include "GreedyProjection.hpp"
 
+#ifndef M_PIl
+#define M_PIl M_PI
+#endif
+
 namespace pdal
 {
 
@@ -69,7 +73,7 @@ void GreedyProjection::addArgs(ProgramArgs& args)
     args.add("num_neighbors", "Number of nearest neighbors to consider",
         nnn_, 100);
     args.add("min_angle", "Minimum angle for created triangles",
-        minimum_angle_, M_PI / 18);  // 10 degrees default
+        minimum_angle_, M_PIl / 18);  // 10 degrees default
     args.add("max_angle", "Maximum angle for created triangles",
         maximum_angle_, 2 * M_PI / 3);  // 120 degrees default
     args.add("eps_angle", "Max normal difference angle for triangulation "
diff --git a/filters/GreedyProjection.hpp b/filters/GreedyProjection.hpp
index ae27712fe..a1c7fa917 100644
--- a/filters/GreedyProjection.hpp
+++ b/filters/GreedyProjection.hpp
@@ -39,9 +39,13 @@
 
 #pragma once
 
-// This is for M_PI on Windows.
+// This is for M_PI on Windows
 #define _USE_MATH_DEFINES
-#include <math.h>
+#include <cmath>
+
+//#define M_PI            3.14159265358979323846
+//#define M_PIl M_PI
+
 
 #include <fstream>
 #include <iostream>
