diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1496d7f97..e9d4d2f0b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -11,6 +11,11 @@ string(TOLOWER ${PROJECT_NAME} PROJECT_NAME_LOWER)
 set(ROOT_DIR "${PROJECT_SOURCE_DIR}")
 include(${ROOT_DIR}/cmake/common.cmake NO_POLICY_SCOPE)
 
+if (MINGW)
+    list(APPEND CMAKE_CXX_IMPLICIT_INCLUDE_DIRECTORIES /usr/x86_64-w64-mingw32/sys-root/mingw/include)
+    add_definitions(-D_USE_MATH_DEFINES)
+endif()
+
 #------------------------------------------------------------------------------
 # internal cmake settings
 #------------------------------------------------------------------------------
@@ -290,6 +290,8 @@ target_link_libraries(${PDAL_BASE_LIB_NAME}
         ${PDAL_ARBITER_LIB_NAME}
         ${PDAL_KAZHDAN_LIB_NAME}
         ${LAZPERF_LIBRARY}
+        -lpsapi
+        -lshlwapi
     INTERFACE
         ${PDAL_LIBDIR}
 )
@@ -312,7 +319,7 @@ endif()
 # specifies linking in libpdal_base.so and libpdal_util.so.  This allows
 # users to link a single library, libpdalcpp
 #
-if (UNIX AND NOT APPLE)
+if ((UNIX OR MINGW) AND NOT APPLE)
     set(LIBNAME ${CMAKE_SHARED_LIBRARY_PREFIX}${PDAL_LIB_NAME})
     install(FILES ${LIBNAME} DESTINATION ${PDAL_LIB_INSTALL_DIR}
         RENAME ${LIBNAME}${CMAKE_SHARED_LIBRARY_SUFFIX})
