diff --git before/cmake/dimension.cmake after/cmake/dimension.cmake
index ebb7222..6554f30 100644
--- before/cmake/dimension.cmake
+++ after/cmake/dimension.cmake
@@ -5,6 +5,6 @@
 set(DIMENSION_INFILE ${PDAL_SRC_DIR}/Dimension.json)
 set(DIMENSION_OUTFILE ${CMAKE_CURRENT_BINARY_DIR}/include/pdal/Dimension.hpp)
 add_custom_command(OUTPUT ${DIMENSION_OUTFILE}
-    COMMAND dimbuilder ${DIMENSION_INFILE} ${DIMENSION_OUTFILE}
+    COMMAND ${PDAL_OUTPUT_BIN_DIR}/dimbuilder ${DIMENSION_INFILE} ${DIMENSION_OUTFILE}
         DEPENDS ${DIMENSION_INFILE} dimbuilder)
 add_custom_target(generate_dimension_hpp DEPENDS ${DIMENSION_OUTFILE})
