--- a/root/core/unix/src/TUnixSystem.cxx	2024-11-14 10:27:26.000000000 +0100
+++ b/root/core/unix/src/TUnixSystem.cxx	2026-01-04 02:08:46.383417890 +0100
@@ -604,17 +604,19 @@
    fSignals    = new TFdSet;
 
    //--- install default handlers
+#ifndef ONLY_SIGALARM_ROOT_HANDLER
    UnixSignal(kSigChild,                 SigHandler);
    UnixSignal(kSigBus,                   SigHandler);
    UnixSignal(kSigSegmentationViolation, SigHandler);
    UnixSignal(kSigIllegalInstruction,    SigHandler);
    UnixSignal(kSigAbort,                 SigHandler);
    UnixSignal(kSigSystem,                SigHandler);
-   UnixSignal(kSigAlarm,                 SigHandler);
    UnixSignal(kSigUrgent,                SigHandler);
    UnixSignal(kSigFloatingException,     SigHandler);
    UnixSignal(kSigWindowChanged,         SigHandler);
    UnixSignal(kSigUser2,                 SigHandler);
+#endif //ONLY_SIGALARM_ROOT_HANDLER
+   UnixSignal(kSigAlarm,                 SigHandler);
 
 #if defined(R__MACOSX)
    // trap loading of all dylibs to register dylib name,
