--- a/opt/bin/x86_64-linux-gnu-libgfortran5-cxx11/x86_64-linux-gnu-g++
+++ b/opt/bin/x86_64-linux-gnu-libgfortran5-cxx11/x86_64-linux-gnu-g++
@@ -27,8 +27,7 @@
 
 export LD_LIBRARY_PATH="/usr/local/lib64:/usr/local/lib:/usr/lib64:/usr/lib:/lib64:/lib:/opt/x86_64-linux-musl/x86_64-linux-musl/lib64:/opt/x86_64-linux-musl/x86_64-linux-musl/lib:/opt/x86_64-linux-gnu/x86_64-linux-gnu/lib64:/opt/x86_64-linux-gnu/x86_64-linux-gnu/lib"
 if [[ " ${ARGS[@]} " == *"-march="* ]]; then
-    echo "BinaryBuilder: Cannot force an architecture via -march" >&2
-    exit 1
+    echo "BinaryBuilder: warning architecture forced via -march" >&2
 fi
 
 if [[ "${ARGS[@]}" =~ "-Ofast"|"-ffast-math"|"-funsafe-math-optimizations" ]]; then
