--- a/Makefile
+++ b/Makefile
@@ -584,16 +584,14 @@ raudio.o : raudio.c raylib.h
 raygui.o : raygui.c
 	$(CC) -c $< $(CFLAGS) $(INCLUDE_PATHS) -D$(PLATFORM) -DRAYGUI_IMPLEMENTATION
 raygui.c:
-	echo #define RAYGUI_IMPLEMENTATION > raygui.c
-	echo #include "$(RAYLIB_MODULE_RAYGUI_PATH)/raygui.h" >> raygui.c
+	@echo '#include "extras/raygui.h"' > raygui.c
 
 # Compile physac module
 # NOTE: physac header should be distributed with raylib.h
 physac.o : physac.c
 	$(CC) -c $< $(CFLAGS) $(INCLUDE_PATHS) -D$(PLATFORM) -DPHYSAC_IMPLEMENTATION
 physac.c:
-	@echo #define PHYSAC_IMPLEMENTATION > physac.c
-	@echo #include "$(RAYLIB_MODULE_PHYSAC_PATH)/physac.h" >> physac.c
+	@echo '#include "extras/physac.h"' >> physac.c
 
 # Compile android_native_app_glue module
 android_native_app_glue.o : $(NATIVE_APP_GLUE)/android_native_app_glue.c
@@ -643,6 +641,7 @@ ifeq ($(ROOT),root)
 		cp --update raymath.h $(RAYLIB_H_INSTALL_PATH)/raymath.h
 		cp --update rlgl.h $(RAYLIB_H_INSTALL_PATH)/rlgl.h
 		cp --update extras/physac.h $(RAYLIB_H_INSTALL_PATH)/physac.h
+		cp --update extras/raygui.h $(RAYLIB_H_INSTALL_PATH)/raygui.h
 		@echo "raylib development files installed/updated!"
     else
 		@echo "This function currently works on GNU/Linux systems. Add yours today (^;"
@@ -671,6 +671,7 @@ ifeq ($(ROOT),root)
 		rm --force --interactive --verbose $(RAYLIB_H_INSTALL_PATH)/raymath.h
 		rm --force --interactive --verbose $(RAYLIB_H_INSTALL_PATH)/rlgl.h
 		rm --force --interactive --verbose $(RAYLIB_H_INSTALL_PATH)/physac.h
+		rm --force --interactive --verbose $(RAYLIB_H_INSTALL_PATH)/raygui.h
 		@echo "raylib development files removed!"
         else
 		@echo "This function currently works on GNU/Linux systems. Add yours today (^;"
