--- a/src/Makefile
+++ b/src/Makefile
@@ -383,9 +383,11 @@ ifeq ($(PLATFORM),PLATFORM_DESKTOP)
         LDFLAGS += -L. -Lsrc -L/usr/local/lib -L$(RAYLIB_RELEASE_PATH)
     endif
     ifeq ($(USE_EXTERNAL_GLFW),TRUE)
-        # Check the version name. If GLFW3 was built manually, it may have produced
-        # a static library known as libglfw3.a. In that case, the name should be -lglfw3
-        LDFLAGS += -lglfw
+	ifeq ($(OS),Windows_NT)
+	        LDFLAGS += -L$(libdir) -lglfw3
+	else
+		LDFLAGS += -lglfw
+	endif
     endif
 endif
 
