--- /dev/null
+++ b/cmake/modules/FindSCOTCH.cmake
@@ -344,17 +344,18 @@
     return 1;
 }
 ")
-check_c_source_runs("${SCOTCH_C_TEST_SCOTCH_Num_4}" SCOTCH_Num_4)
-if(NOT SCOTCH_Num_4)
-  check_c_source_runs("${SCOTCH_C_TEST_SCOTCH_Num_8}" SCOTCH_Num_8)
-  if(NOT SCOTCH_Num_8)
-    set(SCOTCH_INTSIZE -1)
-  else()
-    set(SCOTCH_INTSIZE 8)
-  endif()
-else()
-  set(SCOTCH_INTSIZE 4)
-endif()
+#check_c_source_runs("${SCOTCH_C_TEST_SCOTCH_Num_4}" SCOTCH_Num_4)
+#if(NOT SCOTCH_Num_4)
+#  check_c_source_runs("${SCOTCH_C_TEST_SCOTCH_Num_8}" SCOTCH_Num_8)
+#  if(NOT SCOTCH_Num_8)
+#    set(SCOTCH_INTSIZE -1)
+#  else()
+#    set(SCOTCH_INTSIZE 8)
+#  endif()
+#else()
+#  set(SCOTCH_INTSIZE 4)
+#endif()
+set(SCOTCH_INTSIZE 8) # SCOTCH is always built with 64-bit integers
 set(CMAKE_REQUIRED_INCLUDES "")
 
 # check that SCOTCH has been found
