--- /dev/null
+++ b/cmake/modules/FindElas.cmake
@@ -39,15 +39,25 @@ FIND_PATH(ELAS_INCLUDE_DIR
 
 # Check for elas library (search in common locations in order to prevent
 # modifications in ELAS installation)
-FIND_LIBRARY(ELAS_LIBRARY
-  NAMES Elas Elas${ELAS_LIB_SUFFIX}
+FIND_FILE(ELAS_LIBRARY
+  NAMES libElas.dll
   HINTS ${ELAS_LIBRARY}
   $ENV{ELAS_LIBRARY}
   $ENV{HOME}/lib
   ${ELAS_DIR}/lib
   $ENV{ELAS_DIR}/lib
+  PATH_SUFFIXES bin ../bin lib ../lib
   DOC "The ELAS library"
   )
+#FIND_LIBRARY(ELAS_LIBRARY
+#  NAMES Elas Elas${ELAS_LIB_SUFFIX}
+#  HINTS ${ELAS_LIBRARY}
+#  $ENV{ELAS_LIBRARY}
+#  $ENV{HOME}/lib
+#  ${ELAS_DIR}/lib
+#  $ENV{ELAS_DIR}/lib
+#  DOC "The ELAS library"
+#  )
 
 INCLUDE(FindPackageHandleStandardArgs)
 FIND_PACKAGE_HANDLE_STANDARD_ARGS(ELAS DEFAULT_MSG
