diff --git i/cmake/base-config-ix.cmake w/cmake/base-config-ix.cmake
index d92bc0e71fa1..d9a45a49ac5e 100644
--- i/cmake/base-config-ix.cmake
+++ w/cmake/base-config-ix.cmake
@@ -284,9 +284,9 @@ macro(test_targets)
     elseif("${COMPILER_RT_DEFAULT_TARGET_ARCH}" MATCHES "avr")
       test_target_arch(avr "__AVR__" "--target=avr")
     elseif("${COMPILER_RT_DEFAULT_TARGET_ARCH}" MATCHES "aarch32")
-      test_target_arch(aarch32 "" "-march=armv8-a")
+      test_target_arch(aarch32 "" "")
     elseif("${COMPILER_RT_DEFAULT_TARGET_ARCH}" MATCHES "aarch64")
-      test_target_arch(aarch64 "" "-march=armv8-a")
+      test_target_arch(aarch64 "" "")
     elseif("${COMPILER_RT_DEFAULT_TARGET_ARCH}" MATCHES "riscv32")
       test_target_arch(riscv32 "" "")
     elseif("${COMPILER_RT_DEFAULT_TARGET_ARCH}" MATCHES "riscv64")
