From feb5b50dee8995c3d3e1e20fbf45913ddf9131ec Mon Sep 17 00:00:00 2001
From: Valentin Churavy <v.churavy@gmail.com>
Date: Tue, 7 Feb 2023 15:16:23 -0500
Subject: [PATCH] Disable NVTX on Linux for now

---
 CMakeLists.txt | 8 ++------
 1 file changed, 2 insertions(+), 6 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2503fb9..37a938d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -160,10 +160,6 @@ FIND_PACKAGE(MPI)
 
 message ("This is a MPI build:" ${MPI_FOUND})
 
-# Enable NVTX ranges on Linux
-if(NOT WIN32)
-  set(NVTXRANGE_FLAG "-DNVTX_RANGES;")
-endif()
 
 # Configuration specific nvcc flags
 GET_FILENAME_COMPONENT(CMAKE_CXX_COMPILER_NAME "${CMAKE_CXX_COMPILER}" NAME)
@@ -362,8 +358,8 @@ IF (WIN32)
   target_link_libraries( amgx CUDA::cublas CUDA::cusparse CUDA::cusolver)
   target_link_libraries( amgxsh CUDA::cublas CUDA::cusparse CUDA::cusolver)
 ELSE (WIN32)
-  target_link_libraries( amgx CUDA::cublas CUDA::cusparse CUDA::cusolver CUDA::nvToolsExt m pthread)
-  target_link_libraries( amgxsh CUDA::cublas CUDA::cusparse CUDA::cusolver CUDA::nvToolsExt m pthread)
+  target_link_libraries( amgx CUDA::cublas CUDA::cusparse CUDA::cusolver m pthread)
+  target_link_libraries( amgxsh CUDA::cublas CUDA::cusparse CUDA::cusolver  m pthread)
 ENDIF(WIN32)
 
 if(MPI_FOUND)
-- 
2.37.2

