Author: Andreas Beckmann <anbe@debian.org>
Description: tune distro mode
Forwarded: not-needed

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1420,6 +1420,9 @@

 set(KERNELLIB_HOST_DISTRO_VARIANTS 0)
 if(KERNELLIB_HOST_CPU_VARIANTS STREQUAL "distro")
+  if("${LLC_HOST_CPU}" STREQUAL "GENERIC")
+    set(HOST_CPU_FORCED 0 CACHE INTERNAL "CPU is forced by user")
+  endif()
   if(HOST_CPU_FORCED)
     message(FATAL_ERROR "Cannot build with CPU autodetection distro variants build, and enforce LLC_HOST_CPU at the same time. Please pick one")
   endif()
@@ -1430,8 +1433,13 @@
     endif()
   elseif(POWERPC64LE)
     set(KERNELLIB_HOST_CPU_VARIANTS pwr8 pwr9 pwr10)
-  else()
+  elseif("${LLC_HOST_CPU}" STREQUAL "GENERIC")
+    set(KERNELLIB_HOST_CPU_VARIANTS "")
+  else()
     message(FATAL_ERROR "Don't know what CPU variants to use for kernel library on this platform.")
+  endif()
+  if("${LLC_HOST_CPU}" STREQUAL "GENERIC")
+    list(INSERT KERNELLIB_HOST_CPU_VARIANTS 0 ${LLC_HOST_CPU})
   endif()
   set(KERNELLIB_HOST_DISTRO_VARIANTS 1)
 endif()

