diff --git a/src/makefile_gfortran b/src/makefile_gfortran
index f08f9bd..bb83caa 100644
--- a/src/makefile_gfortran
+++ b/src/makefile_gfortran
@@ -37,6 +37,8 @@ FLEXPART-ETA-BIN = FLEXPART_ETA_BIN
 
 export OMP_NESTED=FALSE
 
+CPATH = ${includedir}
+LIBRARY_PATH = ${libdir}
 ROOT_DIR = . #MD
 
 #F90 = /usr/bin/gfortran
@@ -87,13 +89,13 @@ O_LEV_DBG = g # [0,g]
 # (defined by the machine where you are compiling it)
 #
 ifeq ($(arch), generic)
-	FUSER = -g -fopenmp -march=core-avx2
+	FUSER = -fopenmp
 else ifeq ($(arch), x86-64)
-	FUSER = -g -fopenmp -march=x86-64
+	FUSER = -fopenmp
 else ifeq ($(arch), skylake)
-	FUSER = -g -fopenmp -march=skylake-avx512
+	FUSER = -fopenmp
 else
-	FUSER = -g -fopenmp -march=native -mtune=native
+	FUSER = -fopenmp
 endif
 
 # To compile in serial
@@ -103,7 +105,7 @@ mpi: FUSER := $(FUSER) -Dusempi #-MM -MT -MD
 ## LIBRARIES
 LIBS = -leccodes -leccodes_f90 -lm $(NCOPT) $(ETAOPT) #MD -ljasper
 
-FFLAGS   = $(INC) -O$(O_LEV) -m64 -cpp -mcmodel=large $(NCOPT) $(ETAOPT) $(FUSER)  #-convert little_endian -fpp -WB -check all
+FFLAGS   = $(INC) -O$(O_LEV) -cpp -mcmodel=large $(NCOPT) $(ETAOPT) $(FUSER)  #-convert little_endian -fpp -WB -check all
 
 DBGFLAGS = $(INC) -O$(O_LEV_DBG) -g3 -ggdb3 -cpp -m64 -mcmodel=large -fconvert=little-endian -frecord-marker=4 -fmessage-length=0 -flto=jobserver -O$(O_LEV_DBG) $(NCOPT) $(ETAOPT) -fbacktrace   -Wall  -fdump-core $(FUSER)  #  -ffpe-trap=invalid,overflow,denormal,underflow,zero  -Warray-bounds -fcheck=all
 
