diff --git a/Source/Fortran/makefile_fast b/Source/Fortran/makefile_fast
index 7b0a215..7e17b21 100644
--- a/Source/Fortran/makefile_fast
+++ b/Source/Fortran/makefile_fast
@@ -18,7 +18,7 @@ ECCODES_LIB =  -Bstatic -leccodes_f90 -leccodes -Bdynamic -lm
 EMOSLIB=-lemosR64
 LIB =  $(ECCODES_LIB) $(EMOSLIB)
 
-ECCODES_INCLUDE_DIR=/usr/lib/x86_64-linux-gnu/fortran/gfortran-mod-15
+ECCODES_INCLUDE_DIR=${includedir}
 #/usr/local/include/ #oldstable
 
 INC = -I. -I$(ECCODES_INCLUDE_DIR)
