diff --color -ur flexpart-orig/src/makefile flexpart/src/makefile
--- flexpart-orig/src/makefile	2024-06-28 12:00:25.488456542 +0200
+++ flexpart/src/makefile	2024-06-28 12:08:25.775123497 +0200
@@ -91,13 +91,9 @@
 	F90	  = gfortran
 	MPIF90    = mpifort
 
-	INCPATH1  = /usr/include/
+	INCPATH1  = ${includedir}
 	INCPATH2  = ./
-	LIBPATH1  = /usr/lib/
-
-	INCPATH2  = /opt/eccodes/include/
-#	INCPATH2  = ./
-	LIBPATH2  = /opt/eccodes/lib/
+	LIBPATH1  = ${libdir}
 
 # the following works on a Debian buster (and probably newer) machine
 #	F90	  = gfortran
@@ -128,9 +124,7 @@
 #LIBS = -lgrib_api_f90 -lgrib_api -lm -ljasper $(NCOPT)
 LIBS = -leccodes_f90 -leccodes -lm -ljasper $(NCOPT)
 
-FFLAGS   = -I$(INCPATH1) -I$(INCPATH2) -O$(O_LEV) -g -cpp -m64 -mcmodel=medium \
-           -fconvert=little-endian -frecord-marker=4 -fmessage-length=0 \
-           -flto=jobserver -O$(O_LEV) $(NCOPT) $(FUSER)  #-Warray-bounds -fcheck=all # -march=native
+FFLAGS   = -I$(INCPATH1) -O$(O_LEV) -cpp -mcmodel=large -fconvert=little-endian -frecord-marker=4 -fmessage-length=0 -O$(O_LEV) $(NCOPT) $(FUSER)
 
 DBGFLAGS = -I$(INCPATH1) -I$(INCPATH2) -O$(O_LEV_DBG) -g3 -ggdb3 -cpp \
            -m64 -mcmodel=medium -fconvert=little-endian -frecord-marker=4  \
@@ -138,7 +134,7 @@
            -fbacktrace   -Wall  -fdump-core $(FUSER) -fcheck=all #\
             #  -ffpe-trap=invalid,overflow,denormal,underflow,zero 
 
-LDFLAGS  = $(FFLAGS) -L$(LIBPATH1) -L$(LIBPATH2) -Wl,-rpath,$(LIBPATH1) $(LIBS)
+LDFLAGS  = $(FFLAGS) -L$(LIBPATH1) -Wl,-rpath,$(LIBPATH1) $(LIBS)
 LDDEBUG  = $(DBGFLAGS) -L$(LIBPATH1) $(LIBS)
 
 MODOBJS = \
