--- a/Makefile
+++ b/Makefile
@@ -11,7 +11,7 @@
 #
 # CFLAGS settings for 64 bit Linux/unix systems.
 #
-export CFLAGS=-m64 -march=native -mtune=native -fopenmp -ansi -Wall -DBIT64 -DUSEOPENMP -DSETNUMTHREADS -DUSEGETTIME -I../include
+export CFLAGS=-march=native -mtune=native -fopenmp -ansi -Wall -DUSEOPENMP -DSETNUMTHREADS -DUSEGETTIME -I../include
 #
 # LIBS settings for 64 bit Linux/unix systems.
 #
