diff --git a/src/node_credentials.cc b/src/node_credentials.cc
index 115c2fc877..946f39f7b4 100644
--- a/src/node_credentials.cc
+++ b/src/node_credentials.cc
@@ -13,7 +13,13 @@
 #endif
 #ifdef __linux__
 #include <linux/capability.h>
-#include <sys/auxv.h>
+#include <elf.h>
+#ifdef __LP64__
+#define Elf_auxv_t Elf64_auxv_t
+#else
+#define Elf_auxv_t Elf32_auxv_t
+#endif  // __LP64__
+extern char** environ;
 #include <sys/syscall.h>
 #endif  // __linux__
 
@@ -38,8 +44,15 @@ bool linux_at_secure() {
   // and returns the correct value,  e.g. even in static
   // initialization code in other files.
 #ifdef __linux__
-  static const bool value = getauxval(AT_SECURE);
-  return value;
+  char** envp = environ;
+  while (*envp++ != nullptr) {}
+  Elf_auxv_t* auxv = reinterpret_cast<Elf_auxv_t*>(envp);
+  for (; auxv->a_type != AT_NULL; auxv++) {
+    if (auxv->a_type == AT_SECURE) {
+      static const bool value = auxv->a_un.a_val;
+      return value;
+    }
+  }
 #else
   return false;
 #endif
