diff --git a/test/cctest/test_crypto_clienthello.cc b/test/cctest/test_crypto_clienthello.cc
index 60a8e294c8..e673073dc3 100644
--- a/test/cctest/test_crypto_clienthello.cc
+++ b/test/cctest/test_crypto_clienthello.cc
@@ -16,6 +16,7 @@
 // safety is violated.
 #if !defined(NO_GUARD_PAGE)
 #ifdef __linux__
+#include <malloc.h>
 #include <sys/mman.h>
 #include <unistd.h>
 #if defined(_SC_PAGE_SIZE) && defined(PROT_NONE) && defined(PROT_READ) &&      \
@@ -54,7 +55,7 @@ class OverrunGuardedBuffer {
 #ifdef USE_MPROTECT
     // Place the packet right before a guard page, which, when accessed, causes
     // a segmentation fault.
-    alloc_base = static_cast<uint8_t*>(aligned_alloc(page, 2 * page));
+    alloc_base = static_cast<uint8_t*>(memalign(page, 2 * page));
     EXPECT_NE(alloc_base, nullptr);
     uint8_t* second_page = alloc_base + page;
     EXPECT_EQ(mprotect(second_page, page, PROT_NONE), 0);
