diff --git a/include/cpp-statsd-client/UDPSender.hpp b/include/cpp-statsd-client/UDPSender.hpp
index c7d667a..574b4c9 100644
--- a/include/cpp-statsd-client/UDPSender.hpp
+++ b/include/cpp-statsd-client/UDPSender.hpp
@@ -2,7 +2,9 @@
 #define UDP_SENDER_HPP
 
 #ifdef _WIN32
-#define NOMINMAX
+#ifndef NOMINMAX
+    #define NOMINMAX
+#endif
 #include <io.h>
 #include <winsock2.h>
 #include <ws2tcpip.h>
@@ -13,6 +15,10 @@
 #include <unistd.h>
 #endif
 
+#if defined(__FreeBSD__)
+    #include <netinet/in.h>
+#endif
+
 #include <atomic>
 #include <cmath>
 #include <cstdint>
