diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 4632fdbcf..c76b29884 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -291,7 +291,7 @@ if(PKG_CONFIG_FOUND)
   if(NOT \${_res} EQUAL 0)
     message(FATAL_ERROR \"Configuring libvalhalla.pc failed: \${_err}\")
   endif()")
-  install(FILES ${VALHALLA_BINARY_DIR}/libvalhalla.pc
+  install(FILES ${VALHALLA_BINARY_DIR}/libvalhalla.pc.exe
     DESTINATION "${CMAKE_INSTALL_LIBDIR}/pkgconfig"
     COMPONENT development)
 endif()
