diff --git a/CMakeLists.txt b/CMakeLists.txt
index 3af4452a7..1e9b92004 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -176,6 +176,13 @@ endif()
 # newer protobuf versions require a compat bool
 set(protobuf_MODULE_COMPATIBLE ON CACHE BOOL "")
 find_package(Protobuf REQUIRED)
+
+if(Protobuf_PROTOC_EXECUTABLE)
+  set_target_properties(protobuf::protoc PROPERTIES
+    IMPORTED_LOCATION_RELEASE "${Protobuf_PROTOC_EXECUTABLE}"
+  )
+endif()
+

 # and turn it off again
 message(STATUS "Using protoc from ${Protobuf_PROTOC_EXECUTABLE}")
 message(STATUS "Using pbf headers from ${Protobuf_INCLUDE_DIRS}")
