diff --git a/source/global/management/include/G4Timer.hh b/source/global/management/include/G4Timer.hh
index 022e3557df8..390bc44d855 100644
--- a/source/global/management/include/G4Timer.hh
+++ b/source/global/management/include/G4Timer.hh
@@ -72,7 +72,7 @@
 #ifndef G4TIMER_HH
 #define G4TIMER_HH 1
 
-#if !(defined(WIN32) || defined(__MINGW32__))
+#if !(defined(_WIN32)
 #  include <sys/times.h>
 #  include <unistd.h>
 #else
@@ -82,7 +82,7 @@
 extern "C"
 {
   int sysconf(int);
-};
+}
 
 // Structure returned by times()
 
@@ -97,7 +97,7 @@ struct tms
 extern "C"
 {
   extern clock_t times(struct tms*);
-};
+}
 #endif /* WIN32 */
 
 #include "G4Types.hh"
diff --git a/source/global/management/src/G4Timer.cc b/source/global/management/src/G4Timer.cc
index 461d0141a0a..d45631602e6 100644
--- a/source/global/management/src/G4Timer.cc
+++ b/source/global/management/src/G4Timer.cc
@@ -41,7 +41,7 @@
 #  endif
 #endif
 
-#ifdef WIN32
+#ifdef _WIN32
 #  include <sys/types.h>
 #  include <windows.h>
 
