--- /dev/null
+++ b/src/OSD/OSD_MemInfo.cxx
@@ -143,20 +143,20 @@
   #elif defined(__EMSCRIPTEN__)
   if (IsActive(MemHeapUsage) || IsActive(MemWorkingSet) || IsActive(MemWorkingSetPeak))
   {
-    // /proc/%d/status is not emulated - get more info from mallinfo()
-    const struct mallinfo aMI = mallinfo();
-    if (IsActive(MemHeapUsage))
-    {
-      myCounters[MemHeapUsage] = aMI.uordblks;
-    }
-    if (IsActive(MemWorkingSet))
-    {
-      myCounters[MemWorkingSet] = aMI.uordblks;
-    }
-    if (IsActive(MemWorkingSetPeak))
-    {
-      myCounters[MemWorkingSetPeak] = aMI.usmblks;
-    }
+    //// /proc/%d/status is not emulated - get more info from mallinfo()
+    //const struct mallinfo aMI = mallinfo();
+    //if (IsActive(MemHeapUsage))
+    //{
+    //  myCounters[MemHeapUsage] = aMI.uordblks;
+    //}
+    //if (IsActive(MemWorkingSet))
+    //{
+    //  myCounters[MemWorkingSet] = aMI.uordblks;
+    //}
+    //if (IsActive(MemWorkingSetPeak))
+    //{
+    //  myCounters[MemWorkingSetPeak] = aMI.usmblks;
+    //}
   }
   if (IsActive(MemVirtual))
   {
@@ -172,16 +172,16 @@
       #endif
     #endif

-    #ifdef HAS_MALLINFO
-      #ifdef HAS_MALLINFO2
-    const struct mallinfo2 aMI = mallinfo2();
-      #else
-    const struct mallinfo aMI = mallinfo();
-      #endif
-    myCounters[MemHeapUsage] = aMI.uordblks;
-    #else
-    myCounters[MemHeapUsage] = 0;
-    #endif
+    //#ifdef HAS_MALLINFO
+    //  #ifdef HAS_MALLINFO2
+    //const struct mallinfo2 aMI = mallinfo2();
+    //  #else
+    //const struct mallinfo aMI = mallinfo();
+    //  #endif
+    //myCounters[MemHeapUsage] = aMI.uordblks;
+    //#else
+    //myCounters[MemHeapUsage] = 0;
+    //#endif
   }

   if (!IsActive(MemVirtual) && !IsActive(MemWorkingSet) && !IsActive(MemWorkingSetPeak)