--- /dev/null
+++ b/src/FoundationClasses/TKernel/Standard/Standard_Real.hxx
@@ -164,7 +164,7 @@
 }

 //! Returns the minimum value of a double.
-[[nodiscard]] constexpr double RealFirst()
+[[nodiscard]] constexpr double RealFirst() noexcept
 {
   return -DBL_MAX;
 }
@@ -176,7 +176,7 @@
 }

 //! Returns the maximum value of a double.
-[[nodiscard]] constexpr double RealLast()
+[[nodiscard]] constexpr double RealLast() noexcept
 {
   return DBL_MAX;
 }
