--- /dev/null
+++ b/src/DataExchange/TKDESTEP/STEPConstruct/STEPConstruct_AP203Context.cxx
@@ -122,7 +122,7 @@
     struct tm* time_struct = localtime(&time_clock);
     int        shift       = int(time_struct->tm_gmtoff);
 #else
-    int shift = int(timezone);
+    int shift = int((long long)timezone);
 #endif
     int                     shifth = abs(shift) / 3600;
     int                     shiftm = (abs(shift) - shifth * 3600) / 60;
