commit 4927282330beea6dd3d293478dfc89fb96706335
Author: Benjamin Lorenz <lorenz@math.tu-berlin.de>
Date:   Sat Jun 24 11:10:24 2023 +0200

    centroid_volume: add explicit coord cast

diff --git a/apps/polytope/src/centroid_volume.cc b/apps/polytope/src/centroid_volume.cc
index 1fed900b7d..15938742c8 100644
--- a/apps/polytope/src/centroid_volume.cc
+++ b/apps/polytope/src/centroid_volume.cc
@@ -40,8 +40,8 @@ void centroid_volume(BigObject p, const GenericMatrix<MatrixTop>& Points, const
          b+=*i;
       centroid+=v*b;
    }
-   centroid/=volume*(d+1);
-   volume /= Integer::fac(d);
+   centroid/=volume*static_cast<Coord>(d+1);
+   volume /= static_cast<Coord>(Integer::fac(d));
 
    p.take("CENTROID")<<centroid;
    p.take("VOLUME") << volume;
