diff --git a/perllib/Polymake/Core/CPlusPlusGenerator.pm b/perllib/Polymake/Core/CPlusPlusGenerator.pm
index 5c6a9781c2..5651027e11 100644
--- a/perllib/Polymake/Core/CPlusPlusGenerator.pm
+++ b/perllib/Polymake/Core/CPlusPlusGenerator.pm
@@ -913,7 +913,7 @@ sub build_temp_shared_module {
 
    warn_print( "Compiling temporary shared module, please be patient..." ) if $Verbose::cpp;
 
-   if (system(($Verbose::cpp>1 && "cat $dir/$cpperl_filename.cpperl >&2; ") .
+   if (Interrupts::system(($Verbose::cpp>1 && "cat $dir/$cpperl_filename.cpperl >&2; ") .
               "ninja -C $dir $so_name ".($Verbose::cpp ? ">&2" : ">/dev/null")." 2>$dir/err.log")) {
       if ($Verbose::cpp) {
          die "C++/perl Interface module compilation failed; see the generated code and the error log below.\n\n",
