--- a/libversion/CMakeLists.txt
+++ b/libversion/CMakeLists.txt
@@ -34,11 +34,6 @@
 	C_VISIBILITY_PRESET hidden
 )
 generate_export_header(libversion EXPORT_FILE_NAME export.h)
-if(WIN32)
-	# avoid clash with both c:/windows/system32/version.dll
-	# and static version.lib from the next target
-	set_target_properties(libversion PROPERTIES OUTPUT_NAME libversion)
-endif()

 # static library
 add_library(libversion_static STATIC ${LIBVERSION_SOURCES} ${LIBVERSION_HEADERS})
