--- a/magick/timer.c
+++ b/magick/timer.c
@@ -43,7 +43,7 @@
 #if !defined(CLK_TCK)
 #  define CLK_TCK  sysconf(_SC_CLK_TCK)
 #endif
-#if defined(HAVE_CLOCK_GETTIME)
+#if defined(HAVE_CLOCK_GETTIME) && defined(HAVE_CLOCK_REALTIME)
 #  define NANOSECONDS_PER_SECOND 1000000000.0
 #  if defined(CLOCK_HIGHRES) /* Solaris */
 #    define CLOCK_ID CLOCK_HIGHRES
@@ -124,7 +124,7 @@ MagickExport unsigned int ContinueTimer(TimerInfo *time_info)
 */
 static double ElapsedTime(void)
 {
-#if defined(HAVE_CLOCK_GETTIME)
+#if defined(HAVE_CLOCK_GETTIME) && defined(HAVE_CLOCK_REALTIME)
   struct timespec timer;
   (void) clock_gettime(CLOCK_ID, &timer);

@@ -236,7 +236,7 @@ MagickExport void GetTimerInfo(TimerInfo *time_info)
 */
 MagickExport double GetTimerResolution(void)
 {
-#if defined(HAVE_CLOCK_GETRES)
+#if defined(HAVE_CLOCK_GETRES) && defined(HAVE_CLOCK_REALTIME)
   struct timespec timer;
   (void) clock_getres(CLOCK_ID, &timer);
