diff --git a/src/Offline/DABitGenerator.cpp b/src/Offline/DABitGenerator.cpp
index fefb3cc..9a0f22b 100644
--- a/src/Offline/DABitGenerator.cpp
+++ b/src/Offline/DABitGenerator.cpp
@@ -7,7 +7,7 @@ All rights reserved
 
 #include "DABitGenerator.h"
 #include "DABitMachine.h"
-#include "LSSS/PRSS.cpp"
+#include "LSSS/PRSS.h"
 #include "OT/OT_Thread_Data.h"
 #include "Processor/Processor_IO.h"
 #include "Tools/util_containers.h"
