diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index f4435939b..264bd1b36 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -449,7 +449,7 @@ function(add_exec_test name tgt)
     endif()
 endfunction()
 
-if(EMSCRIPTEN AND LEAN_EMSCRIPTEN_BUILD STREQUAL Main)
+if(NOT BUILD_TESTING OR (EMSCRIPTEN AND LEAN_EMSCRIPTEN_BUILD STREQUAL Main))
 # skip building standalone test binaries
 else()
 add_subdirectory(tests/util)
@@ -478,7 +478,7 @@ endif()
 
 endif()
 
-if(EMSCRIPTEN AND LEAN_EMSCRIPTEN_BUILD STREQUAL Test)
+if(NOT BUILD_TESTING OR (EMSCRIPTEN AND LEAN_EMSCRIPTEN_BUILD STREQUAL Main))
 # skip building tests for main files
 else()
 # Include style check
-- 
2.20.4

