diff --color -ur qtbase-everywhere-src-6.5.2/cmake/QtCompilerOptimization.cmake qtbase-everywhere-src-6.5.2-patched/cmake/QtCompilerOptimization.cmake
--- qtbase-everywhere-src-6.5.2/cmake/QtCompilerOptimization.cmake	2023-07-06 12:02:03.000000000 +0200
+++ qtbase-everywhere-src-6.5.2-patched/cmake/QtCompilerOptimization.cmake	2023-09-17 19:17:41.738579711 +0200
@@ -45,7 +45,7 @@
     set(QT_CFLAGS_RDSEED     "-mrdseed")
     set(QT_CFLAGS_AVX        "-mavx")
     set(QT_CFLAGS_AVX2       "-mavx2")
-    set(QT_CFLAGS_ARCH_HASWELL "-march=haswell")
+    set(QT_CFLAGS_ARCH_HASWELL "-mavx2" "-mf16c" "-mfma" "-mbmi" "-mbmi2" "-mlzcnt")
     set(QT_CFLAGS_AVX512F    "-mavx512f")
     set(QT_CFLAGS_AVX512ER   "-mavx512er")
     set(QT_CFLAGS_AVX512CD   "-mavx512cd")
diff --color -ur qtbase-everywhere-src-6.5.2/src/gui/opengl/qopengl.cpp qtbase-everywhere-src-6.5.2-patched/src/gui/opengl/qopengl.cpp
--- qtbase-everywhere-src-6.5.2/src/gui/opengl/qopengl.cpp	2023-07-06 12:02:03.000000000 +0200
+++ qtbase-everywhere-src-6.5.2-patched/src/gui/opengl/qopengl.cpp	2023-09-17 19:27:14.290816143 +0200
@@ -187,6 +187,14 @@
     return result;
 }
 
+namespace QOSWorkaround {
+    // Due to a mingw bug being confused by static constexpr variables in an exported class,
+    // we cannot use QOperatingSystemVersion::Windows11 in libraries outside of QtCore.
+    // ### TODO Remove this when that problem is fixed.
+    static constexpr QOperatingSystemVersionBase Windows11 { QOperatingSystemVersionBase::Windows,
+                                                             10, 0, 22000 };
+}
+
 // OS term consisting of name and optional version found in
 // under "os" in main array and in "exceptions" lists.
 struct OsTypeTerm
@@ -196,7 +204,7 @@
     static QVersionNumber hostKernelVersion() { return QVersionNumber::fromString(QSysInfo::kernelVersion()); }
     static QString hostOsRelease() {
 #ifdef Q_OS_WIN
-        if (QOperatingSystemVersion::current() >= QOperatingSystemVersion::Windows11)
+        if (QOperatingSystemVersion::current() >= QOSWorkaround::Windows11)
             return u"11"_s;
         return u"10"_s;
 #else
diff --color -ur qtbase-everywhere-src-6.5.2/src/plugins/platforms/windows/qwindowsintegration.cpp qtbase-everywhere-src-6.5.2-patched/src/plugins/platforms/windows/qwindowsintegration.cpp
--- qtbase-everywhere-src-6.5.2/src/plugins/platforms/windows/qwindowsintegration.cpp	2023-07-06 12:02:03.000000000 +0200
+++ qtbase-everywhere-src-6.5.2-patched/src/plugins/platforms/windows/qwindowsintegration.cpp	2023-09-17 20:02:28.197790564 +0200
@@ -651,6 +651,14 @@
     MessageBeep(MB_OK);  // For QApplication
 }
 
+namespace QOSWorkaround {
+    // Due to a mingw bug being confused by static constexpr variables in an exported class,
+    // we cannot use QOperatingSystemVersion::Windows11 in libraries outside of QtCore.
+    // ### TODO Remove this when that problem is fixed.
+    static constexpr QOperatingSystemVersionBase Windows11 { QOperatingSystemVersionBase::Windows,
+                                                             10, 0, 22000 };
+}
+
 void QWindowsIntegration::setApplicationBadge(qint64 number)
 {
     // Clamp to positive numbers, as the Windows API doesn't support negative numbers
@@ -659,7 +667,7 @@
     // Persist, so we can re-apply it on setting changes and Explorer restart
     m_applicationBadgeNumber = number;
 
-    static const bool isWindows11 = QOperatingSystemVersion::current() >= QOperatingSystemVersion::Windows11;
+    static const bool isWindows11 = QOperatingSystemVersion::current() >= QOSWorkaround::Windows11;
 
 #if QT_CONFIG(cpp_winrt)
     // We prefer the native BadgeUpdater API, that allows us to set a number directly,
