diff --color -ur a/cmake/QtCompilerOptimization.cmake b/cmake/QtCompilerOptimization.cmake
--- a/cmake/QtCompilerOptimization.cmake	2026-01-13 11:18:17.000000000 +0100
+++ b/cmake/QtCompilerOptimization.cmake	2026-02-17 22:50:57.547445002 +0100
@@ -49,7 +49,7 @@
     set(QT_CFLAGS_RDSEED     "${__prefix}-mrdseed")
     set(QT_CFLAGS_AVX        "${__prefix}-mavx")
     set(QT_CFLAGS_AVX2       "${__prefix}-mavx2")
-    set(QT_CFLAGS_ARCH_HASWELL "${__prefix}-march=haswell")
+    set(QT_CFLAGS_ARCH_HASWELL "-mavx2" "-mf16c" "-mfma" "-mbmi" "-mbmi2" "-mlzcnt")
     set(QT_CFLAGS_AVX512F    "${__prefix}-mavx512f")
     set(QT_CFLAGS_AVX512ER   "${__prefix}-mavx512er")
     set(QT_CFLAGS_AVX512CD   "${__prefix}-mavx512cd")
diff --color -ur a/src/corelib/global/qlogging.cpp b/src/corelib/global/qlogging.cpp
--- a/src/corelib/global/qlogging.cpp	2026-01-13 11:18:17.000000000 +0100
+++ b/src/corelib/global/qlogging.cpp	2026-02-18 21:02:34.179245237 +0100
@@ -203,7 +203,7 @@
     return false;
 }
 
-#if defined(Q_OS_LINUX) || defined(Q_OS_DARWIN) || defined(Q_OS_FREEBSD) || defined(Q_OS_NETBSD)
+#if (defined(Q_OS_LINUX) && defined(__GLIBC__)) || defined(Q_OS_DARWIN) || defined(Q_OS_FREEBSD) || defined(Q_OS_NETBSD)
 static bool qt_append_thread_name_to(QString &message)
 {
     std::array<char, 16> name{};
diff --color -ur a/src/corelib/global/qsimd_p.h b/src/corelib/global/qsimd_p.h
--- a/src/corelib/global/qsimd_p.h	2026-01-13 11:18:17.000000000 +0100
+++ b/src/corelib/global/qsimd_p.h	2026-02-19 21:58:05.412506514 +0100
@@ -353,6 +353,10 @@
 #define __ARM_FEATURE_CRC32 1
 #endif
 
+#if defined(Q_OS_MACOS) && defined(Q_PROCESSOR_ARM)
+#define __ARM_FEATURE_CRYPTO 1
+#endif
+
 #if defined(Q_PROCESSOR_ARM_64)
 #if defined(Q_CC_CLANG)
 #define QT_FUNCTION_TARGET_STRING_AES        "aes"
