diff --git a/bolt/runtime/instr.cpp b/bolt/runtime/instr.cpp
index 1a8bcf500362..7747c412a2ae 100644
--- a/bolt/runtime/instr.cpp
+++ b/bolt/runtime/instr.cpp
@@ -1495,7 +1495,7 @@ __bolt_instr_data_dump() {
 /// Event loop for our child process spawned during setup to dump profile data
 /// at user-specified intervals
 void watchProcess() {
-  timespec ts, rem;
+  ::timespec ts, rem;
   uint64_t Ellapsed = 0ull;
   uint64_t ppid;
   if (__bolt_instr_wait_forks) {
