--- a/subprojects/libvhost-user/libvhost-user.c
+++ b/subprojects/libvhost-user/libvhost-user.c
@@ -80,6 +80,14 @@
         }                                       \
     } while (0)
 
+/* copied from include/qemu/osdep.h */
+#define QEMU_MADV_INVALID -1
+#ifdef MADV_NOHUGEPAGE
+#define QEMU_MADV_NOHUGEPAGE MADV_NOHUGEPAGE
+#else
+#define QEMU_MADV_NOHUGEPAGE QEMU_MADV_INVALID
+#endif
+
 static inline
 bool has_feature(uint64_t features, unsigned int fbit)
 {
@@ -634,7 +642,7 @@
          */
         ret = madvise((void *)(uintptr_t)dev_region->mmap_addr,
                       dev_region->size + dev_region->mmap_offset,
-                      MADV_NOHUGEPAGE);
+                      QEMU_MADV_NOHUGEPAGE);
         if (ret) {
             /*
              * Note: This can happen legally on kernels that are configured
