diff --git a/meson.build b/meson.build
index 96de1a6ef9..36b315971e 100644
--- a/meson.build
+++ b/meson.build
@@ -2527,7 +2527,7 @@ util_ss.add_all(trace_ss)
 util_ss = util_ss.apply(config_all, strict: false)
 libqemuutil = static_library('qemuutil',
                              sources: util_ss.sources() + stub_ss.sources() + genh,
-                             dependencies: [util_ss.dependencies(), libm, threads, glib, socket, malloc, pixman])
+                             dependencies: [util_ss.dependencies(), libm, threads, glib, socket, malloc, pixman, rt])
 qemuutil = declare_dependency(link_with: libqemuutil,
                               sources: genh + version_res)
 
