diff --git a/cpp/lazperf/portable_endian.hpp b/cpp/lazperf/portable_endian.hpp
index b0e753f32..1883ad881 100644
--- a/cpp/lazperf/portable_endian.hpp
+++ b/cpp/lazperf/portable_endian.hpp
@@ -65,9 +65,9 @@
 #       define be32toh ntohl
 #       define le32toh(x) (x)

-#       define htobe64 htonll
+#       define htobe64(x) ((((uint64_t)htonl(x&0xFFFFFFFF)) << 32) + htonl(x >> 32))
 #       define htole64(x) (x)
-#       define be64toh ntohll
+#       define be64toh(x) ((((uint64_t)ntohl(x&0xFFFFFFFF)) << 32) + ntohl(x >> 32))
 #       define le64toh(x) (x)

 #   elif BYTE_ORDER == BIG_ENDIAN
