Subject: [PATCH] Revert "fix(ffi): don't build C libraries by default"

This reverts commit 1c6637060e36654ddb2fdfccb0d146c7ad527476.
---
diff --git a/Cargo.toml b/Cargo.toml
index f2ee5efe3e..e87137f595 100644
--- b/Cargo.toml
+++ a/Cargo.toml
@@ -19,6 +19,9 @@
   #"build.rs",
 ]
 
+[lib]
+crate-type = ["lib", "staticlib", "cdylib"]
+
 [dependencies]
 bytes = "1"
 futures-core = { version = "0.3", default-features = false }
