--- a/cpp_version/CMakeLists.txt
+++ b/cpp_version/CMakeLists.txt
@@ -26,6 +26,7 @@ find_package(Threads REQUIRED)
 ## Subdirectories and source files
 ## ======================================================================================##
 include_directories(src src/utility src/Forest src/Tree)
+file(GLOB_RECURSE HEADERS ../src/Tree/*.h)
 
 ## ======================================================================================##
 ## Debug and release targets
@@ -64,5 +65,6 @@ target_link_libraries(ranger-bin ranger)
 
 set_property(TARGET ranger PROPERTY PUBLIC_HEADER ${HEADERS})
 
-install(TARGETS ranger LIBRARY DESTINATION lib)
-
+install(TARGETS ranger
+        LIBRARY DESTINATION lib
+        PUBLIC_HEADER DESTINATION include/ranger)
