diff -u a/xxhash.h b/xxhash.h
--- a/xxhash.h     2022-05-24 14:43:40.632328800 -0400
+++ b/xxhash.h     2022-05-24 14:44:30.468301000 -0400
@@ -228,17 +228,7 @@
  * @{
  */
 /* specific declaration modes for Windows */
-#if !defined(XXH_INLINE_ALL) && !defined(XXH_PRIVATE_API)
-#  if defined(WIN32) && defined(_MSC_VER) && (defined(XXH_IMPORT) || defined(XXH_EXPORT))
-#    ifdef XXH_EXPORT
 #      define XXH_PUBLIC_API __declspec(dllexport)
-#    elif XXH_IMPORT
-#      define XXH_PUBLIC_API __declspec(dllimport)
-#    endif
-#  else
-#    define XXH_PUBLIC_API   /* do nothing */
-#  endif
-#endif

 #ifdef XXH_DOXYGEN
 /*!
 
@@ -325,6 +315,21 @@
 #define XXH_VERSION_RELEASE  1
 #define XXH_VERSION_NUMBER  (XXH_VERSION_MAJOR *100*100 + XXH_VERSION_MINOR *100 + XXH_VERSION_RELEASE)

+#ifdef XXH_INLINE_ALL
+#error "XXH_INLINE_ALL is not compatible with dynamic linking"
+#endif
+
+#ifdef XXH_PRIVATE_API
+#error "XXH_PRIVATE_API is not compatible with dynamic linking"
+#endif
+
+#ifdef XXH_IMPORT
+#error "XXH_IMPORT is not compatible with dynamic linking"
+#endif
+
+#ifdef XXH_EXPORT
+#error "XXH_EXPORT is good"
+#endif
 /*!
  * @brief Obtains the xxHash version.
  *
