diff --git a/mlx/backend/metal/metal.h b/mlx/backend/metal/metal.h
index 830df4f4..e5cb65af 100644
--- a/mlx/backend/metal/metal.h
+++ b/mlx/backend/metal/metal.h
@@ -2,6 +2,7 @@
 
 #pragma once
 
+#include <unordered_map>
 #include <variant>
 
 #include "mlx/array.h"
diff --git a/mlx/io.h b/mlx/io.h
index 4805d1c8..23380b23 100644
--- a/mlx/io.h
+++ b/mlx/io.h
@@ -2,6 +2,7 @@
 
 #pragma once
 
+#include <unordered_map>
 #include <variant>
 
 #include "mlx/array.h"
