--- a/meson.build
+++ b/meson.build
@@ -318,22 +318,10 @@
 coretext_deps = []
 if host_machine.system() == 'darwin' and not get_option('coretext').disabled()
   app_services_dep = dependency('appleframeworks', modules: ['ApplicationServices'], required: false)
-  if cpp.has_type('CTFontRef', prefix: '#include <ApplicationServices/ApplicationServices.h>', dependencies: app_services_dep)
     coretext_deps += [app_services_dep]
     conf.set('HAVE_CORETEXT', 1)
   # On iOS CoreText and CoreGraphics are stand-alone frameworks
   # Check for a different symbol to avoid getting cached result
-  else
-    coretext_dep = dependency('appleframeworks', modules: ['CoreText'], required: false)
-    coregraphics_dep = dependency('appleframeworks', modules: ['CoreGraphics'], required: false)
-    corefoundation_dep = dependency('appleframeworks', modules: ['CoreFoundation'], required: false)
-    if cpp.has_type('CTRunRef', prefix: '#include <CoreText/CoreText.h>', dependencies: [coretext_dep, coregraphics_dep, corefoundation_dep])
-      coretext_deps += [coretext_dep, coregraphics_dep, corefoundation_dep]
-      conf.set('HAVE_CORETEXT', 1)
-    elif get_option('coretext').enabled()
-      error('CoreText was enabled explicitly, but required headers or frameworks are missing.')
-    endif
-  endif
 endif
 
 # threads
