diff --git a/source/libnormaliz/general.cpp b/source/libnormaliz/general.cpp
index 6f578be5..226d6474 100644
--- a/source/libnormaliz/general.cpp
+++ b/source/libnormaliz/general.cpp
@@ -25,19 +25,7 @@
 #include <csignal>
 #include "libnormaliz/general.h"
 
-#ifndef _MSC_VER
-#include <sys/time.h>
-#else
-#define WIN32_LEAN_AND_MEAN
-#include <Windows.h>
-#include <stdint.h> // portable: uint64_t   MSVC: __int64
-
-// MSVC defines this in winsock2.h!?
-typedef struct timeval {
-    long tv_sec;
-    long tv_usec;
-} timeval;
-
+#ifdef _MSC_VER
 int gettimeofday(struct timeval * tp, struct timezone * tzp)
 {
     // Note: some broken versions only have 8 trailing zero's, the correct epoch has 9 trailing zero's
diff --git a/source/libnormaliz/general.h b/source/libnormaliz/general.h
index b87624f0..bc425f35 100644
--- a/source/libnormaliz/general.h
+++ b/source/libnormaliz/general.h
@@ -31,6 +31,20 @@
 #include <string>
 #include <vector>
 
+#ifndef _MSC_VER
+#include <sys/time.h>
+#else
+#define WIN32_LEAN_AND_MEAN
+#include <Windows.h>
+#include <stdint.h> // portable: uint64_t   MSVC: __int64
+
+// MSVC defines this in winsock2.h!?
+typedef struct timeval {
+    long tv_sec;
+    long tv_usec;
+} timeval;
+#endif
+
 #include <libnormaliz/dynamic_bitset.h>
 
 #ifndef NMZ_MAKEFILE_CLASSIC
