diff --git a/makefile b/makefile
index 620d241..a7f08a9 100644
--- a/makefile
+++ b/makefile
@@ -1,5 +1,6 @@
 CC = gcc
 CXX = g++
+CXX_BUILD = $(CXX)
 CFLAGS=-fopenmp -O3 -fPIC -mavx -g
 LDFLAGS=-fopenmp -lm -lrt
 
@@ -19,8 +20,8 @@ CvxCompress_Test: CvxCompress_Test.o $(OBJECTS)
 CvxCompress_Test_Dyn: CvxCompress_Test.o libcvxcompress.so
 	$(CXX) $(LDFLAGS) CvxCompress_Test.o  -L. -lcvxcompress -o $@
 
-CvxCompress_GenCode: CvxCompress_GenCode.o CvxCompress.hxx Wavelet_Transform_Slow.o
-	$(CXX) -O2 Wavelet_Transform_Slow.o  CvxCompress_GenCode.o -o CvxCompress_GenCode
+CvxCompress_GenCode: CvxCompress_GenCode.cpp CvxCompress.hxx Wavelet_Transform_Slow.cpp
+	$(CXX_BUILD) -O2 Wavelet_Transform_Slow.cpp CvxCompress_GenCode.cpp -o $@
 
 Test_Compression: Test_Compression.o $(OBJECTS)
 	$(CXX) $(LDFLAGS) $(OBJECTS)  Test_Compression.o -o Test_Compression
