--- a/src/runtime_src/xocl/api/icd/ocl_icd_bindings.h
+++ b/src/runtime_src/xocl/api/icd/ocl_icd_bindings.h
@@ -17,16 +17,16 @@
 #define xocl_api_icd_ocl_icd_bindings_h_
 
 #ifndef CL_TARGET_OPENCL_VERSION
-# define CL_TARGET_OPENCL_VERSION 200
+#define CL_TARGET_OPENCL_VERSION 200
 #endif
 
-#ifdef _WIN32
-# define NOMINMAX
-# include "windows/icd_dispatch.h"
-using _cl_icd_dispatch = KHRicdVendorDispatchRec;
-#else
-# include <ocl_icd.h>
-#endif
+// #ifdef _WIN32
+// # define NOMINMAX
+// # include "windows/icd_dispatch.h"
+// using _cl_icd_dispatch = KHRicdVendorDispatchRec;
+// #else
+#include <ocl_icd.h>
+// #endif
 extern const _cl_icd_dispatch cl_icd_dispatch;
 
 #endif
