diff --git a/CMakeLists.txt b/CMakeLists.txt
index c71a74d4a..29bd1f040 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -78,6 +78,9 @@ set(protobuf_DEBUG_POSTFIX "d"
 mark_as_advanced(protobuf_DEBUG_POSTFIX)
 # User options
 include(${protobuf_SOURCE_DIR}/cmake/protobuf-options.cmake)
+option(protobuf_INSTALL_PROTOBUF "Install libprotobuf" ON)
+option(protobuf_INSTALL_PROTOBUF_LITE "Install libprotobuf-lite" ON)
+option(protobuf_INSTALL_SDK "Install SDK" ON)
 
 if (protobuf_BUILD_SHARED_LIBS)
   # This is necessary for linking in Abseil.
diff --git a/cmake/install.cmake b/cmake/install.cmake
index f7a0d9e17..82f3549f9 100644
--- a/cmake/install.cmake
+++ b/cmake/install.cmake
@@ -5,7 +5,13 @@ configure_file(${CMAKE_CURRENT_SOURCE_DIR}/cmake/protobuf.pc.cmake
 configure_file(${CMAKE_CURRENT_SOURCE_DIR}/cmake/protobuf-lite.pc.cmake
                ${CMAKE_CURRENT_BINARY_DIR}/protobuf-lite.pc @ONLY)
 
-set(_protobuf_libraries libprotobuf-lite libprotobuf)
+set(_protobuf_libraries)
+if (protobuf_INSTALL_PROTOBUF OR protobuf_INSTALL_SDK OR protobuf_BUILD_LIBPROTOC)
+  list(APPEND _protobuf_libraries libprotobuf)
+endif()
+if (protobuf_INSTALL_PROTOBUF_LITE OR protobuf_INSTALL_SDK)
+  list(APPEND _protobuf_libraries libprotobuf-lite)
+endif()
 if (protobuf_BUILD_LIBPROTOC)
     list(APPEND _protobuf_libraries libprotoc)
 endif (protobuf_BUILD_LIBPROTOC)
@@ -41,7 +47,9 @@ if (protobuf_BUILD_PROTOC_BINARIES)
   endif()
 endif (protobuf_BUILD_PROTOC_BINARIES)
 
-install(FILES ${CMAKE_CURRENT_BINARY_DIR}/protobuf.pc ${CMAKE_CURRENT_BINARY_DIR}/protobuf-lite.pc DESTINATION "${CMAKE_INSTALL_LIBDIR}/pkgconfig")
+if (protobuf_INSTALL_SDK)
+  install(FILES ${CMAKE_CURRENT_BINARY_DIR}/protobuf.pc ${CMAKE_CURRENT_BINARY_DIR}/protobuf-lite.pc DESTINATION "${CMAKE_INSTALL_LIBDIR}/pkgconfig")
+endif()
 
 include(${protobuf_SOURCE_DIR}/src/file_lists.cmake)
 set(protobuf_HEADERS
@@ -56,10 +64,12 @@ foreach(_header ${protobuf_HEADERS})
   get_filename_component(_extract_from "${protobuf_SOURCE_DIR}/src/${_header}" ABSOLUTE)
   get_filename_component(_extract_name ${_header} NAME)
   get_filename_component(_extract_to "${CMAKE_INSTALL_INCLUDEDIR}/${_header}" DIRECTORY)
-  install(FILES "${_extract_from}"
-    DESTINATION "${_extract_to}"
-    COMPONENT protobuf-headers
-    RENAME "${_extract_name}")
+  if (protobuf_INSTALL_SDK)
+    install(FILES "${_extract_from}"
+      DESTINATION "${_extract_to}"
+      COMPONENT protobuf-headers
+      RENAME "${_extract_name}")
+  endif()
 endforeach()
 
 # Install configuration
@@ -95,17 +105,19 @@ configure_file(${protobuf_SOURCE_DIR}/cmake/protobuf-generate.cmake
 
 # Allows the build directory to be used as a find directory.
 
-install(EXPORT protobuf-targets
-  DESTINATION "${CMAKE_INSTALL_CMAKEDIR}"
-  NAMESPACE protobuf::
-  COMPONENT protobuf-export
-)
-
-install(DIRECTORY ${CMAKE_BUILD_CMAKEDIR}/
-  DESTINATION "${CMAKE_INSTALL_CMAKEDIR}"
-  COMPONENT protobuf-export
-  PATTERN protobuf-targets.cmake EXCLUDE
-)
+if (protobuf_INSTALL_SDK)
+  install(EXPORT protobuf-targets
+    DESTINATION "${CMAKE_INSTALL_CMAKEDIR}"
+    NAMESPACE protobuf::
+    COMPONENT protobuf-export
+  )
+  
+  install(DIRECTORY ${CMAKE_BUILD_CMAKEDIR}/
+    DESTINATION "${CMAKE_INSTALL_CMAKEDIR}"
+    COMPONENT protobuf-export
+    PATTERN protobuf-targets.cmake EXCLUDE
+  )
+endif()
 
 option(protobuf_INSTALL_EXAMPLES "Install the examples folder" OFF)
 if(protobuf_INSTALL_EXAMPLES)
