From 54ec81fd5f5da27ba489ba833e9c8dbfa215f53a Mon Sep 17 00:00:00 2001
From: Min RK <benjaminrk@gmail.com>
Date: Fri, 20 May 2022 14:34:24 +0200
Subject: [PATCH 1/4] Remove unconditional testing

Normally invoked by ctest
---
 CMakeLists.txt | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index d63fe21..4531d15 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -80,9 +80,6 @@ option(BUILD_LIBSCOTCHMETIS "Build ScotchMeTiS library" ON)
 # Install our MeTiS header stubs
 option(INSTALL_METIS_HEADERS "Install libScotchMeTiS headers in destination directory" ON)
 
-# Allow for testing
-enable_testing()
-
 add_subdirectory(src)
 
 # Install man pages
-- 
2.35.3

