diff --git a/src/libscotch/CMakeLists.txt b/src/libscotch/CMakeLists.txt
index e3b0cb9..006be7d 100644
--- a/src/libscotch/CMakeLists.txt
+++ b/src/libscotch/CMakeLists.txt
@@ -541,7 +549,7 @@ endif()
 # Add thread library
 if(Threads_FOUND)
   target_compile_definitions(scotch PUBLIC COMMON_PTHREAD SCOTCH_PTHREAD)
-  target_link_libraries(scotch PUBLIC Threads::Threads)
+  target_link_libraries(scotch PUBLIC Threads::Threads ${THREADS_PTHREADS_WIN32_LIBRARY})
 endif(Threads_FOUND)
 
 # Include files
@@ -771,7 +779,7 @@ if(BUILD_PTSCOTCH)
 
   if(Threads_FOUND)
     target_compile_definitions(ptscotch PUBLIC SCOTCH_PTHREAD COMMON_PTHREAD)
-    target_link_libraries(ptscotch PUBLIC Threads::Threads)
+    target_link_libraries(ptscotch PUBLIC Threads::Threads ${THREADS_PTHREADS_WIN32_LIBRARY})
   endif(Threads_FOUND)
 
   add_library(ptscotcherr library_error.c)
diff --git a/src/scotch/CMakeLists.txt b/src/scotch/CMakeLists.txt
index 3974f4f..94c0080 100644
--- a/src/scotch/CMakeLists.txt
+++ b/src/scotch/CMakeLists.txt
@@ -69,7 +69,7 @@ function(add_scotch_exe)
   target_compile_definitions(${file_we} PUBLIC "SCOTCH_CODENAME=\"${SCOTCH_CODENAME}\"")
   if(Threads_FOUND)
     target_compile_definitions(${file_we} PUBLIC SCOTCH_PTHREAD COMMON_PTHREAD)
-    target_link_libraries(${file_we} PUBLIC Threads::Threads)
+    target_link_libraries(${file_we} PUBLIC Threads::Threads ${THREADS_PTHREADS_WIN32_LIBRARY})
   endif(Threads_FOUND)
   target_link_libraries(${file_we} PUBLIC scotch scotcherrexit)
 endfunction(add_scotch_exe)
@@ -144,7 +146,7 @@ if(BUILD_PTSCOTCH)
     target_compile_definitions(${file_we} PUBLIC "SCOTCH_CODENAME=\"${SCOTCH_CODENAME}\"")
     if(Threads_FOUND)
       target_compile_definitions(${file_we} PUBLIC SCOTCH_PTHREAD COMMON_PTHREAD)
-      target_link_libraries(${file_we} PUBLIC Threads::Threads)
+      target_link_libraries(${file_we} PUBLIC Threads::Threads ${THREADS_PTHREADS_WIN32_LIBRARY})
     endif(Threads_FOUND)
     target_include_directories(${file_we} PRIVATE ${GENERATED_INCLUDE_DIR})
     target_link_libraries(${file_we} PUBLIC scotch ptscotch ptscotcherrexit)
