diff --git a/src/libscotch/CMakeLists.txt b/src/libscotch/CMakeLists.txt
index 4137e2c..e3c1b45 100644
--- a/src/libscotch/CMakeLists.txt
+++ b/src/libscotch/CMakeLists.txt
@@ -49,15 +49,29 @@
 #  dummysizes  #
 ################
 
-add_executable(dummysizes dummysizes.c)
-set_target_properties(dummysizes PROPERTIES RUNTIME_OUTPUT_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR})
+add_custom_command(
+  OUTPUT  ${CMAKE_CURRENT_BINARY_DIR}/dummysizes
+  COMMAND $ENV{BUILD_CC}
+          ${CMAKE_CURRENT_SOURCE_DIR}/dummysizes.c
+          -o ${CMAKE_CURRENT_BINARY_DIR}/dummysizes
+  DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/dummysizes.c)
+add_custom_target(dummysizes_tgt
+  DEPENDS ${CMAKE_CURRENT_BINARY_DIR}/dummysizes)
+set(DUMMYSIZES_CMD ${CMAKE_CURRENT_BINARY_DIR}/dummysizes)
+set(DUMMYSIZES_DEP dummysizes_tgt)
+
 if(BUILD_PTSCOTCH)
-  add_executable(ptdummysizes dummysizes.c)
-  set_target_properties(ptdummysizes PROPERTIES RUNTIME_OUTPUT_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR})
-  target_link_libraries(ptdummysizes PRIVATE MPI::MPI_C)
-  add_dependencies(ptdummysizes scotch_h)
-  set_target_properties(ptdummysizes PROPERTIES
-    COMPILE_FLAGS -DSCOTCH_PTSCOTCH)
+  add_custom_command(
+    OUTPUT  ${CMAKE_CURRENT_BINARY_DIR}/ptdummysizes
+    COMMAND $ENV{BUILD_CC}
+            -DSCOTCH_PTSCOTCH
+            ${CMAKE_CURRENT_SOURCE_DIR}/dummysizes.c
+            -o ${CMAKE_CURRENT_BINARY_DIR}/ptdummysizes
+    DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/dummysizes.c)
+  add_custom_target(ptdummysizes_tgt
+    DEPENDS ${CMAKE_CURRENT_BINARY_DIR}/ptdummysizes)
+  set(PTDUMMYSIZES_CMD ${CMAKE_CURRENT_BINARY_DIR}/ptdummysizes)
+  set(PTDUMMYSIZES_DEP ptdummysizes_tgt)
 endif()
 
 ##############
@@ -66,8 +80,8 @@ endif()
 
 # Generate scotch.h
 add_custom_command(OUTPUT ${GENERATED_INCLUDE_DIR}/scotch.h
-  COMMAND $<TARGET_FILE:dummysizes> "-s${SCOTCH_NAME_SUFFIX}"
-  ${CMAKE_CURRENT_SOURCE_DIR}/library.h ${GENERATED_INCLUDE_DIR}/scotch.h DEPENDS dummysizes
+  COMMAND ${DUMMYSIZES_CMD} "-s${SCOTCH_NAME_SUFFIX}"
+  ${CMAKE_CURRENT_SOURCE_DIR}/library.h ${GENERATED_INCLUDE_DIR}/scotch.h DEPENDS ${DUMMYSIZES_DEP}
   DEPENDS library.h
   COMMENT "Generate scotch.h")
 add_custom_target(scotch_h
@@ -75,8 +89,8 @@ add_custom_target(scotch_h
 
 # Generate scotchf.h
 add_custom_command(OUTPUT ${GENERATED_INCLUDE_DIR}/scotchf.h
-  COMMAND $<TARGET_FILE:dummysizes>  "-s${SCOTCH_NAME_SUFFIX}"
-  ${CMAKE_CURRENT_SOURCE_DIR}/library_f.h ${GENERATED_INCLUDE_DIR}/scotchf.h DEPENDS dummysizes
+  COMMAND ${DUMMYSIZES_CMD}  "-s${SCOTCH_NAME_SUFFIX}"
+  ${CMAKE_CURRENT_SOURCE_DIR}/library_f.h ${GENERATED_INCLUDE_DIR}/scotchf.h DEPENDS ${DUMMYSIZES_DEP}
   DEPENDS library.h
   COMMENT "Generate scotchf.h")
 add_custom_target(scotchf_h
@@ -89,8 +103,8 @@ add_custom_target(scotchf_h
 if(BUILD_PTSCOTCH)
   # Generate ptscotch.h
   add_custom_command(OUTPUT ${GENERATED_INCLUDE_DIR}/ptscotch.h
-    COMMAND $<TARGET_FILE:ptdummysizes> "-s${SCOTCH_NAME_SUFFIX}"
-    ${CMAKE_CURRENT_SOURCE_DIR}/library_pt.h ${GENERATED_INCLUDE_DIR}/ptscotch.h DEPENDS ptdummysizes
+    COMMAND ${PTDUMMYSIZES_CMD} "-s${SCOTCH_NAME_SUFFIX}"
+    ${CMAKE_CURRENT_SOURCE_DIR}/library_pt.h ${GENERATED_INCLUDE_DIR}/ptscotch.h DEPENDS ${PTDUMMYSIZES_DEP}
     DEPENDS library_pt.h scotch_h
     COMMENT "Generate ptscotch.h")
   add_custom_target(ptscotch_h
@@ -98,8 +112,8 @@ if(BUILD_PTSCOTCH)
 
   # Generate ptscotchf.h
   add_custom_command(OUTPUT ${GENERATED_INCLUDE_DIR}/ptscotchf.h
-    COMMAND $<TARGET_FILE:ptdummysizes> "-s${SCOTCH_NAME_SUFFIX}"
-    ${CMAKE_CURRENT_SOURCE_DIR}/library_pt_f.h ${GENERATED_INCLUDE_DIR}/ptscotchf.h DEPENDS ptdummysizes
+    COMMAND ${PTDUMMYSIZES_CMD} "-s${SCOTCH_NAME_SUFFIX}"
+    ${CMAKE_CURRENT_SOURCE_DIR}/library_pt_f.h ${GENERATED_INCLUDE_DIR}/ptscotchf.h DEPENDS ${PTDUMMYSIZES_DEP}
     DEPENDS library.h
     COMMENT "Generate ptscotchf.h")
   add_custom_target(ptscotchf_h
diff --git a/src/libscotchmetis/CMakeLists.txt b/src/libscotchmetis/CMakeLists.txt
index 107a05d..d4cc9ce 100644
--- a/src/libscotchmetis/CMakeLists.txt
+++ b/src/libscotchmetis/CMakeLists.txt
@@ -45,6 +45,18 @@
 ##                                                        ##
 ############################################################
 
+################
+#  dummysizes  #
+################
+
+set(DUMMYSIZES_CMD ${CMAKE_CURRENT_BINARY_DIR}/../libscotch/dummysizes)
+set(DUMMYSIZES_DEP dummysizes_tgt)
+
+if(BUILD_PTSCOTCH)
+  set(PTDUMMYSIZES_CMD ${CMAKE_CURRENT_BINARY_DIR}/../libscotch/ptdummysizes)
+  set(PTDUMMYSIZES_DEP ptdummysizes_tgt)
+endif()
+
 ####################
 #  libScotchMeTiS  #
 ####################
@@ -54,7 +66,7 @@ set_source_files_properties(${GENERATED_INCLUDE_DIR}/metis.h PROPERTIES GENERATE
 
 # metis.h
 add_custom_command(OUTPUT ${GENERATED_INCLUDE_DIR}/metis.h
-  COMMAND $<TARGET_FILE:dummysizes> ${CMAKE_CURRENT_SOURCE_DIR}/library_metis.h ${GENERATED_INCLUDE_DIR}/metis.h DEPENDS dummysizes
+  COMMAND ${DUMMYSIZES_CMD} ${CMAKE_CURRENT_SOURCE_DIR}/library_metis.h ${GENERATED_INCLUDE_DIR}/metis.h DEPENDS ${DUMMYSIZES_DEP}
   DEPENDS library_metis.h
   COMMENT "Generating metis.h")
 
@@ -63,7 +75,7 @@ add_custom_target(metis_h
 
 # metisf.h
 add_custom_command(OUTPUT ${GENERATED_INCLUDE_DIR}/metisf.h
-  COMMAND $<TARGET_FILE:dummysizes> ${CMAKE_CURRENT_SOURCE_DIR}/library_metis_f.h ${GENERATED_INCLUDE_DIR}/metisf.h DEPENDS dummysizes
+  COMMAND ${DUMMYSIZES_CMD} ${CMAKE_CURRENT_SOURCE_DIR}/library_metis_f.h ${GENERATED_INCLUDE_DIR}/metisf.h DEPENDS ${DUMMYSIZES_DEP}
   DEPENDS library_metis_f.h
   COMMENT "Generating metisf.h")
 
@@ -150,7 +162,7 @@ if(BUILD_PTSCOTCH)
 
   # parmetis.h
   add_custom_command(OUTPUT ${GENERATED_INCLUDE_DIR}/parmetis.h
-    COMMAND $<TARGET_FILE:dummysizes> ${CMAKE_CURRENT_SOURCE_DIR}/library_parmetis.h ${GENERATED_INCLUDE_DIR}/parmetis.h DEPENDS dummysizes
+    COMMAND ${DUMMYSIZES_CMD} ${CMAKE_CURRENT_SOURCE_DIR}/library_parmetis.h ${GENERATED_INCLUDE_DIR}/parmetis.h DEPENDS ${DUMMYSIZES_DEP}
     DEPENDS library_parmetis.h
     COMMENT "Generating parmetis.h")
 
