--- a/src/Makefile.inc
+++ b/src/Makefile.inc
@@ -9,9 +9,9 @@
 CCS             = $(CC)
 CCP             = mpicc
 CCD             = $(CC)
-CFLAGS          = -O3 -fPIC -Drestrict=__restrict -DCOMMON_PTHREAD_BARRIER -DCOMMON_PTHREAD -DSCOTCH_CHECK_AUTO -DCOMMON_RANDOM_FIXED_SEED -DCOMMON_TIMING_OLD -DSCOTCH_RENAME -DCOMMON_FILE_COMPRESS_GZ -DIDXSIZE64
+CFLAGS          = -O3 -fPIC -DCOMMON_RANDOM_FIXED_SEED -DSCOTCH_RENAME -DCOMMON_STUB_FORK -DCOMMON_WINDOWS -DIDXSIZE64
 CLIBFLAGS       = -shared -fPIC
-LDFLAGS         = -lz -lm -pthread
+LDFLAGS         = -lz -lm
 CP              = cp
 LEX             = flex -Pscotchyy -olex.yy.c
 LN              = ln
