diff --git a/config/FindKLU.cmake b/config/FindKLU.cmake

--- a/config/FindKLU.cmake
+++ b/config/FindKLU.cmake
@@ -60,10 +60,6 @@ endif ()

 if (NOT SUITESPARSECONFIG_LIBRARY)
     set(SUITESPARSECONFIG_LIBRARY_NAME suitesparseconfig)
-    # NOTE: no prefix for this library on windows
-    if(WIN32 AND NOT MSYS)
-        set(CMAKE_FIND_LIBRARY_PREFIXES "")
-    endif()
     find_library( SUITESPARSECONFIG_LIBRARY ${SUITESPARSECONFIG_LIBRARY_NAME} ${KLU_LIBRARY_DIR} NO_DEFAULT_PATH)
     mark_as_advanced(SUITESPARSECONFIG_LIBRARY)
 endif ()
