diff --git a/src/portable.cpp b/src/portable.cpp
index a14afdb95..853d9d74c 100644
--- a/src/portable.cpp
+++ b/src/portable.cpp
@@ -642,7 +642,7 @@ std::ofstream Portable::openOutputStream(const QCString &fileName,bool append)
 #if defined(__clang__) && defined(__MINGW32__)
   return std::ofstream(fs::path(fileName.str()).wstring(), mode);
 #else
-  return std::ofstream(fs::path(fileName.str()), mode);
+  return std::ofstream(fs::path(fileName.str()).string(), mode);
 #endif
 }
 
@@ -654,7 +654,7 @@ std::ifstream Portable::openInputStream(const QCString &fileName,bool binary, bo
 #if defined(__clang__) && defined(__MINGW32__)
   return std::ifstream(fs::path(fileName.str()).wstring(), mode);
 #else
-  return std::ifstream(fs::path(fileName.str()), mode);
+  return std::ifstream(fs::path(fileName.str()).string(), mode);
 #endif
 }
 
