diff --git a/src/tracer/Makefile.am b/src/tracer/Makefile.am
index 0521e2cb..506f9546 100644
--- a/src/tracer/Makefile.am
+++ b/src/tracer/Makefile.am
@@ -633,7 +633,7 @@ endif
 HWC_LINKER_FLAGS = $(PAPI_LINKER_FLAGS) $(PMAPI_LINKER_FLAGS)
 
 if HAVE_XML2
-  XML2_LINKER_FLAGS = @XML2_LDFLAGS@ @XML2_RPATH@ @XML2_LIBS@
+  XML2_LINKER_FLAGS = @XML2_LDFLAGS@ @XML2_RPATH@ @XML2_LIBS@ -lxml2
 endif
 if HAVE_LIBDL
   LDL = -ldl
