diff --git src/tracer/wrappers/CUDA/cuda_wrapper_cupti.c src/tracer/wrappers/CUDA/cuda_wrapper_cupti.c
index c07e14f7..f19dfabe 100644
--- src/tracer/wrappers/CUDA/cuda_wrapper_cupti.c
+++ src/tracer/wrappers/CUDA/cuda_wrapper_cupti.c
@@ -43,6 +43,7 @@
 #include <cupti.h>
 #include "cuda_common.h"
 #include "cuda_probe.h"
+#include "cuda_wrapper_cupti.h"
 
 #define LOG_UNTRACKED_CALLBACKS 0
 #define LOG_OTHER_DOMAIN_UNTRACKED_CALLBACKS 0
diff --git src/tracer/wrappers/CUDA/cuda_wrapper_cupti.h src/tracer/wrappers/CUDA/cuda_wrapper_cupti.h
index 164531e8..6eaf196f 100644
--- src/tracer/wrappers/CUDA/cuda_wrapper_cupti.h
+++ src/tracer/wrappers/CUDA/cuda_wrapper_cupti.h
@@ -25,7 +25,7 @@
 #ifndef CUDA_WRAPPER_CUPTI_H_
 #define CUDA_WRAPPER_CUPTI_H_
 
-#if CUPTI_API_VERSION > 12
+#if CUPTI_API_VERSION > 11
 typedef struct cudaConfigureCall_v3020_params_st {
 	dim3 gridDim;
 	dim3 blockDim;
@@ -36,7 +36,11 @@ typedef struct cudaConfigureCall_v3020_params_st {
 typedef struct cudaStreamDestroy_v3020_params_st {
     cudaStream_t stream;
 } cudaStreamDestroy_v3020_params;
-#endif /* CUPTI_API_VERSION > 12 */
+
+typedef struct cudaLaunch_v3020_params_st {
+    const char *func;
+} cudaLaunch_v3020_params;
+#endif /* CUPTI_API_VERSION > 11 */
 
 void Extrae_CUDA_init (int rank);
 
