q--- a/configure.in
+++ b/configure.in
@@ -514,43 +514,9 @@
 
 # Curl library (will be pulled in to the shared CFITSIO library):
 # ---------------------------------------------------------------
-CURL_INC=""
-CURL_LIB=""
-CURL_LIB_PATH=""
 if test "x$ADD_CURL" = xyes; then
-  # Use curl-config to get compiler & linker flags, if available.
-  # (and prefer /usr/bin/curl-config)
-  AC_PATH_PROG([CURLCONFIG], [curl-config], [], [/usr/bin$PATH_SEPARATOR$PATH], [])
-  if test "x$CURLCONFIG" != x; then
-    CURL_LIB=`$CURLCONFIG --libs`
-    CURL_INC=`$CURLCONFIG --cflags`
-    if test "x$CURL_LIB" != x; then
-      LIBS_CURL="$CURL_LIB"
-      # Mac OS: For third-party curl-config, acquire an rpath:
-      if test `echo $host | grep -c apple-darwin` -ne 0 -a `echo $CURL_LIB | grep -c "^-L"` -gt 0; then
-        CURL_LIB_PATH=`echo ${CURL_LIB} | tr " " "\012" | grep "^-L" | tr "\012" " " | sed 's:-L::' | sed 's:[ ]$::'`
-        if test "x$CURL_LIB_PATH" != x; then
-          LIBS_CURL="-Wl,-rpath,$CURL_LIB_PATH $CURL_LIB"
-        fi
-      fi
-      if test `echo $host | grep -c cygwin` -ne 0 -o `echo $host | grep -c mingw32` -ne 0; then
-          LIBS="$CURL_LIB $LIBS"
-      fi
-      AC_DEFINE(CFITSIO_HAVE_CURL)
-    fi
-    if test "x$CURL_INC" != x; then
-      CFLAGS="$CURL_INC $CFLAGS"
-    fi
-  # No curl-config:
-  else
-    AC_MSG_WARN(curl-config not found. Disabling curl support.)
-  # Incomplete stubs for possible future use:
-  #  AC_CHECK_LIB([curl],[main],[],
-  #       [AC_MSG_WARN(Not building curl support for CFITSIO)])
-  #  AC_CHECK_HEADER(curl.h,[])
-  fi
+  PKG_CHECK_MODULES([CURL], [libcurl >= 7.0], [AC_DEFINE(CFITSIO_HAVE_CURL)])
 fi
-AC_SUBST(LIBS_CURL)
 
 # ZLIB (required):
 if test "x$ZLIB_CHECK" = xyes; then
