--- a/OpenLSTO/M2DO_FEA/include/M2DO_FEA.h
+++ b/OpenLSTO/M2DO_FEA/include/M2DO_FEA.h
@@ -14,8 +14,8 @@
 #include <cassert>
 #include <set>
 
-#include <../../vendor/eigen3/Eigen/Dense>
-#include <../../vendor/eigen3/Eigen/Sparse>
+#include <Eigen/Dense>
+#include <Eigen/Sparse>
 
 using namespace std ;
 using namespace Eigen ;

--- a/OpenLSTO/M2DO_FEA/src/utility_functions.cpp	2021-08-04 17:24:02.124496503 -0400
+++ b/OpenLSTO/M2DO_FEA/src/utility_functions2.cpp	2021-08-04 17:24:16.564322003 -0400
@@ -10,8 +10,8 @@
 #include <cassert>
 #include <set>
 
-#include <../vendor/eigen3/Eigen/Dense>
-#include <../vendor/eigen3/Eigen/Sparse>
+#include <Eigen/Dense>
+#include <Eigen/Sparse>
 
 template <class T>
 void print_vector (vector<T> v) {