diff --git a/CMakeLists.txt b/CMakeLists.txt
index 40cafbfbd..7779eaf98 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1528,6 +1528,7 @@ IF(NOT TARGET cpuinfo)
     "${CPUINFO_SOURCE_DIR}"
     "${CMAKE_BINARY_DIR}/cpuinfo")
 ENDIF()
+SET_PROPERTY(TARGET cpuinfo PROPERTY POSITION_INDEPENDENT_CODE ON)
 TARGET_LINK_LIBRARIES(XNNPACK PRIVATE cpuinfo)
 
 # ---[ Configure pthreadpool
@@ -1539,6 +1540,7 @@ IF(NOT TARGET pthreadpool)
     "${PTHREADPOOL_SOURCE_DIR}"
     "${CMAKE_BINARY_DIR}/pthreadpool")
 ENDIF()
+SET_PROPERTY(TARGET pthreadpool PROPERTY POSITION_INDEPENDENT_CODE ON)
 TARGET_LINK_LIBRARIES(XNNPACK PUBLIC pthreadpool)
 
 # ---[ Configure FXdiv
