diff --git a/src/tbb/allocator.cpp b/src/tbb/allocator.cpp
index bb7dc99..e610ade 100644
--- a/src/tbb/allocator.cpp
+++ b/src/tbb/allocator.cpp
@@ -25,7 +25,7 @@
 #include <cstdlib>
 
 #if _WIN32 || _WIN64
-#include <Windows.h>
+#include <windows.h>
 #else
 #include <dlfcn.h>
 #endif /* _WIN32||_WIN64 */

diff --git a/src/tbb/dynamic_link.h b/src/tbb/dynamic_link.h
index a41493e..f2499da 100644
--- a/src/tbb/dynamic_link.h
+++ b/src/tbb/dynamic_link.h
@@ -27,7 +27,7 @@
 
 #include <cstddef>
 #if _WIN32
-#include <Windows.h>
+#include <windows.h>
 #endif /* _WIN32 */
 
 namespace tbb {
