From fae3817dc7d946c141d84993fd7e7e45b326c3db Mon Sep 17 00:00:00 2001
From: Pepijn de Vos <pepijndevos@gmail.com>
Date: Fri, 20 Aug 2021 16:39:10 +0200
Subject: [PATCH] musl

---
 UART.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/UART.cpp b/UART.cpp
index 36d4124..76b1636 100644
--- a/UART.cpp
+++ b/UART.cpp
@@ -47,6 +47,10 @@
 
 //asm/termios.h seems to conflict with sys/ioctl.h and termios.h
 //so just pull these by hand
+//define __THROW to avoid build issue when it's not available from the libc
+#ifndef __THROW
+# define __THROW
+#endif
 extern "C" int tcflush (int __fd, int __queue_selector) __THROW;
 extern "C" int ioctl (int __fd, unsigned long int __request, ...) __THROW;
 
-- 
2.32.0

