commit ca4f63950c7800faf953d4019bea9bd33755a085
Author: Max Horn <max@quendi.de>
Date:   2020-11-09 12:00:38 +0100

    Ensure alloca can be used on FreeBSD

diff --git a/getopt.c b/getopt.c
index 6be0aaf..d6e4819 100644
--- a/getopt.c
+++ b/getopt.c
@@ -30,6 +30,7 @@
 #endif
 
 #include <stdio.h>
+#include <stdlib.h>
 
 /* Comment out all this code if we are using the GNU C Library, and are not
    actually compiling the library itself.  This code is part of the GNU C
diff --git a/nettle-internal.h b/nettle-internal.h
index 0b0d25c..e2ca7a2 100644
--- a/nettle-internal.h
+++ b/nettle-internal.h
@@ -43,6 +43,7 @@
  * fix maximum size, and abort if we ever need anything larger. */
 
 #if HAVE_ALLOCA
+#include <stdlib.h>
 # define TMP_DECL(name, type, max) type *name
 # define TMP_ALLOC(name, size) (name = alloca(sizeof (*name) * (size)))
 #else /* !HAVE_ALLOCA */
