From 7f7171e68a420991b537d3e9e63263a0b2871618 Mon Sep 17 00:00:00 2001
From: Peter Williams <peter@newton.cx>
Date: Wed, 21 Sep 2022 11:56:33 -0400
Subject: [PATCH] gio: properly guard use of utimensat()

Closes #2766.
---
 gio/glocalfileinfo.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/gio/glocalfileinfo.c b/gio/glocalfileinfo.c
index 6f186957ae..c9e5c8b64c 100644
--- a/gio/glocalfileinfo.c
+++ b/gio/glocalfileinfo.c
@@ -2800,6 +2800,7 @@ set_mtime_atime (char                       *filename,
       return FALSE;
     }
 
+#if defined (HAVE_UTIMENSAT)
   res = utimensat (AT_FDCWD, filename, times_n, 0);
   if (res == -1)
     {
@@ -2811,6 +2812,7 @@ set_mtime_atime (char                       *filename,
                    g_strerror (errsv));
       return FALSE;
     }
+#endif
   return TRUE;
 }
 #endif
-- 
GitLab

