--- a/src/partitioner/PartitionerManager.cpp
+++ b/src/partitioner/PartitionerManager.cpp
@@ -1,7 +1,6 @@
 #include "PartitionerManager.hpp"
 #include "PartitionerFlowCutter.hpp"
-#include "PartitionerKahypar.hpp"
 #include "PartitionerNone.hpp"
 #include "PartitionerShatter.hpp"
 #include <iostream>
@@ -38,9 +37,9 @@
     case PARTITIONER_FLOW_CUTTER:
       return new PartitionerFlowCutter(infoHyperGraph, out);
     case PARTITIONER_KAHYPAR:
-      return new PartitionerKahypar(infoHyperGraph, out);
+      throw std::runtime_error("Partitioner Kahypar not supported");
     case PARTITIONER_PATOH:
-      return new PartitionerPatoh(infoHyperGraph, out);
+      throw std::runtime_error("Partitioner PaToH not supported");
     case PARTITIONER_NONE:
       return new PartitionerNone(infoHyperGraph, out);
     case PARTITIONER_SHATTER:
