--- a/src/problem/circuit/ParserCircuit.cpp
+++ b/src/problem/circuit/ParserCircuit.cpp
@@ -107,7 +107,7 @@
     if (line.rfind("c w ", 0) == 0) {
       std::stringstream ss(line);
       std::string tmp;
       ss >> tmp >> tmp;
-      u_int32_t var;
+      uint32_t var;
       double weight;
       ss >> var >> weight;
       weights[var] = weight;
--- a/3rdParty/bipe/src/solver/Solver.hpp
+++ b/3rdParty/bipe/src/solver/Solver.hpp
@@ -51,7 +51,7 @@
 namespace bipe {
 
 /**
- * @brief u_int32_t variable
+ * @brief uint32_t variable
  *
  */
 using Var = uint32_t;
