Author: Tim Besard <tim.besard@gmail.com>
Description: don't expose all symbols from linked libraries

--- AddLLVM.cmake  2025-05-26 19:07:44.899946198 +0200
+++ AddLLVM.cmake  2025-05-26 19:07:10.042745235 +0200
@@ -797,6 +797,14 @@

+
+  set(llvm_original_libs ${llvm_libs})
+  message(STATUS "Transforming LLVM component links to -hidden-l for target ${name}")
+  set(llvm_libs "")
+  foreach(component_name_item ${llvm_original_libs})
+    list(APPEND llvm_libs "-Wl,-hidden-l${component_name_item}")
+  endforeach()
+
   target_link_libraries(${name} ${libtype}
       ${ARG_LINK_LIBS}
       ${lib_deps}
