diff --git a/mlx/array.cpp b/mlx/array.cpp
index 2c70b9c8..6e18bd90 100644
--- a/mlx/array.cpp
+++ b/mlx/array.cpp
@@ -1,5 +1,6 @@
 // Copyright © 2023-2024 Apple Inc.
 #include <functional>
+#include <unordered_map>
 
 #include "mlx/array.h"
 #include "mlx/ops.h"
diff --git a/mlx/backend/metal/metal.h b/mlx/backend/metal/metal.h
index 830df4f4..e5cb65af 100644
--- a/mlx/backend/metal/metal.h
+++ b/mlx/backend/metal/metal.h
@@ -2,6 +2,7 @@
 
 #pragma once
 
+#include <unordered_map>
 #include <variant>
 
 #include "mlx/array.h"
diff --git a/mlx/graph_utils.h b/mlx/graph_utils.h
index d70228da..affb5e07 100644
--- a/mlx/graph_utils.h
+++ b/mlx/graph_utils.h
@@ -2,6 +2,8 @@
 
 #pragma once
 
+#include <unordered_map>
+
 #include "mlx/array.h"
 
 namespace mlx::core {
diff --git a/mlx/io.h b/mlx/io.h
index 4805d1c8..23380b23 100644
--- a/mlx/io.h
+++ b/mlx/io.h
@@ -2,6 +2,7 @@
 
 #pragma once
 
+#include <unordered_map>
 #include <variant>
 
 #include "mlx/array.h"
