diff --git a/CMakeLists.txt b/CMakeLists.txt
index 54d974a2..0f9b9bb7 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -202,20 +202,7 @@ endif()
 
 find_package(MPI)
 if(MPI_FOUND)
-  execute_process(
-    COMMAND zsh "-c" "mpirun --version"
-    OUTPUT_VARIABLE MPI_VERSION
-    ERROR_QUIET)
-  if(${MPI_VERSION} MATCHES ".*Open MPI.*")
-    target_include_directories(mlx PRIVATE ${MPI_INCLUDE_PATH})
-  elseif(MPI_VERSION STREQUAL "")
-    set(MPI_FOUND FALSE)
-    message(
-      WARNING "MPI found but mpirun is not available. Building without MPI.")
-  else()
-    set(MPI_FOUND FALSE)
-    message(WARNING "MPI which is not OpenMPI found. Building without MPI.")
-  endif()
+  target_include_directories(mlx PRIVATE ${MPI_INCLUDE_PATH})
 endif()
 
 add_subdirectory(${CMAKE_CURRENT_LIST_DIR}/mlx)
