diff -ru a/LIBRARY/src/bid_functions.h b/LIBRARY/src/bid_functions.h
--- a/LIBRARY/src/bid_functions.h	2024-05-07 19:30:17
+++ b/LIBRARY/src/bid_functions.h	2025-10-15 17:44:41
@@ -52,7 +52,7 @@
 // we will use gcc types defined in bid_gcc_intrinsics.h.
 #include "bid_gcc_intrinsics.h"
 
-#define BID_ALIGN(n) __attribute__ ((aligned(n)))
+#define BID_ALIGN(n) // Julia does not support user specified alignment of structs so make this a NOP
 #else
 typedef char BID_SINT8;
 typedef unsigned char BID_UINT8;
@@ -77,11 +77,11 @@
 #if defined _M_IX86 && !defined __INTEL_COMPILER        // Win IA-32, MS compiler
 #define BID_ALIGN(n)
 #else
-#define BID_ALIGN(n) __declspec(align(n))
+#define BID_ALIGN(n) // Julia does not support user specified alignment of structs so make this a NOP
 #endif
 #else
 #if !defined HPUX_OS
-#define BID_ALIGN(n) __attribute__ ((aligned(n)))
+#define BID_ALIGN(n) // Julia does not support user specified alignment of structs so make this a NOP
 #else
 #define BID_ALIGN(n)
 #endif
diff -ru a/TESTS/test_bid_functions.h b/TESTS/test_bid_functions.h
--- a/TESTS/test_bid_functions.h	2024-05-07 19:30:18
+++ b/TESTS/test_bid_functions.h	2025-10-15 17:45:02
@@ -52,7 +52,7 @@
 // we will use gcc types defined in bid_gcc_intrinsics.h.
 #include "bid_gcc_intrinsics.h"
 
-#define BID_ALIGN(n) __attribute__ ((aligned(n)))
+#define BID_ALIGN(n) // Julia does not support user specified alignment of structs so make this a NOP
 #else
 typedef char BID_SINT8;
 typedef unsigned char BID_UINT8;
@@ -77,11 +77,11 @@
 #if defined _M_IX86 && !defined __INTEL_COMPILER        // Win IA-32, MS compiler
 #define BID_ALIGN(n)
 #else
-#define BID_ALIGN(n) __declspec(align(n))
+#define BID_ALIGN(n) // Julia does not support user specified alignment of structs so make this a NOP
 #endif
 #else
 #if !defined HPUX_OS
-#define BID_ALIGN(n) __attribute__ ((aligned(n)))
+#define BID_ALIGN(n) // Julia does not support user specified alignment of structs so make this a NOP
 #else
 #define BID_ALIGN(n)
 #endif
