diff --git a/src/client/linux/dump_writer_common/thread_info.h b/src/client/linux/dump_writer_common/thread_info.h
index fb216fa6..4a9ccaa2 100644
--- a/src/client/linux/dump_writer_common/thread_info.h
+++ b/src/client/linux/dump_writer_common/thread_info.h
@@ -33,6 +33,24 @@
 #include <sys/ucontext.h>
 #include <sys/user.h>
 
+// Patch for older glibc's not containing this structure
+#if defined(__aarch64__) && __GLIBC__ == 2 && __GLIBC_MINOR__ < 20
+struct user_regs_struct
+{
+  unsigned long long regs[31];
+  unsigned long long sp;
+  unsigned long long pc;
+  unsigned long long pstate;
+};
+
+struct user_fpsimd_struct
+{
+  __uint128_t  vregs[32];
+  unsigned int fpsr;
+  unsigned int fpcr;
+};
+#endif
+
 #include "client/linux/dump_writer_common/raw_context_cpu.h"
 #include "common/memory_allocator.h"
 #include "google_breakpad/common/minidump_format.h"
