diff --git a/CMakeLists.txt b/CMakeLists.txt
index bf38bfc..d8f68ec 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -149,7 +149,9 @@ configure_file(
   ${CMAKE_CURRENT_SOURCE_DIR}/include/amd_comgr.h.in
   ${CMAKE_CURRENT_BINARY_DIR}/include/amd_comgr.h @ONLY)
 
-include(bc2h)
+#include(bc2h)
+add_executable(bc2h IMPORTED)
+set_target_properties(bc2h PROPERTIES IMPORTED_LOCATION ${CMAKE_CURRENT_BINARY_DIR}/bc2h)
 include(opencl_pch)
 include(DeviceLibs)
 
