diff --git a/CMakeLists.txt b/CMakeLists.txt
index bf38bfc..8cc6908 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -291,13 +291,6 @@ if (NOT UNIX)
     PRIVATE version)
 endif()
 
-enable_testing()
-add_custom_target(check-comgr COMMAND ${CMAKE_CTEST_COMMAND} DEPENDS amd_comgr)
-if (NOT CMAKE_SOURCE_DIR STREQUAL CMAKE_CURRENT_SOURCE_DIR)
-  set_property(GLOBAL APPEND PROPERTY LLVM_ADDITIONAL_TEST_TARGETS check-comgr)
-endif()
-add_subdirectory(test)
-
 if(CMAKE_SOURCE_DIR STREQUAL CMAKE_CURRENT_SOURCE_DIR)
   # Add packaging directives for amd_comgr
   set(CPACK_PACKAGE_NAME comgr)
