diff --git a/src/comgr-objdump.cpp b/src/comgr-objdump.cpp
index 1fdb1df..260db7e 100644
--- a/src/comgr-objdump.cpp
+++ b/src/comgr-objdump.cpp
@@ -174,9 +174,9 @@ cl::opt<bool> SectionHeaders("section-headers",
 static cl::alias SectionHeadersShort("headers",
                                      cl::desc("Alias for --section-headers"),
                                      cl::aliasopt(SectionHeaders));
-static cl::alias SectionHeadersShorter("h",
+/*static cl::alias SectionHeadersShorter("h",
                                        cl::desc("Alias for --section-headers"),
-                                       cl::aliasopt(SectionHeaders));
+                                       cl::aliasopt(SectionHeaders));*/
 
 cl::list<std::string>
     FilterSections("section",
