diff --git a/configure.ac b/configure.ac
index 1df2e99..c790d5b 100644
--- a/configure.ac
+++ b/configure.ac
@@ -9,6 +9,18 @@ AC_PROG_CC
 AC_CHECK_HEADER([getopt.h],
     [AC_DEFINE([HAVE_GETOPT_H], 1, [Define to 1 if you have <getopt.h>.])],
 		[AC_MSG_ERROR([utftex needs getopt support to run. Please install it.])])
+AC_MSG_CHECKING([if libtool needs -no-undefined flag to build shared libraries])
+case "$host_os" in
+  cygwin*|msys*|mingw*)
+    ## Add in the -no-undefined flag to LDFLAGS for libtool.
+    AC_MSG_RESULT([yes])
+    LDFLAGS="$LDFLAGS -no-undefined"
+    ;;
+  *)
+    ## Don't add in anything.
+    AC_MSG_RESULT([no])
+    ;;
+esac
 AC_DEFINE([UTFTEXVERSION], ["1.11"], [utftex version string])
 AC_CONFIG_SRCDIR([src/boxes.c])
 AC_CONFIG_HEADERS([src/config.h])
