diff -ruN Birch.orig/libraries/Standard/src/system/stdio.birch Birch/libraries/Standard/src/system/stdio.birch
--- Birch.orig/libraries/Standard/src/system/stdio.birch	2020-11-24 01:37:30.887366775 +0100
+++ Birch/libraries/Standard/src/system/stdio.birch	2020-11-24 01:38:08.909768931 +0100
@@ -1,3 +1,9 @@
+hpp{{
+  #undef stdin
+  #undef stdout
+  #undef stderr
+}}
+
 /*
  * Returns the file associated with stdin.
  */
