--- fflas-ffpack-2.5.0.orig/fflas-ffpack/config-blas.h	2024-08-04 15:41:34
+++ fflas-ffpack-2.5.0/fflas-ffpack/config-blas.h	2024-08-04 17:00:24
@@ -128,6 +128,8 @@
     void strmm_ (const char*, const char*, const char*, const char*, const int*, const int*, const float*, const float*, const int*, float*, const int*);
     void sgemm_ (const char*, const char*, const int*, const int*, const int*, const float*, const float*, const int*, const float*, const int*, const float*, float*, const int*);
     void dgemm_ (const char*, const char*, const int*, const int*, const int*, const double*, const double*, const int*, const double*, const int*, const double*, double*, const int*);
+    void ssyrk_ (const char *, const char *, int, int, float, const float  *, int, float, float *, int); // missing?
+    void dsyrk_ (const char *, const char *, int, int, double, const double *, int, double, double *, int);
 }
 
 // define C wrappers
