diff -ru libaom-3.12.1.orig/aom_dsp/grain_table.c libaom-3.12.1/aom_dsp/grain_table.c
--- libaom-3.12.1.orig/aom_dsp/grain_table.c	2025-04-11 16:37:49
+++ libaom-3.12.1/aom_dsp/grain_table.c	2025-07-18 12:14:46
@@ -30,6 +30,7 @@
  *  cCr <ar_coeff_cr_0> ....
  * E <start-time> ...
  */
+#include <inttypes.h>
 #include <string.h>
 #include <stdio.h>
 #include "aom_dsp/aom_dsp_common.h"
diff -ru libaom-3.12.1.orig/apps/aomdec.c libaom-3.12.1/apps/aomdec.c
--- libaom-3.12.1.orig/apps/aomdec.c	2025-04-11 16:37:49
+++ libaom-3.12.1/apps/aomdec.c	2025-07-18 12:13:29
@@ -9,6 +9,7 @@
  * PATENTS file, you can obtain it at www.aomedia.org/license/patent.
  */
 
+#include <inttypes.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <stdarg.h>
diff -ru libaom-3.12.1.orig/apps/aomenc.c libaom-3.12.1/apps/aomenc.c
--- libaom-3.12.1.orig/apps/aomenc.c	2025-04-11 16:37:49
+++ libaom-3.12.1/apps/aomenc.c	2025-07-18 12:14:52
@@ -14,6 +14,7 @@
 #include "config/aom_config.h"
 
 #include <assert.h>
+#include <inttypes.h>
 #include <limits.h>
 #include <math.h>
 #include <stdarg.h>
diff -ru libaom-3.12.1.orig/av1/common/av1_txfm.c libaom-3.12.1/av1/common/av1_txfm.c
--- libaom-3.12.1.orig/av1/common/av1_txfm.c	2025-04-11 16:37:49
+++ libaom-3.12.1/av1/common/av1_txfm.c	2025-07-18 12:13:29
@@ -9,6 +9,8 @@
  * PATENTS file, you can obtain it at www.aomedia.org/license/patent.
  */
 
+#include <inttypes.h>
+
 #include "config/aom_dsp_rtcd.h"
 #include "config/av1_rtcd.h"
 
diff -ru libaom-3.12.1.orig/av1/encoder/partition_search.c libaom-3.12.1/av1/encoder/partition_search.c
--- libaom-3.12.1.orig/av1/encoder/partition_search.c	2025-04-11 16:37:49
+++ libaom-3.12.1/av1/encoder/partition_search.c	2025-07-18 12:13:29
@@ -10,6 +10,7 @@
  */
 
 #include <float.h>
+#include <inttypes.h>
 
 #include "config/aom_config.h"
 
diff -ru libaom-3.12.1.orig/av1/encoder/tx_search.c libaom-3.12.1/av1/encoder/tx_search.c
--- libaom-3.12.1.orig/av1/encoder/tx_search.c	2025-04-11 16:37:49
+++ libaom-3.12.1/av1/encoder/tx_search.c	2025-07-18 12:13:29
@@ -9,6 +9,8 @@
  * PATENTS file, you can obtain it at www.aomedia.org/license/patent.
  */
 
+#include <inttypes.h>
+
 #include "av1/common/cfl.h"
 #include "av1/common/reconintra.h"
 #include "av1/encoder/block.h"
diff -ru libaom-3.12.1.orig/examples/multilayer_metadata.cc libaom-3.12.1/examples/multilayer_metadata.cc
--- libaom-3.12.1.orig/examples/multilayer_metadata.cc	2025-04-11 16:37:49
+++ libaom-3.12.1/examples/multilayer_metadata.cc	2025-07-18 12:23:22
@@ -12,12 +12,12 @@
 #include "examples/multilayer_metadata.h"
 
 #include <assert.h>
-#include <inttypes.h>
 #include <limits.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
 
+#include <cinttypes>
 #include <cmath>
 #include <fstream>
 #include <iostream>
diff -ru libaom-3.12.1.orig/examples/noise_model.c libaom-3.12.1/examples/noise_model.c
--- libaom-3.12.1.orig/examples/noise_model.c	2025-04-11 16:37:49
+++ libaom-3.12.1/examples/noise_model.c	2025-07-18 12:14:41
@@ -38,6 +38,7 @@
  *    --auto-alt-ref=2 --bit-depth=8 --film-grain-table=film_grain.tbl \
  *    -o denoised_with_grain_params.ivf denoised.854_480.yuv
  */
+#include <inttypes.h>
 #include <math.h>
 #include <stdio.h>
 #include <stdlib.h>
diff -ru libaom-3.12.1.orig/examples/svc_encoder_rtc.cc libaom-3.12.1/examples/svc_encoder_rtc.cc
--- libaom-3.12.1.orig/examples/svc_encoder_rtc.cc	2025-04-11 16:37:49
+++ libaom-3.12.1/examples/svc_encoder_rtc.cc	2025-07-18 12:21:51
@@ -13,13 +13,13 @@
 //  encoding scheme for RTC video applications.
 
 #include <assert.h>
-#include <inttypes.h>
 #include <limits.h>
 #include <math.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
 
+#include <cinttypes>
 #include <memory>
 
 #include "config/aom_config.h"
diff -ru libaom-3.12.1.orig/test/wiener_test.cc libaom-3.12.1/test/wiener_test.cc
--- libaom-3.12.1.orig/test/wiener_test.cc	2025-04-11 16:37:49
+++ libaom-3.12.1/test/wiener_test.cc	2025-07-18 12:13:29
@@ -9,6 +9,7 @@
  * PATENTS file, you can obtain it at www.aomedia.org/license/patent.
  */
 
+#include <cinttypes>
 #include <tuple>
 #include <utility>
 #include <vector>
