--- before/whisper.cpp/CMakeLists.txt	2023-02-08 16:43:45.440934329 -0500
+++ after/whisper.cpp/CMakeLists.txt	2023-02-08 16:45:07.143100006 -0500
@@ -225,11 +225,13 @@
 target_compile_definitions(${TARGET} PUBLIC
     ${WHISPER_EXTRA_FLAGS}
     )
+set_target_properties(${TARGET} PROPERTIES PUBLIC_HEADER "whisper.h")
 
 install(TARGETS ${TARGET}
     LIBRARY DESTINATION lib
     ARCHIVE DESTINATION lib/static
     RUNTIME DESTINATION bin
+    PUBLIC_HEADER DESTINATION include
     )
 
 #
