diff --git a/lib/cdt/Makefile.am b/lib/cdt/Makefile.am
index 2f8455168..6101edf29 100644
--- a/lib/cdt/Makefile.am
+++ b/lib/cdt/Makefile.am
@@ -6,7 +6,7 @@ pdfdir = $(pkgdatadir)/doc/pdf
 pkgextraincludedir = $(pkgincludedir)
 pkgconfigdir = $(libdir)/pkgconfig
 
-AM_CPPFLAGS = -I$(top_srcdir)/lib
+AM_CPPFLAGS = -I$(top_srcdir)/lib -DEXPORT_CDT
 
 pkginclude_HEADERS = cdt.h
 noinst_HEADERS = dthdr.h
diff --git a/lib/cgraph/Makefile.am b/lib/cgraph/Makefile.am
index f5c78e429..e087f2c33 100644
--- a/lib/cgraph/Makefile.am
+++ b/lib/cgraph/Makefile.am
@@ -5,7 +5,7 @@ CGRAPH_VERSION="6:0:0"
 pdfdir = $(pkgdatadir)/doc/pdf
 pkgconfigdir = $(libdir)/pkgconfig
 
-AM_CPPFLAGS = -I$(top_srcdir)/lib -I$(top_srcdir)/lib/cdt
+AM_CPPFLAGS = -I$(top_srcdir)/lib -I$(top_srcdir)/lib/cdt -DEXPORT_CGRAPH -DEXPORT_AGXBUF -DEXPORT_CGHDR
 
 pkginclude_HEADERS = cgraph.h
 noinst_HEADERS = agxbuf.h cghdr.h itos.h likely.h sprint.h strcasecmp.h \
diff --git a/lib/circogen/Makefile.am b/lib/circogen/Makefile.am
index 056e61c43..be4526828 100644
--- a/lib/circogen/Makefile.am
+++ b/lib/circogen/Makefile.am
@@ -9,7 +9,8 @@ AM_CPPFLAGS = \
         -I$(top_srcdir)/lib/pack \
         -I$(top_srcdir)/lib/pathplan \
         -I$(top_srcdir)/lib/cgraph \
-        -I$(top_srcdir)/lib/cdt
+        -I$(top_srcdir)/lib/cdt \
+        -DNEATO_EXPORTS
 
 noinst_HEADERS = block.h blockpath.h blocktree.h circo.h \
 	circpos.h circular.h deglist.h edgelist.h nodelist.h \
diff --git a/lib/common/Makefile.am b/lib/common/Makefile.am
index 16d81e876..76e879408 100644
--- a/lib/common/Makefile.am
+++ b/lib/common/Makefile.am
@@ -8,7 +8,8 @@ AM_CPPFLAGS = \
 	-I$(top_srcdir)/lib/xdot \
 	-I$(top_srcdir)/lib/pathplan \
 	-I$(top_srcdir)/lib/cgraph \
-	-I$(top_srcdir)/lib/cdt $(GD_INCLUDES) $(EXPAT_INCLUDES) $(Z_INCLUDES)
+	-I$(top_srcdir)/lib/cdt $(GD_INCLUDES) $(EXPAT_INCLUDES) $(Z_INCLUDES) \
+	-DGVC_EXPORTS
 
 if WITH_WIN32
 AM_CFLAGS = -D_BLD_gvc=1
diff --git a/lib/common/memory.h b/lib/common/memory.h
index 15b08e441..7bcb328cd 100644
--- a/lib/common/memory.h
+++ b/lib/common/memory.h
@@ -26,16 +26,12 @@ extern "C" {
 #define RALLOC(size,ptr,type) ((type*)grealloc(ptr,(size)*sizeof(type)))
 #define ZALLOC(size,ptr,type,osize) (ptr? (type*)zrealloc(ptr,size,sizeof(type),osize):(type*)zmalloc((size)*sizeof(type)))
 #ifdef GVDLL
+#ifdef GVC_EXPORTS
 #define MEMORY_API __declspec(dllexport)
 #else
-#ifdef _WIN32
-#ifndef GVC_EXPORTS
 #define MEMORY_API __declspec(dllimport)
 #endif
-#endif
-
-#endif
-#ifndef MEMORY_API
+#else
 #define MEMORY_API extern
 #endif
 
diff --git a/lib/fdpgen/Makefile.am b/lib/fdpgen/Makefile.am
index 1f8d2fb40..ad97419a4 100644
--- a/lib/fdpgen/Makefile.am
+++ b/lib/fdpgen/Makefile.am
@@ -7,7 +7,8 @@ AM_CPPFLAGS = \
         -I$(top_srcdir)/lib/pack \
         -I$(top_srcdir)/lib/pathplan \
         -I$(top_srcdir)/lib/cgraph \
-        -I$(top_srcdir)/lib/cdt
+        -I$(top_srcdir)/lib/cdt \
+        -DNEATO_EXPORTS
 
 noinst_HEADERS = comp.h dbg.h fdp.h grid.h tlayout.h xlayout.h \
 	clusteredges.h
diff --git a/lib/gvc/Makefile.am b/lib/gvc/Makefile.am
index 71f7baa69..0cae3d1f0 100644
--- a/lib/gvc/Makefile.am
+++ b/lib/gvc/Makefile.am
@@ -11,7 +11,8 @@ AM_CPPFLAGS = \
 	-I$(top_srcdir)/lib/pathplan \
 	-I$(top_srcdir)/lib/cgraph \
 	-I$(top_srcdir)/lib/cdt \
-	$(INCLTDL) -DGVLIBDIR=\"$(pkglibdir)\"
+	$(INCLTDL) -DGVLIBDIR=\"$(pkglibdir)\" \
+	-DGVC_EXPORTS
 
 if WITH_WIN32
 AM_CFLAGS = -D_BLD_gvc=1
@@ -48,12 +49,10 @@ libgvc_C_la_SOURCES = gvrender.c gvlayout.c gvdevice.c gvloadimage.c \
 
 libgvc_C_la_LIBADD = \
 	$(top_builddir)/lib/pack/libpack_C.la \
-	$(top_builddir)/lib/xdot/libxdot_C.la \
 	$(top_builddir)/lib/label/liblabel_C.la \
 	$(top_builddir)/lib/common/libcommon_C.la
 libgvc_C_la_DEPENDENCIES = \
 	$(top_builddir)/lib/pack/libpack_C.la \
-	$(top_builddir)/lib/xdot/libxdot_C.la \
 	$(top_builddir)/lib/label/liblabel_C.la \
 	$(top_builddir)/lib/common/libcommon_C.la
 if WITH_ORTHO
diff --git a/lib/gvpr/Makefile.am b/lib/gvpr/Makefile.am
index 087c270b9..53e3a1c75 100644
--- a/lib/gvpr/Makefile.am
+++ b/lib/gvpr/Makefile.am
@@ -11,7 +11,8 @@ AM_CPPFLAGS = \
 	-I$(top_builddir)/lib/common \
 	-I$(top_srcdir)/lib/cgraph \
 	-I$(top_srcdir)/lib/cdt \
-	-DDFLT_GVPRPATH="\".$(PATH_SEPARATOR)$(pkgdatadir)/gvpr\""
+	-DDFLT_GVPRPATH="\".$(PATH_SEPARATOR)$(pkgdatadir)/gvpr\"" \
+	-DEXPORT_GVPR
 
 pkginclude_HEADERS = gvpr.h
 noinst_HEADERS = actions.h compile.h gdefs.h gprdata.inc gprstate.h parse.h \
diff --git a/lib/label/Makefile.am b/lib/label/Makefile.am
index 8f6e1a41e..753fc81be 100644
--- a/lib/label/Makefile.am
+++ b/lib/label/Makefile.am
@@ -4,7 +4,8 @@ AM_CPPFLAGS = \
 	-I$(top_srcdir)/lib \
 	-I$(top_srcdir)/lib/common \
 	-I$(top_srcdir)/lib/cgraph \
-    -I$(top_srcdir)/lib/cdt
+    -I$(top_srcdir)/lib/cdt \
+    -DGVC_EXPORTS
 
 noinst_HEADERS = xlabels.h index.h node.h rectangle.h split.q.h
 noinst_LTLIBRARIES = liblabel_C.la
diff --git a/lib/neatogen/Makefile.am b/lib/neatogen/Makefile.am
index c67e72227..253334e8e 100644
--- a/lib/neatogen/Makefile.am
+++ b/lib/neatogen/Makefile.am
@@ -7,7 +7,8 @@ AM_CPPFLAGS = \
         -I$(top_srcdir)/lib/pack \
         -I$(top_srcdir)/lib/pathplan \
         -I$(top_srcdir)/lib/cgraph \
-        -I$(top_srcdir)/lib/cdt $(IPSEPCOLA_INCLUDES) $(GTS_CFLAGS)
+        -I$(top_srcdir)/lib/cdt $(IPSEPCOLA_INCLUDES) $(GTS_CFLAGS) \
+        -DNEATO_EXPORTS
 
 noinst_LTLIBRARIES = libneatogen_C.la
 
diff --git a/lib/neatogen/neatoprocs.h b/lib/neatogen/neatoprocs.h
index 8eec1516e..9ddb8c16c 100644
--- a/lib/neatogen/neatoprocs.h
+++ b/lib/neatogen/neatoprocs.h
@@ -15,6 +15,16 @@ extern "C" {
 #endif
 #include <neatogen/adjust.h>
 
+#ifdef _WIN32
+#ifdef NEATO_EXPORTS
+#define NEATOGEN_API __declspec(dllexport)
+#else
+#define NEATOGEN_API __declspec(dllimport)
+#endif
+#else
+#define NEATOGEN_API extern
+#endif
+
     extern void avoid_cycling(graph_t *, Agnode_t *, double *);
     extern int checkStart(graph_t * G, int nG, int);
     extern Agnode_t *choose_node(graph_t *, int);
@@ -37,11 +47,11 @@ extern "C" {
     extern void makeSpline(edge_t *, Ppoly_t **, int, boolean);
     extern void make_spring(graph_t *, Agnode_t *, Agnode_t *, double);
     extern void move_node(graph_t *, int, Agnode_t *);
-    extern int init_nop(graph_t * g, int);
+    NEATOGEN_API int init_nop(graph_t * g, int);
     extern void neato_cleanup(graph_t * g);
     extern node_t *neato_dequeue(void);
     extern void neato_enqueue(node_t *);
-    extern void neato_init_node(node_t * n);
+    NEATOGEN_API void neato_init_node(node_t * n);
     extern void neato_layout(Agraph_t * g);
     extern int Plegal_arrangement(Ppoly_t ** polys, int n_polys);
     extern void randompos(Agnode_t *, int);
@@ -62,7 +72,7 @@ extern "C" {
     extern void neato_translate(Agraph_t * g);
     extern boolean neato_set_aspect(graph_t * g);
     extern void toggle(int);
-    extern int user_pos(Agsym_t *, Agsym_t *, Agnode_t *, int);
+    NEATOGEN_API int user_pos(Agsym_t *, Agsym_t *, Agnode_t *, int);
     extern double **new_array(int i, int j, double val);
     extern void free_array(double **rv);
     extern int matinv(double **A, double **Ainv, int n);
diff --git a/lib/osage/Makefile.am b/lib/osage/Makefile.am
index 1c73b781f..46a99b29b 100644
--- a/lib/osage/Makefile.am
+++ b/lib/osage/Makefile.am
@@ -7,7 +7,8 @@ AM_CPPFLAGS = \
 	-I$(top_srcdir)/lib/pack \
 	-I$(top_srcdir)/lib/pathplan \
 	-I$(top_srcdir)/lib/cgraph \
-	-I$(top_srcdir)/lib/cdt
+	-I$(top_srcdir)/lib/cdt \
+	-DNEATO_EXPORTS
 
 noinst_HEADERS = osage.h
 noinst_LTLIBRARIES = libosage_C.la
diff --git a/lib/pack/Makefile.am b/lib/pack/Makefile.am
index 84eb37849..68d0ffdbc 100644
--- a/lib/pack/Makefile.am
+++ b/lib/pack/Makefile.am
@@ -9,7 +9,8 @@ AM_CPPFLAGS = \
 	-I$(top_srcdir)/lib/gvc \
 	-I$(top_srcdir)/lib/pathplan \
 	-I$(top_srcdir)/lib/cgraph \
-	-I$(top_srcdir)/lib/cdt
+	-I$(top_srcdir)/lib/cdt \
+	-DGVC_EXPORTS
 
 if WITH_WIN32
 AM_CFLAGS = -D_BLD_gvc=1
diff --git a/lib/patchwork/Makefile.am b/lib/patchwork/Makefile.am
index bc9a7b2e4..36f5702b5 100644
--- a/lib/patchwork/Makefile.am
+++ b/lib/patchwork/Makefile.am
@@ -7,7 +7,8 @@ AM_CPPFLAGS = \
 	-I$(top_srcdir)/lib/pack \
 	-I$(top_srcdir)/lib/pathplan \
 	-I$(top_srcdir)/lib/cgraph \
-	-I$(top_srcdir)/lib/cdt
+	-I$(top_srcdir)/lib/cdt \
+	-DNEATO_EXPORTS
 
 noinst_HEADERS = patchwork.h tree_map.h
 noinst_LTLIBRARIES = libpatchwork_C.la
diff --git a/lib/pathplan/Makefile.am b/lib/pathplan/Makefile.am
index e4ef47000..06af4cc44 100644
--- a/lib/pathplan/Makefile.am
+++ b/lib/pathplan/Makefile.am
@@ -5,7 +5,7 @@ PATHPLAN_VERSION="4:0:0"
 pdfdir = $(pkgdatadir)/doc/pdf
 pkgconfigdir = $(libdir)/pkgconfig
 
-AM_CPPFLAGS = -I$(top_srcdir)/lib
+AM_CPPFLAGS = -I$(top_srcdir)/lib -DPATHPLAN_EXPORTS
 
 if WITH_WIN32
 AM_CPPFLAGS += -O0
diff --git a/lib/sfdpgen/Makefile.am b/lib/sfdpgen/Makefile.am
index 908ef7231..a86c27814 100644
--- a/lib/sfdpgen/Makefile.am
+++ b/lib/sfdpgen/Makefile.am
@@ -7,7 +7,8 @@ AM_CPPFLAGS = \
 	-I$(top_srcdir)/lib/pack \
 	-I$(top_srcdir)/lib/pathplan \
 	-I$(top_srcdir)/lib/cgraph \
-	-I$(top_srcdir)/lib/cdt
+	-I$(top_srcdir)/lib/cdt \
+	-DNEATO_EXPORTS
 
 noinst_HEADERS = sfdpinternal.h spring_electrical.h \
 	sparse_solve.h post_process.h \
diff --git a/lib/twopigen/Makefile.am b/lib/twopigen/Makefile.am
index 8eb651e6a..195065702 100644
--- a/lib/twopigen/Makefile.am
+++ b/lib/twopigen/Makefile.am
@@ -7,7 +7,8 @@ AM_CPPFLAGS = \
 	-I$(top_srcdir)/lib/pack \
 	-I$(top_srcdir)/lib/pathplan \
 	-I$(top_srcdir)/lib/cgraph \
-	-I$(top_srcdir)/lib/cdt
+	-I$(top_srcdir)/lib/cdt \
+	-DNEATO_EXPORTS
 
 noinst_HEADERS = circle.h
 noinst_LTLIBRARIES = libtwopigen_C.la
diff --git a/lib/xdot/Makefile.am b/lib/xdot/Makefile.am
index acf2bdf23..0e542171f 100644
--- a/lib/xdot/Makefile.am
+++ b/lib/xdot/Makefile.am
@@ -7,7 +7,7 @@ XDOT_VERSION="4:0:0"
 pdfdir = $(pkgdatadir)/doc/pdf
 pkginclude_HEADERS = xdot.h
 
-AM_CPPFLAGS = -I$(top_srcdir)/lib
+AM_CPPFLAGS = -I$(top_srcdir)/lib -DEXPORT_XDOT
 
 noinst_LTLIBRARIES = libxdot_C.la
 lib_LTLIBRARIES = libxdot.la
