diff --git a/Makefile.am b/Makefile.am
index e2f3244..2f0b6c2 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -167,9 +167,9 @@ librsvg_@RSVG_API_MAJOR_VERSION@_la_LDFLAGS = \
 	$(AM_LDFLAGS)
 
 librsvg_@RSVG_API_MAJOR_VERSION@_la_LIBADD = \
-	$(LIBRSVG_LIBS) 	\
-	$(LIBM)			\
-	$(RUST_LIB)
+	$(RUST_LIB)     \
+	$(LIBRSVG_LIBS)	\
+	$(LIBM)
 
 librsvgincdir = $(includedir)/librsvg-$(RSVG_API_VERSION)/librsvg
 librsvginc_HEADERS = 	\
