commit 42352e948cbd1b8df4b26766415de7020e7b8ee6 (HEAD -> master)
Author: Keno Fischer <keno@juliacomputing.com>
Date:   Sat Dec 11 02:39:59 2021 -0500

    signal_x86_64: Handle build with old glibc

diff --git a/dlls/ntdll/unix/signal_x86_64.c b/dlls/ntdll/unix/signal_x86_64.c
index 68e0c7ce66e..16f232dd366 100644
--- a/dlls/ntdll/unix/signal_x86_64.c
+++ b/dlls/ntdll/unix/signal_x86_64.c
@@ -64,6 +64,13 @@
 #ifdef __APPLE__
 # include <mach/mach.h>
 #endif
+#ifdef __linux__
+#if (((__GLIBC__ * 100) + __GLIBC_MINOR__) >= 216)
+#ifndef AT_HWCAP2
+#define AT_HWCAP2 26
+#endif
+#endif
+#endif
 
 #define NONAMELESSUNION
 #define NONAMELESSSTRUCT
@@ -2961,7 +2968,9 @@ void signal_init_process(void)
         {
             fs32_sel = (sel << 3) | 3;
             syscall_flags |= SYSCALL_HAVE_PTHREAD_TEB;
+#ifdef AT_HWCAP2
             if (getauxval( AT_HWCAP2 ) & 2) syscall_flags |= SYSCALL_HAVE_WRFSGSBASE;
+#endif
         }
         else ERR_(seh)( "failed to allocate %%fs selector\n" );
     }
