From 355567871e97b68ad38a69821ecb63f66d2cc6a2 Mon Sep 17 00:00:00 2001
From: Marco Matthies <71844+marcom@users.noreply.github.com>
Date: Mon, 19 Jun 2023 10:16:26 +0200
Subject: [PATCH] Fix missing include for size_t, breaks parsing of header
 files

---
 src/ViennaRNA/constraints/soft.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/ViennaRNA/constraints/soft.h b/src/ViennaRNA/constraints/soft.h
index 88f6ada1..85c597d6 100644
--- a/src/ViennaRNA/constraints/soft.h
+++ b/src/ViennaRNA/constraints/soft.h
@@ -25,6 +25,8 @@
  */
 typedef struct  vrna_sc_s vrna_sc_t;
 
+#include <stdlib.h>
+
 #include <ViennaRNA/datastructures/basic.h>
 #include <ViennaRNA/fold_compound.h>
 #include <ViennaRNA/constraints/basic.h>
-- 
2.34.1

