diff --git a/src/runtime_src/tools/xclbinutil/aie-pdi-transform/src/CMakeLists.txt b/src/runtime_src/tools/xclbinutil/aie-pdi-transform/src/CMakeLists.txt
index 1354017cc..6cf6f3330 100644
--- a/src/runtime_src/tools/xclbinutil/aie-pdi-transform/src/CMakeLists.txt
+++ b/src/runtime_src/tools/xclbinutil/aie-pdi-transform/src/CMakeLists.txt
@@ -1,7 +1,7 @@
 # SPDX-License-Identifier: Apache-2.0
 # Copyright (C) 2018-2022 Xilinx, Inc. All rights Reserved.
 # Copyright (C) 2022-2026 Advanced Micro Devices, Inc. All rights Reserved.
-if(NOT WIN32)
+if(NOT MSVC)
   add_compile_options(-Wextra -fvisibility=default)
 else()
   add_compile_options(/wd4244 /wd4267 /wd4477 /wd4245)
