diff --git a/cmake/install.cmake b/cmake/install.cmake
index 90bae8b5..ae474459 100644
--- a/cmake/install.cmake
+++ b/cmake/install.cmake
@@ -62,10 +62,10 @@ ENDIF()
 #
 
 SET(INSTALL_BINDIR_DEFAULT "bin")
-SET(INSTALL_LIBDIR_DEFAULT "lib/mariadb")
+SET(INSTALL_LIBDIR_DEFAULT "lib")
 SET(INSTALL_PCDIR_DEFAULT "lib/pkgconfig")
 SET(INSTALL_INCLUDEDIR_DEFAULT "include/mariadb")
-SET(INSTALL_DOCDIR_DEFAULT "docs")
+SET(INSTALL_DOCDIR_DEFAULT "share/docs/mariadb")
 SET(INSTALL_MANDIR_DEFAULT "man")
 IF(NOT IS_SUBPROJECT)
   SET(INSTALL_PLUGINDIR_DEFAULT "lib/mariadb/plugin")
diff --git a/libmariadb/CMakeLists.txt b/libmariadb/CMakeLists.txt
index 7faf8275..953180af 100644
--- a/libmariadb/CMakeLists.txt
+++ b/libmariadb/CMakeLists.txt
@@ -466,6 +466,13 @@ SET_TARGET_PROPERTIES(libmariadb PROPERTIES VERSION
  ${CPACK_PACKAGE_VERSION_MAJOR}
  SOVERSION ${CPACK_PACKAGE_VERSION_MAJOR})
 
+IF(MINGW)
+  SET_TARGET_PROPERTIES(libmariadb PROPERTIES
+                        OUTPUT_NAME mariadb
+                        RUNTIME_OUTPUT_NAME libmariadb
+                        ARCHIVE_OUTPUT_NAME mariadb)
+ENDIF()
+
 IF(NOT WIN32)
   SET_TARGET_PROPERTIES(mariadbclient PROPERTIES OUTPUT_NAME "${LIBMARIADB_STATIC_NAME}")
 ENDIF()
@@ -476,7 +483,9 @@ INSTALL(TARGETS mariadbclient
 IF(WIN32)
 INSTALL(TARGETS libmariadb
         COMPONENT SharedLibraries
-        DESTINATION ${INSTALL_LIBDIR})
+        RUNTIME DESTINATION ${INSTALL_BINDIR}
+        LIBRARY DESTINATION ${INSTALL_LIBDIR}
+        ARCHIVE DESTINATION ${INSTALL_LIBDIR})
 ELSE()
 # in cmake 3.12+ we can use
 #INSTALL(TARGETS libmariadb LIBRARY DESTINATION ${INSTALL_LIBDIR}
