--- a/include/mysql.h
+++ b/include/mysql.h
@@ -515,7 +515,7 @@ typedef struct {
 } MARIADB_X509_INFO;


-struct st_mysql_client_plugin *
+struct st_mysql_client_plugin * STDCALL
 mysql_load_plugin(struct st_mysql *mysql, const char *name, int type,
                   int argc, ...);
 struct st_mysql_client_plugin * STDCALL
--- a/include/mysql/client_plugin.h
+++ b/include/mysql/client_plugin.h
@@ -196,7 +196,7 @@ struct st_mysql_client_plugin
   @retval
   a pointer to the loaded plugin, or NULL in case of a failure
 */
-struct st_mysql_client_plugin *
+struct st_mysql_client_plugin * STDCALL
 mysql_load_plugin(struct st_mysql *mysql, const char *name, int type,
                   int argc, ...);

