From e018f266b3e857a8847fa61a272305801ebaa376 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Mos=C3=A8=20Giordano?= <mose@gnu.org>
Date: Wed, 24 Aug 2022 00:47:49 +0100
Subject: [PATCH 1/1] Define `strerror_r` macro also for MinGW

---
 include/ma_global.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/ma_global.h b/include/ma_global.h
index b8ef7e16..5c3c494a 100644
--- a/include/ma_global.h
+++ b/include/ma_global.h
@@ -33,8 +33,8 @@
 #if _MSC_VER < 1900
 #define snprintf _snprintf
 #endif
-#define strerror_r(errno,buf,len) strerror_s(buf,len,errno)
 #endif
+#define strerror_r(errno,buf,len) strerror_s(buf,len,errno)
 #define STDCALL __stdcall 
 #endif
 
-- 
2.34.2

