diff --git a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -167,9 +167,6 @@ IF(GIT_EXECUTABLE AND EXISTS ${CC_SOURCE_DIR}/.git)
    IF(RES EQUAL 0)
     STRING(REGEX REPLACE "\n$" "" CC_SOURCE_REVISION "${OUT}")
   ENDIF()
-  EXECUTE_PROCESS(
-    COMMAND ${GIT_EXECUTABLE} submodule update --init --recursive
-            WORKING_DIRECTORY ${CC_SOURCE_DIR})
 ENDIF()

 # various defines for generating include/mysql_version.h
