--- a/meson.build
+++ b/meson.build
@@ -2620,8 +2620,8 @@
 specific_ss.add_all(when: 'CONFIG_LINUX_USER', if_true: linux_user_ss)
 
 # needed for fuzzing binaries
-subdir('tests/qtest/libqos')
-subdir('tests/qtest/fuzz')
+#subdir('tests/qtest/libqos')
+#subdir('tests/qtest/fuzz')
 
 # accel modules
 tcg_real_module_ss = ss.source_set()
@@ -3107,7 +3107,7 @@
 subdir('tools')
 subdir('pc-bios')
 subdir('docs')
-subdir('tests')
+#subdir('tests')
 if gtk.found()
   subdir('po')
 endif
