diff --git before/topcom-0.17.8/external/Makefile after/topcom-0.17.8/external/Makefile
index a22acdf..fc0a4d5 100644
--- before/topcom-0.17.8/external/Makefile
+++ after/topcom-0.17.8/external/Makefile
@@ -35,9 +35,9 @@ CDD_DIR      = $(CDD_NAME)-$(CDD_VERSION)
 
 all: gmp cdd
 
-gmp: $(GMP_DONE)
+gmp:;@true
 
-cdd: $(CDD_DONE)
+cdd:;@true
 
 $(GMP_DONE): $(GMP_DIR)
 	@echo ==============================================================================
