--- a/server/TracyPopcnt.hpp
+++ b/server/TracyPopcnt.hpp
@@ -4,11 +4,7 @@
 #include <limits.h>
 #include <stdint.h>

-#if defined _WIN64
-#  include <intrin.h>
-#  define TracyCountBits __popcnt64
-#  define TracyLzcnt __lzcnt64
-#elif defined __GNUC__ || defined __clang__
+#if defined __GNUC__ || defined __clang__
 static inline uint64_t TracyCountBits( uint64_t i )
 {
     return uint64_t( __builtin_popcountll( i ) );
@@ -17,6 +13,10 @@
 {
     return uint64_t( __builtin_clzll( i ) );
 }
+#elif defined _WIN64
+#  include <intrin.h>
+#  define TracyCountBits __popcnt64
+#  define TracyLzcnt __lzcnt64
 #else
 static inline uint64_t TracyCountBits( uint64_t i )
 {
