--- a/update/src/OfflineSymbolResolverDbgHelper.cpp
+++ b/update/src/OfflineSymbolResolverDbgHelper.cpp
@@ -73,7 +73,7 @@ class SymbolResolver
                 newEntry.name = "[unknown] + " + std::to_string(offset);
             }
 
-            IMAGEHLP_LINE lineInfo = { 0 };
+            IMAGEHLP_LINE64 lineInfo = { 0 };
             lineInfo.SizeOfStruct = sizeof(IMAGEHLP_LINE64);
             DWORD displaceMent = 0;
             if ( SymGetLineFromAddr64( m_procHandle, address, &displaceMent, &lineInfo ) )
