diff --git a/common/unix.mk b/common/unix.mk
index f836bdfc..136c7382 100644
--- a/common/unix.mk
+++ b/common/unix.mk
@@ -16,6 +16,11 @@ else ifeq (0,$(shell ld -ltbb -o /dev/null 2>/dev/null; echo $$?))
 endif
 endif

+UNAME_OS := $(shell uname -o)
+ifeq ($(UNAME_OS),Cygwin)
+       LIBS += -lole32 -lws2_32 -luuid
+endif
+
 OBJDIRBASE := obj/$(BUILD)
 OBJDIR := $(OBJDIRBASE)/o/o/o

diff --git a/profiler/build/unix/legacy.mk b/profiler/build/unix/legacy.mk
index 24765f1a..aed688e5 100644
--- a/profiler/build/unix/legacy.mk
+++ b/profiler/build/unix/legacy.mk
@@ -2,30 +2,41 @@ CFLAGS +=
 CXXFLAGS := $(CFLAGS) -std=c++17
 DEFINES += -DIMGUI_ENABLE_FREETYPE
 INCLUDES := $(shell pkg-config --cflags glfw3 freetype2 capstone) -I../../../imgui
-LIBS := $(shell pkg-config --libs glfw3 freetype2 capstone) -lpthread -ldl
+LIBS := $(shell pkg-config --libs freetype2 capstone) -lglfw3dll -lpthread

 PROJECT := Tracy
 IMAGE := $(PROJECT)-$(BUILD)

-FILTER := ../../../nfd/nfd_win.cpp
+UNAME_OS := $(shell uname -o)
+
+ifeq ($(UNAME_OS),Cygwin)
+	CXXFLAGS += -Wa,-mbig-obj
+	FILTER :=
+else
+	FILTER := ../../../nfd/nfd_win.cpp
+	LIBS += -ldl
+endif
+
 include ../../../common/src-from-vcxproj.mk

 ifdef TRACY_NO_FILESELECTOR
 	CXXFLAGS += -DTRACY_NO_FILESELECTOR
 else
 	UNAME := $(shell uname -s)
-	ifeq ($(UNAME),Darwin)
-		SRC3 += ../../../nfd/nfd_cocoa.m
-		LIBS +=  -framework CoreFoundation -framework AppKit -framework UniformTypeIdentifiers
-	else
-		ifdef TRACY_GTK_FILESELECTOR
-			SRC += ../../../nfd/nfd_gtk.cpp
-			INCLUDES += $(shell pkg-config --cflags gtk+-3.0)
-			LIBS += $(shell pkg-config --libs gtk+-3.0)
+	ifneq ($(UNAME_OS),Cygwin)
+		ifeq ($(UNAME),Darwin)
+			SRC3 += ../../../nfd/nfd_cocoa.m
+			LIBS +=  -framework CoreFoundation -framework AppKit -framework UniformTypeIdentifiers
 		else
-			SRC += ../../../nfd/nfd_portal.cpp
-			INCLUDES += $(shell pkg-config --cflags dbus-1)
-			LIBS += $(shell pkg-config --libs dbus-1)
+			ifdef TRACY_GTK_FILESELECTOR
+				SRC += ../../../nfd/nfd_gtk.cpp
+				INCLUDES += $(shell pkg-config --cflags gtk+-3.0)
+				LIBS += $(shell pkg-config --libs gtk+-3.0)
+			else
+				SRC += ../../../nfd/nfd_portal.cpp
+				INCLUDES += $(shell pkg-config --cflags dbus-1)
+				LIBS += $(shell pkg-config --libs dbus-1)
+			endif
 		endif
 	endif
 endif
diff --git a/server/TracyPopcnt.hpp b/server/TracyPopcnt.hpp
index 945324fd..07b98fd8 100644
--- a/server/TracyPopcnt.hpp
+++ b/server/TracyPopcnt.hpp
@@ -4,11 +4,7 @@
 #include <limits.h>
 #include <stdint.h>

-#if defined _WIN64
-#  include <intrin.h>
-#  define TracyCountBits __popcnt64
-#  define TracyLzcnt __lzcnt64
-#elif defined __GNUC__ || defined __clang__
+#if defined __GNUC__ || defined __clang__
 static inline uint64_t TracyCountBits( uint64_t i )
 {
     return uint64_t( __builtin_popcountll( i ) );
@@ -17,6 +13,10 @@ static inline uint64_t TracyLzcnt( uint64_t i )
 {
     return uint64_t( __builtin_clzll( i ) );
 }
+#elif defined _WIN64
+#  include <intrin.h>
+#  define TracyCountBits __popcnt64
+#  define TracyLzcnt __lzcnt64
 #else
 static inline uint64_t TracyCountBits( uint64_t i )
 {
