diff --git before/cmake/FindSQLite3.cmake after/cmake/FindSQLite3.cmake
index c368920..9c99ae5 100644
--- before/cmake/FindSQLite3.cmake
+++ after/cmake/FindSQLite3.cmake
@@ -19,7 +19,7 @@
 FIND_PATH(SQLITE3_INCLUDE_DIR NAMES sqlite3.h)
 
 # Look for the library.
-FIND_LIBRARY(SQLITE3_LIBRARY NAMES sqlite)
+FIND_LIBRARY(SQLITE3_LIBRARY NAMES sqlite3)
 
 # Handle the QUIETLY and REQUIRED arguments and set SQLITE3_FOUND to TRUE if all listed variables are TRUE.
 INCLUDE(FindPackageHandleStandardArgs)