diff --git a/doc/news/changes/minor/20260327Ehlert b/doc/news/changes/minor/20260327Ehlert
new file mode 100644
index 0000000000..cbbc8661c8
--- /dev/null
+++ b/doc/news/changes/minor/20260327Ehlert
@@ -0,0 +1,5 @@
+Extended: parallel::distributed::Triangulation::copy_triangulation() now allows to
+control whether to copy also the triangulation settings or to use a developer provided
+setting.
+This can be useful for multigrid computations.
+(Vivienne Ehlert, Martin Kronbichler, 2026/03/27)
diff --git a/include/deal.II/distributed/tria.h b/include/deal.II/distributed/tria.h
index 95b1d13ba3..855c18a5b1 100644
--- a/include/deal.II/distributed/tria.h
+++ b/include/deal.II/distributed/tria.h
@@ -478,6 +478,19 @@ namespace parallel
       copy_triangulation(
         const dealii::Triangulation<dim, spacedim> &other_tria) override;
 
+      /**
+       * Implementation of the same function as in the base class but instead
+       * of copying the settings of the other triangulation use the provided
+       * parameter as settings for the copy.
+       *
+       * @note This function can be used to copy a serial Triangulation to a
+       * parallel::distributed::Triangulation but only if the serial
+       * Triangulation has never been refined.
+       */
+      void
+      copy_triangulation(const dealii::Triangulation<dim, spacedim> &other_tria,
+                         const Settings                              settings);
+
       /**
        * Create a triangulation as documented in the base class.
        *
diff --git a/include/deal.II/multigrid/mg_transfer_global_coarsening.templates.h b/include/deal.II/multigrid/mg_transfer_global_coarsening.templates.h
index e4fc51fbb4..3f43c14d3e 100644
--- a/include/deal.II/multigrid/mg_transfer_global_coarsening.templates.h
+++ b/include/deal.II/multigrid/mg_transfer_global_coarsening.templates.h
@@ -94,7 +94,27 @@ namespace MGTransferGlobalCoarseningTools
       {
         // copy triangulation
         auto new_tria = create_new_empty_triangulation();
+
+#ifdef DEAL_II_WITH_P4EST
+        if constexpr (dim > 1 && dim == spacedim)
+          {
+            if (auto new_distr_tria = dynamic_cast<
+                  parallel::distributed::Triangulation<dim, spacedim> *>(
+                  &*new_tria))
+              new_distr_tria->copy_triangulation(
+                *coarse_grid_triangulations[l],
+                parallel::distributed::Triangulation<dim>::default_setting);
+            else
+              new_tria->copy_triangulation(*coarse_grid_triangulations[l]);
+          }
+        else
+          {
+            new_tria->copy_triangulation(*coarse_grid_triangulations[l]);
+          }
+#else
         new_tria->copy_triangulation(*coarse_grid_triangulations[l]);
+#endif
+
         new_tria->set_mesh_smoothing(mesh_smoothing);
 
         // coarsen mesh
diff --git a/source/distributed/tria.cc b/source/distributed/tria.cc
index 7170666ed2..a2523dbe6e 100644
--- a/source/distributed/tria.cc
+++ b/source/distributed/tria.cc
@@ -3923,6 +3923,23 @@ namespace parallel
     DEAL_II_CXX20_REQUIRES((concepts::is_valid_dim_spacedim<dim, spacedim>))
     void Triangulation<dim, spacedim>::copy_triangulation(
       const dealii::Triangulation<dim, spacedim> &other_tria)
+    {
+      if (const dealii::parallel::distributed::Triangulation<dim, spacedim>
+            *other_distributed =
+              dynamic_cast<const dealii::parallel::distributed::
+                             Triangulation<dim, spacedim> *>(&other_tria))
+        copy_triangulation(other_tria, other_distributed->settings);
+      else
+        copy_triangulation(other_tria, default_setting);
+    }
+
+
+
+    template <int dim, int spacedim>
+    DEAL_II_CXX20_REQUIRES((concepts::is_valid_dim_spacedim<dim, spacedim>))
+    void Triangulation<dim, spacedim>::copy_triangulation(
+      const dealii::Triangulation<dim, spacedim> &other_tria,
+      const Settings                              settings)
     {
       Assert(
         (dynamic_cast<
@@ -3952,8 +3969,9 @@ namespace parallel
               dynamic_cast<const dealii::parallel::distributed::
                              Triangulation<dim, spacedim> *>(&other_tria))
         {
+          // use settings from the provided parameter instead of copying them
+          this->settings = settings;
           // copy parallel distributed specifics
-          settings = other_distributed->settings;
           triangulation_has_content =
             other_distributed->triangulation_has_content;
           coarse_cell_to_p4est_tree_permutation =
