diff --git a/frame/include/bli_macro_defs.h b/frame/include/bli_macro_defs.h
index e35e48c8a..fb1e8901f 100644
--- a/frame/include/bli_macro_defs.h
+++ b/frame/include/bli_macro_defs.h
@@ -87,10 +87,10 @@
 #define PASTEMACT(ch1, ch2, ch3, ch4)   bli_ ## ch1 ## ch2 ## _ ## ch3 ## _ ## ch4
 // name-mangling macros.
 #ifdef BLIS_ENABLE_NO_UNDERSCORE_API
-#define PASTEF770(name)                                                  name
-#define PASTEF77(ch1,name)                                        ch1 ## name
-#define PASTEF772(ch1,ch2,name)                            ch1 ## ch2 ## name
-#define PASTEF773(ch1,ch2,ch3,name)                 ch1 ## ch2 ## ch3 ## name
+#define PASTEF770(name)                                                  name ## _64
+#define PASTEF77(ch1,name)                                        ch1 ## name ## _64
+#define PASTEF772(ch1,ch2,name)                            ch1 ## ch2 ## name ## _64
+#define PASTEF773(ch1,ch2,ch3,name)                 ch1 ## ch2 ## ch3 ## name ## _64
 #else
 #define PASTEF770(name)                                            name ## _
 #define PASTEF77(ch1,name)                                  ch1 ## name ## _
