diff --git a/CMakeLists.txt b/CMakeLists.txt
index c1a57bb..704fc5b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -109,9 +109,7 @@ endif(DACE_MEMORY_MODEL_U STREQUAL "STATIC")
 
 
 # Run checks for platform-specific functions
-include(CheckSafeStrings)
-check_safe_strings(HAVE_SAFE_STRINGS)
-
+set(HAVE_SAFE_STRINGS FALSE)
 
 # Pass options to source by configuring config.h file in core
 configure_file(core/include/dace/config.h.in core/include/dace/config.h)
