diff --git before/CMakeLists.txt after/CMakeLists.txt
index 7d78484..f3391bc 100644
--- before/CMakeLists.txt
+++ after/CMakeLists.txt
@@ -166,7 +166,7 @@ endif()
 # This cmake target alias will be defined by either: 
 # a) libnabo sources if built as a git submodule in the same project than this library, or
 # b) by libnabo-targets.cmake, included by find_package(libnabo) above.
-set(libnabo_LIBRARIES libnabo::nabo)
+#set(libnabo_LIBRARIES libnabo::nabo)
 
 #--------------------
 # DEPENDENCY: OpenMP (optional)
