diff -ruN readline-8.1/signals.c readline-8.1-fixed/signals.c
--- readline-8.1/signals.c	2021-01-15 16:08:14.477184153 +0800
+++ readline-8.1-fixed/signals.c	2021-01-15 16:07:04.733781000 +0800
@@ -266,8 +266,10 @@
     case SIGQUIT:
 #endif
 
+#if defined (HAVE_POSIX_SIGNALS)
       if (block_sig)
 	sigprocmask (SIG_BLOCK, &set, &oset);
+#endif
 
       rl_echo_signal_char (sig);
       rl_cleanup_after_signal ();
