--- a/rlwinsize.h
+++ b/rlwinsize.h
@@ -54,6 +54,11 @@
 #if defined (M_UNIX) && !defined (_SCO_DS) && !defined (tcflow)
 #  define tcflow(fd, action)	ioctl(fd, TCXONC, action)
 #endif
+
+/* Forward-declare `struct winsize`. We'll never use it on Windows. */
+#if defined(_WIN32)
+struct winsize;
+#endif
 
 extern int _rl_tcgetwinsize (int, struct winsize *);
 extern void _rl_tcsetwinsize (int, struct winsize *);
