--- sys_linux/g77_gcc.conf.orig
+++ sys_linux/g77_gcc.conf
@@ -34,7 +34,7 @@
 # Mandatory.
 # The FORTRAN compiler to use.
  
-   FCOMPL="g77"
+   FCOMPL="gfortran"
 
 # Mandatory.
 # The FORTRAN compiler flags to use when compiling the pgplot library.
@@ -117,7 +117,7 @@
 # library-specification flags used to specify these libraries to
 # $SHARED_LD
  
-   SHARED_LIB_LIBS=""
+   SHARED_LIB_LIBS="-lpng -lz"
 
 # Optional:
 # Compiler name used on Next systems to compile objective-C files.
