From 49306a57098f1eaebb700eed29025e6f472a799a Mon Sep 17 00:00:00 2001
From: "Maarten L. Hekkelman" <maarten@hekkelman.com>
Date: Tue, 13 Jun 2023 10:15:38 +0200
Subject: [PATCH] Fix export of libcifpp dependency

---
 CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0da7343..2ec2caf 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -118,7 +118,7 @@ add_library(dssp
 
 add_library(dssp::dssp ALIAS dssp)
 
-target_link_libraries(dssp PRIVATE cifpp::cifpp)
+target_link_libraries(dssp PUBLIC cifpp::cifpp)
 
 target_include_directories(dssp
 	PUBLIC
@@ -128,7 +128,7 @@ target_include_directories(dssp
 
 add_executable(mkdssp ${PROJECT_SOURCE_DIR}/src/mkdssp.cpp)
 
-target_link_libraries(mkdssp PRIVATE dssp cifpp::cifpp libmcfp::libmcfp dssp::dssp)
+target_link_libraries(mkdssp PRIVATE libmcfp::libmcfp dssp::dssp)
 
 if(USE_RSRC)
 	mrc_target_resources(mkdssp
-- 
2.34.1

