diff --git a/lib/cmpbuf.c b/lib/cmpbuf.c
index 7cee45a..2b466b0 100644
--- a/lib/cmpbuf.c
+++ b/lib/cmpbuf.c
@@ -71,8 +71,8 @@ block_read (int fd, char *buf, size_t nbytes)
              ancient AIX hosts that set errno to EINTR after uncaught
              SIGCONT.  See <news:1r77ojINN85n@ftp.UU.NET>
              (1993-04-22).  */
-          if (! SA_RESTART && errno == EINTR)
-            continue;
+          // if (! SA_RESTART && errno == EINTR)
+          //   continue;
 
           return SIZE_MAX;
         }
