--- a/src/util.c
+++ b/src/util.c
@@ -45,6 +45,16 @@
 # define SA_RESTART 0
 #endif
 
+/* Out-of-range substitutes make a good fallback for uncatchable
+   signals.  */
+#ifndef SIGKILL
+# define SIGKILL (-1)
+#endif
+#ifndef SIGSTOP
+# define SIGSTOP (-1)
+#endif
+
+
 char const pr_program[] = PR_PROGRAM;
 
 /* Queue up one-line messages to be printed at the end,
@@ -297,7 +307,15 @@
 #ifdef SIGALRM
     SIGALRM,
 #endif
-    SIGHUP, SIGINT, SIGPIPE,
+#ifdef SIGHUP
+      SIGHUP, 
+#endif
+#ifdef SIGINT 
+      SIGINT, 
+#endif
+#ifdef SIGPIPE 
+      SIGPIPE, 
+#endif
 #ifdef SIGQUIT
     SIGQUIT,
 #endif
