diff -ur a/cmake/ecbuild_bundle.cmake b/cmake/ecbuild_bundle.cmake
--- a/cmake/ecbuild_bundle.cmake	2026-01-19 14:09:07.000000000 +0100
+++ b/cmake/ecbuild_bundle.cmake	2026-02-17 14:04:57.775325689 +0100
@@ -159,7 +159,7 @@
       if( DEFINED ${PNAME}_SOURCE )
         ecbuild_critical( "ecbuild_bundle called with SOURCE for project ${_PAR_PROJECT} but ${PNAME}_SOURCE is defined" )
       endif()
-      execute_process( COMMAND ${CMAKE_COMMAND} -E create_symlink ${_PAR_SOURCE} ${PROJECT_SOURCE_DIR}/${_PAR_PROJECT} )
+      execute_process( COMMAND ${CMAKE_COMMAND} -E copy ${_PAR_SOURCE} ${PROJECT_SOURCE_DIR}/${_PAR_PROJECT} )
     endif()
 
     if( NOT EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/${_PAR_PROJECT} OR NOT EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/${_PAR_PROJECT}/CMakeLists.txt )
diff -ur a/cmake/ecbuild_declare_project.cmake b/cmake/ecbuild_declare_project.cmake
--- a/cmake/ecbuild_declare_project.cmake	2026-01-19 14:09:07.000000000 +0100
+++ b/cmake/ecbuild_declare_project.cmake	2026-02-17 14:04:57.775325689 +0100
@@ -194,7 +194,7 @@
 
   # make sure nothing breaks if INSTALL_LIB_DIR is not lib
   if( NOT INSTALL_LIB_DIR STREQUAL "lib" AND NOT EXISTS ${CMAKE_BINARY_DIR}/${INSTALL_LIB_DIR} )
-    execute_process( COMMAND ${CMAKE_COMMAND} -E create_symlink lib ${CMAKE_BINARY_DIR}/${INSTALL_LIB_DIR} )
+    execute_process( COMMAND ${CMAKE_COMMAND} -E copy lib ${CMAKE_BINARY_DIR}/${INSTALL_LIB_DIR} )
   endif()
 
   # ecbuild_debug_var( CMAKE_INSTALL_RPATH )
diff -ur a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2026-01-19 14:07:07.000000000 +0100
+++ b/CMakeLists.txt	2026-02-17 14:02:11.648341364 +0100
@@ -342,7 +342,7 @@
     # Symbols need to be explicitly exported on Windows so we can link to dlls.
     set( CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS TRUE )
     set( ECCODES_ON_WINDOWS 1 )
-
+    if( MSVC )
     # Suppress compliler warnings - see ECC-850
     # Suppress warnings about using 'insecure' functions. Fixing this would require changes all over
     # the codebase which would damage portability.
@@ -351,6 +351,7 @@
     ecbuild_add_c_flags("/D_CRT_NONSTDC_NO_DEPRECATE")
     # Suppress C4267: warns about possible loss of data when converting 'size_t' to 'int'.
     ecbuild_add_c_flags("/wd4267")
+    endif()
 endif()
 
 ###############################################################################
@@ -384,7 +385,7 @@
 
 add_definitions( -DHAVE_ECCODES_CONFIG_H )
 
-if( CMAKE_CXX_COMPILER_ID STREQUAL "GNU" )
+if( CMAKE_CXX_COMPILER_ID STREQUAL "GNU" AND NOT MINGW )
     ecbuild_add_c_flags("-pedantic")
     ecbuild_add_cxx_flags("-pedantic")
 endif()
diff -ur a/definitions/CMakeLists.txt b/definitions/CMakeLists.txt
--- a/definitions/CMakeLists.txt	2026-01-19 14:07:07.000000000 +0100
+++ b/definitions/CMakeLists.txt	2026-02-17 14:03:00.845099708 +0100
@@ -45,7 +45,7 @@
 # link to the definitions. See GRIB-786
 file( MAKE_DIRECTORY ${CMAKE_BINARY_DIR}/${INSTALL_DATA_DIR} )
 if( NOT EXISTS "${CMAKE_BINARY_DIR}/${ECCODES_DEFINITION_SUFF}" )
-  execute_process( COMMAND "${CMAKE_COMMAND}" "-E" "create_symlink"
+  execute_process( COMMAND "${CMAKE_COMMAND}" "-E" "copy"
     "${CMAKE_CURRENT_SOURCE_DIR}"
     "${CMAKE_BINARY_DIR}/${ECCODES_DEFINITION_SUFF}" )
 endif()
diff -ur a/fortran/CMakeLists.txt b/fortran/CMakeLists.txt
--- a/fortran/CMakeLists.txt	2026-01-19 14:07:07.000000000 +0100
+++ b/fortran/CMakeLists.txt	2026-02-17 14:04:57.775325689 +0100
@@ -96,7 +96,7 @@
     #add_custom_command(
     #    OUTPUT ${_eccodes_mod}
     #    WORKING_DIRECTORY ${CMAKE_Fortran_MODULE_DIRECTORY}
-    #    #COMMAND ${CMAKE_COMMAND} -E create_symlink "${CMAKE_Fortran_MODULE_DIRECTORY}/grib_api.mod" ${_eccodes_mod}
+    #    #COMMAND ${CMAKE_COMMAND} -E copy "${CMAKE_Fortran_MODULE_DIRECTORY}/grib_api.mod" ${_eccodes_mod}
     #    #COMMAND ${CMAKE_COMMAND} -E copy "${CMAKE_Fortran_MODULE_DIRECTORY}/grib_api.mod" ${_eccodes_mod}
     #    COMMAND "/usr/bin/sh" "-c" "if test -f GRIB_API.mod; then ln -s GRIB_API.mod ECCODES.mod; fi"
     #    COMMAND "/usr/bin/sh" "-c" "if test -f grib_api.mod; then ln -s grib_api.mod eccodes.mod; fi"
diff -ur a/fortran/eccodes_f90_int.f90 b/fortran/eccodes_f90_int.f90
--- a/fortran/eccodes_f90_int.f90	2026-01-19 14:07:07.000000000 +0100
+++ b/fortran/eccodes_f90_int.f90	2026-02-17 14:03:43.598514402 +0100
@@ -100,7 +100,6 @@
       codes_get_string, &
       codes_get_byte_array, &
       codes_get_int_array, &
-      codes_get_long_array, &
       codes_get_real4_array, &
       codes_get_real8_array
   end interface codes_get
diff -ur a/fortran/grib_fortran.cc b/fortran/grib_fortran.cc
--- a/fortran/grib_fortran.cc	2026-01-19 14:07:07.000000000 +0100
+++ b/fortran/grib_fortran.cc	2026-02-17 14:10:03.192623390 +0100
@@ -903,7 +903,7 @@
                 grib_context_log(context,GRIB_LOG_FATAL,"grib_f_open_file_: posix_memalign unable to allocate io_buffer\n");
             }
 #else
-            iobuf = (void*)malloc(context->io_buffer_size);
+            iobuf = (char*)malloc(context->io_buffer_size);
             if (!iobuf) {
                 grib_context_log(context,GRIB_LOG_FATAL,"grib_f_open_file_: Unable to allocate io_buffer\n");
             }
diff -ur a/ifs_samples/CMakeLists.txt b/ifs_samples/CMakeLists.txt
--- a/ifs_samples/CMakeLists.txt	2026-01-19 14:07:07.000000000 +0100
+++ b/ifs_samples/CMakeLists.txt	2026-02-17 14:04:57.765325669 +0100
@@ -14,7 +14,7 @@
   execute_process( COMMAND ${CMAKE_COMMAND} -E make_directory ${destination_build} )
   foreach( _file ${_samples} )
     get_filename_component(_filename ${_file} NAME)
-    execute_process( COMMAND ${CMAKE_COMMAND} -E create_symlink
+    execute_process( COMMAND ${CMAKE_COMMAND} -E copy
        ${_file} ${destination_build}/${_filename} )
   endforeach()
 endfunction()
diff -ur a/samples/CMakeLists.txt b/samples/CMakeLists.txt
--- a/samples/CMakeLists.txt	2026-01-19 14:07:07.000000000 +0100
+++ b/samples/CMakeLists.txt	2026-02-17 14:04:57.775325689 +0100
@@ -16,7 +16,7 @@
 # link to the samples in the build directory. See GRIB-786
 file( MAKE_DIRECTORY ${CMAKE_BINARY_DIR}/${INSTALL_DATA_DIR} )
 if( NOT EXISTS "${CMAKE_BINARY_DIR}/${ECCODES_SAMPLES_SUFF}" )
-  execute_process( COMMAND "${CMAKE_COMMAND}" "-E" "create_symlink"
+  execute_process( COMMAND "${CMAKE_COMMAND}" "-E" "copy"
     "${CMAKE_CURRENT_SOURCE_DIR}"
     "${CMAKE_BINARY_DIR}/${ECCODES_SAMPLES_SUFF}" )
 endif()
diff -ur a/src/eccodes/accessor/DataG1SecondOrderGeneralExtendedPacking.cc b/src/eccodes/accessor/DataG1SecondOrderGeneralExtendedPacking.cc
--- a/src/eccodes/accessor/DataG1SecondOrderGeneralExtendedPacking.cc	2026-01-19 14:07:07.000000000 +0100
+++ b/src/eccodes/accessor/DataG1SecondOrderGeneralExtendedPacking.cc	2026-02-17 14:11:33.509484719 +0100
@@ -20,7 +20,7 @@
 #define MAX_NUMBER_OF_GROUPS 65534
 #define EFDEBUG              0
 
-static const unsigned long nbits[64] = {
+static const size_t nbits[64] = {
     0x1, 0x2, 0x4, 0x8,
     0x10, 0x20, 0x40, 0x80,
     0x100, 0x200, 0x400, 0x800,
@@ -41,7 +41,7 @@
 
 long number_of_bits(grib_handle* h, unsigned long x)
 {
-    const unsigned long* n = nbits;
+    const size_t* n = nbits;
     const int count        = sizeof(nbits) / sizeof(nbits[0]);
     long i                 = 0;
     while (x >= *n) {
diff -ur a/src/eccodes/grib_api_internal.h b/src/eccodes/grib_api_internal.h
--- a/src/eccodes/grib_api_internal.h	2026-01-19 14:07:07.000000000 +0100
+++ b/src/eccodes/grib_api_internal.h	2026-02-17 14:12:07.962893046 +0100
@@ -87,7 +87,9 @@
         #define M_PI 3.14159265358979323846
     #endif
 
+    #ifdef _MSC_VER
     #define R_OK 04 /* Needed for Windows */
+    #endif
 
     #ifndef F_OK
         #define F_OK 0
diff -ur a/src/eccodes/grib_context.cc b/src/eccodes/grib_context.cc
--- a/src/eccodes/grib_context.cc	2026-01-19 14:07:07.000000000 +0100
+++ b/src/eccodes/grib_context.cc	2026-02-17 14:12:35.242952682 +0100
@@ -413,7 +413,7 @@
         /* On UNIX, when we read from a file we get exactly what is in the file on disk.
          * But on Windows a file can be opened in binary or text mode. In binary mode the system behaves exactly as in UNIX.
          */
-#ifdef ECCODES_ON_WINDOWS
+#ifdef _MSC_VER
         _set_fmode(_O_BINARY);
 #endif
 
diff -ur a/tools/grib_tools.cc b/tools/grib_tools.cc
--- a/tools/grib_tools.cc	2026-01-19 14:07:07.000000000 +0100
+++ b/tools/grib_tools.cc	2026-02-17 14:37:54.022770498 +0100
@@ -665,7 +665,7 @@
     int stat_val = 0;
     int err = 0;
 
-#ifndef ECCODES_ON_WINDOWS
+#ifdef _MSC_VER
     stat_val = lstat(path, &s);
 #else
     stat_val = stat(path, &s);
diff -ur a/tools/grib_tools.h b/tools/grib_tools.h
--- a/tools/grib_tools.h	2026-01-19 14:07:07.000000000 +0100
+++ b/tools/grib_tools.h	2026-02-17 14:29:20.875238300 +0100
@@ -160,7 +160,7 @@
     int headers_only;
     int skip_all;
     grib_values tolerance[MAX_KEYS];
-    off_t infile_offset;
+    long long infile_offset;
     int json_output;
 } grib_runtime_options;
 
