--- libffi-3.2.1-orig/configure.ac	2021-11-18 00:13:42.753131022 -0500
+++ libffi-3.2.1/configure.ac	2021-11-18 00:14:43.014058771 -0500
@@ -590,26 +590,7 @@
     AC_DEFINE(USING_PURIFY, 1, [Define this if you are using Purify and want to suppress spurious messages.])
   fi)
 
-# These variables are only ever used when we cross-build to X86_WIN32.
-# And we only support this with GCC, so...
-if test "x$GCC" = "xyes"; then
-  if test -n "$with_cross_host" &&
-     test x"$with_cross_host" != x"no"; then
-    toolexecdir="${exec_prefix}"/'$(target_alias)'
-    toolexeclibdir="${toolexecdir}"/lib
-  else
-    toolexecdir="${libdir}"/gcc-lib/'$(target_alias)'
-    toolexeclibdir="${libdir}"
-  fi
-  multi_os_directory=`$CC $CFLAGS -print-multi-os-directory`
-  case $multi_os_directory in
-    .) ;; # Avoid trailing /.
-    ../*) toolexeclibdir=$toolexeclibdir/$multi_os_directory ;;
-  esac
-  AC_SUBST(toolexecdir)
-else
-  toolexeclibdir="${libdir}"
-fi
+toolexeclibdir="${libdir}"
 AC_SUBST(toolexeclibdir)
 
 AC_CONFIG_COMMANDS(include, [test -d include || mkdir include])

