--- libffi-clean/src/aarch64/sysv.S     2014-11-08 13:47:24.000000000 +0100
+++ libffi-3.2.1/src/aarch64/sysv.S     2024-12-14 13:17:42.329889690 +0100
@@ -100,8 +100,8 @@
 
 #define ffi_call_SYSV_FS (8 * 4)
 
-        .cfi_startproc
 CNAME(ffi_call_SYSV):
+        .cfi_startproc
         stp     x29, x30, [sp, #-16]!
        cfi_adjust_cfa_offset (16)
         cfi_rel_offset (x29, 0)
@@ -247,8 +247,8 @@
 #ifdef __APPLE__
         .align 2
 #endif
-        .cfi_startproc
 CNAME(ffi_closure_SYSV):
+    .cfi_startproc
         stp     x29, x30, [sp, #-16]!
        cfi_adjust_cfa_offset (16)
         cfi_rel_offset (x29, 0)
