diff --git a/configure.ac b/configure.ac
index 16fd2ac5..b3d79fa4 100755
--- a/configure.ac
+++ b/configure.ac
@@ -479,20 +479,6 @@ if test "x$F77" = "xg95"; then
   LDFLAGS="-Wl,-framework,Accelerate $LDFLAGS"
 fi
 
-dnl *********************************************************************
-dnl checks for libraries now
-
-dnl This check explicitly tests for fortran libraries needed to link C++ with
-dnl fortran source files.  It *might* also be necessary if all Fortran libraries
-dnl we need (LAPACK, BLAS, etc.) are static libraries.  It should not be
-dnl necessary for normal shared libraries, but should also not be harmful.
-dnl The result of this macro call is that all libraries needed to link with
-dnl static Fortran objects will be put in FLIBS, which we later tack on to
-dnl LIBS.
-if test "$FORTRAN_BLAS" != "no"; then
-  AC_F77_LIBRARY_LDFLAGS
-fi
-
 AC_SEARCH_LIBS(sqrt, [m])
 
 dnl find a valid blas and lapack library
