diff --git a/configure.ac b/configure.ac
index d3453a72..0bf380d1 100755
--- a/configure.ac
+++ b/configure.ac
@@ -479,13 +479,6 @@ if test "x$F77" = "xg95"; then
   LDFLAGS="-Wl,-framework,Accelerate $LDFLAGS"
 fi
 
-dnl *********************************************************************
-dnl checks for libraries now
-dnl Replace `main' with a function in -lm:
-if test "$FORTRAN_BLAS" != "no"; then
-  AC_F77_LIBRARY_LDFLAGS
-fi
-
 AC_SEARCH_LIBS(sqrt, [m])
 
 dnl find a valid blas and lapack library
diff --git a/src/Makefile.am b/src/Makefile.am
index b60a533d..a4d24e5d 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -82,8 +82,10 @@ XYCELINK = $(MSLINK)
 # build library (static & shared) for linking (w/libtool) into other codes
 lib_LTLIBRARIES = libxyce.la
 libxyce_la_SOURCES = $(XYCESOURCES)
-libxyce_la_LIBADD = $(XYCELINK)
-libxyce_la_LDFLAGS = $(AM_LDFLAGS) @XYCELIBS@
+libxyce_la_LIBADD = $(XYCELINK) @XYCELIBS@
+libxyce_la_LDFLAGS = $(AM_LDFLAGS) -no-undefined
+# XYCESOURCE may be empty - make sure libtool still knows that this is a C++ link
+libxyce_la_LIBTOOLFLAGS = --tag=CXX
 
 if BUILD_XYCE_BINARY
 
