From a4379af4e3d8a9075c34f774fbb0bb88c8ccca08 Mon Sep 17 00:00:00 2001
From: Pablo Antolin <pablo.antolin@epfl.ch>
Date: Tue, 8 Oct 2024 08:30:05 +0200
Subject: [PATCH] Adding constructor to xarraySlice

---
 algoim/xarray.hpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/algoim/xarray.hpp b/algoim/xarray.hpp
index 68439eb..bba9446 100644
--- a/algoim/xarray.hpp
+++ b/algoim/xarray.hpp
@@ -69,6 +69,8 @@ namespace algoim
         T* ptr;
         int len;
 
+        xarraySlice(T *ptr, int len) : ptr(ptr), len(len){};
+
         xarraySlice(xarraySlice&) = delete;
         xarraySlice(xarraySlice&&) = delete;
 
-- 
2.25.1

