diff --git a/configure.ac b/configure.ac
index 6e54f62..00dadf1 100644
--- a/configure.ac
+++ b/configure.ac
@@ -619,33 +619,8 @@ if test "$have_w32_system" = yes; then
     fi
     force_use_syscfg=yes
 fi
-if test x"$gl_use_threads" = xno; then
-  lock_obj_h_generated=yes
-  if test ! -d src; then mkdir src; fi
-  host=$host $srcdir/src/gen-lock-obj.sh --disable-threads \
-    >src/lock-obj-pub.native.h
-  AC_MSG_NOTICE([generated src/lock-obj-pub.native.h for $host])
-elif test x$cross_compiling = xyes; then
-  case $host in
-    *-*-gnu* | *-*-linux-gnu* | *-*-linux-musl*)
-    AC_CHECK_TOOL(OBJDUMP, [objdump])
-    if test -n "$OBJDUMP"; then
-      lock_obj_h_generated=yes
-      if test ! -d src; then mkdir src; fi
-      LOCK_ABI_VERSION=1 host=$host host_alias=$host_alias \
-          CC=$CC OBJDUMP=$OBJDUMP \
-          ac_ext=$ac_ext ac_objext=$ac_objext \
-          AWK=$AWK $srcdir/src/gen-lock-obj.sh \
-          >src/lock-obj-pub.native.h
-      AC_MSG_NOTICE([generated src/lock-obj-pub.native.h using $host_alias-objdump and $AWK])
-    else
-      force_use_syscfg=yes
-    fi
-    ;;
-    *)
+if test x$cross_compiling = xyes; then
     force_use_syscfg=yes
-    ;;
-  esac
 fi
 
 AM_CONDITIONAL(HAVE_W32_SYSTEM, test "$have_w32_system" = yes)
