--- a/mpi.h
+++ b/mpi.h
@@ -1899,7 +1899,74 @@
 int PMPI_T_source_get_info(int source_index, char *name, int *name_len, char *desc, int *desc_len, MPI_T_source_order *ordering, MPI_Count *ticks_per_second, MPI_Count *max_ticks, MPI_Info *info);
 int PMPI_T_source_get_num(int *num_sources);
 int PMPI_T_source_get_timestamp(int source_index, MPI_Count *timestamp);
+
+/* Fortran declarations */
+
+typedef int MPI_Fint;
+
+typedef MPI_Status MPI_F08_Status;
+
+#define MPI_F_STATUS_IGNORE     ((MPI_Fint*)MPI_STATUS_IGNORE)
+#define MPI_F_STATUSES_IGNORE   ((MPI_Fint*)MPI_STATUSES_IGNORE)
+#define MPI_F08_STATUS_IGNORE   ((MPI_F08_Status*)MPI_STATUS_IGNORE)
+#define MPI_F08_STATUSES_IGNORE ((MPI_F08_Status*)MPI_STATUSES_IGNORE)
+
+int PMPI_Status_f2c(const MPI_Fint *f_status, MPI_Status *c_status);
+int PMPI_Status_c2f(const MPI_Status *c_status, MPI_Fint *f_status);
+int PMPI_Status_f082c(const MPI_F08_Status *f08_status, MPI_Status *c_status);
+int PMPI_Status_c2f08(const MPI_Status *c_status, MPI_F08_Status *f08_status);
+
+int MPI_Status_f2c(const MPI_Fint *f_status, MPI_Status *c_status);
+int MPI_Status_c2f(const MPI_Status *c_status, MPI_Fint *f_status);
+int MPI_Status_f082c(const MPI_F08_Status *f08_status, MPI_Status *c_status);
+int MPI_Status_c2f08(const MPI_Status *c_status, MPI_F08_Status *f08_status);
+
+MPI_Comm PMPI_Comm_f2c(MPI_Fint comm);
+MPI_Fint PMPI_Comm_c2f(MPI_Comm comm);
+MPI_Errhandler PMPI_Errhandler_f2c(MPI_Fint errhandler);
+MPI_Fint PMPI_Errhandler_c2f(MPI_Errhandler errhandler);
+MPI_File PMPI_File_f2c(MPI_Fint file);
+MPI_Fint PMPI_File_c2f(MPI_File file);
+MPI_Group PMPI_Group_f2c(MPI_Fint group);
+MPI_Fint PMPI_Group_c2f(MPI_Group group);
+MPI_Info PMPI_Info_f2c(MPI_Fint info);
+MPI_Fint PMPI_Info_c2f(MPI_Info info);
+MPI_Message PMPI_Message_f2c(MPI_Fint message);
+MPI_Fint PMPI_Message_c2f(MPI_Message message);
+MPI_Op PMPI_Op_f2c(MPI_Fint op);
+MPI_Fint PMPI_Op_c2f(MPI_Op op);
+MPI_Request PMPI_Request_f2c(MPI_Fint request);
+MPI_Fint PMPI_Request_c2f(MPI_Request request);
+MPI_Session PMPI_Session_f2c(MPI_Fint session);
+MPI_Fint PMPI_Session_c2f(MPI_Session session);
+MPI_Datatype PMPI_Type_f2c(MPI_Fint datatype);
+MPI_Fint PMPI_Type_c2f(MPI_Datatype datatype);
+MPI_Win PMPI_Win_f2c(MPI_Fint win);
+MPI_Fint PMPI_Win_c2f(MPI_Win win);
+
+MPI_Comm MPI_Comm_f2c(MPI_Fint comm);
+MPI_Fint MPI_Comm_c2f(MPI_Comm comm);
+MPI_Errhandler MPI_Errhandler_f2c(MPI_Fint errhandler);
+MPI_Fint MPI_Errhandler_c2f(MPI_Errhandler errhandler);
+MPI_File MPI_File_f2c(MPI_Fint file);
+MPI_Fint MPI_File_c2f(MPI_File file);
+MPI_Group MPI_Group_f2c(MPI_Fint group);
+MPI_Fint MPI_Group_c2f(MPI_Group group);
+MPI_Info MPI_Info_f2c(MPI_Fint info);
+MPI_Fint MPI_Info_c2f(MPI_Info info);
+MPI_Message MPI_Message_f2c(MPI_Fint message);
+MPI_Fint MPI_Message_c2f(MPI_Message message);
+MPI_Op MPI_Op_f2c(MPI_Fint op);
+MPI_Fint MPI_Op_c2f(MPI_Op op);
+MPI_Request MPI_Request_f2c(MPI_Fint request);
+MPI_Fint MPI_Request_c2f(MPI_Request request);
+MPI_Session MPI_Session_f2c(MPI_Fint session);
+MPI_Fint MPI_Session_c2f(MPI_Session session);
+MPI_Datatype MPI_Type_f2c(MPI_Fint datatype);
+MPI_Fint MPI_Type_c2f(MPI_Datatype datatype);
+MPI_Win MPI_Win_f2c(MPI_Fint win);
+MPI_Fint MPI_Win_c2f(MPI_Win win);
 
 #if defined(__cplusplus)
 }
 #endif
