diff --git a/shared/offline_compiler/source/utilities/linux/safety_guard_linux.h b/shared/offline_compiler/source/utilities/linux/safety_guard_linux.h
index ed8a1472a..38953e9e7 100644
--- a/shared/offline_compiler/source/utilities/linux/safety_guard_linux.h
+++ b/shared/offline_compiler/source/utilities/linux/safety_guard_linux.h
@@ -10,7 +10,6 @@
 
 #include <cstdio>
 #include <cstdlib>
-#include <execinfo.h>
 #include <setjmp.h>
 #include <signal.h>
 
@@ -28,19 +27,6 @@ class SafetyGuardLinux {
     }
 
     static void sigAction(int sigNum, siginfo_t *info, void *ucontext) {
-        const int callstackDepth = 30;
-        void *addresses[callstackDepth];
-        char **callstack;
-        int backtraceSize = 0;
-
-        backtraceSize = backtrace(addresses, callstackDepth);
-        callstack = backtrace_symbols(addresses, backtraceSize);
-
-        for (int i = 0; i < backtraceSize; ++i) {
-            printf("[%d]: %s\n", i, callstack[i]);
-        }
-
-        free(callstack);
         longjmp(jmpbuf, 1);
     }
 
