From c38fbc46798b0d6538107474f5b3c51de035ea4c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Mos=C3=A8=20Giordano?= <mose@gnu.org>
Date: Tue, 6 Jul 2021 00:31:00 +0100
Subject: [PATCH 1/1] Fix whole-archive linker options for macOS

---
 CMakeLists.txt | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5c32d62a..fa5f660b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -582,6 +582,9 @@ target_link_libraries(SHOTSolver SHOTTasks)
 if(MSYS OR MINGW)
     # MinGW cross compilation will not work if "-Wl,--whole-archive" is defined below
     target_link_libraries(SHOTSolver SHOTModelingInterfaces)
+elseif(APPLE)
+    # need these whole-archive flags to ensure that EntryPointGAMS symbols will be present in library
+    target_link_libraries(SHOTSolver "-Wl,-all_load" SHOTModelingInterfaces)
 else()
     # need these whole-archive flags to ensure that EntryPointGAMS symbols will be present in library
     target_link_libraries(SHOTSolver "-Wl,--whole-archive" SHOTModelingInterfaces "-Wl,--no-whole-archive")
-- 
2.26.3

