diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5c32d62a..e5e2ac25 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -451,6 +451,8 @@ target_link_libraries(SHOTDualStrategy SHOTModel)
 
 # Creates the tasks library
 file(GLOB_RECURSE TASK_SOURCES "${PROJECT_SOURCE_DIR}/src/Tasks/*.cpp")
+list(REMOVE_ITEM TASK_SOURCES "${PROJECT_SOURCE_DIR}/src/Tasks/TaskBase.h")
+list(REMOVE_ITEM TASK_SOURCES "${PROJECT_SOURCE_DIR}/src/Tasks/TaskBase.cpp")
 add_library(SHOTTasks STATIC ${TASK_SOURCES})
 target_link_libraries(SHOTTasks SHOTPrimalStrategy)
 target_link_libraries(SHOTTasks SHOTDualStrategy)
