From fd86e1d5f258109d05792c648e280d99d9a702dc Mon Sep 17 00:00:00 2001
From: SpaceIm <30052553+SpaceIm@users.noreply.github.com>
Date: Sat, 28 Mar 2020 19:04:42 +0100
Subject: [PATCH] fix mingw build (#185)

---
 src/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 90f4501..932ba88 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -204,7 +204,7 @@ set_target_properties(${SIDX_C_LIB_NAME}
     PROPERTIES VERSION "${SIDX_LIB_VERSION}"
                SOVERSION "${SIDX_LIB_SOVERSION}" )
 
-if(WIN32)
+if(MSVC)
     target_compile_options(${SIDX_LIB_NAME} PRIVATE "/wd4068")
     target_compile_options(${SIDX_C_LIB_NAME} PRIVATE "/wd4068")
 
-- 
2.40.0

