diff --git a/CMakeLists.txt b/CMakeLists.txt
index c34ee8d..859ba6c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -144,11 +144,7 @@ if(WIN32)
   set(DEFAULT_DATA_SUBDIR .)
   set(DEFAULT_INCLUDE_SUBDIR include)
 
-  if (MSVC)
-    set(DEFAULT_BIN_SUBDIR bin)
-  else()
-    set(DEFAULT_BIN_SUBDIR .)
-  endif()
+  set(DEFAULT_BIN_SUBDIR bin)
 else()
   # Common locations for Unix and Mac OS X
   set(DEFAULT_BIN_SUBDIR bin)
