--- a/src/libscotch/library_dgraph_stat.c
+++ b/src/libscotch/library_dgraph_stat.c
@@ -65,7 +65,11 @@
 */
 
 static int                  dgraphstatblentab[2] = { 7, 3 };
-static MPI_Datatype         dgraphstattypetab[2] = { GNUM_MPI, MPI_DOUBLE };
+static MPI_Datatype         dgraphstattypetab[2];
+__attribute__((__constructor__)) static void init_dgraphstattypetab() {
+  dgraphstattypetab[0] = GNUM_MPI;
+  dgraphstattypetab[1] = MPI_DOUBLE;
+}
 
 /************************************/
 /*                                  */
