--- a/packages/zoltan/src/Utilities/Communication/comm_default.c
+++ b/packages/zoltan/src/Utilities/Communication/comm_default.c
@@ -36,7 +36,10 @@
 
 #else
 static pthread_mutex_t zoltan_global_mpi_lock;
-static MPI_Comm Zoltan_Global_MPI_Comm = MPI_COMM_WORLD; // CHECK: ALLOW MPI_COMM_WORLD
+static MPI_Comm Zoltan_Global_MPI_Comm;
+__attribute__((__constructor__)) static void init_Zoltan_Global_MPI_Comm() {
+  Zoltan_Global_MPI_Comm = MPI_COMM_WORLD; // CHECK: ALLOW MPI_COMM_WORLD
+}
 
 /* Function to set the default communicator */
 void zoltan_initialize_global_comm(MPI_Comm comm) {
