diff --git before/Jinja2Cpp/thirdparty/internal_deps.cmake after/Jinja2Cpp/thirdparty/internal_deps.cmake
index e09cee0..6f147e2 100644
--- before/Jinja2Cpp/thirdparty/internal_deps.cmake
+++ after/Jinja2Cpp/thirdparty/internal_deps.cmake
@@ -76,10 +76,10 @@ if (JINJA2CPP_BUILD_TESTS)
 endif()

 install (FILES
-        thirdparty/nonstd/expected-lite/include/nonstd/expected.hpp
-        thirdparty/nonstd/variant-lite/include/nonstd/variant.hpp
-        thirdparty/nonstd/optional-lite/include/nonstd/optional.hpp
-        thirdparty/nonstd/string-view-lite/include/nonstd/string_view.hpp
+        ${expected-lite_SOURCE_DIR}/include/nonstd/expected.hpp
+        ${variant-lite_SOURCE_DIR}/include/nonstd/variant.hpp
+        ${optional-lite_SOURCE_DIR}/include/nonstd/optional.hpp
+        ${string-view-lite_SOURCE_DIR}/include/nonstd/string_view.hpp
     DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/nonstd)

 install (TARGETS RapidJson