--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -174,7 +174,7 @@
     # If theses warnings are interpreted as errors, the test fails even though we want it to pass.
     # This behaviour caused a serious bug and did not initialize our shared memory. See https://github.com/DLR-AMR/t8code/issues/1985.
     set( T8CODE_OLD_CMAKE_REQUIRED_FLAGS "${CMAKE_REQUIRED_FLAGS}" )
-    if (WIN32)
+    if (WIN32 AND NOT MINGW)
         set(CMAKE_REQUIRED_FLAGS "${CMAKE_REQUIRED_FLAGS} /w" )
     else()
         set(CMAKE_REQUIRED_FLAGS "${CMAKE_REQUIRED_FLAGS} -w" )
--- a/sc/cmake/config.cmake
+++ b/sc/cmake/config.cmake
@@ -184,7 +184,7 @@
 check_symbol_exists(gettimeofday sys/time.h SC_HAVE_GETTIMEOFDAY)
 
 if(WIN32)
-  set(WINSOCK_LIBRARIES wsock32 ws2_32 Iphlpapi)
+  set(WINSOCK_LIBRARIES wsock32 ws2_32 iphlpapi)
 endif()
 
 check_include_file(libgen.h SC_HAVE_LIBGEN_H)
