--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -108,9 +108,9 @@
 
 if( T8CODE_ENABLE_MPI )
     if( T8CODE_BUILD_FORTRAN_INTERFACE )
-      find_package( MPI 3.0 COMPONENTS C Fortran REQUIRED )
+      find_package( MPI COMPONENTS C Fortran REQUIRED )
     else()
-      find_package( MPI 3.0 COMPONENTS C REQUIRED )
+      find_package( MPI COMPONENTS C REQUIRED )
     endif()
 
     if( NOT MPIEXEC_EXECUTABLE )
