--- a/src/t8_cmesh/t8_cmesh_vertex_connectivity/t8_cmesh_vertex_conn_tree_to_vertex.hxx
+++ b/src/t8_cmesh/t8_cmesh_vertex_connectivity/t8_cmesh_vertex_conn_tree_to_vertex.hxx
@@ -112,7 +112,7 @@
     /* We copy the data directly, hence set data_persiss to 0 */
     const int data_persists = 0;
-    t8_debugf ("Setting %i global vertices for global tree %li.\n", num_vertices, global_tree);
+    t8_debugf ("Setting %i global vertices for global tree %lli.\n", num_vertices, global_tree);
     t8_cmesh_set_attribute_gloidx_array (cmesh, global_tree, t8_get_package_id (),
                                          T8_CMESH_GLOBAL_VERTICES_ATTRIBUTE_KEY, global_tree_vertices, num_vertices,
                                          data_persists);
--- a/src/t8_cmesh/t8_cmesh_vertex_connectivity/t8_cmesh_vertex_conn_vertex_to_tree.hxx
+++ b/src/t8_cmesh/t8_cmesh_vertex_connectivity/t8_cmesh_vertex_conn_vertex_to_tree.hxx
@@ -115,7 +115,7 @@
     try {
       return vertex_to_tree.at (global_vertex_id);
     } catch (const std::out_of_range& e) {
-      t8_errorf ("ERROR: Could not find vertex %li for cmesh.\n", global_vertex_id);
+      t8_errorf ("ERROR: Could not find vertex %lli for cmesh.\n", global_vertex_id);
       SC_ABORTF ("Caught exception 'out of range': %s\n", e.what ());
     }
   }
--- a/src/t8_cmesh/t8_cmesh_io/t8_cmesh_readmshfile.cxx
+++ b/src/t8_cmesh/t8_cmesh_io/t8_cmesh_readmshfile.cxx
@@ -450,7 +450,7 @@
       return std::nullopt;
     }
     /* Fill the node with the entries in the file */
-    retval = sscanf (line, "%li %lf %lf %lf", &index, &coords[0], &coords[1], &coords[2]);
+    retval = sscanf (line, "%lli %lf %lf %lf", &index, &coords[0], &coords[1], &coords[2]);
     if (retval != 4) {
       t8_global_errorf ("Error reading node file after node %li.\n", (long) last_index);
       free (line);
@@ -461,7 +461,7 @@
     /* If second value is false then the node was already in the hash table.
      * This case should not occur. */
     if (emplaced.second == false) {
-      t8_global_errorf ("Node %li defined more than once.\n", index);
+      t8_global_errorf ("Node %lli defined more than once.\n", index);
       free (line);
       return std::nullopt;
     }
@@ -741,7 +741,7 @@
       for (int i_node = 0; i_node < num_nodes; i_node++) {
         const int t8_vertex_num = t8_msh_tree_vertex_to_t8_vertex_num[eclass][i_node];
         T8_ASSERT (strcmp (line_modify, "\0"));
-        retval = sscanf (line_modify, "%li", &node_indices[t8_vertex_num]);
+        retval = sscanf (line_modify, "%lli", &node_indices[t8_vertex_num]);
         if (retval != 1) {
           t8_global_errorf ("Premature end of line while reading tree.\n");
           t8_debugf ("The line is %s", line);
@@ -754,7 +754,7 @@
         Node.index = node_indices[t8_vertex_num];
         const auto found_node = vertices.find (Node);
         if (found_node == vertices.end ()) {
-          t8_global_errorf ("Could not find Node %li.\n", node_indices[t8_vertex_num]);
+          t8_global_errorf ("Could not find Node %lli.\n", node_indices[t8_vertex_num]);
           free (line);
           t8_cmesh_destroy (&cmesh);
           return std::nullopt;
@@ -1560,7 +1560,7 @@
         for (int i_node = 0; i_node < num_nodes; i_node++) {
           const int t8_vertex_num = t8_msh_tree_vertex_to_t8_vertex_num[eclass][i_node];
           T8_ASSERT (strcmp (line_modify, "\0"));
-          retval = sscanf (line_modify, "%li", &node_indices[t8_vertex_num]);
+          retval = sscanf (line_modify, "%lli", &node_indices[t8_vertex_num]);
           if (retval != 1) {
             t8_global_errorf ("Premature end of line while reading tree.\n");
             t8_debugf ("The line is %s", line);
