--- a/configure
+++ b/configure
@@ -7693,7 +7693,7 @@
 printf "%s\n" "$as_me: WARNING: Aqua can only be used when CoreFoundation is available" >&2;}
 	    tk_aqua=no
 	fi
-	if test ! -d /System/Library/Frameworks/Cocoa.framework; then
+	if test ! -d /opt/${host}/${host}/sys-root/System/Library/Frameworks/Cocoa.framework; then
 	    { printf "%s\n" "$as_me:${as_lineno-$LINENO}: WARNING: Aqua can only be used when Cocoa is available" >&5
 printf "%s\n" "$as_me: WARNING: Aqua can only be used when Cocoa is available" >&2;}
 	    tk_aqua=no
