diff --git a/amdocl/glibc_functions.cpp b/amdocl/glibc_functions.cpp
index 908ed75..8187a43 100644
--- a/amdocl/glibc_functions.cpp
+++ b/amdocl/glibc_functions.cpp
@@ -27,11 +27,11 @@ extern "C" {
 #endif // __cplusplus
 
 #if defined(_LP64)
-asm (".symver memcpy, memcpy@GLIBC_2.2.5");
+/*asm (".symver memcpy, memcpy@GLIBC_2.2.5");
 void *__wrap_memcpy(void *dest, const void *src, size_t n)
 {
     return memcpy(dest, src, n);
-}
+}*/
 #endif // _LP64
 
 #if defined(__cplusplus)
