diff --git a/thread/thread.cpp b/thread/thread.cpp
index feb25765..92f47539 100644
--- a/thread/thread.cpp
+++ b/thread/thread.cpp
@@ -121,7 +121,7 @@ void Thread::resume() {
 
 namespace details {
 
-__thread Thread* thread_ __attribute__((tls_model("initial-exec")));
+__thread Thread* thread_;
 
 }  // namespace details
 
diff --git a/thread/thread.hpp b/thread/thread.hpp
index 790bbfb3..8a9e0660 100644
--- a/thread/thread.hpp
+++ b/thread/thread.hpp
@@ -186,7 +186,7 @@ namespace details {
 
 #if defined(__linux__)
 
-extern __thread Thread* thread_ __attribute__((tls_model("initial-exec")));
+extern __thread Thread* thread_;
 
 static inline Thread* currentThread() { return thread_; }
 
