diff --git a/sysdeps/generic/ldconfig.h b/sysdeps/generic/ldconfig.h
index 94f008bd90..546bd0f6ba 100644
--- a/sysdeps/generic/ldconfig.h
+++ b/sysdeps/generic/ldconfig.h
@@ -28,6 +28,7 @@
 #define FLAG_ELF			0x0001
 #define FLAG_ELF_LIBC5			0x0002
 #define FLAG_ELF_LIBC6			0x0003
+#define FLAG_ELF_MUSL			0x0004
 #define FLAG_REQUIRED_MASK		0xff00
 #define FLAG_SPARC_LIB64		0x0100
 #define FLAG_IA64_LIB64			0x0200
