diff --git a/c/include/nvtx3/nvtxDetail/nvtxImpl.h b/c/include/nvtx3/nvtxDetail/nvtxImpl.h
index fa1ca8e..3639e20 100644
--- a/NVTX/c/include/nvtx3/nvtxDetail/nvtxImpl.h
+++ b/NVTX/c/include/nvtx3/nvtxDetail/nvtxImpl.h
@@ -73,9 +73,6 @@
 extern "C" {
 #endif /* __cplusplus */
 
-#ifdef __GNUC__
-#pragma GCC visibility push(hidden)
-#endif
 
 /* ---- Forward declare all functions referenced in globals ---- */
 
@@ -433,8 +433,5 @@ NVTX_LINKONCE_DEFINE_FUNCTION void NVTX_API NVTX_VERSIONED_IDENTIFIER(nvtxEtiSet
 
 #include "nvtxInit.h"
 
-#ifdef __GNUC__
-#pragma GCC visibility pop
-#endif
 
 #ifdef __cplusplus
 } /* extern "C" */
 
