diff --color --unified --color=auto -r gnuplot-6.0.4.orig/Makefile.am gnuplot-6.0.4/Makefile.am
--- gnuplot-6.0.4.orig/Makefile.am	2025-11-30 23:15:20
+++ gnuplot-6.0.4/Makefile.am	2026-01-17 14:17:11
@@ -7,6 +7,8 @@
 PATCHLEVEL PGPKEYS README RELEASE_NOTES \
 win
 
+dist_doc_DATA = FAQ.pdf README RELEASE_NOTES Copyright NEWS BUGS
+
 DISTCLEANFILES = stamp-h
 
 ACLOCAL_AMFLAGS = -I m4
diff --color --unified --color=auto -r gnuplot-6.0.4.orig/config/mingw/Makefile gnuplot-6.0.4/config/mingw/Makefile
--- gnuplot-6.0.4.orig/config/mingw/Makefile	2025-12-16 21:48:52
+++ gnuplot-6.0.4/config/mingw/Makefile	2026-01-17 14:20:47
@@ -872,27 +872,27 @@
 	-$(HHC) chm/wgnuplot.hhp
 	-$(CP) chm/wgnuplot.chm wgnuplot.chm
 
-htmlhelp.def: /c/Windows/System32/hhctrl.ocx
-ifndef MINGW64
-ifndef CLANG64
-	pexports $^ > $@
-else
-	gendef - $^ > $@
-endif
-else
-	gendef - $^ > $@
-endif
+# htmlhelp.def: /c/Windows/System32/hhctrl.ocx
+# ifndef MINGW64
+# ifndef CLANG64
+# 	pexports $^ > $@
+# else
+# 	gendef - $^ > $@
+# endif
+# else
+# 	gendef - $^ > $@
+# endif
 
-prntvpt.def: /c/Windows/System32/prntvpt.dll
-ifndef MINGW64
-ifndef CLANG64
-	pexports $^ > $@
-else
-	gendef - $^ > $@
-endif
-else
-	gendef - $^ > $@
-endif
+# prntvpt.def: /c/Windows/System32/prntvpt.dll
+# ifndef MINGW64
+# ifndef CLANG64
+# 	pexports $^ > $@
+# else
+# 	gendef - $^ > $@
+# endif
+# else
+# 	gendef - $^ > $@
+# endif
 
 # Working directory for patched Japanese sources 
 JA = ja/
diff --color --unified --color=auto -r gnuplot-6.0.4.orig/configure.ac gnuplot-6.0.4/configure.ac
--- gnuplot-6.0.4.orig/configure.ac	2025-12-12 05:12:31
+++ gnuplot-6.0.4/configure.ac	2026-01-17 14:17:11
@@ -38,6 +38,8 @@
 AC_C_STRINGIZE
 AC_PROG_LN_S
 
+AC_CANONICAL_HOST
+
 if test "${build}" != "${host}"
 then
   CC=${CC-${host_alias-gcc}}
@@ -51,6 +53,8 @@
   CC_FOR_BUILD="${CC}"
   AC_PROG_CC
 
+  build_mingw=no
+
   # We must set the default linker to the linker used by gcc for the correct
   # operation of libtool.  If LD is not defined and we are using gcc, try to
   # set the LD default to the ld used by gcc.
@@ -60,7 +64,10 @@
     then
       case $build in
       *-*-mingw*)
-        gcc_prog_ld=`$CC -print-prog-name=ld 2>&1 | tr -d '\015'` ;;
+        gcc_prog_ld=`$CC -print-prog-name=ld 2>&1 | tr -d '\015'`
+        AC_MSG_NOTICE([MINGW environment detected, executing MINGW type build])
+        #AC_CONFIG_LINKS([src/config.h:config/config.mgw])
+        build_mingw=yes ;;
       *)
         gcc_prog_ld=`$CC -print-prog-name=ld 2>&1` ;;
       esac
@@ -999,6 +1006,8 @@
     AC_DEFINE(USE_POLAR_GRID,1,
                [ Define to support polar gridded surface plots])
   )
+
+AM_CONDITIONAL(BUILD_MINGW, test "${build_mingw}" = yes)
 
 dnl Enable subsystem to generate statistical summary of file contents
 AC_ARG_ENABLE(stats,dnl
diff --color --unified --color=auto -r gnuplot-6.0.4.orig/demo/plugin/Makefile.am gnuplot-6.0.4/demo/plugin/Makefile.am
--- gnuplot-6.0.4.orig/demo/plugin/Makefile.am	2025-11-30 23:15:20
+++ gnuplot-6.0.4/demo/plugin/Makefile.am	2026-01-17 14:17:11
@@ -4,7 +4,7 @@
 noinst_PROGRAMS = demo_plugin.so
 
 demo_plugin_so_SOURCES = demo_plugin.c gnuplot_plugin.h
-demo_plugin_so_CFLAGS = -rdynamic -fPIC -I$(top_srcdir)/src
+demo_plugin_so_CFLAGS = -fPIC -I$(top_srcdir)/src
 demo_plugin_so_LDFLAGS = -shared
 
 # Note: Solaris SunPro C compiler needs these flags instead
diff --color --unified --color=auto -r gnuplot-6.0.4.orig/docs/Makefile.am gnuplot-6.0.4/docs/Makefile.am
--- gnuplot-6.0.4.orig/docs/Makefile.am	2025-11-30 23:15:20
+++ gnuplot-6.0.4/docs/Makefile.am	2026-01-17 14:17:11
@@ -48,6 +48,11 @@
 MP =
 endif
 
+if BUILD_MINGW
+AM_CPPFLAGS += -DWINDOWS_NO_GUI
+dist_doc_DATA = gnuplot.pdf
+endif
+
 # Documentation will be prepared for all terminals in CORETERM
 # even if the terminal itself is not included in the build.
 
diff --color --unified --color=auto -r gnuplot-6.0.4.orig/src/Makefile.am gnuplot-6.0.4/src/Makefile.am
--- gnuplot-6.0.4.orig/src/Makefile.am	2025-11-30 23:15:20
+++ gnuplot-6.0.4/src/Makefile.am	2026-01-17 14:17:11
@@ -99,6 +99,27 @@
 # Hercules and original pc graphics driver code
 # corgraph.asm header.mac hrcgraph.asm pcgraph.asm lineproc.mac
 
+# MINGW specific build section
+
+
+AM_CPPFLAGS += -DHAVE_GDIPLUS -DHAVE_DWRITE -DWIN32
+AM_CPPFLAGS += -O2 -pipe -DUNICODE -D_UNICODE -Wno-unused-function
+AM_CPPFLAGS += -D_Windows -DHAVE_CONFIG_H
+AM_CPPFLAGS += -D__USE_MINGW_ANSI_STDIO=1
+AM_CPPFLAGS += -DUSE_MOUSE=1 -DWIN_IPC
+
+AM_LDFLAGS = 
+
+gnuplot_SOURCES += win/winmain.c win/wgnuplib.c win/wgraph.c win/wprinter.c \
+win/wpause.c win/wgdiplus.cpp win/wtext.h plot.h\
+gpexecute.c
+
+gnuplot_CPPFLAGS = $(AM_CPPFLAGS) -DPIPES -DWGP_CONSOLE
+gnuplot_LDADD += -lkernel32 -lgdi32 -lwinspool -lcomdlg32 -lcomctl32 \
+ -ladvapi32 -lshell32 -lmsimg32 -lgdiplus -lshlwapi -ld2d1 -ld3d11 -ldwrite \
+ -lwindowscodecs -lole32 -lgobject-2.0
+gnuplot_LDFLAGS = $(AM_LDFLAGS) -mconsole -municode
+
 if BUILD_WXWIDGETS
 gnuplot_SOURCES += wxterminal/wxt_gui.cpp
 endif
@@ -109,6 +130,41 @@
 
 if BUILD_BITMAP
 gnuplot_SOURCES += bitmap.c
+
+#wgnuplot section
+bin_PROGRAMS += wgnuplot
+
+menudir = $(prefix)/bin
+menu_DATA = win/wgnuplot.mnu
+
+wgnuplot_SOURCES = $(gnuplot_SOURCES)
+wgnuplot_SOURCES += win/wtext.c win/screenbuf.c win/wmenu.c win/wredirect.cpp
+wgnuplot_CPPFLAGS = $(AM_CPPFLAGS) -DPIPES
+wgnuplot_LDADD = $(gnuplot_LDADD) wgplt_res.$(O)
+wgnuplot_LDFLAGS = $(AM_LDFLAGS) -mwindows --enable-auto-import -municode
+
+#wgnuplot_pipes section
+bin_PROGRAMS += wgnuplot_pipes
+
+wgnuplot_pipes_SOURCES = $(gnuplot_SOURCES)
+wgnuplot_pipes_SOURCES += win/wtext.c win/screenbuf.c win/wmenu.c win/wredirect.cpp
+wgnuplot_pipes_CPPFLAGS = $(AM_CPPFLAGS) -DPIPES
+wgnuplot_pipes_LDADD = $(gnuplot_LDADD) wgplt_res.$(O)
+wgnuplot_pipes_LDFLAGS = $(AM_LDFLAGS) -mwindows -municode
+
+RCFLAGS = --include-dir=$(srcdir)/win \
+	--define __WIN32__ --define __WIN95__ \
+	--define __GNUWIN32__ --use-temp-file
+RCOUT = wgplt_res.$(O)
+RES2COFF = echo wgplt_res.$(O)
+
+wgplt_res.$(O): win/wgnuplot.rc win/wgnuplib.rc win/wresourc.h win/texticon.ico win/grpicon.ico win/wgnuplot.exe.manifest win/wgnuplot.exe.manifest64
+	echo Making RESOURCE FILE
+	windres $(RCFLAGS) $< wgplt_res.$(O)
+	echo wgplt_res.$(O)
+
+
+
 endif
 
 DISTCLEANFILES = timestamp.h
diff --color --unified --color=auto -r gnuplot-6.0.4.orig/src/command.c gnuplot-6.0.4/src/command.c
--- gnuplot-6.0.4.orig/src/command.c	2025-12-12 05:12:31
+++ gnuplot-6.0.4/src/command.c	2026-01-17 14:17:11
@@ -107,7 +107,7 @@
 static TBOOLEAN pause_internal;    /* flag to indicate not to use a dialog box */
 #endif /* OS2_IPC */
 
-#ifndef _WIN32
+#ifndef _MSC_VER
 # include "help.h"
 #endif
 
@@ -3365,7 +3365,7 @@
 
 
 #ifdef NO_GIH
-#ifdef _WIN32
+#ifdef _MSC_VER
 void
 help_command()
 {
diff --color --unified --color=auto -r gnuplot-6.0.4.orig/src/win/winmain.c gnuplot-6.0.4/src/win/winmain.c
--- gnuplot-6.0.4.orig/src/win/winmain.c	2025-11-01 17:36:35
+++ gnuplot-6.0.4/src/win/winmain.c	2026-01-17 14:17:11
@@ -372,7 +372,8 @@
     else
 	lang = GetLanguageCode();
 
-    filename = (LPTSTR) malloc((_tcslen(szModuleName) + _tcslen(name) + _tcslen(lang) + _tcslen(ext) + 1) * sizeof(TCHAR));
+    //filename = (LPTSTR) malloc((_tcslen(szModuleName) + _tcslen(name) + _tcslen(lang) + _tcslen(ext) + 1) * sizeof(TCHAR));
+    filename = (LPTSTR) malloc(1024 * sizeof(TCHAR));
     if (filename) {
 	_tcscpy(filename, szModuleName);
 	_tcscat(filename, name);
