From d9a4913cf45eb87885b72d2126f7491ac94f58e9 Mon Sep 17 00:00:00 2001
From: Keno Fischer <keno@juliacomputing.com>
Date: Wed, 24 Mar 2021 14:43:17 -0400
Subject: [PATCH] Add missing rule for ngicon.rc

---
 src/Makefile.am | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/src/Makefile.am b/src/Makefile.am
index ede7a9e69..0e064dc0c 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -192,9 +192,12 @@ endif
 endif
 
 if WINRESOURCE
-ngspice_LDADD += ngicon.o
+ngspice_SOURCES += ngicon.rc
 endif
 
+.rc.lo:
+	$(LIBTOOL) $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) --tag=RC --mode=compile $(RC) $(RCFLAGS) $< -o $@
+
 if OLDAPPS
 ## nutmeg:
 
@@ -301,7 +304,7 @@ ngmakeidx_LDADD =  \
 	frontend/libfte.la \
 	frontend/parser/libparser.la \
 	misc/libmisc.la
-    
+
 ## create index for online help:
 
 ngspice.idx: ngmakeidx$(EXEEXT) $(srcdir)/ngspice.txt
-- 
2.25.1

