--- a/src/daemon/meson.build
+++ b/src/daemon/meson.build
@@ -29,7 +29,6 @@
   install: true,
   install_rpath : '$ORIGIN/pulseaudio',
   include_directories : [configinc, topinc],
-  link_args : ['-ffast-math'],
   link_with : [libpulsecore],
   dependencies : [ltdl_dep, cap_dep, dbus_dep, libsystemd_dep, dl_dep, libintl_dep, platform_dep, platform_socket_dep, libpulsecommon_dep, libpulse_dep],
   c_args : pa_c_args,
