--- a/runtime/CMakeLists.txt
+++ b/runtime/CMakeLists.txt
@@ -61,7 +61,7 @@
       GIT_TAG               "v3.1.1"
       SOURCE_DIR            ${UTFCPP_DIR}
       UPDATE_DISCONNECTED   1
-      CMAKE_ARGS            -DCMAKE_INSTALL_PREFIX=${UTFCPP_DIR}/install -DUTF8_TESTS=off -DUTF8_SAMPLES=off
+      CMAKE_ARGS            -DCMAKE_INSTALL_PREFIX=${UTFCPP_DIR}/install -DUTF8_TESTS=off -DUTF8_SAMPLES=off -DCMAKE_TOOLCHAIN_FILE=$ENV{CMAKE_TARGET_TOOLCHAIN}
       STEP_TARGETS          build)
 
     include_directories(
