diff --git a/src/sys/dll/cxx/demangle.cxx b/src/sys/dll/cxx/demangle.cxx
index 0c7a1e32aa3..fc3b5437371 100644
--- a/src/sys/dll/cxx/demangle.cxx
+++ b/src/sys/dll/cxx/demangle.cxx
@@ -17,6 +17,6 @@ PetscErrorCode PetscDemangleSymbol(const char mangledName[], char **name)
   newname = __cxxabiv1::__cxa_demangle(mangledName, NULL, NULL, &status);
   if (status) {
     PetscCheck(status != -1, PETSC_COMM_SELF, PETSC_ERR_MEM, "Failed to allocate memory for symbol %s", mangledName);
-    PetscCheck(status == -2, PETSC_ERR_LIB, "Demangling failed for symbol %s", mangledName);
+    PetscCheck(status == -2, PETSC_COMM_SELF, PETSC_ERR_LIB, "Demangling failed for symbol %s", mangledName);
     /* Mangled name is not a valid name under the C++ ABI mangling rules */
     PetscCall(PetscStrallocpy(mangledName, name));
     PetscFunctionReturn(PETSC_SUCCESS);