diff --git a/config/BuildSystem/config/packages/BlasLapack.py b/config/BuildSystem/config/packages/BlasLapack.py
index bbae50c096..9ad0a994eb 100644
--- a/config/BuildSystem/config/packages/BlasLapack.py
+++ b/config/BuildSystem/config/packages/BlasLapack.py
@@ -128,7 +128,8 @@ class Configure(config.package.Package):
       blasLibrary   = [blasLibrary]
 
     # allow a user-specified suffix to be appended to BLAS/LAPACK symbols
-    self.suffix = self.argDB.get('with-blaslapack-suffix', '')
+    # argDB return a list when with-blaslapack-suffix is used
+    self.suffix = self.argDB.get('with-blaslapack-suffix', '')[0]
     # allow user to dictate which BLAS/LAPACK mangling to use (some BLAS/LAPACK libraries, like on Apple, provide several)
     if 'known-blaslapack-mangling' in self.argDB:
       mangling = self.argDB['known-blaslapack-mangling']
