From 38383de3d05778645f65a450b03a7429b8d49a69 Mon Sep 17 00:00:00 2001
From: Valentin Churavy <v.churavy@gmail.com>
Date: Tue, 7 Feb 2023 16:25:56 -0500
Subject: [PATCH] don't build tests either

---
 CMakeLists.txt | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8b0a308..fb0a30a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -390,6 +390,3 @@ install(TARGETS amgxsh DESTINATION "lib")
 # build examples
 
 # build unit tests
-if(EXISTS ${CMAKE_SOURCE_DIR}/tests)
-   add_subdirectory(tests)
-endif(EXISTS ${CMAKE_SOURCE_DIR}/tests)
-- 
2.37.2

