commit 2bc4d33779124193c487fb93df3d02d733369bc1
Author: Elliot Saba <staticfloat@gmail.com>
Date:   Fri Jul 9 22:01:42 2021 +0000

    Fix `soapy` test C++ issues

diff --git a/libxtrx/soapy/CMakeLists.txt b/libxtrx/soapy/CMakeLists.txt
index edfe657..407fef9 100644
--- a/libxtrx/soapy/CMakeLists.txt
+++ b/libxtrx/soapy/CMakeLists.txt
@@ -1,9 +1,9 @@
 set(CMAKE_CXX_STANDARD 11)
-project(soapyxtrx C CXX)
+project(soapyxtrx CXX)
 
 find_package(SoapySDR REQUIRED)
 
-add_executable(test_xtrx_soapy test_xtrx_soapy.c)
+add_executable(test_xtrx_soapy test_xtrx_soapy.cc)
 target_link_libraries(test_xtrx_soapy ${SoapySDR_LIBRARIES})
 
 SOAPY_SDR_MODULE_UTIL(
diff --git a/soapy/test_xtrx_soapy.c b/soapy/test_xtrx_soapy.cc
similarity index 99%
rename from soapy/test_xtrx_soapy.c
rename to soapy/test_xtrx_soapy.cc
index 7221410..0435380 100644
--- a/libxtrx/soapy/test_xtrx_soapy.c
+++ b/libxtrx/soapy/test_xtrx_soapy.cc
@@ -74,7 +74,7 @@ int main(void)
     SoapySDRDevice_activateStream(sdr, rxStream, 0, 0, 0); //start streaming
 
     //create a re-usable buffer for rx samples
-    complex float buff[1024];
+    _Complex float buff[1024];
 
     //receive some samples
     for (size_t i = 0; i < 10; i++)
