diff --git before/aeron/CMakeLists.txt after/aeron/CMakeLists.txt
index a28de49..f2cf458 100755
--- before/aeron/CMakeLists.txt
+++ after/aeron/CMakeLists.txt
@@ -134,6 +134,7 @@ if (UNIX)
         message("Set -Ofast for ${CMAKE_CXX_COMPILER_ID} ${CMAKE_CXX_COMPILER_VERSION}")
         add_compile_options($<$<CONFIG:Release>:-Ofast>)
     endif ()
+    unset(CMAKE_C_STANDARD)
 endif ()
 
 # platform specific flags
