diff -ru at-spi2-core-2.57.1.old/atspi/atspi-event-listener.c at-spi2-core-2.57.1/atspi/atspi-event-listener.c
--- at-spi2-core-2.57.1.old/atspi/atspi-event-listener.c	2025-08-01 19:07:24
+++ at-spi2-core-2.57.1/atspi/atspi-event-listener.c	2025-08-07 12:27:41
@@ -22,11 +22,14 @@
  * Boston, MA 02110-1301, USA.
  */
 
+#define _BSD_SOURCE
+
 #include "atspi-accessible-private.h"
 #include "atspi-private.h"
 #include <ctype.h>
 #include <string.h>
 #include <strings.h>
+#include <sys/time.h>
 
 static struct timeval window_filter_time;
 
