diff --git a/src/work_mgga_inc.c b/src/work_mgga_inc.c
index f4b18d5..99eacc0 100644
--- a/src/work_mgga_inc.c
+++ b/src/work_mgga_inc.c
@@ -91,7 +91,10 @@ WORK_MGGA(ORDER_TXT, SPIN_TXT)
       my_sigma[1] = (my_sigma[1] <= +s_ave ? my_sigma[1] : +s_ave);
     }
 
-    FUNC(ORDER_TXT, SPIN_TXT)(p, ip, my_rho, my_sigma, &VAR(lapl, ip, 0), my_tau, out);
+    // Tweak the code here to bypass an ICE in the aarch64-linux-gnu-libgfortran3 target,
+    // at the cost of a warning for the temporary const double* -> double* conversion.
+    double *lapl_ptr = &VAR(lapl, ip, 0);
+    FUNC(ORDER_TXT, SPIN_TXT)(p, ip, my_rho, my_sigma, lapl_ptr, my_tau, out);
 
     /* check for NaNs */
 #ifdef XC_DEBUG
