diff --git a/CMakeLists.txt b/CMakeLists.txt
index da0395e..240e294 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -303,9 +303,6 @@ if (${CMAKE_SYSTEM_PROCESSOR} MATCHES "arm" OR ${CMAKE_SYSTEM_PROCESSOR} MATCHES
     if (MSVC)
         # TODO [llama.cpp]: arm msvc?
     else()
-        if (${CMAKE_SYSTEM_PROCESSOR} MATCHES "aarch64")
-            add_compile_options(-mcpu=native)
-        endif()
         # TODO [llama.cpp]: armv6,7,8 version specific flags
     endif()
 elseif (${CMAKE_SYSTEM_PROCESSOR} MATCHES "^(x86_64|i686|AMD64)$")
@@ -452,3 +449,5 @@ if (NOT RWKV_STANDALONE)
 elseif()
     set_property(TARGET ggml PROPERTY GGML_STANDALONE ON)
 endif()
+
+install(TARGETS rwkv)
