diff --git a/Makefile b/Makefile
index 908c578..b389bed 100644
--- a/Makefile
+++ b/Makefile
@@ -229,8 +229,8 @@ endif
 # Target compatibility with GLIBC 2.3.4
 # 1) _GNU_SOURCE avoids using isoc99_fscanf, limiting binary portability to recent GLIBC.
 # 2) -fno-stack-protector avoids using a feature requiring GLBIC 2.4
-STD_CFLAGS += -m$(march) -D_GNU_SOURCE -fno-stack-protector
-STD_LDFLAGS += -m$(march) -Wl,--rpath,\$$ORIGIN/ -Wl,--as-needed
+STD_CFLAGS += -D_GNU_SOURCE -fno-stack-protector
+STD_LDFLAGS += -Wl,--rpath,\$$ORIGIN/ -Wl,--as-needed
 endif
 
 # Convert back DISALBE_*="no" flags to be empty
diff --git a/make/dll.mak b/make/dll.mak
index 7895c75..cfda8c2 100644
--- a/make/dll.mak
+++ b/make/dll.mak
@@ -55,25 +55,21 @@ BINDIR = bin/$(ARCH)
 # Mac OS X on Intel 32 bit processor
 ifeq ($(ARCH),maci)
 DLL_SUFFIX := dylib
-DLL_LDFLAGS += -m32
 endif
 
 # Mac OS X on Intel 64 bit processor
 ifeq ($(ARCH),maci64)
 DLL_SUFFIX := dylib
-DLL_LDFLAGS += -m64
 endif
 
 # Linux-32
 ifeq ($(ARCH),glnx86)
 DLL_SUFFIX := so
-DLL_LDFLAGS += -m32
 endif
 
 # Linux-64
 ifeq ($(ARCH),glnxa64)
 DLL_SUFFIX := so
-DLL_LDFLAGS += -m64
 endif
 
 # --------------------------------------------------------------------
