diff --git a/src/test_threads.c b/src/test_threads.c
index 3557121..99ba88b 100644
--- a/src/test_threads.c
+++ b/src/test_threads.c
@@ -22,7 +22,7 @@ the terms of the BSD license (see the COPYING file).
 #endif
 
 #if defined(VL_THREADS_WIN)
-#include <Windows.h>
+#include <windows.h>
 #endif
 
 #if defined(VL_THREADS_POSIX)
diff --git a/toolbox/misc/vl_getpid.c b/toolbox/misc/vl_getpid.c
index 821e7ab..3c9e361 100644
--- a/toolbox/misc/vl_getpid.c
+++ b/toolbox/misc/vl_getpid.c
@@ -15,7 +15,7 @@ the terms of the BSD license (see the COPYING file).
 #include <vl/generic.h>
 
 #ifdef VL_OS_WIN
-#include <Windows.h>
+#include <windows.h>
 #else
 #include <unistd.h>
 #endif
diff --git a/vl/generic.c b/vl/generic.c
index c6f84a9..ade63cd 100644
--- a/vl/generic.c
+++ b/vl/generic.c
@@ -776,7 +776,7 @@ calling @c omp_set_num_threads() in the application. Note that:
 #include <math.h>
 
 #if defined(VL_OS_WIN)
-#include <Windows.h>
+#include <windows.h>
 #endif
 
 #if ! defined(VL_DISABLE_THREADS) && defined(VL_THREADS_POSIX)
