--- /dev/null
+++ b/CMakeLists.txt
@@ -1,6 +1,6 @@
-cmake_minimum_required(VERSION 2.8)
+cmake_minimum_required(VERSION 3.2)
 set(CMAKE_BUILD_TYPE  "Release")
-SET(CMAKE_C_FLAGS   "-w -g -O3 -funroll-loops")
+SET(CMAKE_C_FLAGS     "-w -g -O3 -funroll-loops")
 set(CMAKE_MACOSX_RPATH 1)
 set(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
 
@@ -29,7 +29,10 @@ TARGET_LINK_LIBRARIES( Commons ${LIBRARIES})
 
 INSTALL(
   TARGETS Commons
-  RUNTIME DESTINATION "$ENV{HOME}/lib"
-  LIBRARY DESTINATION "$ENV{HOME}/lib")
+  RUNTIME DESTINATION "bin"
+  LIBRARY DESTINATION "bin")
+
+INSTALL(
+  FILES ${header_files}
+  DESTINATION "include")
 
-INSTALL(               FILES   ${header_files} DESTINATION "$ENV{HOME}/include")
