--- /dev/null
+++ b/CMakeLists.txt
@@ -1,6 +1,6 @@
-cmake_minimum_required(VERSION 2.8)
+cmake_minimum_required(VERSION 3.2)
 set(CMAKE_BUILD_TYPE  "Release")
-SET(CMAKE_CXX_FLAGS   "-w -g -O3")
+SET(CMAKE_C_FLAGS     "-w -g -O3")
 set(CMAKE_MACOSX_RPATH 1)
 set(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
 
@@ -11,24 +11,26 @@ project(Elas)
 file(                  GLOB_RECURSE source_files sources/*)
 file(                  GLOB_RECURSE header_files sources/*.h)
 ADD_LIBRARY(           Elas SHARED ${source_files})
-find_library(          Commons NAMES Commons HINTS "$ENV{HOME}/lib")
-INCLUDE_DIRECTORIES(  "$ENV{HOME}/include")
-LINK_DIRECTORIES(     "$ENV{HOME}/lib")
+find_file(             Commons NAMES libCommons.dll PATH_SUFFIXES bin ../bin)
+# INCLUDE_DIRECTORIES(  "include")
+# LINK_DIRECTORIES(     "lib")
 target_link_libraries( Elas ${Commons})
+
 INSTALL(
   TARGETS Elas
-  LIBRARY  DESTINATION "$ENV{HOME}/lib"
-  RUNTIME  DESTINATION "$ENV{HOME}/lib")
-INSTALL(               FILES ${header_files} DESTINATION "$ENV{HOME}/include")
+  LIBRARY  DESTINATION "bin"
+  RUNTIME  DESTINATION "bin")
+
+INSTALL(
+  FILES ${header_files}
+  DESTINATION "include")
 
 #Executable
 project(main)
 add_executable(        elastic sources/elastic.c)
 target_link_libraries( elastic ${Commons} Elas)
-INSTALL(               TARGETS elastic RUNTIME DESTINATION "$ENV{HOME}/bin")
-
-
- 
-
 
+INSTALL(
+  TARGETS elastic
+  RUNTIME DESTINATION "bin")
 
--- /dev/null
+++ b/sources/elastic.c
@@ -27,8 +27,10 @@ static void excfun(int sigid) {
   switch(sigid) {
     case SIGABRT:
       fprintf(stdout,"  Abnormal stop\n");  break;
+#ifdef SIGBUS
     case SIGBUS:
       fprintf(stdout,"  Code error...\n");  break;
+#endif
     case SIGFPE:
       fprintf(stdout,"  Floating-point exception\n"); break;
     case SIGILL:
@@ -341,7 +343,9 @@ int main(int argc,char **argv) {
   signal(SIGSEGV,excfun);
   signal(SIGTERM,excfun);
   signal(SIGINT,excfun);
+#ifdef SIGBUS
   signal(SIGBUS,excfun);
+#endif
 
   /* init structure */
   memset(&lsst.info,0,sizeof(Info));
