--- a/src/picosat/picosat.c
+++ b/src/picosat/picosat.c
@@ -8144,10 +8144,13 @@ picosat_stats (PS * ps)
            ps->prefix, picosat_max_bytes_allocated (ps) / (double) (1 << 20));
 }

+#if defined(WIN32) || defined(_WIN32) || defined(__WIN32) || defined(WIN32_) || defined(WIN32__) || defined(__WIN32__) /* mingw does not include sys/resource.h */
+#define NGETRUSAGE
+#endif
+
 #ifndef NGETRUSAGE
 #include <sys/time.h>
 #include <sys/resource.h>
-#include <sys/unistd.h>
 #endif

 double
