--- a/CMakeLists.txt	2022-12-22 13:34:39
+++ b/CMakeLists.txt	2022-12-22 13:35:40
@@ -569,11 +569,13 @@
 endif()


+if (NOT CMAKE_CROSSCOMPILING)
 include(CheckFloatPrecision)
 check_float_precision()
 if (HAVE__FPU_SETCW)
     add_definitions(-DYALSAT_FPU)
     message(STATUS "Found FPU code for yalsat: fpu_control.h, _FPU_SINGLE, _FPU_DOUBLE")
 endif()
+endif()

 option(WEIGHTED_SAMPLING "Allow for weighted sampling" OFF)
