From 9184fafd3efb8ed5f6abec6f9c5692bd2ffd1fe5 Mon Sep 17 00:00:00 2001
From: Julian P Samaroo <jpsamaroo@jpsamaroo.me>
Date: Tue, 18 Jan 2022 13:32:28 -0600
Subject: [PATCH] [lld/CMake] Respect LLVM_TOOLS_INSTALL_DIR

---
 lld/cmake/modules/AddLLD.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lld/cmake/modules/AddLLD.cmake b/lld/cmake/modules/AddLLD.cmake
index 23df4131240..49ea01cb57e 100644
--- a/lld/cmake/modules/AddLLD.cmake
+++ b/lld/cmake/modules/AddLLD.cmake
@@ -22,7 +22,7 @@ macro(add_lld_library name)
       ${export_to_lldtargets}
       LIBRARY DESTINATION lib${LLVM_LIBDIR_SUFFIX}
       ARCHIVE DESTINATION lib${LLVM_LIBDIR_SUFFIX}
-      RUNTIME DESTINATION bin)
+      RUNTIME DESTINATION ${LLVM_TOOLS_INSTALL_DIR})
 
     if (${ARG_SHARED} AND NOT CMAKE_CONFIGURATION_TYPES)
       add_llvm_install_targets(install-${name}
-- 
2.33.0

