diff --git a/c/include/nvtx3/nvtxDetail/nvtxImpl.h b/c/include/nvtx3/nvtxDetail/nvtxImpl.h
index 590ce90..c728fa7 100644
--- a/NVTX/c/include/nvtx3/nvtxDetail/nvtxImpl.h
+++ b/NVTX/c/include/nvtx3/nvtxDetail/nvtxImpl.h
@@ -23,6 +23,10 @@
 #include <android/api-level.h> 
 #endif
 
+#if defined(__linux__)
+#include <linux/limits.h>
+#endif
+
 #if defined(__linux__) || defined(__CYGWIN__)
 #include <sched.h>
 #endif
