diff --git before/qr_mumps/CMakeLists.txt after/qr_mumps/CMakeLists.txt
index 6b7a300..9197f8f 100644
--- before/qr_mumps/CMakeLists.txt
+++ after/qr_mumps/CMakeLists.txt
@@ -416,10 +416,10 @@ message("=======================================================================
 
 add_subdirectory(include)
 add_subdirectory(src)
-add_subdirectory(examples/C)
-add_subdirectory(examples/Fortran)
+# add_subdirectory(examples/C)
+# add_subdirectory(examples/Fortran)
 add_subdirectory(timing)
-add_subdirectory(testing)
+# add_subdirectory(testing)
 
 
 enable_testing()
