diff --git a/src/cWrapper.f95 b/src/cWrapper.f95
index 6bfc7adc..01b2fe18 100644
--- a/src/cWrapper.f95
+++ b/src/cWrapper.f95
@@ -304,7 +304,7 @@
         real(kind=c_double), dimension(theta_d0,theta_d1),intent(out) :: theta
         real(kind=c_double), dimension(theta_d0,theta_d1),intent(out) :: phi
         integer(kind=c_int), value,intent(in) :: lmax
-        integer(kind=c_double), optional,intent(in) :: radius
+        real(kind=c_double), optional,intent(in) :: radius
         integer(kind=c_int), intent(out) :: n
         integer(kind=c_int), optional,intent(in) :: sampling
         integer(kind=c_int), optional,intent(in) :: lmax_calc
