--- meson.build
+++ meson.build
@@ -33,10 +33,10 @@
 		'-Wold-style-definition', # nop
 		'-Wpointer-arith',
 		'-Wstrict-prototypes',
-		'-Wimplicit-fallthrough',
+#		'-Wimplicit-fallthrough',
 		'-Wmissing-prototypes',
-		'-Wno-unknown-warning-option',
-		'-Wno-unused-command-line-argument',
+#		'-Wno-unknown-warning-option',
+#		'-Wno-unused-command-line-argument',
 		'-Wvla',
 		'-Wl,--exclude-libs=ALL',
 		'-D_XOPEN_SOURCE=700',
@@ -153,7 +153,7 @@
 	link_with: private_lib,
 	include_directories: [include_directories('.', 'include')],
 	install: true,
-	link_args: symbols_flag,
+	link_args: ['-lrt', symbols_flag],
 	link_depends: symbols_file,
 )
 
@@ -173,6 +173,7 @@
 		[ server_files, 'seatd/seatd.c' ],
 		include_directories: [include_directories('.', 'include')],
 		install: true,
+		link_args: ['-lrt'],
 	)
 endif
 
