--- meson.build
+++ meson.build
@@ -153,7 +153,7 @@
 	link_with: private_lib,
 	include_directories: [include_directories('.', 'include')],
 	install: true,
-	link_args: symbols_flag,
+	link_args: ['-lrt', symbols_flag],
 	link_depends: symbols_file,
 )
 
@@ -173,6 +173,7 @@
 		[ server_files, 'seatd/seatd.c' ],
 		include_directories: [include_directories('.', 'include')],
 		install: true,
+		link_args: ['-lrt'],
 	)
 endif
 
