--- a/configure.ac
+++ b/configure.ac
@@ -359,11 +359,11 @@
     ;;
   *-*-mingw*)
     FPIC=
-    LIBNAME=
+    LIBNAME="libmpdec.dll"
     LIBIMPORT="libmpdec.dll.a"
-    LIBSONAME=
-    LIBSHARED="libmpdec-4.dll"
-    CONFIGURE_LDFLAGS="-shared -Wl,--out-implib,$LIBIMPORT"
+    LIBSONAME="libmpdec-4.dll"
+    LIBSHARED="libmpdec-4.0.0.dll"
+    CONFIGURE_LDFLAGS="-shared"
     ENABLE_MINGW="yes"
     ;;
   *-*-msys)
