--- hdf5-1.14.0.old/config/commence.am
+++ hdf5-1.14.0.new/config/commence.am
@@ -37,7 +37,7 @@
 LIBH5JNI=$(top_builddir)/java/src/jni/libhdf5_java.la
 LIBH5TOOLS=$(top_builddir)/tools/lib/libh5tools.la
 LIBH5_HL=$(top_builddir)/hl/src/libhdf5_hl.la
-LIBH5F_HL=$(top_builddir)/hl/fortran/src/libhdf5hl_fortran.la
+LIBH5F_HL=$(top_builddir)/hl/fortran/src/libhdf5_hl_fortran.la
 LIBH5CPP_HL=$(top_builddir)/hl/c++/src/libhdf5_hl_cpp.la
 
 # Install directories that automake doesn't know about
