--- a/hl/fortran/src/Makefile
+++ a/hl/fortran/src/Makefile
@@ -1472,10 +1472,11 @@
 # H5HL_buildiface.F90 generates all the APIs that have a KIND type associated
 # with them.
 
-H5LTff_gen.F90: H5HL_buildiface$(EXEEXT)
-	$(RUNSERIAL) ./H5HL_buildiface$(EXEEXT)
+H5LTff_gen.F90:
+	cat ../../../../saved/H5LTff_gen.F90 >H5LTff_gen.F90
 
-H5TBff_gen.F90: H5HL_buildiface$(EXEEXT)
+H5TBff_gen.F90:
+	cat ../../../../saved/H5TBff_gen.F90 >H5TBff_gen.F90
 # Hardcode the dependencies of these files. There isn't a known way of
 # determining this automagically (like we do with the C files). So, when
 # doing a parallel make, some modules could be made way before the
