--- hdf5-1.14.0.old/src/H5private.h	2022-12-13 14:52:35
+++ hdf5-1.14.0.new/src/H5private.h	2023-04-10 11:51:05
@@ -1068,7 +1068,11 @@
 #define HDmemset(X, C, Z) memset(X, C, Z)
 #endif
 #ifndef HDmkdir
-#define HDmkdir(S, M) mkdir(S, M)
+#  if defined _WIN32
+#    define HDmkdir(S, M) mkdir(S)
+#  else
+#    define HDmkdir(S, M) mkdir(S, M)
+#  endif
 #endif
 #ifndef HDmkfifo
 #define HDmkfifo(S, M) mkfifo(S, M)
