--- a/src/Makefile
+++ a/src/Makefile
@@ -2175,26 +2175,16 @@
 # Things should have been all set during H5detect making.
 # Remove the generated .c file if errors occur unless HDF5_Make_Ignore
 # is set to ignore the error.
-H5Tinit.c: H5detect$(EXEEXT)
-	@if $(AM_V_P); then set -x; else echo "  GEN      H5Tinit.c"; fi;       \
-	LD_LIBRARY_PATH="$$LD_LIBRARY_PATH`echo $(LDFLAGS) |                    \
-	    sed -e 's/-L/:/g' -e 's/ //g'`"                                     \
-	$(RUNSERIAL) ./H5detect$(EXEEXT)  $@  ||                               	\
-	    (test $$HDF5_Make_Ignore && echo "*** Error ignored") ||            \
-	    ($(RM) $@ ; exit 1)
+H5Tinit.c:
+	cat ../../saved/H5Tinit.c >$@
 
 # Build configuration header file generation
 # The LD_LIBRARY_PATH setting is a kludge.
 # Things should have been all set during H5make_libsettings making.
 # Remove the generated .c file if errors occur unless HDF5_Make_Ignore
 # is set to ignore the error.
-H5lib_settings.c: H5make_libsettings$(EXEEXT) libhdf5.settings
-	@if $(AM_V_P); then set -x; else echo "  GEN      H5lib_settings.c"; fi;    \
-	LD_LIBRARY_PATH="$$LD_LIBRARY_PATH`echo $(LDFLAGS) |                        \
-	    sed -e 's/-L/:/g' -e 's/ //g'`"                                         \
-	$(RUNSERIAL) ./H5make_libsettings$(EXEEXT)  $@  ||                          \
-	    (test $$HDF5_Make_Ignore && echo "*** Error ignored") ||                \
-	    ($(RM) $@ ; exit 1)
+H5lib_settings.c:
+	echo 'char H5libhdf5_settings[]="";' >$@
 
 # Error header generation
 #
