diff -ru hdf5-1.14.0.old/src/H5private.h hdf5-1.14.0.new/src/H5private.h
--- hdf5-1.14.0.old/src/H5private.h	2022-12-13 14:52:35
+++ hdf5-1.14.0.new/src/H5private.h	2023-04-10 11:19:29
@@ -1367,7 +1367,7 @@
 #define HDstrncpy(X, Y, Z) strncpy(X, Y, Z)
 #endif
 #ifndef HDstrndup
-#define HDstrndup(S, N) strndup(S, N)
+#define HDstrndup(S, N) H5_strndup(S, N)
 #endif
 #ifndef HDstrpbrk
 #define HDstrpbrk(X, Y) strpbrk(X, Y)
$ diff -u hdf5-1.14.0.old/src/H5system.c hdf5-1.14.0.new/src/H5system.c
--- hdf5-1.14.0.old/src/H5system.c	2022-12-13 14:52:35
+++ hdf5-1.14.0.new/src/H5system.c	2023-04-18 11:33:50
@@ -918,6 +918,7 @@

     FUNC_LEAVE_NOAPI(ret_value)
 } /* end H5_expand_windows_env_vars() */
+#endif /* H5_HAVE_WIN32_API */

 /*-------------------------------------------------------------------------
  * Function:    H5_strndup
@@ -961,7 +962,6 @@
 done:
     FUNC_LEAVE_NOAPI(ret_value)
 }
-#endif /* H5_HAVE_WIN32_API */

 /* dirname() and basename() are not easily ported to Windows and basename
  * behavior varies depending on if you get POSIX vs. GNU. As a more
