diff --git a/src/macro/Driver.cc b/src/macro/Driver.cc
index c0419e02..760f592a 100644
--- a/src/macro/Driver.cc
+++ b/src/macro/Driver.cc
@@ -41,7 +41,7 @@ Driver::parse(const string &file_arg, const string &basename_arg, const istream
     }
 
   // Handle empty files
-  if (modfile.rdbuf()->in_avail() == 0)
+  if (modfile.peek() == std::istream::traits_type::eof())
     return;
 
   stringstream file_with_endl;
