diff --git a/src/DynareMain.cc b/src/DynareMain.cc
index 723d948a..c026a773 100644
--- a/src/DynareMain.cc
+++ b/src/DynareMain.cc
@@ -40,7 +40,7 @@
    Function can be found in: MacroExpandModFile.cc
 */
 stringstream
-macroExpandModFile(const string &filename, const string &basename, const istream &modfile,
+macroExpandModFile(const string &filename, const string &basename, istream &modfile,
                    bool debug, bool save_macro, string save_macro_file, bool line_macro,
                    const vector<pair<string, string>> &defines,
                    vector<filesystem::path> paths);
diff --git a/src/MacroExpandModFile.cc b/src/MacroExpandModFile.cc
index c39a9331..ae142504 100644
--- a/src/MacroExpandModFile.cc
+++ b/src/MacroExpandModFile.cc
@@ -26,7 +26,7 @@
 #include "macro/Driver.hh"
 
 stringstream
-macroExpandModFile(const string &filename, const string &basename, const istream &modfile,
+macroExpandModFile(const string &filename, const string &basename, istream &modfile,
                    bool debug, bool save_macro, string save_macro_file, bool line_macro,
                    const vector<pair<string, string>> &defines,
                    vector<filesystem::path> paths)
diff --git a/src/macro/Driver.cc b/src/macro/Driver.cc
index c0419e02..90d19eb9 100644
--- a/src/macro/Driver.cc
+++ b/src/macro/Driver.cc
@@ -23,7 +23,7 @@
 using namespace macro;
 
 void
-Driver::parse(const string &file_arg, const string &basename_arg, const istream &modfile,
+Driver::parse(const string &file_arg, const string &basename_arg, istream &modfile,
               bool debug, const vector<pair<string, string>> &defines,
               Environment &env, vector<filesystem::path> &paths, ostream &output)
 {
@@ -41,7 +41,7 @@ Driver::parse(const string &file_arg, const string &basename_arg, const istream
     }
 
   // Handle empty files
-  if (modfile.rdbuf()->in_avail() == 0)
+  if (modfile.peek() == std::istream::traits_type::eof())
     return;
 
   stringstream file_with_endl;
diff --git a/src/macro/Driver.hh b/src/macro/Driver.hh
index 4d600a80..7cf59909 100644
--- a/src/macro/Driver.hh
+++ b/src/macro/Driver.hh
@@ -84,7 +84,7 @@ namespace macro
 
     //! Starts parsing a file, modifies `env`, `paths` and `output`
     //! as they are modified by various macro directives
-    void parse(const string &file, const string &basename, const istream &modfile,
+    void parse(const string &file, const string &basename, istream &modfile,
                bool debug, const vector<pair<string, string>> &defines,
                Environment &env, vector<filesystem::path> &paths, ostream &output);
 
