diff --git a/src/Makefile.am b/src/Makefile.am
index 45d6d7e5..9115ef05 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -69,7 +69,7 @@ EXTRA_DIST = \
 
 # The -I. is for <FlexLexer.h>
 dynare_preprocessor_CPPFLAGS = $(BOOST_CPPFLAGS) -I.
-dynare_preprocessor_LDFLAGS = $(AM_LDFLAGS) $(BOOST_LDFLAGS)
+dynare_preprocessor_LDFLAGS = $(AM_LDFLAGS) $(BOOST_LDFLAGS) -static-libgcc 
 dynare_preprocessor_LDADD = macro/libmacro.a
 
 # -Ca flag comes from hitting a hard-coded size limit.
