From 6b5a9cc87d16d7556287f085a0bbd0cab66f5cf5 Mon Sep 17 00:00:00 2001
From: Cristy <mikayla-grace@urban-warrior.org>
Date: Sat, 6 Apr 2019 17:52:25 -0400
Subject: [PATCH] https://github.com/ImageMagick/ImageMagick/pull/1538

---
 coders/pdf.c | 1 +
 coders/ps.c  | 1 +
 2 files changed, 2 insertions(+)

diff --git a/coders/pdf.c b/coders/pdf.c
index ec385e5bd2..425beb97f7 100644
--- a/coders/pdf.c
+++ b/coders/pdf.c
@@ -243,6 +243,7 @@ static MagickBooleanType InvokePDFDelegate(const MagickBooleanType verbose,
       SetArgsStart(command,args_start);
       (void) fputs(args_start,stdout);
     }
+  interpreter=(gs_main_instance *) NULL;
   errors=(char *) NULL;
   status=(ghost_info->new_instance)(&interpreter,(void *) &errors);
   if (status < 0)
diff --git a/coders/ps.c b/coders/ps.c
index 828f6cfcbb..475b1bdda5 100644
--- a/coders/ps.c
+++ b/coders/ps.c
@@ -233,6 +233,7 @@ static MagickBooleanType InvokePostscriptDelegate(
       SetArgsStart(command,args_start);
       (void) fputs(args_start,stdout);
     }
+  interpreter=(gs_main_instance *) NULL;
   errors=(char *) NULL;
   status=(ghost_info->new_instance)(&interpreter,(void *) &errors);
   if (status < 0)
