diff --git before/MDOODZ7.0.git/MDLIB/CMakeLists.txt after/MDOODZ7.0.git/MDLIB/CMakeLists.txt
index 2c43df5..f309682 100644
--- before/MDOODZ7.0.git/MDLIB/CMakeLists.txt
+++ after/MDOODZ7.0.git/MDLIB/CMakeLists.txt
@@ -24,4 +24,6 @@ add_library(mdoodz SHARED
         ThermalRoutines.c
         ThermalSolver.c)
 target_link_libraries(mdoodz ${DEPENDENCIES})
-target_include_directories(mdoodz PUBLIC ${CMAKE_CURRENT_SOURCE_DIR}/include)
\ No newline at end of file
+target_include_directories(mdoodz PUBLIC ${CMAKE_CURRENT_SOURCE_DIR}/include)
+
+install(TARGETS mdoodz)