--- a/extCutPlan.cpp
+++ b/extCutPlan.cpp
@@ -60,8 +60,10 @@
     out_all_subproblems_py_completely_solved = false;
     out_number_of_subproblems_py_completely_solved = 0;
     
-    out_number_of_integer_solution_repetitions = 0;
-    out_integer_solution_history.desallocate();
+    #if MRQ_DEBUG_MODE
+        out_number_of_integer_solution_repetitions = 0;
+        out_integer_solution_history.desallocate();
+    #endif
     
     out_number_of_master_relaxation_solved = 0;
 }
