From 7234fd1a537971f2cbb4cd784dd3021e3d533776 Mon Sep 17 00:00:00 2001
From: Hans Schoenemann <hannes@mathematik.uni-kl.de>
Date: Fri, 8 Jan 2021 17:10:03 +0100
Subject: [PATCH] fix: add n_InitMPZ and n_MPZ for Zn

---
 libpolys/coeffs/rmodulon.cc | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/libpolys/coeffs/rmodulon.cc b/libpolys/coeffs/rmodulon.cc
index 72ba1ae01..b67f44786 100644
--- a/libpolys/coeffs/rmodulon.cc
+++ b/libpolys/coeffs/rmodulon.cc
@@ -862,6 +862,19 @@ nMapFunc nrnSetMap(const coeffs src, const coeffs dst)
   return NULL;      // default
 }
 
+static number nrnInitMPZ(mpz_t m, const coeffs r)
+{
+  mpz_ptr erg = (mpz_ptr)omAllocBin(gmp_nrz_bin);
+  mpz_init_set(erg,m);
+  mpz_mod(erg, erg, r->modNumber);
+  return (number) erg;
+}
+
+static void nrnMPZ(mpz_t m, number &n, const coeffs)
+{
+  mpz_init_set(m, (mpz_ptr)n);
+}
+
 /*
  * set the exponent (allocate and init tables) (TODO)
  */
@@ -1021,6 +1034,8 @@ BOOLEAN nrnInitChar (coeffs r, void* p)
   r->nCoeffIsEqual = nrnCoeffIsEqual;
   r->cfKillChar    = nrnKillChar;
   r->cfQuot1       = nrnQuot1;
+  r->cfInitMPZ     = nrnInitMPZ;
+  r->cfMPZ         = nrnMPZ;
 #if SI_INTEGER_VARIANT==2
   r->cfWriteFd     = nrzWriteFd;
   r->cfReadFd      = nrzReadFd;
-- 
2.28.0

