diff --git a/Make.inc b/Make.inc
index dfc2594c40..72fb4b4b01 100644
--- a/Make.inc
+++ b/Make.inc
@@ -774,7 +774,7 @@ else ifeq (cygwin, $(shell $(CC) -dumpmachine | cut -d\- -f3))
 $(error "cannot build julia with cygwin-target compilers. set XC_HOST to i686-w64-mingw32 or x86_64-w64-mingw32 for mingw cross-compile")
 else ifeq (msys, $(shell $(CC) -dumpmachine | cut -d\- -f3))
 $(error "cannot build julia with msys-target compilers. please see the README.windows document for instructions on setting up mingw-w64 compilers")
-else ifneq (,$(findstring MSYS,$(shell uname)))
+else ifneq (,$(findstring MSYS,$(BUILD_OS)))
 $(error "cannot build julia from a msys shell. please launch a mingw shell instead by setting MSYSTEM=MINGW64")
 endif
 
