--- a/include/openPMD/RecordComponent.tpp
+++ b/include/openPMD/RecordComponent.tpp
@@ -78,7 +78,7 @@
     for( auto const& dimensionSize : extent )
         numPoints *= dimensionSize;
 
-#if defined(__clang_major__) && __clang_major__ < 7
+#if defined(__clang_major__)
     auto newData =
         std::shared_ptr<T>(new T[numPoints], [](T *p) { delete[] p; });
     loadChunk(newData, offset, extent);
@@ -377,7 +377,7 @@
         std::move( extent ),
         []( size_t size )
         {
-#if defined(__clang_major__) && __clang_major__ < 7
+#if defined(__clang_major__)
             return std::shared_ptr< T >{
                 new T[ size ], []( auto * ptr ) { delete[] ptr; } };
 #else
--- a/include/openPMD/backend/PatchRecordComponent.hpp
+++ b/include/openPMD/backend/PatchRecordComponent.hpp
@@ -144,7 +144,7 @@
 inline std::shared_ptr<T> PatchRecordComponent::load()
 {
     uint64_t numPoints = getExtent()[0];
-#if defined(__clang_major__) && __clang_major__ < 7
+#if defined(__clang_major__)
     auto newData =
         std::shared_ptr<T>(new T[numPoints], [](T *p) { delete[] p; });
     load(newData);
