diff --git a/CMakeLists.txt b/CMakeLists.txt
index 49506e4..0431738 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.8.4)
+cmake_minimum_required(VERSION 3.16)
 project(QLever C CXX)
 
 # C/C++ Versions
@@ -51,6 +51,8 @@ include_directories(third_party/antlr4/runtime/Cpp/runtime/src)
 ################################
 # Threading
 ################################
+set(CMAKE_THREAD_PREFER_PTHREAD TRUE)
+set(THREADS_PREFER_PTHREAD_FLAG TRUE)
 find_package(Threads REQUIRED)
 
 #################################
@@ -111,8 +113,9 @@ set(CMAKE_C_FLAGS_DEBUG "${CMAKE_C_FLAGS_DEBUG}")
 ################################
 # GTEST
 ################################
-add_subdirectory(third_party/googletest/googletest EXCLUDE_FROM_ALL)
-include_directories(third_party/googletest/googletest/include)
+# add_subdirectory(third_party/googletest/googletest EXCLUDE_FROM_ALL)
+# include_directories(third_party/googletest/googletest/include)
+include_directories(${GTEST_INCLUDE_DIR})
 
 ################################
 # NLOHNMANN-JSON
@@ -211,8 +214,8 @@ add_subdirectory(src/parser)
 add_subdirectory(src/engine)
 add_subdirectory(src/index)
 add_subdirectory(src/util)
-enable_testing()
-add_subdirectory(test)
+# enable_testing()
+# add_subdirectory(test)
 
 
 
@@ -222,26 +225,26 @@ configure_file(src/web/script.js script.js)
 
 
 add_executable(IndexBuilderMain src/index/IndexBuilderMain.cpp)
-target_link_libraries(IndexBuilderMain index ${CMAKE_THREAD_LIBS_INIT})
+target_link_libraries(IndexBuilderMain PUBLIC index Threads::Threads)
 
 add_executable(CreatePatternsMain src/index/CreatePatternsMain.cpp src/util/ConstexprSmallString.h)
-target_link_libraries(CreatePatternsMain index ${CMAKE_THREAD_LIBS_INIT})
+target_link_libraries(CreatePatternsMain PUBLIC index Threads::Threads)
 
 add_executable(ServerMain src/ServerMain.cpp)
-target_link_libraries (ServerMain engine ${CMAKE_THREAD_LIBS_INIT})
+target_link_libraries(ServerMain PUBLIC engine Threads::Threads)
 
 add_executable(PrefixHeuristicEvaluatorMain src/PrefixHeuristicEvaluatorMain.cpp)
-target_link_libraries (PrefixHeuristicEvaluatorMain index ${CMAKE_THREAD_LIBS_INIT})
+target_link_libraries(PrefixHeuristicEvaluatorMain PUBLIC index Threads::Threads)
 
 add_executable(TurtleParserMain src/TurtleParserMain.cpp)
-target_link_libraries(TurtleParserMain parser ${CMAKE_THREAD_LIBS_INIT} absl::flat_hash_map)
+target_link_libraries(TurtleParserMain PUBLIC parser absl::flat_hash_map Threads::Threads)
 
 add_executable(VocabularyMergerMain src/VocabularyMergerMain.cpp)
-target_link_libraries(VocabularyMergerMain index ${CMAKE_THREAD_LIBS_INIT})
+target_link_libraries(VocabularyMergerMain PUBLIC index Threads::Threads)
 
 add_executable(PermutationExporterMain src/index/PermutationExporterMain.cpp)
-target_link_libraries(PermutationExporterMain index ${CMAKE_THREAD_LIBS_INIT})
+target_link_libraries(PermutationExporterMain PUBLIC index Threads::Threads)
 
 #add_executable(TextFilterComparison src/experiments/TextFilterComparison.cpp)
-#target_link_libraries (TextFilterComparison experiments)
+#target_link_libraries(TextFilterComparison experiments)
 
diff --git a/src/index/CMakeLists.txt b/src/index/CMakeLists.txt
index bba049c..654d589 100644
--- a/src/index/CMakeLists.txt
+++ b/src/index/CMakeLists.txt
@@ -14,4 +14,4 @@ add_library(index
         PrefixHeuristic.cpp PrefixHeuristic.h
         CompressedRelation.h CompressedRelation.cpp)
 
-target_link_libraries(index parser ${STXXL_LIBRARIES} ${ICU_LIBRARIES} absl::flat_hash_map absl::flat_hash_set zstd)
+target_link_libraries(index PUBLIC parser ${STXXL_LIBRARIES} ${ICU_LIBRARIES} absl::flat_hash_map absl::flat_hash_set zstd Threads::Threads)
