diff --git a/CMakeLists.txt b/CMakeLists.txt
index 49506e4..b61ed9f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.8.4)
+cmake_minimum_required(VERSION 3.14)
 project(QLever C CXX)
 
 # C/C++ Versions
@@ -44,9 +44,10 @@ endif ()
 ###############################
 # ANTLR CPP RUNTIME FOR THE SPARQL PARSER
 ###############################
-add_subdirectory(third_party/antlr4/runtime/Cpp EXCLUDE_FROM_ALL)
-target_compile_options(antlr4_static PUBLIC -Wno-attributes -Wno-implicit-fallthrough -Wno-ambiguous-reversed-operator)
-include_directories(third_party/antlr4/runtime/Cpp/runtime/src)
+# add_subdirectory(third_party/antlr4/runtime/Cpp EXCLUDE_FROM_ALL)
+# target_compile_options(antlr4_static PUBLIC -Wno-attributes -Wno-implicit-fallthrough -Wno-ambiguous-reversed-operator)
+# include_directories(third_party/antlr4/runtime/Cpp/runtime/src)
+include_directories(${ANTLR_INCLUDE_DIR})
 
 ################################
 # Threading
@@ -111,8 +112,9 @@ set(CMAKE_C_FLAGS_DEBUG "${CMAKE_C_FLAGS_DEBUG}")
 ################################
 # GTEST
 ################################
-add_subdirectory(third_party/googletest/googletest EXCLUDE_FROM_ALL)
-include_directories(third_party/googletest/googletest/include)
+# add_subdirectory(third_party/googletest/googletest EXCLUDE_FROM_ALL)
+# include_directories(third_party/googletest/googletest/include)
+include_directories(${GTEST_INCLUDE_DIR})
 
 ################################
 # NLOHNMANN-JSON
@@ -211,8 +213,8 @@ add_subdirectory(src/parser)
 add_subdirectory(src/engine)
 add_subdirectory(src/index)
 add_subdirectory(src/util)
-enable_testing()
-add_subdirectory(test)
+# enable_testing()
+# add_subdirectory(test)
 
 
 
diff --git a/src/parser/sparqlParser/CMakeLists.txt b/src/parser/sparqlParser/CMakeLists.txt
index 43fc825..2d93f6a 100644
--- a/src/parser/sparqlParser/CMakeLists.txt
+++ b/src/parser/sparqlParser/CMakeLists.txt
@@ -4,4 +4,4 @@ add_library(sparqlParser SparqlQleverVisitor.h SparqlQleverVisitor.cpp
         generated/SparqlAutomaticListener.h generated/SparqlAutomaticListener.cpp
         generated/SparqlAutomaticParser.h generated/SparqlAutomaticParser.cpp
         generated/SparqlAutomaticVisitor.h generated/SparqlAutomaticVisitor.cpp)
-target_link_libraries(sparqlParser antlr4_static sparqlExpressions rdfEscaping absl::flat_hash_map)
+target_link_libraries(sparqlParser antlr4_runtime sparqlExpressions rdfEscaping absl::flat_hash_map)
diff --git a/src/parser/sparqlParser/generated/CMakeLists.txt b/src/parser/sparqlParser/generated/CMakeLists.txt
index 1c26fe6..69e2981 100644
--- a/src/parser/sparqlParser/generated/CMakeLists.txt
+++ b/src/parser/sparqlParser/generated/CMakeLists.txt
@@ -5,4 +5,4 @@ add_library(antlrSparqlParser
         SparqlAutomaticParser.h SparqlAutomaticParser.cpp
         SparqlAutomaticVisitor.h SparqlAutomaticVisitor.cpp)
 
-target_link_libraries(antlrSparqlParser antlr4_static)
\ No newline at end of file
+target_link_libraries(antlrSparqlParser antlr4_runtime)
diff --git a/src/util/HttpServer/HttpParser/CMakeLists.txt b/src/util/HttpServer/HttpParser/CMakeLists.txt
index b8b728f..064e965 100644
--- a/src/util/HttpServer/HttpParser/CMakeLists.txt
+++ b/src/util/HttpServer/HttpParser/CMakeLists.txt
@@ -4,4 +4,4 @@ add_library(httpParser AcceptHeaderQleverVisitor.h AcceptHeaderQleverVisitor.cpp
         generated/AcceptHeaderListener.h generated/AcceptHeaderListener.cpp
         generated/AcceptHeaderParser.h generated/AcceptHeaderParser.cpp
         generated/AcceptHeaderVisitor.h generated/AcceptHeaderVisitor.cpp)
-target_link_libraries(httpParser antlr4_static)
+target_link_libraries(httpParser antlr4_runtime)
