diff --git a/CMakeLists.txt b/CMakeLists.txt
index 49506e4..362a176 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.8.4)
+cmake_minimum_required(VERSION 3.14)
 project(QLever C CXX)
 
 # C/C++ Versions
@@ -47,6 +47,7 @@ endif ()
 add_subdirectory(third_party/antlr4/runtime/Cpp EXCLUDE_FROM_ALL)
 target_compile_options(antlr4_static PUBLIC -Wno-attributes -Wno-implicit-fallthrough -Wno-ambiguous-reversed-operator)
 include_directories(third_party/antlr4/runtime/Cpp/runtime/src)
+# include_directories(${ANTLR_INCLUDE_DIR})
 
 ################################
 # Threading
@@ -111,8 +112,9 @@ set(CMAKE_C_FLAGS_DEBUG "${CMAKE_C_FLAGS_DEBUG}")
 ################################
 # GTEST
 ################################
-add_subdirectory(third_party/googletest/googletest EXCLUDE_FROM_ALL)
-include_directories(third_party/googletest/googletest/include)
+# add_subdirectory(third_party/googletest/googletest EXCLUDE_FROM_ALL)
+# include_directories(third_party/googletest/googletest/include)
+include_directories(${GTEST_INCLUDE_DIR})
 
 ################################
 # NLOHNMANN-JSON
@@ -211,8 +213,8 @@ add_subdirectory(src/parser)
 add_subdirectory(src/engine)
 add_subdirectory(src/index)
 add_subdirectory(src/util)
-enable_testing()
-add_subdirectory(test)
+# enable_testing()
+# add_subdirectory(test)
 
 
 
