diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9b40ee29..8c3c9a8a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -87,7 +87,8 @@ endif()
 ######################################
 # BOOST
 ######################################
-find_package(Boost 1.74 COMPONENTS iostreams program_options REQUIRED)
+find_package(ZLIB)
+find_package(Boost 1.74 REQUIRED COMPONENTS iostreams program_options REQUIRED)
 include_directories(${Boost_INCLUDE_DIR})
 
 
@@ -239,7 +240,7 @@ add_executable(CreatePatternsMain src/index/CreatePatternsMain.cpp src/util/Cons
 target_link_libraries(CreatePatternsMain index ${CMAKE_THREAD_LIBS_INIT})
 
 add_executable(ServerMain src/ServerMain.cpp)
-target_link_libraries (ServerMain engine ${CMAKE_THREAD_LIBS_INIT} boost_program_options)
+target_link_libraries (ServerMain engine ${CMAKE_THREAD_LIBS_INIT} boost_program_options Boost::iostreams ZLIB::ZLIB)
 
 add_executable(PrefixHeuristicEvaluatorMain src/PrefixHeuristicEvaluatorMain.cpp)
 target_link_libraries (PrefixHeuristicEvaluatorMain index ${CMAKE_THREAD_LIBS_INIT})
