diff --git a/Makefile b/Makefile
index a9e7844..66fa1ea 100644
--- a/Makefile
+++ b/Makefile
@@ -114,7 +114,7 @@ ifeq ($(USE_SSL),1)
     SSL_LDFLAGS+=-L$(OPENSSL_PREFIX)/lib
   endif
 
-  SSL_LDFLAGS+=-lssl -lcrypto
+  SSL_LDFLAGS+=-lssl -lcrypto -L. -lhiredis
 endif
 
 ifeq ($(uname_S),FreeBSD)
@@ -175,7 +175,7 @@ $(STLIBNAME): $(OBJ)
 	$(STLIB_MAKE_CMD) $(STLIBNAME) $(OBJ)
 
 #################### SSL building rules start ####################
-$(SSL_DYLIBNAME): $(SSL_OBJ)
+$(SSL_DYLIBNAME): $(SSL_OBJ) $(DYLIBNAME)
 	$(SSL_DYLIB_MAKE_CMD) $(DYLIB_PLUGIN) -o $(SSL_DYLIBNAME) $(SSL_OBJ) $(REAL_LDFLAGS) $(LDFLAGS) $(SSL_LDFLAGS)
 
 $(SSL_STLIBNAME): $(SSL_OBJ)
