diff --git a/imagequant-sys/Cargo.toml b/imagequant-sys/Cargo.toml
index 50cb6e9..a71ab09 100644
--- a/imagequant-sys/Cargo.toml
+++ b/imagequant-sys/Cargo.toml
@@ -9,15 +9,15 @@ repository = "https://github.com/ImageOptim/libimagequant"
 documentation = "https://docs.rs/imagequant"
 categories = ["multimedia::images"]
 keywords = ["quantization", "palette", "image", "dither", "quant"]
-include = ["COPYRIGHT", "src/*.rs", "build.rs", "*.h", "README.md", "Cargo.toml"]
+include = ["COPYRIGHT", "src/*.rs", "*.h", "README.md", "Cargo.toml"]
 readme = "README.md"
 edition = "2021"
 links = "imagequant"
-build = "build.rs"
+# build = "build.rs" # Disable for BB cross compilation
 rust-version = "1.63"

 [lib]
-crate-type = ["staticlib", "lib"]
+crate-type = ["staticlib", "lib", "cdylib"]
 doctest = false
 path = "src/ffi.rs"
 name = "imagequant_sys"
