diff --git a/imagequant-sys/Cargo.toml b/imagequant-sys/Cargo.toml
index 50cb6e9..54d5769 100644
--- a/imagequant-sys/Cargo.toml
+++ b/imagequant-sys/Cargo.toml
@@ -13,11 +13,11 @@ include = ["COPYRIGHT", "src/*.rs", "build.rs", "*.h", "README.md", "Cargo.toml"
 readme = "README.md"
 edition = "2021"
 links = "imagequant"
-build = "build.rs"
+# build = "build.rs" # disabled for BinaryBuilder rust-version = "1.63"

 [lib]
-crate-type = ["staticlib", "lib"]
+crate-type = ["staticlib", "lib", "cdylib"]
 doctest = false
 path = "src/ffi.rs"
 name = "imagequant_sys"
