diff --git before/likwid-5.2.1/Makefile after/likwid-5.2.1/Makefile
index a196775..8077581 100644
--- before/likwid-5.2.1/Makefile
+++ after/likwid-5.2.1/Makefile
@@ -484,7 +484,9 @@ install: install_daemon install_freq install_appdaemon
 	@chmod 755 $(LIBPREFIX)
 	@install -m 755 $(TARGET_LIB) $(LIBPREFIX)/$(TARGET_LIB).$(VERSION).$(RELEASE)
 	@install -m 755 liblikwidpin.so $(LIBPREFIX)/liblikwidpin.so.$(VERSION).$(RELEASE)
-	@install -m 755 $(TARGET_HWLOC_LIB) $(LIBPREFIX)/$(shell basename $(TARGET_HWLOC_LIB)).$(VERSION).$(RELEASE)
+	@if [ "$(USE_INTERNAL_HWLOC)" = "true" ]; then \
+		install -m 755 $(TARGET_HWLOC_LIB) $(LIBPREFIX)/$(shell basename $(TARGET_HWLOC_LIB)).$(VERSION).$(RELEASE); \
+	fi
 	@if [ "$(LUA_INTERNAL)" = "true" ]; then \
 		install -m 755 $(TARGET_LUA_LIB) $(LIBPREFIX)/$(shell basename $(TARGET_LUA_LIB)).$(VERSION).$(RELEASE); \
 	fi
@@ -492,8 +494,10 @@ install: install_daemon install_freq install_appdaemon
 	@cd $(LIBPREFIX) && ln -fs $(TARGET_LIB).$(VERSION).$(RELEASE) $(TARGET_LIB).$(VERSION)
 	@cd $(LIBPREFIX) && ln -fs $(PINLIB).$(VERSION).$(RELEASE) $(PINLIB)
 	@cd $(LIBPREFIX) && ln -fs $(PINLIB).$(VERSION).$(RELEASE) $(PINLIB).$(VERSION)
-	@cd $(LIBPREFIX) && ln -fs $(shell basename $(TARGET_HWLOC_LIB)).$(VERSION).$(RELEASE) $(shell basename $(TARGET_HWLOC_LIB))
-	@cd $(LIBPREFIX) && ln -fs $(shell basename $(TARGET_HWLOC_LIB)).$(VERSION).$(RELEASE) $(shell basename $(TARGET_HWLOC_LIB)).$(VERSION)
+	@if [ "$(USE_INTERNAL_HWLOC)" = "true" ]; then \
+		cd $(LIBPREFIX) && ln -fs $(shell basename $(TARGET_HWLOC_LIB)).$(VERSION).$(RELEASE) $(shell basename $(TARGET_HWLOC_LIB)); \
+		cd $(LIBPREFIX) && ln -fs $(shell basename $(TARGET_HWLOC_LIB)).$(VERSION).$(RELEASE) $(shell basename $(TARGET_HWLOC_LIB)).$(VERSION); \
+	fi
 	@if [ "$(LUA_INTERNAL)" = "true" ]; then \
 		cd $(LIBPREFIX) && ln -fs $(shell basename $(TARGET_LUA_LIB)).$(VERSION).$(RELEASE) $(shell basename $(TARGET_LUA_LIB)); \
 		cd $(LIBPREFIX) && ln -fs $(shell basename $(TARGET_LUA_LIB)).$(VERSION).$(RELEASE) $(shell basename $(TARGET_LUA_LIB)).$(VERSION); \
diff --git before/likwid-5.2.1/config.mk after/likwid-5.2.1/config.mk
index aa623fd..2b4e984 100644
--- before/likwid-5.2.1/config.mk
+++ after/likwid-5.2.1/config.mk
@@ -16,7 +16,7 @@ PREFIX ?= /usr/local#NO SPACE
 # Set the default mode for MSR access.
 # This can usually be overriden on the commandline.
 # Valid values are: direct, accessdaemon and perf_event
-ACCESSMODE = accessdaemon#NO SPACE
+ACCESSMODE = perf_event#NO SPACE
 
 # Build Fortran90 module interface for Marker API. Adopt Fortran compiler
 # in ./make/include_<COMPILER>.mk if necessary. Default: ifort (even for
@@ -67,7 +67,7 @@ INSTALLED_ACCESSDAEMON = $(INSTALLED_PREFIX)/sbin/likwid-accessD#NO SPACE
 
 # Build the setFrequencies daemon to allow users setting the CPU and Uncore
 # frequency
-BUILDFREQ = true#NO SPACE
+BUILDFREQ = false#NO SPACE
 # Paths for frequencie deaemon after installation
 FREQDAEMON = $(PREFIX)/sbin/likwid-setFreq#NO SPACE
 INSTALLED_FREQDAEMON = $(INSTALLED_PREFIX)/sbin/likwid-setFreq#NO SPACE
diff --git before/likwid-5.2.1/make/include_GCC.mk after/likwid-5.2.1/make/include_GCC.mk
index 72850a1..7bf80a1 100644
--- before/likwid-5.2.1/make/include_GCC.mk
+++ after/likwid-5.2.1/make/include_GCC.mk
@@ -26,6 +26,7 @@ SHARED_LFLAGS = -shared -fvisibility=hidden
 DEFINES  = -DPAGE_ALIGNMENT=4096
 DEFINES  += -DLIKWID_MONITOR_LOCK
 DEFINES  += -DDEBUGLEV=0
+DEFINES  += -DHAS_SCHEDAFFINITY
 
 INCLUDES =
 LIBS     = -lm -lrt
