diff --git before/likwid-5.2.1/config.mk after/likwid-5.2.1/config.mk
index aa623fd..0d6e395 100644
--- before/likwid-5.2.1/config.mk
+++ after/likwid-5.2.1/config.mk
@@ -11,12 +11,12 @@
 COMPILER = GCC#NO SPACE
 
 # Path were to install likwid
-PREFIX ?= /usr/local#NO SPACE
+PREFIX ?= /workspace/destdir#NO SPACE
 
 # Set the default mode for MSR access.
 # This can usually be overriden on the commandline.
 # Valid values are: direct, accessdaemon and perf_event
-ACCESSMODE = accessdaemon#NO SPACE
+ACCESSMODE = perf_event#NO SPACE
 
 # Build Fortran90 module interface for Marker API. Adopt Fortran compiler
 # in ./make/include_<COMPILER>.mk if necessary. Default: ifort (even for
@@ -67,7 +67,7 @@ INSTALLED_ACCESSDAEMON = $(INSTALLED_PREFIX)/sbin/likwid-accessD#NO SPACE
 
 # Build the setFrequencies daemon to allow users setting the CPU and Uncore
 # frequency
-BUILDFREQ = true#NO SPACE
+BUILDFREQ = false#NO SPACE
 # Paths for frequencie deaemon after installation
 FREQDAEMON = $(PREFIX)/sbin/likwid-setFreq#NO SPACE
 INSTALLED_FREQDAEMON = $(INSTALLED_PREFIX)/sbin/likwid-setFreq#NO SPACE
