diff --git a/lib/kerasify/keras_model.h b/lib/kerasify/keras_model.h
index e912c44..5a98cbe 100644
--- a/lib/kerasify/keras_model.h
+++ b/lib/kerasify/keras_model.h
@@ -13,6 +13,7 @@
 #include <numeric>
 #include <string>
 #include <vector>
+#include <cstdio>
 
 #define KASSERT(x, ...)                                                        \
     if (!(x)) {                                                                \
