diff --git a/cpu_features/CMakeLists.txt b/cpu_features/CMakeLists.txt
index ecd9729af..6d6d5544e 100644
--- a/cpu_features/CMakeLists.txt
+++ b/cpu_features/CMakeLists.txt
@@ -149,9 +149,9 @@ add_library(CpuFeature::cpu_features ALIAS cpu_features)
 # program : list_cpu_features
 #
 
-add_executable(list_cpu_features ${PROJECT_SOURCE_DIR}/src/utils/list_cpu_features.c)
-target_link_libraries(list_cpu_features PRIVATE cpu_features)
-add_executable(CpuFeature::list_cpu_features ALIAS list_cpu_features)
+# add_executable(list_cpu_features ${PROJECT_SOURCE_DIR}/src/utils/list_cpu_features.c)
+# target_link_libraries(list_cpu_features PRIVATE cpu_features)
+# add_executable(CpuFeature::list_cpu_features ALIAS list_cpu_features)
 
 #
 # ndk_compat
@@ -219,7 +219,7 @@ endif()
 #
 
 include(GNUInstallDirs)
-install(TARGETS cpu_features list_cpu_features
+install(TARGETS cpu_features
   EXPORT CpuFeaturesTargets
   PUBLIC_HEADER DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/cpu_features
   ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
diff --git a/cpu_features/Makefile b/cpu_features/Makefile
index ec2de5a1f..c9b34478f 100644
--- a/cpu_features/Makefile
+++ b/cpu_features/Makefile
@@ -3,7 +3,7 @@
 # Makefile added by Tim Davis
 
 go:
-	cmake -S. -Bbuild -DBUILD_TESTING=OFF -DCMAKE_BUILD_TYPE=Release
+	cmake -S. -Bbuild -DBUILD_TESTING=OFF -DCMAKE_BUILD_TYPE=Release -DBUILD_SHARED_LIBS=OFF -DCMAKE_TOOLCHAIN_FILE=$(CMAKE_TARGET_TOOLCHAIN)
 	cmake --build build --config Release -j
 
 distclean: purge