diff --git a/SPEX/SPEX_Left_LU/Lib/Makefile b/SPEX/SPEX_Left_LU/Lib/Makefile
index 72e207a..7032800 100644
--- a/SPEX/SPEX_Left_LU/Lib/Makefile
+++ b/SPEX/SPEX_Left_LU/Lib/Makefile
@@ -22,7 +22,7 @@ SUITESPARSE ?= $(realpath $(CURDIR)/../../..)
 include ../../../SuiteSparse_config/SuiteSparse_config.mk
 
 # uncomment for extra error checking:
-CFLAGS += -Wall -Wextra -Wpedantic -Werror
+CFLAGS += -Wall -Wextra -Wpedantic
 
 # SPEX_Left LU depends on SuiteSparse_config, AMD, COLAMD, M, GMP, and MPFR
 LDLIBS += -lspexutil -lsuitesparseconfig -lamd -lcolamd -lm -lgmp -lmpfr
diff --git a/SPEX/SPEX_Util/Lib/Makefile b/SPEX/SPEX_Util/Lib/Makefile
index 754ab87..7ac6e88 100644
--- a/SPEX/SPEX_Util/Lib/Makefile
+++ b/SPEX/SPEX_Util/Lib/Makefile
@@ -22,7 +22,7 @@ SUITESPARSE ?= $(realpath $(CURDIR)/../../..)
 include ../../../SuiteSparse_config/SuiteSparse_config.mk
 
 # uncomment for extra error checking:
-CFLAGS += -Wall -Wextra -Wpedantic -Werror
+CFLAGS += -Wall -Wextra -Wpedantic
 
 # SPEX_Util depends on SuiteSparse_config, AMD, COLAMD, M, GMP, and MPFR
 LDLIBS += -lsuitesparseconfig -lamd -lcolamd -lm -lgmp -lmpfr
