From 83452dae259c4e6652b96bed1106dad03d3a8fac Mon Sep 17 00:00:00 2001
From: Jamie Mair <JamieMair@users.noreply.github.com>
Date: Mon, 3 Jun 2024 14:28:31 +0100
Subject: [PATCH 1/4] Removed gc-sections to allow cross compile

---
 cmake/MujocoLinkOptions.cmake | 14 +-------------
 1 file changed, 1 insertion(+), 13 deletions(-)

diff --git a/cmake/MujocoLinkOptions.cmake b/cmake/MujocoLinkOptions.cmake
index 242767f9..0a5bf246 100644
--- a/cmake/MujocoLinkOptions.cmake
+++ b/cmake/MujocoLinkOptions.cmake
@@ -45,18 +45,6 @@ function(get_mujoco_extra_link_options OUTPUT_VAR)
           set(EXTRA_LINK_OPTIONS ${EXTRA_LINK_OPTIONS} -fuse-ld=gold)
         endif()
       endif()
-
-      set(CMAKE_REQUIRED_FLAGS ${EXTRA_LINK_OPTIONS} "-Wl,--gc-sections")
-      check_c_source_compiles("int main() {}" SUPPORTS_GC_SECTIONS)
-      if(SUPPORTS_GC_SECTIONS)
-        set(EXTRA_LINK_OPTIONS ${EXTRA_LINK_OPTIONS} -Wl,--gc-sections)
-      else()
-        set(CMAKE_REQUIRED_FLAGS ${EXTRA_LINK_OPTIONS} "-Wl,-dead_strip")
-        check_c_source_compiles("int main() {}" SUPPORTS_DEAD_STRIP)
-        if(SUPPORTS_DEAD_STRIP)
-          set(EXTRA_LINK_OPTIONS ${EXTRA_LINK_OPTIONS} -Wl,-dead_strip)
-        endif()
-      endif()
     endif()
   endif()
 
@@ -64,4 +52,4 @@ function(get_mujoco_extra_link_options OUTPUT_VAR)
       ${EXTRA_LINK_OPTIONS}
       PARENT_SCOPE
   )
-endfunction()
+endfunction()
\ No newline at end of file
-- 
2.34.1


From 0f283fc37bbb0203116cd96169e354c8b29d5ee4 Mon Sep 17 00:00:00 2001
From: Jamie Mair <JamieMair@users.noreply.github.com>
Date: Tue, 4 Jun 2024 11:20:34 +0100
Subject: [PATCH 2/4] Continued macos fixes

---
 CMakeLists.txt                  | 1 +
 src/engine/engine_util_errmem.c | 6 +++++-
 2 files changed, 6 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index c1ac72a7..a6ee9fc6 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -50,6 +50,7 @@ endif()
 
 include(MujocoOptions)
 include(MujocoMacOS)
+enforce_mujoco_macosx_min_version()
 include(MujocoDependencies)
 
 set(MUJOCO_HEADERS
diff --git a/src/engine/engine_util_errmem.c b/src/engine/engine_util_errmem.c
index 7c63c1ce..eee75139 100644
--- a/src/engine/engine_util_errmem.c
+++ b/src/engine/engine_util_errmem.c
@@ -33,7 +33,11 @@ static inline void* mju_alignedMalloc(size_t size, size_t align) {
 #ifdef _WIN32
   return _aligned_malloc(size, align);
 #elif defined(__STDC_VERSION__) && __STDC_VERSION__ >= 201112L
-  return aligned_alloc(align, size);
+  void* ptr = NULL;
+  if (posix_memalign(&ptr, align, size) != 0) {
+      return NULL;
+  }
+  return ptr;
 #endif
 }
 
-- 
2.34.1


From d128d96281b1df99b0066bcea77392a17cdb937f Mon Sep 17 00:00:00 2001
From: Jamie Mair <JamieMair@users.noreply.github.com>
Date: Tue, 4 Jun 2024 11:39:20 +0100
Subject: [PATCH 3/4] Ensured that C++17 is being used directly for macos

---
 CMakeLists.txt | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index a6ee9fc6..5b6e222b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -53,6 +53,16 @@ include(MujocoMacOS)
 enforce_mujoco_macosx_min_version()
 include(MujocoDependencies)
 
+set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+set(CMAKE_CXX_EXTENSIONS OFF)
+
+# Ensure you're linking against the correct standard library
+if (CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -stdlib=libc++")
+  set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -stdlib=libc++")
+endif()
+
 set(MUJOCO_HEADERS
     include/mujoco/mjdata.h
     include/mujoco/mjexport.h
-- 
2.34.1


From ea8ddc7687bdb5abea972c14d93be36bae07eaa9 Mon Sep 17 00:00:00 2001
From: Jamie Mair <JamieMair@users.noreply.github.com>
Date: Tue, 4 Jun 2024 12:20:25 +0100
Subject: [PATCH 4/4] Edited the standards used

---
 CMakeLists.txt | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5b6e222b..d701ee7c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -54,14 +54,16 @@ enforce_mujoco_macosx_min_version()
 include(MujocoDependencies)
 
 set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_C_STANDARD 17)
+set(CMAKE_C_STANDARD_REQUIRED ON)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 set(CMAKE_CXX_EXTENSIONS OFF)
 
 # Ensure you're linking against the correct standard library
-if (CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
-  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -stdlib=libc++")
-  set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -stdlib=libc++")
-endif()
+# if (CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
+#   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -stdlib=libc++")
+#   set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -stdlib=libc++")
+# endif()
 
 set(MUJOCO_HEADERS
     include/mujoco/mjdata.h
-- 
2.34.1
