diff --git a/src/sparse_rref/thread_pool.hpp b/src/sparse_rref/thread_pool.hpp
index e9bb7ce..d116d70 100644
--- a/src/sparse_rref/thread_pool.hpp
+++ b/src/sparse_rref/thread_pool.hpp
@@ -96,7 +96,7 @@ import std;
     #endif
 #endif
 
-#if defined(__linux__)
+#if defined(__linux__) || defined(__FreeBSD__)
     // On Linux, <sys/sysmacros.h> defines macros called `major` and `minor`. We undefine them here so the `version` struct can work.
     #ifdef major
         #undef major
