diff --git a/CMakeLists.txt b/CMakeLists.txt
index dea4acb..832ae9b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -16,6 +16,8 @@ option(ENABLE_COMPACT_MONOMIAL_MAP "Use a more compact map for lookups in the mo
 option(ENABLE_STORE_TRACE "Enable a (very expensive) trace of each access (operator[]) into a store." OFF)
 option(ENABLE_STORE_DUMP "Enable a (very expensive) dump when deconstructing a store." OFF)
 option(ENABLE_NATIVE "Enable -march=native to target the host CPU (creates non-portable binaries)" ON)
+option(ENABLE_TEST "Build f4ncgb_test program" ON)
+option(ENABLE_SIGNAL "Enable signal handling for printing statistics" ON)
 
 set(MAX_BLOCKS 16 CACHE STRING "Maximum number of blocks to compile for. Must be integral. Changes compilation time (duplicates implementation for each block).")
 
@@ -106,6 +108,10 @@ if(SHARED_LIB)
   set_property(TARGET f4ncgb_bin PROPERTY INSTALL_RPATH "$ORIGIN/../${CMAKE_INSTALL_LIBDIR}")
 endif()
 
+if (NOT ENABLE_TEST)
+   set_target_properties(f4ncgb_test PROPERTIES EXCLUDE_FROM_ALL 1)
+endif()
+
 # needed for pthread flag
 target_link_libraries(f4ncgb_bin PUBLIC ${CMAKE_THREAD_LIBS_INIT} )
 
@@ -155,6 +161,13 @@ else()
   message(STATUS "Store dumping is disabled")
 endif()
 
+if(ENABLE_SIGNAL)
+   message(STATUS "Signal handling is enabled")
+   target_compile_definitions(f4ncgb_bin PUBLIC F4NCGB_ENABLE_SIGNAL)
+else()
+   message(STATUS "Signal handling is disabled")
+endif()
+
 if(CMAKE_BUILD_TYPE MATCHES "Sanitize")
   target_compile_options(f4ncgb PRIVATE -fsanitize=address -fsanitize=undefined -g)
   target_link_options(f4ncgb PRIVATE -fsanitize=address -fsanitize=undefined)
diff --git a/src/f4ncgb.cpp b/src/f4ncgb.cpp
index 62478a1..cf9499f 100644
--- a/src/f4ncgb.cpp
+++ b/src/f4ncgb.cpp
@@ -51,6 +51,7 @@ using namespace f4ncgb;
 
 static std::function<void()> print_statistics_fun;
 
+#ifdef F4NCGB_ENABLE_SIGNAL
 static void
 handle_signal(int signal) {
   if(signal == SIGUSR1) {
@@ -61,6 +62,7 @@ handle_signal(int signal) {
     }
   }
 }
+#endif
 
 /**
     Main Function of f4ncgb.
@@ -69,7 +71,9 @@ handle_signal(int signal) {
 */
 int
 main(int argc, char** argv) {
+#ifdef F4NCGB_ENABLE_SIGNAL
   signal(SIGUSR1, handle_signal);
+#endif
 
   // clang-format off
   po::options_description desc("f4ncgb, version " F4NCGB_VERSION "\n"
