--- a/src/f4.hpp.orig
+++ b/src/f4.hpp
@@ -530,7 +530,7 @@
           C& c = cofactor.c;
           mon_id a = cofactor.a();
           mon_id b = cofactor.b();
-          for(auto& cofactor_i : cofactors[(uint)cofactor.i()]) {
+          for(auto& cofactor_i : cofactors[(size_t)cofactor.i()]) {
             C cc;
             mpq_mul(cc.data(), c.data(), cofactor_i.c.data());
             mon_id aa = mons.get_product_id(a, cofactor_i.a());
