diff --git a/crypto/fipsmodule/cpucap/cpu_ppc64le.c b/crypto/fipsmodule/cpucap/cpu_ppc64le.c
index 6cad2c644..17286facd 100644
--- a/crypto/fipsmodule/cpucap/cpu_ppc64le.c
+++ b/crypto/fipsmodule/cpucap/cpu_ppc64le.c
@@ -17,6 +17,7 @@
 #if defined(OPENSSL_PPC64LE)
 
 #include <sys/auxv.h>
+#include <linux/auxvec.h>
 
 #if !defined(PPC_FEATURE2_HAS_VCRYPTO)
 // PPC_FEATURE2_HAS_VCRYPTO was taken from section 4.1.2.3 of the “OpenPOWER
