diff --git a/src/components/cuda/tests/Makefile b/src/components/cuda/tests/Makefile
index 557c7753b..9b7f85ac0 100644
--- a/src/components/cuda/tests/Makefile
+++ b/src/components/cuda/tests/Makefile
@@ -15,7 +15,7 @@ TESTS_NOCTX = concurrent_profiling_noCuCtx pthreads_noCuCtx \
 NVCC = $(PAPI_CUDA_ROOT)/bin/nvcc
 
 PAPI_FLAG = -DPAPI    # Comment this line for tests to run without PAPI profiling
-NVCFLAGS = -g -ccbin='$(CC)' $(PAPI_FLAG)
+NVCFLAGS = -g -ccbin='$(CXX)' $(PAPI_FLAG)
 ifeq ($(BUILD_SHARED_LIB),yes)
        NVCFLAGS += -Xcompiler -fpic
 endif
