diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7831bb4d..2198b6de 100755
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -12,7 +12,7 @@ set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
 # Options
 #############################
 option(BUILD_HIPIFY_CLANG "Enable building the CUDA->HIP converter" OFF)
-option(__HIP_ENABLE_PCH "Enable/Disable pre-compiled hip headers" ON)
+option(__HIP_ENABLE_PCH "Enable/Disable pre-compiled hip headers" OFF)
 
 if(__HIP_ENABLE_PCH)
   set(_pchStatus 1)
