diff --git a/bin/hipcc b/bin/hipcc
index 17a80755..b9f1c51f 100755
--- a/bin/hipcc
+++ b/bin/hipcc
@@ -682,9 +682,8 @@ if ($HIP_PLATFORM eq "amd") {
     }

     if ($hasHIP) {
-        if ($DEVICE_LIB_PATH ne "$ROCM_PATH/amdgcn/bitcode") {
-            $HIPCXXFLAGS .= " --hip-device-lib-path=\"$DEVICE_LIB_PATH\"";
-        }
+        $HIPCXXFLAGS .= " --hip-device-lib-path=$DEVICE_LIB_PATH";
+        $HIPCXXFLAGS .= " --rocm-path=$ROCM_PATH";
         $HIPCXXFLAGS .= " -fhip-new-launch-api";
     }
     if (not $isWindows) {
