diff --git a/bin/hipcc b/bin/hipcc
index b9f1c51f..1844a408 100755
--- a/bin/hipcc
+++ b/bin/hipcc
@@ -681,6 +681,18 @@ if ($HIP_PLATFORM eq "amd") {
         }
     }

+    # BinaryBuilder compile flags
+    $SYSTEM_TARGET = $ENV{'target'};
+    $HIPCXXFLAGS .= " -isystem " . glob "/opt/$SYSTEM_TARGET/$SYSTEM_TARGET/include/c++/*";
+    $HIPCXXFLAGS .= " -isystem " . glob "/opt/$SYSTEM_TARGET/$SYSTEM_TARGET/include/c++/*/$SYSTEM_TARGET";
+    $HIPCXXFLAGS .= " --sysroot=/opt/$SYSTEM_TARGET/$SYSTEM_TARGET/sys-root";
+
+    # BinaryBuilder link flags
+    $HIPLDFLAGS .= " --sysroot=/opt/$SYSTEM_TARGET/$SYSTEM_TARGET/sys-root";
+    $HIPLDFLAGS .= " -B" . glob "/opt/$SYSTEM_TARGET/lib/gcc/$SYSTEM_TARGET/*";
+    $HIPLDFLAGS .= " -L" . glob "/opt/$SYSTEM_TARGET/lib/gcc/$SYSTEM_TARGET/*";
+    $HIPLDFLAGS .= " -L/opt/$SYSTEM_TARGET/$SYSTEM_TARGET/lib64";
+
     if ($hasHIP) {
         $HIPCXXFLAGS .= " --hip-device-lib-path=$DEVICE_LIB_PATH";
         $HIPCXXFLAGS .= " --rocm-path=$ROCM_PATH";
