diff --git a/bin/hipcc b/bin/hipcc
index 17baf93a..3d579783 100755
--- a/bin/hipcc
+++ b/bin/hipcc
@@ -163,7 +163,7 @@ if (defined $HIP_ROCCLR_HOME) {
     }
 }
 
-if (defined $HIP_COMPILER and $HIP_PLATFORM eq "hcc" and $HIP_COMPILER eq "clang") {
+if (defined $HIP_COMPILER and $HIP_PLATFORM eq "rocclr" and $HIP_COMPILER eq "clang") {
   if (!defined $HIP_CLANG_PATH) {
     $HIP_CLANG_PATH = "$ROCM_PATH/llvm/bin";
   }
@@ -191,7 +191,7 @@ $setStdLib = 0;  # TODO - set to 0
 
 $default_amdgpu_target = 1;
 
-if ($HIP_PLATFORM eq "hcc" and $HIP_COMPILER eq "clang") {
+if ($HIP_PLATFORM eq "rocclr" and $HIP_COMPILER eq "clang") {
     $HIPCC="$HIP_CLANG_PATH/clang++";
 
     # If $HIPCC clang++ is not compiled, use clang instead
@@ -470,7 +470,7 @@ foreach $arg (@ARGV)
          $targetsStr .=  substr($arg, length($targetOpt));
          $default_amdgpu_target = 0;
          # hip-clang does not accept --amdgpu-target= options.
-         if ($HIP_PLATFORM eq "hcc" and $HIP_COMPILER eq "clang") {
+         if ($HIP_PLATFORM eq "rocclr" and $HIP_COMPILER eq "clang") {
              $swallowArg = 1;
          }
     }
@@ -690,13 +690,13 @@ foreach $arg (@ARGV)
                 $needCXXFLAGS = 1;
                 if ($HIP_COMPILE_CXX_AS_HIP eq '0' or $HIP_COMPILER ne "clang") {
                     $hasCXX = 1;
-                } elsif ($HIP_PLATFORM eq "hcc" and $HIP_COMPILER eq "clang") {
+                } elsif ($HIP_PLATFORM eq "rocclr" and $HIP_COMPILER eq "clang") {
                     $hasHIP = 1;
                     $toolArgs .= " -x hip";
                 }
             } elsif ((($arg =~ /\.cu$/ or $arg =~ /\.cuh$/) and $HIP_COMPILE_CXX_AS_HIP ne '0') or ($arg =~ /\.hip$/)) {
                 $needCXXFLAGS = 1;
-                if ($HIP_PLATFORM eq "hcc" and $HIP_COMPILER eq "clang") {
+                if ($HIP_PLATFORM eq "rocclr" and $HIP_COMPILER eq "clang") {
                     $hasHIP = 1;
                     $toolArgs .= " -x hip";
                 } else {
@@ -822,7 +822,7 @@ if ($needHipHcc) {
 # Reason is that NVCC uses the file extension to determine whether to compile in CUDA mode or
 # pass-through CPP mode.
 
-if ($HIP_PLATFORM eq "hcc" and $HIP_COMPILER eq "clang") {
+if ($HIP_PLATFORM eq "rocclr" and $HIP_COMPILER eq "clang") {
     # Set default optimization level to -O3 for hip-clang.
     if ($optArg eq "") {
         $HIPCXXFLAGS .= " -O3";
