diff --git a/lib/fold.c b/lib/fold.c
index ff609d6..d26ce9d 100644
--- a/lib/fold.c
+++ b/lib/fold.c
@@ -24,11 +24,7 @@
 
 /*@unused@*/
 static char rcsid[] UNUSED = "$Id: fold.c,v 1.38 2007/12/19 10:27:42 ivo Exp $";
-#ifdef __GNUC__
-#define INLINE inline
-#else
 #define INLINE
-#endif
 
 #define PAREN
 
diff --git a/lib/subopt.c b/lib/subopt.c
index f2a04f9..148218c 100644
--- a/lib/subopt.c
+++ b/lib/subopt.c
@@ -191,7 +191,7 @@ PUBLIC  double print_energy = 9999; /* printing threshold for use with logML */
 extern	int circ;
 PUBLIC	SOLUTION *subopt_circ(char *seq, char *sequence, int delta, FILE *fp);
 PRIVATE int *fM2;	 /* energies of M2 */
-PUBLIC	int	Fc, FcH, FcI, FcM;		/* parts of the exterior loop energies */
+PRIVATE	int	Fc, FcH, FcI, FcM;		/* parts of the exterior loop energies */
 
 PRIVATE void encode_seq(char *sequence) {
   unsigned int i,l;
