From 86baf9cd28c20ad3589546639b44b776edef1684 Mon Sep 17 00:00:00 2001
From: Alexander Barth <barth.alexander@gmail.com>
Date: Wed, 2 Mar 2022 14:24:23 +0100
Subject: [PATCH 2/3] include getopt.h if available

---
 ncdump/ncvalidator.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/ncdump/ncvalidator.c b/ncdump/ncvalidator.c
index dead0d03..8f275b10 100644
--- a/ncdump/ncvalidator.c
+++ b/ncdump/ncvalidator.c
@@ -60,6 +60,9 @@ THIS SOFTWARE.
 */
 
 #include "config.h"
+#ifdef HAVE_GETOPT_H
+#include <getopt.h>
+#endif
 #include <stdio.h>
 #include <stdlib.h>
 #include <sys/types.h>  /* open() */
-- 
2.25.1

