From bdafbf13ade3d75feeed07b9190c718e092ac3f6 Mon Sep 17 00:00:00 2001
From: Alexander Barth <barth.alexander@gmail.com>
Date: Fri, 13 May 2022 13:44:35 +0200
Subject: [PATCH] save rc in globalstate->rcinfo.triples

---
 libdispatch/drc.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/libdispatch/drc.c b/libdispatch/drc.c
index 64d34bd5..d39f133a 100644
--- a/libdispatch/drc.c
+++ b/libdispatch/drc.c
@@ -513,6 +513,7 @@ NC_rcfile_insert(const char* key, const char* value, const char* hostport)
     
     if(rc == NULL) {
 	rc = nclistnew();
+	globalstate->rcinfo.triples = rc;
 	if(rc == NULL) {ret = NC_ENOMEM; goto done;}
     }
     triple = rclocate(key,hostport);
-- 
2.25.1

