diff --git a/include/extractor/extraction_helper_functions.hpp b/include/extractor/extraction_helper_functions.hpp
index d9e0f9b44..97dafaf8a 100644
--- a/include/extractor/extraction_helper_functions.hpp
+++ b/include/extractor/extraction_helper_functions.hpp
@@ -1,7 +1,7 @@
 #ifndef EXTRACTION_HELPER_FUNCTIONS_HPP
 #define EXTRACTION_HELPER_FUNCTIONS_HPP
 
-#include <boost/spirit/include/phoenix.hpp>
+#include <boost/phoenix.hpp>
 #include <boost/spirit/include/qi.hpp>
 
 #include <boost/algorithm/string/replace.hpp>
diff --git a/include/server/api/base_parameters_grammar.hpp b/include/server/api/base_parameters_grammar.hpp
index 827855acf..885d1d594 100644
--- a/include/server/api/base_parameters_grammar.hpp
+++ b/include/server/api/base_parameters_grammar.hpp
@@ -8,7 +8,7 @@
 #include "engine/polyline_compressor.hpp"
 
 #include <boost/optional.hpp>
-#include <boost/spirit/include/phoenix.hpp>
+#include <boost/phoenix.hpp>
 #include <boost/spirit/include/qi.hpp>
 
 #include <limits>
diff --git a/include/server/api/match_parameter_grammar.hpp b/include/server/api/match_parameter_grammar.hpp
index 9893b01db..d65a64f84 100644
--- a/include/server/api/match_parameter_grammar.hpp
+++ b/include/server/api/match_parameter_grammar.hpp
@@ -4,7 +4,7 @@
 #include "server/api/route_parameters_grammar.hpp"
 #include "engine/api/match_parameters.hpp"
 
-#include <boost/spirit/include/phoenix.hpp>
+#include <boost/phoenix.hpp>
 #include <boost/spirit/include/qi.hpp>
 
 namespace osrm
diff --git a/include/server/api/nearest_parameter_grammar.hpp b/include/server/api/nearest_parameter_grammar.hpp
index 8327a9503..aa593b926 100644
--- a/include/server/api/nearest_parameter_grammar.hpp
+++ b/include/server/api/nearest_parameter_grammar.hpp
@@ -4,7 +4,7 @@
 #include "server/api/base_parameters_grammar.hpp"
 #include "engine/api/nearest_parameters.hpp"
 
-#include <boost/spirit/include/phoenix.hpp>
+#include <boost/phoenix.hpp>
 #include <boost/spirit/include/qi.hpp>
 
 namespace osrm
diff --git a/include/server/api/route_parameters_grammar.hpp b/include/server/api/route_parameters_grammar.hpp
index df5fb52df..9d04ffef6 100644
--- a/include/server/api/route_parameters_grammar.hpp
+++ b/include/server/api/route_parameters_grammar.hpp
@@ -4,7 +4,7 @@
 #include "server/api/base_parameters_grammar.hpp"
 #include "engine/api/route_parameters.hpp"
 
-#include <boost/spirit/include/phoenix.hpp>
+#include <boost/phoenix.hpp>
 #include <boost/spirit/include/qi.hpp>
 
 namespace osrm
diff --git a/include/server/api/table_parameter_grammar.hpp b/include/server/api/table_parameter_grammar.hpp
index c88f85299..6daf8ac6b 100644
--- a/include/server/api/table_parameter_grammar.hpp
+++ b/include/server/api/table_parameter_grammar.hpp
@@ -4,7 +4,7 @@
 #include "server/api/base_parameters_grammar.hpp"
 #include "engine/api/table_parameters.hpp"
 
-#include <boost/spirit/include/phoenix.hpp>
+#include <boost/phoenix.hpp>
 #include <boost/spirit/include/qi.hpp>
 
 namespace osrm
diff --git a/include/server/api/tile_parameter_grammar.hpp b/include/server/api/tile_parameter_grammar.hpp
index 8c22b3837..7ac7626f8 100644
--- a/include/server/api/tile_parameter_grammar.hpp
+++ b/include/server/api/tile_parameter_grammar.hpp
@@ -6,7 +6,7 @@
 #include "engine/hint.hpp"
 #include "engine/polyline_compressor.hpp"
 
-#include <boost/spirit/include/phoenix.hpp>
+#include <boost/phoenix.hpp>
 #include <boost/spirit/include/qi.hpp>
 
 #include <string>
diff --git a/include/server/api/trip_parameter_grammar.hpp b/include/server/api/trip_parameter_grammar.hpp
index ecb14db50..bc075e000 100644
--- a/include/server/api/trip_parameter_grammar.hpp
+++ b/include/server/api/trip_parameter_grammar.hpp
@@ -4,7 +4,7 @@
 #include "server/api/route_parameters_grammar.hpp"
 #include "engine/api/trip_parameters.hpp"
 
-#include <boost/spirit/include/phoenix.hpp>
+#include <boost/phoenix.hpp>
 #include <boost/spirit/include/qi.hpp>
 
 namespace osrm
diff --git a/include/storage/serialization.hpp b/include/storage/serialization.hpp
index d83abfd9a..35190d6af 100644
--- a/include/storage/serialization.hpp
+++ b/include/storage/serialization.hpp
@@ -10,7 +10,7 @@
 #include "storage/tar.hpp"
 
 #include <boost/assert.hpp>
-#include <boost/function_output_iterator.hpp>
+#include <boost/iterator/function_output_iterator.hpp>
 #include <boost/iterator/function_input_iterator.hpp>
 
 #include <cmath>
diff --git a/include/storage/shared_data_index.hpp b/include/storage/shared_data_index.hpp
index 41ae6dabb..8cafc572f 100644
--- a/include/storage/shared_data_index.hpp
+++ b/include/storage/shared_data_index.hpp
@@ -3,7 +3,7 @@
 
 #include "storage/shared_datatype.hpp"
 
-#include <boost/function_output_iterator.hpp>
+#include <boost/iterator/function_output_iterator.hpp>
 
 #include <type_traits>
 #include <unordered_map>
diff --git a/include/updater/csv_file_parser.hpp b/include/updater/csv_file_parser.hpp
index c9841798e..91fbfceb5 100644
--- a/include/updater/csv_file_parser.hpp
+++ b/include/updater/csv_file_parser.hpp
@@ -14,7 +14,7 @@
 #include <boost/exception/diagnostic_information.hpp>
 #include <boost/filesystem.hpp>
 #include <boost/iostreams/device/mapped_file.hpp>
-#include <boost/spirit/include/phoenix.hpp>
+#include <boost/phoenix.hpp>
 #include <boost/spirit/include/qi.hpp>
 
 #include <exception>
diff --git a/include/util/indexed_data.hpp b/include/util/indexed_data.hpp
index 5ce5bd4b3..a98033175 100644
--- a/include/util/indexed_data.hpp
+++ b/include/util/indexed_data.hpp
@@ -8,7 +8,7 @@
 #include "util/vector_view.hpp"
 
 #include <boost/assert.hpp>
-#include <boost/function_output_iterator.hpp>
+#include <boost/iterator/function_output_iterator.hpp>
 
 #include <array>
 #include <iterator>
diff --git a/mypatch.patch b/mypatch.patch
new file mode 100644
index 000000000..8e3adec97
--- /dev/null
+++ b/mypatch.patch
@@ -0,0 +1,155 @@
+diff --git a/include/extractor/extraction_helper_functions.hpp b/include/extractor/extraction_helper_functions.hpp
+index d9e0f9b44..97dafaf8a 100644
+--- a/include/extractor/extraction_helper_functions.hpp
++++ b/include/extractor/extraction_helper_functions.hpp
+@@ -1,7 +1,7 @@
+ #ifndef EXTRACTION_HELPER_FUNCTIONS_HPP
+ #define EXTRACTION_HELPER_FUNCTIONS_HPP
+ 
+-#include <boost/spirit/include/phoenix.hpp>
++#include <boost/phoenix.hpp>
+ #include <boost/spirit/include/qi.hpp>
+ 
+ #include <boost/algorithm/string/replace.hpp>
+diff --git a/include/server/api/base_parameters_grammar.hpp b/include/server/api/base_parameters_grammar.hpp
+index 827855acf..885d1d594 100644
+--- a/include/server/api/base_parameters_grammar.hpp
++++ b/include/server/api/base_parameters_grammar.hpp
+@@ -8,7 +8,7 @@
+ #include "engine/polyline_compressor.hpp"
+ 
+ #include <boost/optional.hpp>
+-#include <boost/spirit/include/phoenix.hpp>
++#include <boost/phoenix.hpp>
+ #include <boost/spirit/include/qi.hpp>
+ 
+ #include <limits>
+diff --git a/include/server/api/match_parameter_grammar.hpp b/include/server/api/match_parameter_grammar.hpp
+index 9893b01db..d65a64f84 100644
+--- a/include/server/api/match_parameter_grammar.hpp
++++ b/include/server/api/match_parameter_grammar.hpp
+@@ -4,7 +4,7 @@
+ #include "server/api/route_parameters_grammar.hpp"
+ #include "engine/api/match_parameters.hpp"
+ 
+-#include <boost/spirit/include/phoenix.hpp>
++#include <boost/phoenix.hpp>
+ #include <boost/spirit/include/qi.hpp>
+ 
+ namespace osrm
+diff --git a/include/server/api/nearest_parameter_grammar.hpp b/include/server/api/nearest_parameter_grammar.hpp
+index 8327a9503..aa593b926 100644
+--- a/include/server/api/nearest_parameter_grammar.hpp
++++ b/include/server/api/nearest_parameter_grammar.hpp
+@@ -4,7 +4,7 @@
+ #include "server/api/base_parameters_grammar.hpp"
+ #include "engine/api/nearest_parameters.hpp"
+ 
+-#include <boost/spirit/include/phoenix.hpp>
++#include <boost/phoenix.hpp>
+ #include <boost/spirit/include/qi.hpp>
+ 
+ namespace osrm
+diff --git a/include/server/api/route_parameters_grammar.hpp b/include/server/api/route_parameters_grammar.hpp
+index df5fb52df..9d04ffef6 100644
+--- a/include/server/api/route_parameters_grammar.hpp
++++ b/include/server/api/route_parameters_grammar.hpp
+@@ -4,7 +4,7 @@
+ #include "server/api/base_parameters_grammar.hpp"
+ #include "engine/api/route_parameters.hpp"
+ 
+-#include <boost/spirit/include/phoenix.hpp>
++#include <boost/phoenix.hpp>
+ #include <boost/spirit/include/qi.hpp>
+ 
+ namespace osrm
+diff --git a/include/server/api/table_parameter_grammar.hpp b/include/server/api/table_parameter_grammar.hpp
+index c88f85299..6daf8ac6b 100644
+--- a/include/server/api/table_parameter_grammar.hpp
++++ b/include/server/api/table_parameter_grammar.hpp
+@@ -4,7 +4,7 @@
+ #include "server/api/base_parameters_grammar.hpp"
+ #include "engine/api/table_parameters.hpp"
+ 
+-#include <boost/spirit/include/phoenix.hpp>
++#include <boost/phoenix.hpp>
+ #include <boost/spirit/include/qi.hpp>
+ 
+ namespace osrm
+diff --git a/include/server/api/tile_parameter_grammar.hpp b/include/server/api/tile_parameter_grammar.hpp
+index 8c22b3837..7ac7626f8 100644
+--- a/include/server/api/tile_parameter_grammar.hpp
++++ b/include/server/api/tile_parameter_grammar.hpp
+@@ -6,7 +6,7 @@
+ #include "engine/hint.hpp"
+ #include "engine/polyline_compressor.hpp"
+ 
+-#include <boost/spirit/include/phoenix.hpp>
++#include <boost/phoenix.hpp>
+ #include <boost/spirit/include/qi.hpp>
+ 
+ #include <string>
+diff --git a/include/server/api/trip_parameter_grammar.hpp b/include/server/api/trip_parameter_grammar.hpp
+index ecb14db50..bc075e000 100644
+--- a/include/server/api/trip_parameter_grammar.hpp
++++ b/include/server/api/trip_parameter_grammar.hpp
+@@ -4,7 +4,7 @@
+ #include "server/api/route_parameters_grammar.hpp"
+ #include "engine/api/trip_parameters.hpp"
+ 
+-#include <boost/spirit/include/phoenix.hpp>
++#include <boost/phoenix.hpp>
+ #include <boost/spirit/include/qi.hpp>
+ 
+ namespace osrm
+diff --git a/include/updater/csv_file_parser.hpp b/include/updater/csv_file_parser.hpp
+index c9841798e..91fbfceb5 100644
+--- a/include/updater/csv_file_parser.hpp
++++ b/include/updater/csv_file_parser.hpp
+@@ -14,7 +14,7 @@
+ #include <boost/exception/diagnostic_information.hpp>
+ #include <boost/filesystem.hpp>
+ #include <boost/iostreams/device/mapped_file.hpp>
+-#include <boost/spirit/include/phoenix.hpp>
++#include <boost/phoenix.hpp>
+ #include <boost/spirit/include/qi.hpp>
+ 
+ #include <exception>
+diff --git a/src/server/api/url_parser.cpp b/src/server/api/url_parser.cpp
+index 1a10f66c3..9d5087e94 100644
+--- a/src/server/api/url_parser.cpp
++++ b/src/server/api/url_parser.cpp
+@@ -2,7 +2,7 @@
+ #include "engine/polyline_compressor.hpp"
+ 
+ #include <boost/fusion/include/adapt_struct.hpp>
+-#include <boost/spirit/include/phoenix.hpp>
++#include <boost/phoenix.hpp>
+ #include <boost/spirit/include/qi.hpp>
+ #include <boost/spirit/repository/include/qi_iter_pos.hpp>
+ 
+diff --git a/src/util/conditional_restrictions.cpp b/src/util/conditional_restrictions.cpp
+index b748f1162..15ca91e37 100644
+--- a/src/util/conditional_restrictions.cpp
++++ b/src/util/conditional_restrictions.cpp
+@@ -1,7 +1,7 @@
+ #include "util/conditional_restrictions.hpp"
+ 
+ #include <boost/fusion/include/adapt_struct.hpp>
+-#include <boost/spirit/include/phoenix.hpp>
++#include <boost/phoenix.hpp>
+ #include <boost/spirit/include/qi.hpp>
+ 
+ namespace osrm
+diff --git a/src/util/opening_hours.cpp b/src/util/opening_hours.cpp
+index e3ce583d9..79feda61b 100644
+--- a/src/util/opening_hours.cpp
++++ b/src/util/opening_hours.cpp
+@@ -1,6 +1,6 @@
+ #include "util/opening_hours.hpp"
+ 
+-#include <boost/spirit/include/phoenix.hpp>
++#include <boost/phoenix.hpp>
+ #include <boost/spirit/include/qi.hpp>
+ 
+ #include <boost/io/ios_state.hpp>
diff --git a/src/engine/routing_algorithms/alternative_path_mld.cpp b/src/engine/routing_algorithms/alternative_path_mld.cpp
index 7020e8a8a..aa48621f0 100644
--- a/src/engine/routing_algorithms/alternative_path_mld.cpp
+++ b/src/engine/routing_algorithms/alternative_path_mld.cpp
@@ -13,7 +13,7 @@
 #include <utility>
 #include <vector>
 
-#include <boost/function_output_iterator.hpp>
+#include <boost/iterator/function_output_iterator.hpp>
 
 namespace osrm
 {
diff --git a/src/extractor/location_dependent_data.cpp b/src/extractor/location_dependent_data.cpp
index 60186c2d6..1aa59affc 100644
--- a/src/extractor/location_dependent_data.cpp
+++ b/src/extractor/location_dependent_data.cpp
@@ -8,7 +8,7 @@
 #include <rapidjson/istreamwrapper.h>
 
 #include <boost/filesystem.hpp>
-#include <boost/function_output_iterator.hpp>
+#include <boost/iterator/function_output_iterator.hpp>
 #include <boost/geometry/algorithms/equals.hpp>
 
 #include <fstream>
diff --git a/src/server/api/url_parser.cpp b/src/server/api/url_parser.cpp
index 1a10f66c3..9d5087e94 100644
--- a/src/server/api/url_parser.cpp
+++ b/src/server/api/url_parser.cpp
@@ -2,7 +2,7 @@
 #include "engine/polyline_compressor.hpp"
 
 #include <boost/fusion/include/adapt_struct.hpp>
-#include <boost/spirit/include/phoenix.hpp>
+#include <boost/phoenix.hpp>
 #include <boost/spirit/include/qi.hpp>
 #include <boost/spirit/repository/include/qi_iter_pos.hpp>
 
diff --git a/src/tools/components.cpp b/src/tools/components.cpp
index ea4f21dbf..7431bbca5 100644
--- a/src/tools/components.cpp
+++ b/src/tools/components.cpp
@@ -11,7 +11,7 @@
 #include "util/typedefs.hpp"
 
 #include <boost/filesystem.hpp>
-#include <boost/function_output_iterator.hpp>
+#include <boost/iterator/function_output_iterator.hpp>
 
 #include <tbb/parallel_sort.h>
 
diff --git a/src/util/conditional_restrictions.cpp b/src/util/conditional_restrictions.cpp
index b748f1162..15ca91e37 100644
--- a/src/util/conditional_restrictions.cpp
+++ b/src/util/conditional_restrictions.cpp
@@ -1,7 +1,7 @@
 #include "util/conditional_restrictions.hpp"
 
 #include <boost/fusion/include/adapt_struct.hpp>
-#include <boost/spirit/include/phoenix.hpp>
+#include <boost/phoenix.hpp>
 #include <boost/spirit/include/qi.hpp>
 
 namespace osrm
diff --git a/src/util/opening_hours.cpp b/src/util/opening_hours.cpp
index e3ce583d9..79feda61b 100644
--- a/src/util/opening_hours.cpp
+++ b/src/util/opening_hours.cpp
@@ -1,6 +1,6 @@
 #include "util/opening_hours.hpp"
 
-#include <boost/spirit/include/phoenix.hpp>
+#include <boost/phoenix.hpp>
 #include <boost/spirit/include/qi.hpp>
 
 #include <boost/io/ios_state.hpp>
diff --git a/unit_tests/storage/tar.cpp b/unit_tests/storage/tar.cpp
index 955aa0f2a..8a26423ad 100644
--- a/unit_tests/storage/tar.cpp
+++ b/unit_tests/storage/tar.cpp
@@ -3,7 +3,7 @@
 #include "../common/range_tools.hpp"
 #include "../common/temporary_file.hpp"
 
-#include <boost/function_output_iterator.hpp>
+#include <boost/iterator/function_output_iterator.hpp>
 #include <boost/iterator/function_input_iterator.hpp>
 #include <boost/test/unit_test.hpp>
 
