--- a/src/darkunistd.h
+++ b/src/darkunistd.h
@@ -7,7 +7,7 @@
 *  Please add functionality as needed
 */
 
-#include <Winsock2.h>
+#include <winsock2.h>
 #include <direct.h> /* for _getcwd() and _chdir() */
 #include "getopt.h"
 #include <io.h>
--- a/src/getopt.h
+++ b/src/getopt.h
@@ -67,7 +67,7 @@
 #include <stdlib.h>
 #include <string.h>
 #define WIN32_LEAN_AND_MEAN
-#include <Windows.h>
+#include <windows.h>
 
 #ifdef __cplusplus
 extern "C" {
--- a/src/gettimeofday.h
+++ b/src/gettimeofday.h
@@ -1,9 +1,9 @@
 #ifdef _WIN32
 #define WIN32_LEAN_AND_MEAN
-#include <Windows.h>
-#include <Winsock2.h>
+#include <windows.h>
+#include <winsock2.h>
 #include <stdint.h>
-#include < time.h >
+#include <time.h>
 #include "darknet.h"
 
 #define CLOCK_REALTIME (1)
