--- a/src/benchmark/LAGraph_demo.h	
+++ b/src/benchmark/LAGraph_demo.h
@@ -1101,7 +1101,7 @@
     char msg [LAGRAPH_MSG_LEN] ;
     msg [0] = '\0' ;
 
-    #ifdef __linux__
+    #ifdef __GLIBC__
     // Use mallopt to speedup malloc and free on Linux.  Otherwise, it can take
     // several seconds to free a large block of memory.  For this to be
     // effective, demo_init must be called before calling malloc/free, and
