diff --git a/src/util.h b/src/util.h
index ce26160bd..1f1c3b820 100644
--- a/src/util.h
+++ b/src/util.h
@@ -331,14 +331,14 @@ void libxc_free(void *ptr);
 #else
 
 template <class int_type>
-auto libxc_malloc(const int_type size){
+void* libxc_malloc(const int_type size){
   void * mem;
   cudaMallocManaged(&mem, size);
   return mem;
 }
 
 template <class int_type1, class int_type2>
-auto libxc_calloc(const int_type1 size1, const int_type2 size2){
+void* libxc_calloc(const int_type1 size1, const int_type2 size2){
   void * mem;
   cudaMallocManaged(&mem, size1*size2);
   cudaMemset(mem, 0, size1*size2);
