diff --git a/third_party/llvm-10.0/llvm/lib/Support/ErrorHandling.cpp b/third_party/llvm-10.0/llvm/lib/Support/ErrorHandling.cpp
index a946302..b6548e0 100644
--- a/third_party/llvm-10.0/llvm/lib/Support/ErrorHandling.cpp
+++ b/third_party/llvm-10.0/llvm/lib/Support/ErrorHandling.cpp
@@ -29,9 +29,9 @@
 #include <mutex>
 #include <new>
 
-#if defined(HAVE_UNISTD_H)
-# include <unistd.h>
-#endif
+// #if defined(HAVE_UNISTD_H)
+#include <unistd.h>
+// #endif
 #if defined(_MSC_VER)
 # include <io.h>
 # include <fcntl.h>
