diff --git a/install/extlibs_makefile b/install/extlibs_makefile
index 9c765ad..5865c21 100644
--- a/install/extlibs_makefile
+++ b/install/extlibs_makefile
@@ -59,7 +59,7 @@ libfox:
         mkdir ../FoX; \
 	(gzip -dc ../archive/fox.tgz | (cd ../FoX; tar -xvf -)); \
 	cd ../FoX/fox/; export FC=$(F90); export FCFLAGS="$(FOX_FLAGS)"; \
-	./configure --prefix=$(TOPDIR)/FoX ;\
+	update_configure_scripts ;\
+	./configure --build=$(MACHTYPE) --host=$(target)  --prefix=$(TOPDIR)/FoX ;\
     touch cp_test; \
     if cp -p cp_test cp_test_1; then \
         echo "cp -p works"; \
