diff -ru q-e-qe-7.1.old/install/extlibs_makefile q-e-qe-7.1.new/install/extlibs_makefile
--- q-e-qe-7.1.old/install/extlibs_makefile	2022-06-14 12:16:48.000000000 -0400
+++ q-e-qe-7.1.new/install/extlibs_makefile	2022-08-01 10:35:45.000000000 -0400
@@ -38,7 +38,8 @@
 	$(call update_submodule,external,fox)
 	if test ! -d ../FoX ; then \
   cd $(TOPDIR)/external/fox; \
-    ./configure --prefix=$(TOPDIR)/FoX FC=$(F90) FCFLAGS="$(FOX_FLAGS)"; \
+    update_configure_scripts; \
+    ./configure --build=$(MACHTYPE) --host=$(target) --prefix=$(TOPDIR)/FoX FC=$(F90) FCFLAGS="$(FOX_FLAGS)"; \
     touch cp_test; \
     if cp -p cp_test cp_test_1; then \
         rm cp_test_1; \
