diff --git before/openfoam/etc/bashrc after/openfoam/etc/bashrc
index 857ffaa..903154d 100644
--- before/openfoam/etc/bashrc
+++ after/openfoam/etc/bashrc
@@ -99,7 +99,7 @@ export WM_COMPILE_OPTION=Opt
 #   HPMPI | CRAY-MPICH | FJMPI | QSMPI | SGIMPI | INTELMPI | USERMPI
 # Specify SYSTEMOPENMPI1, SYSTEMOPENMPI2 for internal tracking (if desired)
 # Can also use INTELMPI-xyz etc and define your own wmake rule
-export WM_MPLIB=SYSTEMOPENMPI
+export WM_MPLIB=SYSTEMMPI
 
 
 #------------------------------------------------------------------------------

diff --git before/openfoam/wmake/rules/General/Gcc/c after/openfoam/wmake/rules/General/Gcc/c
index e43a85b..0db22f1 100644
--- before/openfoam/wmake/rules/General/Gcc/c
+++ after/openfoam/wmake/rules/General/Gcc/c
@@ -3,7 +3,7 @@
 #------------------------------------------------------------------------------
 SUFFIXES   += .c
 
-cc         := gcc$(COMPILER_VERSION)
+cc         := gcc$(COMPILER_VERSION) -Wl,-rpath-link=/workspace/srcdir/openfoam/platforms/linux64GccDPInt32Opt/lib -Wl,-rpath-link=/workspace/srcdir/openfoam/platforms/linux64GccDPInt32Opt/lib/dummy -Wl,-rpath-link=/workspace/srcdir/openfoam/platforms/linux64GccDPInt32Opt/lib/mpi-system
 
 cARCH      :=
 cDBUG      :=
diff --git before/openfoam/wmake/rules/General/Gcc/c++ after/openfoam/wmake/rules/General/Gcc/c++
index 1ccc6dd..c802039 100644
--- before/openfoam/wmake/rules/General/Gcc/c++
+++ after/openfoam/wmake/rules/General/Gcc/c++
@@ -6,7 +6,7 @@
 #------------------------------------------------------------------------------
 SUFFIXES   += .C .cc .cpp .cxx
 
-CC         := g++$(COMPILER_VERSION) -std=c++11
+CC         := g++$(COMPILER_VERSION) -std=c++11 -Wl,-rpath-link=/workspace/srcdir/openfoam/platforms/linux64GccDPInt32Opt/lib -Wl,-rpath-link=/workspace/srcdir/openfoam/platforms/linux64GccDPInt32Opt/lib/dummy -Wl,-rpath-link=/workspace/srcdir/openfoam/platforms/linux64GccDPInt32Opt/lib/mpi-system
 
 c++ARCH    :=
 c++DBUG    :=
