diff --git before/openfoam/etc/bashrc after/openfoam/etc/bashrc
index 548f726..f5316b7 100644
--- before/openfoam/etc/bashrc
+++ after/openfoam/etc/bashrc
@@ -103,7 +103,7 @@ export WM_COMPILE_OPTION=Opt
 #   HPMPI | CRAY-MPICH | FJMPI | QSMPI | SGIMPI | INTELMPI | USERMPI
 # Specify SYSTEMOPENMPI1, SYSTEMOPENMPI2 for internal tracking (if desired)
 # Can also use INTELMPI-xyz etc and define your own wmake rule
-export WM_MPLIB=SYSTEMOPENMPI
+export WM_MPLIB=SYSTEMMPI
 
 
 #------------------------------------------------------------------------------
diff --git before/openfoam/etc/config.sh/scotch after/openfoam/etc/config.sh/scotch
index bb7261a..07a62b2 100644
--- before/openfoam/etc/config.sh/scotch
+++ after/openfoam/etc/config.sh/scotch
@@ -1,42 +1,2 @@
-#----------------------------------*-sh-*--------------------------------------
-# =========                 |
-# \\      /  F ield         | OpenFOAM: The Open Source CFD Toolbox
-#  \\    /   O peration     |
-#   \\  /    A nd           | www.openfoam.com
-#    \\/     M anipulation  |
-#------------------------------------------------------------------------------
-#     Copyright (C) 2011-2016 OpenFOAM Foundation
-#     Copyright (C) 2017-2022 OpenCFD Ltd.
-#------------------------------------------------------------------------------
-# License
-#     This file is part of OpenFOAM, distributed under GPL-3.0-or-later.
-#
-# File
-#     etc/config.sh/scotch
-#     - sourced during wmake process only.
-#
-# Description
-#     Setup for SCOTCH include/libraries (usually ThirdParty installation).
-#
-#     To disable its use:               SCOTCH_VERSION=scotch-none
-#     For system-wide installations:    SCOTCH_VERSION=scotch-system
-#
-#     For central installations not located under ThirdParty:
-#       1. use scotch-system
-#       2. and provide full path for SCOTCH_ARCH_PATH
-#
-# Note
-#     No csh version. This file is only used by wmake.
-#
-#     Can also disable by renaming/removing this file or by creating an empty
-#     file with the same name at a user or site location.
-#
-# regressions noted in scotch-v7.0.1
-#------------------------------------------------------------------------------
-# USER EDITABLE PART: Changes made here may be lost with the next upgrade
-
-SCOTCH_VERSION=scotch_6.1.0
-export SCOTCH_ARCH_PATH=$WM_THIRD_PARTY_DIR/platforms/$WM_ARCH$WM_COMPILER$WM_PRECISION_OPTION$WM_LABEL_OPTION/$SCOTCH_VERSION
-
-# END OF (NORMAL) USER EDITABLE PART
-#------------------------------------------------------------------------------
+export SCOTCH_VERSION=6.1.0
+export SCOTCH_ARCH_PATH=/workspace/destdir
diff --git before/openfoam/wmake/rules/General/Gcc/c after/openfoam/wmake/rules/General/Gcc/c
index e43a85b..0db22f1 100644
--- before/openfoam/wmake/rules/General/Gcc/c
+++ after/openfoam/wmake/rules/General/Gcc/c
@@ -3,7 +3,7 @@
 #------------------------------------------------------------------------------
 SUFFIXES   += .c
 
-cc         := gcc$(COMPILER_VERSION)
+cc         := gcc$(COMPILER_VERSION) -Wl,-rpath-link=/workspace/srcdir/openfoam/platforms/linux64GccDPInt32Opt/lib -Wl,-rpath-link=/workspace/srcdir/openfoam/platforms/linux64GccDPInt32Opt/lib/dummy -Wl,-rpath-link=/workspace/srcdir/openfoam/platforms/linux64GccDPInt32Opt/lib/mpi-system
 
 cARCH      :=
 cDBUG      :=
diff --git before/openfoam/wmake/rules/General/Gcc/c++ after/openfoam/wmake/rules/General/Gcc/c++
index d874790..75d2418 100644
--- before/openfoam/wmake/rules/General/Gcc/c++
+++ after/openfoam/wmake/rules/General/Gcc/c++
@@ -6,7 +6,7 @@
 #------------------------------------------------------------------------------
 SUFFIXES   += .C .cc .cpp .cxx
 
-CC         := g++$(COMPILER_VERSION) -std=c++14
+CC         := g++$(COMPILER_VERSION) -std=c++14 -Wl,-rpath-link=/workspace/srcdir/openfoam/platforms/linux64GccDPInt32Opt/lib -Wl,-rpath-link=/workspace/srcdir/openfoam/platforms/linux64GccDPInt32Opt/lib/dummy -Wl,-rpath-link=/workspace/srcdir/openfoam/platforms/linux64GccDPInt32Opt/lib/mpi-system
 
 c++ARCH    :=
 c++DBUG    :=
