diff --git a/test/test-bz106632.c b/test/test-bz106632.c
index daa0c1e..20324fd 100644
--- a/test/test-bz106632.c
+++ b/test/test-bz106632.c
@@ -27,6 +27,9 @@
 #endif
 #include <stdlib.h>
 #include <dirent.h>
+#ifdef _WIN32
+#define mkdir(path,mode) _mkdir(path)
+#endif
 #ifndef HAVE_STRUCT_DIRENT_D_TYPE
 #include <sys/types.h>
 #include <sys/stat.h>
diff --git a/test/test-migration.c b/test/test-migration.c
index b7d96dc..fcfdf81 100644
--- a/test/test-migration.c
+++ b/test/test-migration.c
@@ -31,6 +31,9 @@
 #include <unistd.h>
 #include <sys/types.h>
 #include <dirent.h>
+#ifdef _WIN32
+#define mkdir(path,mode) _mkdir(path)
+#endif
 #ifndef HAVE_STRUCT_DIRENT_D_TYPE
 #include <sys/types.h>
 #include <sys/stat.h>
