--- a/lib/libsystem.c
+++ b/lib/libsystem.c
@@ -263,8 +263,10 @@
         return "SIGQUIT";
     case SIGILL: /* POSIX.1 */
         return "SIGILL";
+#ifdef SIGTRAP
     case SIGTRAP:
         return "SIGTRAP";
+#endif
     case SIGABRT: /* POSIX.1 */
         return "SIGABRT";
     case SIGBUS:
@@ -301,14 +303,22 @@
     case SIGURG:
         return "SIGURG";
 #endif
+#ifdef SIGXCPU
     case SIGXCPU:
         return "SIGXCPU";
+#endif
+#ifdef SIGXFSZ
     case SIGXFSZ:
         return "SIGXFSZ";
+#endif
+#ifdef SIGVTALRM
     case SIGVTALRM:
         return "SIGVTALRM";
+#endif
+#ifdef SIGPROF
     case SIGPROF:
         return "SIGPROF";
+#endif
 #ifdef SIGWINCH
     case SIGWINCH:
         return "SIGWINCH";
--- a/converter/ppm/ppmtompeg/psocket.c
+++ b/converter/ppm/ppmtompeg/psocket.c
@@ -9,6 +9,7 @@

============================================================================*/

-#define _XOPEN_SOURCE 500 /* Make sure stdio.h contains pclose() */
+#define _XOPEN_SOURCE 700 /* Make sure stdio.h contains pclose() */
 /* _ALL_SOURCE is needed on AIX to make the C library include the 
    socket services (e.g. define struct sockaddr) 
 
