--- a/cmake/Modules/AddCompilerRT.cmake
+++ b/cmake/Modules/AddCompilerRT.cmake
@@ -420,11 +420,11 @@
           endif()
         endif()
         if (NEED_EXPLICIT_ADHOC_CODESIGN)
-          add_custom_command(TARGET ${libname}
-            POST_BUILD
-            COMMAND codesign --sign - $<TARGET_FILE:${libname}>
-            WORKING_DIRECTORY ${COMPILER_RT_OUTPUT_LIBRARY_DIR}
-          )
+          # add_custom_command(TARGET ${libname}
+          #   POST_BUILD
+          #   COMMAND codesign --sign - $<TARGET_FILE:${libname}>
+          #   WORKING_DIRECTORY ${COMPILER_RT_OUTPUT_LIBRARY_DIR}
+          # )
         endif()
       endif()
     endif()
