diff --git a/src/Makefile b/src/Makefile
index 9eb21e9..ab6ef75 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -158,7 +158,7 @@ all: atomsk
 # Compile the program
 # Note that the preprocessor must be called (-cpp)
 atomsk: message modes compute general
-	$(FORTRAN) $(FFLAGS) -cpp -o $(BIN) $(OBJ)$(SEP)*.o atomsk.f90 -I $(OBJ) $(LAPACK)
+	$(FORTRAN) $(FFLAGS) -cpp -o $(bindir)/atomsk$(exeext) $(OBJ)$(SEP)*.o atomsk.f90 -I $(OBJ) $(LAPACK)
 	@echo ""
 	@echo "    \o/ Compilation was successful!"
 	@echo ""
@@ -169,7 +169,7 @@ atomsk: message modes compute general
 # Compile the program as a library
 # Note that the preprocessor must be called (-cpp)
 lib: message compute modes
-	$(FORTRAN) $(FFLAGS) -cpp -shared -o atomsk.so $(OBJ)$(SEP)*.o
+	$(FORTRAN) $(FFLAGS) -cpp -shared -o $(libdir)/libatomsk.$(dlext) $(OBJ)$(SEP)*.o -I $(OBJ) $(LAPACK)
 	@echo ""
 	@echo "    \o/ Atomsk was compiled as a shared library file, 'atomsk.so'."
 	@echo ""
