diff --git a/CMakeLists.txt b/CMakeLists.txt
index 69176d6..c806fe1 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -69,7 +69,7 @@ endif()
 
 # Add specific VulkanSDK Lib & Include for glslang (tested on Linux Ubuntu 22.04 LTS)
 if(NOT WIN32)
-	include_directories($ENV{VULKAN_SDK}/include/glslang/Include)
+	include_directories(${Vulkan_INCLUDE_DIRS}/glslang/Include)
 	link_directories($ENV{VULKAN_SDK}/lib)
 endif()
 
