--- a/3rdparty/ippicv/ippicv_lnx/iw/src/iw_core.c
+++ b/3rdparty/ippicv/ippicv_lnx/iw/src/iw_core.c
@@ -60,7 +60,7 @@
         #endif
     #endif
         #ifndef __APPLE__
-            #include <malloc.h>
+            #include <stdlib.h>
         #endif
 #endif
 
--- a/3rdparty/ippicv/ippicv_lnx/iw/src/iw_own.c
+++ b/3rdparty/ippicv/ippicv_lnx/iw/src/iw_own.c
@@ -29,7 +29,7 @@
 #endif
 #include <stdlib.h>
 #ifndef __APPLE__
-#include <malloc.h>
+#include <stdlib.h>
 #endif
 #endif
 
