diff --git a/modules/julia/CMakeLists.txt b/modules/julia/CMakeLists.txt
index fcf380d7..09154aa7 100644
--- a/modules/julia/CMakeLists.txt
+++ b/modules/julia/CMakeLists.txt
@@ -1,46 +1,8 @@
-if(NOT HAVE_JULIA)
-    message(STATUS "Julia not found. Not compiling Julia Bindings. ${HAVE_JULIA}")
-    ocv_module_disable(julia)
-elseif(NOT PYTHON_DEFAULT_EXECUTABLE)
+if(NOT PYTHON_DEFAULT_EXECUTABLE)
    message(WARNING "Python required for Julia bindings...disabling")
    ocv_module_disable(julia)
 endif()
 
-# WARN_MIXED_PRECISION
-macro(WARN_MIXED_PRECISION COMPILER_BITNESS JULIA_BITNESS)
-    set(MSG "Your compiler is ${COMPILER_BITNESS}-bit")
-    set(MSG "${MSG} but your version of Julia is ${JULIA_BITNESS}-bit.")
-    set(MSG "${MSG} To build Julia bindings, please switch to a ${JULIA_BITNESS}-bit compiler.")
-    message(WARNING ${MSG})
-endmacro()
-
-
-math(EXPR ARCH "${CMAKE_SIZEOF_VOID_P} * 8")
-if (${ARCH} EQUAL 32 AND ${Julia_WORD_SIZE} MATCHES "64")
-    warn_mixed_precision("32" "64")
-    ocv_module_disable(julia)
-    return()
-elseif (${ARCH} EQUAL 64 AND NOT ${Julia_WORD_SIZE} MATCHES "64")
-    warn_mixed_precision("64" "32")
-    ocv_module_disable(julia)
-    return()
-endif()
-
-if(NOT JlCxx_DIR)
-    execute_process(
-        COMMAND "${Julia_EXECUTABLE}" --startup-file=no -e "using CxxWrap; print(CxxWrap.CxxWrapCore.prefix_path())"
-        OUTPUT_VARIABLE JlCxx_DIR
-    )
-endif()
-
-if(JlCxx_DIR)
-    if(EXISTS ${JlCxx_DIR}/JlCxxConfig.cmake)
-    else()
-        message(STATUS "JlCxx found but not source build - disabling Julia module")
-        ocv_module_disable(julia)
-    endif()
-endif()
-
 find_package(JlCxx QUIET)
 
 if(NOT JlCxx_FOUND)
@@ -83,9 +45,35 @@ file(COPY ${HDR_PARSER_PATH} DESTINATION ${CMAKE_CURRENT_BINARY_DIR}/gen)
 
 message(STATUS "Generating Julia Binding Files")
 
+include("${OpenCV_SOURCE_DIR}/cmake/OpenCVBindingsPreprocessorDefinitions.cmake")
+
+ocv_bindings_generator_populate_preprocessor_definitions(
+  OPENCV_MODULES_BUILD
+  opencv_preprocessor_defs
+)
+
+set(__config_str
+"{
+  \"preprocessor_definitions\": {
+${opencv_preprocessor_defs}
+  }
+}")
+
+set(JSON_CONFIG_FILE_PATH "${CMAKE_CURRENT_BINARY_DIR}/gen_python_config.json")
+if(EXISTS "${JSON_CONFIG_FILE_PATH}")
+  file(READ "${JSON_CONFIG_FILE_PATH}" __content)
+else()
+  set(__content "")
+endif()
+if(NOT "${__content}" STREQUAL "${__config_str}")
+  file(WRITE "${JSON_CONFIG_FILE_PATH}" "${__config_str}")
+endif()
+unset(__config_str)
+
+
 execute_process(
                 WORKING_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/gen"
-                COMMAND ${PYTHON_DEFAULT_EXECUTABLE} "${CMAKE_CURRENT_BINARY_DIR}/gen/gen_all.py" ${CMAKE_SOURCE_DIR}/modules ${OPENCV_MODULES_BUILD}
+                COMMAND ${PYTHON_DEFAULT_EXECUTABLE} "${CMAKE_CURRENT_BINARY_DIR}/gen/gen_all.py" ${CMAKE_SOURCE_DIR}/modules "${JSON_CONFIG_FILE_PATH}" ${OPENCV_MODULES_BUILD}
                 )
 
 file(COPY ${CMAKE_CURRENT_BINARY_DIR}/gen/cpp_files/ DESTINATION ${CMAKE_CURRENT_BINARY_DIR}/gen/autogen_cpp)
@@ -94,11 +82,11 @@ file(COPY ${CMAKE_CURRENT_BINARY_DIR}/gen/jl_cxx_files/ DESTINATION ${CMAKE_CURR
 ocv_glob_module_sources()
 ocv_module_include_directories()
 ocv_warnings_disable(CMAKE_CXX_FLAGS -Wmissing-prototypes -Wmissing-declarations)
-
 ocv_add_library(${the_module} SHARED ${OPENCV_MODULE_${the_module}_HEADERS}
                                      ${OPENCV_MODULE_${the_module}_SOURCES}
                                      "${CMAKE_CURRENT_BINARY_DIR}/gen/autogen_cpp/cv_core.cpp")
 
+
 if(NOT JULIA_PKG_INSTALL_PATH)
     set(JULIA_PKG_INSTALL_PATH ${CMAKE_BINARY_DIR})
 endif()
@@ -113,23 +101,9 @@ install(DIRECTORY package/ DESTINATION ${JULIA_PKG_INSTALL_PATH})
 install(TARGETS ${the_module} LIBRARY DESTINATION ${JULIA_PKG_INSTALL_PATH}/OpenCV/src/lib)
 
 
-if(JULIA_PKG_INSTALL_ENV)
-    set(JULIA_PKG_EXECS "Pkg.activate(${JULIA_PKG_INSTALL_ENV});${JULIA_PKG_EXECS}")
-endif()
-
-
-set(JULIA_COMMAND "\"${Julia_EXECUTABLE}\" -e \"using Pkg; ${JULIA_PKG_EXECS}\"")
 # message(STATUS "Installing Julia bindings using ${JULIA_COMMAND}")
 
 
-if(NOT INSTALL_CREATE_DISTRIB)
-    install(CODE "
-    set(JULIA_PKG_PATH \"${JULIA_PKG_INSTALL_PATH}/OpenCV\")
-    execute_process(COMMAND \"${Julia_EXECUTABLE}\" ${CMAKE_CURRENT_LIST_DIR}/package/install_package.jl \${JULIA_PKG_PATH} OUTPUT_VARIABLE JULIA_INSTALL_OUT)
-    message(STATUS \"Install output: \${JULIA_INSTALL_OUT}\")
-    ")
-endif()
-
 
 # ocv_create_module()
 
diff --git a/modules/julia/gen/gen3_cpp.py b/modules/julia/gen/gen3_cpp.py
index 5019585f..23537198 100644
--- a/modules/julia/gen/gen3_cpp.py
+++ b/modules/julia/gen/gen3_cpp.py
@@ -74,7 +74,7 @@ class ClassInfo(ClassInfo):
     def get_cpp_code_header(self):
         if self.ismap:
             return 'mod.map_type<%s>("%s");\n'%(self.name, self.mapped_name)
-        if not self.base:
+        elif not self.base:
             return 'mod.add_type<%s>("%s");\n' % (self.name, self.mapped_name)
         else:
             return 'mod.add_type<%s>("%s", jlcxx::julia_base_type<%s>());\n' % (self.name, self.mapped_name, self.base)
@@ -226,8 +226,8 @@ class FuncVariant(FuncVariant):
 
 
 
-def gen(srcfiles):
-    namespaces, default_values = gen_tree(srcfiles)
+def gen(srcfiles, preprocessor_definitions):
+    namespaces, default_values = gen_tree(srcfiles, preprocessor_definitions)
     cpp_code = StringIO()
     include_code = StringIO()
     nsi = sorted(namespaces.items(), key =lambda x: x[0])
@@ -327,6 +327,8 @@ struct SuperType<%s>
 srcfiles = hdr_parser.opencv_hdr_list
 if len(sys.argv) > 1:
     srcfiles = [l.strip() for l in sys.argv[1].split(';')]
+    
+with open(sys.argv[2], "r") as fh:
+    config_json = json.load(fh)
 
-
-gen(srcfiles)
+gen(srcfiles, config_json['preprocessor_definitions'])
diff --git a/modules/julia/gen/gen3_julia.py b/modules/julia/gen/gen3_julia.py
index 2ce99f3e..87dfff18 100644
--- a/modules/julia/gen/gen3_julia.py
+++ b/modules/julia/gen/gen3_julia.py
@@ -36,8 +36,8 @@ class FuncVariant(FuncVariant):
         return 'const %s = OpenCV.%s_%s' %(self.mapped_name, ns, self.mapped_name)
 
 
-def gen(srcfiles):
-    namespaces, _ = gen_tree(srcfiles)
+def gen(srcfiles, preprocessor_definitions):
+    namespaces, _ = gen_tree(srcfiles, preprocessor_definitions)
 
     jl_code = StringIO()
     for name, ns in namespaces.items():
@@ -95,5 +95,7 @@ srcfiles = hdr_parser.opencv_hdr_list
 if len(sys.argv) > 1:
     srcfiles = [l.strip() for l in sys.argv[1].split(';')]
 
+with open(sys.argv[2], "r") as fh:
+    config_json = json.load(fh)
 
-gen(srcfiles)
+gen(srcfiles, config_json['preprocessor_definitions'])
diff --git a/modules/julia/gen/gen3_julia_cxx.py b/modules/julia/gen/gen3_julia_cxx.py
index 8597ff85..a28e22d8 100755
--- a/modules/julia/gen/gen3_julia_cxx.py
+++ b/modules/julia/gen/gen3_julia_cxx.py
@@ -167,8 +167,8 @@ class FuncVariant(FuncVariant):
 
 
 
-def gen(srcfiles):
-    namespaces, _ = gen_tree(srcfiles)
+def gen(srcfiles, preprocessor_definitions):
+    namespaces, _ = gen_tree(srcfiles, preprocessor_definitions)
 
     jl_code = StringIO()
     for name, ns in namespaces.items():
@@ -241,4 +241,7 @@ srcfiles = hdr_parser.opencv_hdr_list
 if len(sys.argv) > 1:
     srcfiles = [l.strip() for l in sys.argv[1].split(';')]
 
-gen(srcfiles)
+with open(sys.argv[2], "r") as fh:
+    config_json = json.load(fh)
+
+gen(srcfiles, config_json['preprocessor_definitions'])
diff --git a/modules/julia/gen/gen_all.py b/modules/julia/gen/gen_all.py
index 494c9e70..0b4431f8 100644
--- a/modules/julia/gen/gen_all.py
+++ b/modules/julia/gen/gen_all.py
@@ -13,6 +13,7 @@ mod_path = sys.argv[1]
 
 
 hdr_list = [
+mod_path+"/core/include/opencv2/core/version.hpp",
 mod_path+"/core/include/opencv2/core.hpp",
 mod_path+"/core/include/opencv2/core/base.hpp",
 mod_path+"/core/include/opencv2/core/bindings_utils.hpp",
@@ -21,7 +22,7 @@ mod_path+"/core/include/opencv2/core/persistence.hpp",
 mod_path+"/core/include/opencv2/core/types.hpp",
 mod_path+"/core/include/opencv2/core/utility.hpp"]
 
-for module in sys.argv[2:]:
+for module in sys.argv[3:]:
     if module=='opencv_imgproc':
         hdr_list.append(mod_path+"/imgproc/include/opencv2/imgproc.hpp")
     elif module =='opencv_dnn':
@@ -39,6 +40,6 @@ if not os.path.exists('autogen_cpp'):
     os.makedirs('autogen_cpp')
     os.makedirs('autogen_jl')
 
-subprocess.call([sys.executable, 'gen3_cpp.py', str(';'.join(hdr_list))])
-subprocess.call([sys.executable, 'gen3_julia_cxx.py', str(';'.join(hdr_list))])
-subprocess.call([sys.executable, 'gen3_julia.py', str(';'.join(hdr_list))])
+subprocess.call([sys.executable, 'gen3_cpp.py', str(';'.join(hdr_list)), sys.argv[2]])
+subprocess.call([sys.executable, 'gen3_julia_cxx.py', str(';'.join(hdr_list)), sys.argv[2]])
+subprocess.call([sys.executable, 'gen3_julia.py', str(';'.join(hdr_list)), sys.argv[2]])
diff --git a/modules/julia/gen/parse_tree.py b/modules/julia/gen/parse_tree.py
index 5b044bf1..0eaf5d38 100644
--- a/modules/julia/gen/parse_tree.py
+++ b/modules/julia/gen/parse_tree.py
@@ -116,11 +116,13 @@ class ClassProp(object):
         if "/RW" in decl[3]:
             self.readonly = False
 
+mapped_types = {"cv::RotatedRect"}  # Global variable
+
 class ClassInfo(object):
     def __init__(self, name, decl=None):
         self.name = name
         self.mapped_name = normalize_class_name(name)
-        self.ismap = False  #CV_EXPORTS_W_MAP
+        self.ismap = (name in mapped_types)  #CV_EXPORTS_W_MAP
         self.isalgorithm = False    #if class inherits from cv::Algorithm
         self.methods = {}   #Dictionary of methods
         self.props = []     #Collection of ClassProp associated with this class
@@ -429,8 +431,8 @@ def add_enum(name, decl):
 
 
 
-def gen_tree(srcfiles):
-    parser = hdr_parser.CppHeaderParser(generate_umat_decls=False, generate_gpumat_decls=False)
+def gen_tree(srcfiles, preprocessor_definitions):
+    parser = hdr_parser.CppHeaderParser(generate_umat_decls=False, generate_gpumat_decls=False, preprocessor_definitions=preprocessor_definitions)
 
     allowed_func_list = []
 
