diff --git before/entwine/util/executor.cpp after/entwine/util/executor.cpp
index 9ee43f4..a4a941b 100644
--- before/entwine/util/executor.cpp
+++ after/entwine/util/executor.cpp
@@ -18,6 +18,7 @@
 #include <pdal/StageFactory.hpp>
 #include <pdal/io/BufferReader.hpp>
 #include <pdal/io/LasReader.hpp>
+#include <pdal/io/LasHeader.hpp>
 
 #include <entwine/third/arbiter/arbiter.hpp>
 #include <entwine/types/schema.hpp>
