diff --git a/entwine/third/arbiter/arbiter.cpp b/entwine/third/arbiter/arbiter.cpp
index 5637ce2..d923106 100644
--- a/entwine/third/arbiter/arbiter.cpp
+++ b/entwine/third/arbiter/arbiter.cpp
@@ -918,7 +918,7 @@ const drivers::Http& Endpoint::getHttpDriver() const
 #include <sys/stat.h>
 #else
 #define UNICODE
-#include <Shlwapi.h>
+#include <shlwapi.h>
 #include <iterator>
 #include <locale>
 #include <codecvt>
