diff --git a/kernels/armsve/bli_kernels_armsve.h b/kernels/armsve/bli_kernels_armsve.h
index 46a39e85..d2711cb2 100644
--- a/kernels/armsve/bli_kernels_armsve.h
+++ b/kernels/armsve/bli_kernels_armsve.h
@@ -45,7 +45,7 @@ GEMM_UKR_PROT( dcomplex, z, gemm_armsve_asm_2vx10_unindexed )
 //GEMMSUP_KER_PROT( double,   d, gemmsup_rv_armsve_10x2v_unindexed )
 
 // Use SVE intrinsics only for referred cases.
-#if !defined(BLIS_FAMILY_A64FX)
+#if 1
 PACKM_KER_PROT( double,   d, packm_armsve256_int_8x10 )
 #endif
 PACKM_KER_PROT( double,   d, packm_armsve512_asm_16x10 )
