diff --git a/frame/include/bli_macro_defs.h b/frame/include/bli_macro_defs.h
index 8af3f5a2..ac10d383 100644
--- a/frame/include/bli_macro_defs.h
+++ b/frame/include/bli_macro_defs.h
@@ -70,15 +70,15 @@
 #define PASTECH(...) PASTECH_(__VA_ARGS__)(__VA_ARGS__)
 
 // Fortran-77 name-mangling macros.
-#define PASTEF770_(op)                         op  ## _
-#define PASTEF771_(ch,op)                      ch  ## op ## _
-#define PASTEF772_(ch1,ch2,op)                 ch1 ## ch2 ## op ## _
-#define PASTEF773_(ch1,ch2,ch3,op)             ch1 ## ch2 ## ch3 ## op ## _
-#define PASTEF774_(ch1,ch2,ch3,ch4,op)         ch1 ## ch2 ## ch3 ## ch4 ## op ## _
-#define PASTEF775_(ch1,ch2,ch3,ch4,ch5,op)     ch1 ## ch2 ## ch3 ## ch4 ## ch5 ## op ## _
-#define PASTEF776_(ch1,ch2,ch3,ch4,ch5,ch6,op) ch1 ## ch2 ## ch3 ## ch4 ## ch5 ## ch6 ## op ## _
-
-#define PASTEF77__(arg1,arg2,arg3,arg4,arg5,arg6,arg7,arg8,...) PASTEF77 ## arg8 ## _
+#define PASTEF770_(op)                         op  ## _64_
+#define PASTEF771_(ch,op)                      ch  ## op ## _64_
+#define PASTEF772_(ch1,ch2,op)                 ch1 ## ch2 ## op ## _64_
+#define PASTEF773_(ch1,ch2,ch3,op)             ch1 ## ch2 ## ch3 ## op ## _64_
+#define PASTEF774_(ch1,ch2,ch3,ch4,op)         ch1 ## ch2 ## ch3 ## ch4 ## op ## _64_
+#define PASTEF775_(ch1,ch2,ch3,ch4,ch5,op)     ch1 ## ch2 ## ch3 ## ch4 ## ch5 ## op ## _64_
+#define PASTEF776_(ch1,ch2,ch3,ch4,ch5,ch6,op) ch1 ## ch2 ## ch3 ## ch4 ## ch5 ## ch6 ## op ## _64_
+
+#define PASTEF77__(arg1,arg2,arg3,arg4,arg5,arg6,arg7,arg8,...) PASTEF77 ## arg8 ## _
 #define PASTEF77_(...) PASTEF77__(__VA_ARGS__, 6, 5, 4, 3, 2, 1, 0, XXX)
 #define PASTEF77(...) PASTEF77_(__VA_ARGS__)(__VA_ARGS__)
 
