--- a/workspace/destdir/CMake/OpenFHETargets-release.cmake
+++ b/workspace/destdir/CMake/OpenFHETargets-release.cmake
@@ -9,31 +9,31 @@ set(CMAKE_IMPORT_FILE_VERSION 1)
 set_property(TARGET OPENFHEcore APPEND PROPERTY IMPORTED_CONFIGURATIONS RELEASE)
 set_target_properties(OPENFHEcore PROPERTIES
   IMPORTED_IMPLIB_RELEASE "${_IMPORT_PREFIX}/lib/libOPENFHEcore.dll.a"
-  IMPORTED_LOCATION_RELEASE "${_IMPORT_PREFIX}/lib/libOPENFHEcore.dll"
+  IMPORTED_LOCATION_RELEASE "${_IMPORT_PREFIX}/bin/libOPENFHEcore.dll"
   )

 list(APPEND _IMPORT_CHECK_TARGETS OPENFHEcore )
-list(APPEND _IMPORT_CHECK_FILES_FOR_OPENFHEcore "${_IMPORT_PREFIX}/lib/libOPENFHEcore.dll.a" "${_IMPORT_PREFIX}/lib/libOPENFHEcore.dll" )
+list(APPEND _IMPORT_CHECK_FILES_FOR_OPENFHEcore "${_IMPORT_PREFIX}/lib/libOPENFHEcore.dll.a" "${_IMPORT_PREFIX}/bin/libOPENFHEcore.dll" )

 # Import target "OPENFHEpke" for configuration "Release"
 set_property(TARGET OPENFHEpke APPEND PROPERTY IMPORTED_CONFIGURATIONS RELEASE)
 set_target_properties(OPENFHEpke PROPERTIES
   IMPORTED_IMPLIB_RELEASE "${_IMPORT_PREFIX}/lib/libOPENFHEpke.dll.a"
-  IMPORTED_LOCATION_RELEASE "${_IMPORT_PREFIX}/lib/libOPENFHEpke.dll"
+  IMPORTED_LOCATION_RELEASE "${_IMPORT_PREFIX}/bin/libOPENFHEpke.dll"
   )

 list(APPEND _IMPORT_CHECK_TARGETS OPENFHEpke )
-list(APPEND _IMPORT_CHECK_FILES_FOR_OPENFHEpke "${_IMPORT_PREFIX}/lib/libOPENFHEpke.dll.a" "${_IMPORT_PREFIX}/lib/libOPENFHEpke.dll" )
+list(APPEND _IMPORT_CHECK_FILES_FOR_OPENFHEpke "${_IMPORT_PREFIX}/lib/libOPENFHEpke.dll.a" "${_IMPORT_PREFIX}/bin/libOPENFHEpke.dll" )

 # Import target "OPENFHEbinfhe" for configuration "Release"
 set_property(TARGET OPENFHEbinfhe APPEND PROPERTY IMPORTED_CONFIGURATIONS RELEASE)
 set_target_properties(OPENFHEbinfhe PROPERTIES
   IMPORTED_IMPLIB_RELEASE "${_IMPORT_PREFIX}/lib/libOPENFHEbinfhe.dll.a"
-  IMPORTED_LOCATION_RELEASE "${_IMPORT_PREFIX}/lib/libOPENFHEbinfhe.dll"
+  IMPORTED_LOCATION_RELEASE "${_IMPORT_PREFIX}/bin/libOPENFHEbinfhe.dll"
   )

 list(APPEND _IMPORT_CHECK_TARGETS OPENFHEbinfhe )
-list(APPEND _IMPORT_CHECK_FILES_FOR_OPENFHEbinfhe "${_IMPORT_PREFIX}/lib/libOPENFHEbinfhe.dll.a" "${_IMPORT_PREFIX}/lib/libOPENFHEbinfhe.dll" )
+list(APPEND _IMPORT_CHECK_FILES_FOR_OPENFHEbinfhe "${_IMPORT_PREFIX}/lib/libOPENFHEbinfhe.dll.a" "${_IMPORT_PREFIX}/bin/libOPENFHEbinfhe.dll" )

 # Commands beyond this point should not need to know the version.
 set(CMAKE_IMPORT_FILE_VERSION)

