--- a/src/binfhe/CMakeLists.txt
+++ b/src/binfhe/CMakeLists.txt
@@ -26,7 +26,9 @@ if ( BUILD_SHARED )
 	set_property(TARGET OPENFHEbinfhe PROPERTY RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
 	install(TARGETS OPENFHEbinfhe
 		EXPORT OpenFHETargets
-		DESTINATION lib)
+		RUNTIME DESTINATION bin
+		LIBRARY DESTINATION lib
+		ARCHIVE DESTINATION lib)
 endif()
 
 if( BUILD_STATIC )
--- a/src/core/CMakeLists.txt
+++ b/src/core/CMakeLists.txt
@@ -28,7 +28,9 @@ if ( BUILD_SHARED )
 	set_property(TARGET OPENFHEcore PROPERTY RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
 	install(TARGETS OPENFHEcore
 		EXPORT OpenFHETargets
-		DESTINATION lib)
+		RUNTIME DESTINATION bin
+		LIBRARY DESTINATION lib
+		ARCHIVE DESTINATION lib)
 endif()
 
 
--- a/src/pke/CMakeLists.txt
+++ b/src/pke/CMakeLists.txt
@@ -26,7 +26,9 @@ if( BUILD_SHARED )
 	set_property(TARGET OPENFHEpke PROPERTY RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
 	install(TARGETS OPENFHEpke
 		EXPORT OpenFHETargets
-		DESTINATION lib)
+		RUNTIME DESTINATION bin
+		LIBRARY DESTINATION lib
+		ARCHIVE DESTINATION lib)
 endif()
 
 if( BUILD_STATIC )
