--- a/src/binfhe/CMakeLists.txt
+++ b/src/binfhe/CMakeLists.txt
@@ -24,7 +24,10 @@ if(BUILD_SHARED)
     set_property(TARGET OPENFHEbinfhe PROPERTY VERSION ${BINFHE_VERSION})
     set_property(TARGET OPENFHEbinfhe PROPERTY SOVERSION ${BINFHE_VERSION_MAJOR})
     set_property(TARGET OPENFHEbinfhe PROPERTY RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
-    install(TARGETS OPENFHEbinfhe EXPORT OpenFHETargets DESTINATION lib)
+    install(TARGETS OPENFHEbinfhe EXPORT OpenFHETargets
+        RUNTIME DESTINATION bin
+        LIBRARY DESTINATION lib
+        ARCHIVE DESTINATION lib)
 endif()
 
 if(BUILD_STATIC)
--- a/src/core/CMakeLists.txt
+++ b/src/core/CMakeLists.txt
@@ -25,7 +25,10 @@ if(BUILD_SHARED)
     set_property(TARGET OPENFHEcore PROPERTY VERSION ${CORE_VERSION})
     set_property(TARGET OPENFHEcore PROPERTY SOVERSION ${CORE_VERSION_MAJOR})
     set_property(TARGET OPENFHEcore PROPERTY RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
-    install(TARGETS OPENFHEcore EXPORT OpenFHETargets DESTINATION lib)
+    install(TARGETS OPENFHEcore EXPORT OpenFHETargets
+        RUNTIME DESTINATION bin
+        LIBRARY DESTINATION lib
+        ARCHIVE DESTINATION lib)
 endif()
 
 if(BUILD_STATIC)
--- a/src/pke/CMakeLists.txt
+++ b/src/pke/CMakeLists.txt
@@ -24,7 +24,10 @@ if(BUILD_SHARED)
     set_property(TARGET OPENFHEpke PROPERTY VERSION ${PKE_VERSION})
     set_property(TARGET OPENFHEpke PROPERTY SOVERSION ${PKE_VERSION_MAJOR})
     set_property(TARGET OPENFHEpke PROPERTY RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
-    install(TARGETS OPENFHEpke EXPORT OpenFHETargets DESTINATION lib)
+    install(TARGETS OPENFHEpke EXPORT OpenFHETargets
+        RUNTIME DESTINATION bin
+        LIBRARY DESTINATION lib
+        ARCHIVE DESTINATION lib)
 endif()
 
 if(BUILD_STATIC)
