diff --git a/math/minuit2/CMakeLists.txt b/math/minuit2/CMakeLists.txt
index 8ff0b0f8c4a..f9f440d873b 100644
--- a/math/minuit2/CMakeLists.txt
+++ b/math/minuit2/CMakeLists.txt
@@ -5,6 +5,7 @@
 # For the list of contributors see $ROOTSYS/README/CREDITS.
 
 cmake_minimum_required(VERSION 3.1)
+project(Minuit2)
 
 option(minuit2_mpi "Enable support for MPI in Minuit2")
 option(minuit2_omp "Enable support for OpenMP in Minuit2")
